--- UnpackedTarball/mysqlcppconn/driver/nativeapi/mysql_client_api.cpp	2014-07-30 13:41:37.000000000 +0200
+++ UnpackedTarball/mysqlcppconn/driver/nativeapi/mysql_client_api.cpp	2014-09-17 22:12:26.000000000 +0200
@@ -32,10 +32,10 @@
 
 #ifdef MYSQLCLIENT_STATIC_BINDING
 /* MySQL client library is linked */
-# include "libmysql_static_proxy.h"
+ #include "libmysql_static_proxy.h"
 #else
 /* MySQL client library will be dynamically loaded */
-# include "libmysql_dynamic_proxy.h"
+ #include "libmysql_dynamic_proxy.h"
 #endif
 
 namespace sql
@@ -76,9 +76,9 @@
 #include "binding_config.h"
 
 #ifdef MYSQLCLIENT_STATIC_BINDING
-# include "libmysql_static_proxy.cpp"
+ #include "libmysql_static_proxy.cpp"
 #else
-# include "libmysql_dynamic_proxy.cpp"
+ #include "libmysql_dynamic_proxy.cpp"
 #endif
 
 /*
