diff --git a/sw/CppunitTest_sw_htmlexport.mk b/sw/CppunitTest_sw_htmlexport.mk
index 23cdef1..69803cd 100644
--- a/sw/CppunitTest_sw_htmlexport.mk
+++ b/sw/CppunitTest_sw_htmlexport.mk
@@ -15,23 +15,48 @@ $(eval $(call gb_CppunitTest_add_exception_objects,sw_htmlexport, \
     sw/qa/extras/htmlexport/htmlexport \
 ))
 
+$(eval $(call gb_CppunitTest_set_componentfile,sw_htmlexport,sw/util/sw))
+
+$(eval $(call gb_CppunitTest_use_library_objects,sw_htmlexport,sw))
+
 $(eval $(call gb_CppunitTest_use_libraries,sw_htmlexport, \
+    $(call gb_Helper_optional,AVMEDIA,avmedia) \
+    basegfx \
     comphelper \
     cppu \
 	cppuhelper \
+    drawinglayer \
+    editeng \
 	i18nlangtag \
+    i18nutil \
+    lng \
     sal \
-    sw \
+    salhelper \
+	sax \
+    sb \
+    sfx \
+    sot \
+    svl \
+    svt \
+    svx \
+    svxcore \
     test \
+    tk \
 	tl \
+    ucbhelper \
     unotest \
     utl \
+    $(if $(filter TRUE,$(DISABLE_SCRIPTING)),, \
+        vbahelper) \
     vcl \
+    xmlreader \
+    xo \
 	$(gb_UWINAPI) \
 ))
 
 $(eval $(call gb_CppunitTest_use_externals,sw_htmlexport,\
 	boost_headers \
+	icuuc \
     libxml2 \
 ))
 
@@ -71,7 +96,6 @@ $(eval $(call gb_CppunitTest_use_components,sw_htmlexport,\
 	sc/util/scfilt \
     package/util/package2 \
     sax/source/expatwrap/expwrap \
-    sw/util/sw \
     sw/util/swd \
     sw/util/msword \
     sfx2/util/sfx \
diff --git a/sw/qa/extras/htmlexport/htmlexport.cxx b/sw/qa/extras/htmlexport/htmlexport.cxx
index 2db20ef..d9a8d15 100644
--- a/sw/qa/extras/htmlexport/htmlexport.cxx
+++ b/sw/qa/extras/htmlexport/htmlexport.cxx
@@ -32,6 +32,12 @@ public:
         m_eUnit(FUNIT_NONE)
     {}
 
+    void setUp() SAL_OVERRIDE
+    {
+        SwModelTestBase::setUp();
+        SwGlobals::ensure();
+    }
+
 private:
     bool mustCalcLayoutOf(const char* filename) SAL_OVERRIDE
     {
