--- misc/build/postgresql-9.1.1/src/interfaces/libpq/Makefile	2011-09-22 23:57:57.000000000 +0200
+++ misc/build/postgresql-9.1.1.patched/src/interfaces/libpq/Makefile	2011-12-15 09:02:18.000000000 +0100
@@ -148,3 +148,6 @@
 maintainer-clean: distclean maintainer-clean-lib
 	$(MAKE) -C test $@
 	rm -f libpq-dist.rc
+
+libpq-flags.mk:
+	@printf '%s\n' 'LIBPQ_DEP_LIBS+=$(SHLIB_LINK)' > '$@'
--- misc/build/postgresql-9.1.1/src/interfaces/libpq/win32.mak	2011-12-14 14:28:59.000000000 +0100
+++ misc/build/postgresql-9.1.1.patched/src/interfaces/libpq/win32.mak	2011-12-15 09:11:37.000000000 +0100
@@ -11,14 +11,12 @@
 !ENDIF
 
 !IFDEF DEBUG
-OPT=/Od /Zi /MDd
+OPT=/Od /Zi
 LOPT=/DEBUG
-DEBUGDEF=/D _DEBUG
-OUTFILENAME=libpqd
+OUTFILENAME=libpq
 !ELSE
 OPT=/O2 /MD
 LOPT=
-DEBUGDEF=/D NDEBUG
 OUTFILENAME=libpq
 !ENDIF
 
@@ -67,18 +66,11 @@
 CPP=cl.exe
 RSC=rc.exe
 
-!IFDEF DEBUG
-OUTDIR=.\Debug
-INTDIR=.\Debug
-CPP_OBJS=.\Debug/
-!ELSE
-OUTDIR=.\Release
-INTDIR=.\Release
-CPP_OBJS=.\Release/
-!ENDIF
-
+OUTDIR=.
+INTDIR=.
+CPP_OBJS=./
 
-ALL : config "$(OUTDIR)\$(OUTFILENAME).lib" "$(OUTDIR)\$(OUTFILENAME).dll"
+ALL : config "$(OUTDIR)\$(OUTFILENAME).lib"
 
 CLEAN :
 	-@erase "$(INTDIR)\getaddrinfo.obj"
@@ -178,10 +170,11 @@
 "$(OUTDIR)" :
     if not exist "$(OUTDIR)/$(NULL)" mkdir "$(OUTDIR)"
 
-CPP_PROJ=/nologo /W3 /EHsc $(OPT) /I "..\..\include" /I "..\..\include\port\win32" /I "..\..\include\port\win32_msvc" /I "..\..\port" /I. /I "$(SSL_INC)" \
+CPP_PROJ=/nologo /W3 /EHsc $(OPT) /I "..\..\include" /I "..\..\include\port\win32" /I "..\..\include\port\win32_msvc" /I "..\..\port" /I. $(SOLARINC) /I $(WORKDIR)/UnpackedTarball/openssl/include \
  /D "FRONTEND" $(DEBUGDEF) \
  /D "WIN32" /D "_WINDOWS" /Fp"$(INTDIR)\libpq.pch" \
  /Fo"$(INTDIR)\\" /Fd"$(INTDIR)\\" /FD /c  \
+ /D "_CRT_NONSTDC_NO_DEPRECATE" \
  /D "_CRT_SECURE_NO_DEPRECATE" $(ADD_DEFINES)
 
 !IFDEF USE_SSL
@@ -222,7 +215,7 @@
 <<
 
 "$(INTDIR)\libpq.res" : "$(INTDIR)" libpq-dist.rc
-	$(RSC) $(RSC_PROJ) libpq-dist.rc
+	$(RSC) $(SOLARINC) $(RSC_PROJ) libpq-dist.rc
 
 
 "$(OUTDIR)\$(OUTFILENAME).dll" : "$(OUTDIR)" "$(INTDIR)\libpq.res"

