--- misc/commons-httpclient-3.1/src/java/org/apache/commons/httpclient/ConnectMethod.java	2012-05-07 12:34:01.000000000 -0500
+++ misc/build/commons-httpclient-3.1/src/java/org/apache/commons/httpclient/ConnectMethod.java	2012-05-07 12:37:50.000000000 -0500
@@ -38,7 +38,7 @@
 /**
  * Establishes a tunneled HTTP connection via the CONNECT method.
  *
- * @author Ortwin Gl???ck
+ * @author Ortwin Glueck
  * @author dIon Gillard
  * @author <a href="mailto:mbowler@GargoyleSoftware.com">Mike Bowler</a>
  * @author <a href="mailto:oleg@ural.ru">Oleg Kalnichevski</a>
diff -ru misc/commons-httpclient-3.1/src/java/org/apache/commons/httpclient/HttpClient.java misc/build/commons-httpclient-3.1/src/java/org/apache/commons/httpclient/HttpClient.java
--- misc/commons-httpclient-3.1/src/java/org/apache/commons/httpclient/HttpClient.java	2012-05-07 12:34:01.000000000 -0500
+++ misc/build/commons-httpclient-3.1/src/java/org/apache/commons/httpclient/HttpClient.java	2012-05-07 12:37:50.000000000 -0500
@@ -48,7 +48,7 @@
  * @author <a href="mailto:rwaldhoff@apache.org">Rodney Waldhoff</a>
  * @author Sean C. Sullivan
  * @author <a href="mailto:dion@apache.org">dIon Gillard</a>
- * @author Ortwin Gl?ck
+ * @author Ortwin Glueck
  * @author <a href="mailto:becke@u.washington.edu">Michael Becke</a>
  * @author <a href="mailto:mbowler@GargoyleSoftware.com">Mike Bowler</a>
  * @author Sam Maloney
diff -ru misc/commons-httpclient-3.1/src/java/org/apache/commons/httpclient/HttpClientError.java misc/build/commons-httpclient-3.1/src/java/org/apache/commons/httpclient/HttpClientError.java
--- misc/commons-httpclient-3.1/src/java/org/apache/commons/httpclient/HttpClientError.java	2012-05-07 12:34:01.000000000 -0500
+++ misc/build/commons-httpclient-3.1/src/java/org/apache/commons/httpclient/HttpClientError.java	2012-05-07 12:37:50.000000000 -0500
@@ -33,7 +33,7 @@
 /**
  * Signals that an error has occurred.
  * 
- * @author Ortwin Gl?ck
+ * @author Ortwin Glueck
  * @version $Revision: 480424 $ $Date: 2006-11-29 06:56:49 +0100 (Wed, 29 Nov 2006) $
  * @since 3.0
  */
diff -ru misc/commons-httpclient-3.1/src/java/org/apache/commons/httpclient/HttpContentTooLargeException.java misc/build/commons-httpclient-3.1/src/java/org/apache/commons/httpclient/HttpContentTooLargeException.java
--- misc/commons-httpclient-3.1/src/java/org/apache/commons/httpclient/HttpContentTooLargeException.java	2012-05-07 12:34:01.000000000 -0500
+++ misc/build/commons-httpclient-3.1/src/java/org/apache/commons/httpclient/HttpContentTooLargeException.java	2012-05-07 12:37:50.000000000 -0500
@@ -3,7 +3,7 @@
 /**
  * Signals that the response content was larger than anticipated. 
  * 
- * @author Ortwin Glück
+ * @author Ortwin Glueck
  */
 public class HttpContentTooLargeException extends HttpException {
     private int maxlen;
diff -ru misc/commons-httpclient-3.1/src/java/org/apache/commons/httpclient/WireLogInputStream.java misc/build/commons-httpclient-3.1/src/java/org/apache/commons/httpclient/WireLogInputStream.java
--- misc/commons-httpclient-3.1/src/java/org/apache/commons/httpclient/WireLogInputStream.java	2012-05-07 12:34:01.000000000 -0500
+++ misc/build/commons-httpclient-3.1/src/java/org/apache/commons/httpclient/WireLogInputStream.java	2012-05-07 12:37:50.000000000 -0500
@@ -37,7 +37,7 @@
 /**
  * Logs all data read to the wire LOG.
  *
- * @author Ortwin Glï¿½ck
+ * @author Ortwin Glueck
  * @author <a href="mailto:mbowler@GargoyleSoftware.com">Mike Bowler</a>
  * @author <a href="mailto:oleg@ural.ru">Oleg Kalnichevski</a>
  * 
diff -ru misc/commons-httpclient-3.1/src/java/org/apache/commons/httpclient/auth/BasicScheme.java misc/build/commons-httpclient-3.1/src/java/org/apache/commons/httpclient/auth/BasicScheme.java
--- misc/commons-httpclient-3.1/src/java/org/apache/commons/httpclient/auth/BasicScheme.java	2012-05-07 12:34:01.000000000 -0500
+++ misc/build/commons-httpclient-3.1/src/java/org/apache/commons/httpclient/auth/BasicScheme.java	2012-05-07 12:37:50.000000000 -0500
@@ -46,7 +46,7 @@
  * @author <a href="mailto:remm@apache.org">Remy Maucherat</a>
  * @author Rodney Waldhoff
  * @author <a href="mailto:jsdever@apache.org">Jeff Dever</a>
- * @author Ortwin Gl?ck
+ * @author Ortwin Glueck
  * @author Sean C. Sullivan
  * @author <a href="mailto:adrian@ephox.com">Adrian Sutton</a>
  * @author <a href="mailto:mbowler@GargoyleSoftware.com">Mike Bowler</a>
diff -ru misc/commons-httpclient-3.1/src/java/org/apache/commons/httpclient/auth/DigestScheme.java misc/build/commons-httpclient-3.1/src/java/org/apache/commons/httpclient/auth/DigestScheme.java
--- misc/commons-httpclient-3.1/src/java/org/apache/commons/httpclient/auth/DigestScheme.java	2012-05-07 12:34:01.000000000 -0500
+++ misc/build/commons-httpclient-3.1/src/java/org/apache/commons/httpclient/auth/DigestScheme.java	2012-05-07 12:37:50.000000000 -0500
@@ -67,7 +67,7 @@
  * @author <a href="mailto:remm@apache.org">Remy Maucherat</a>
  * @author Rodney Waldhoff
  * @author <a href="mailto:jsdever@apache.org">Jeff Dever</a>
- * @author Ortwin Gl?ck
+ * @author Ortwin Glueck
  * @author Sean C. Sullivan
  * @author <a href="mailto:adrian@ephox.com">Adrian Sutton</a>
  * @author <a href="mailto:mbowler@GargoyleSoftware.com">Mike Bowler</a>
diff -ru misc/commons-httpclient-3.1/src/java/org/apache/commons/httpclient/auth/HttpAuthenticator.java misc/build/commons-httpclient-3.1/src/java/org/apache/commons/httpclient/auth/HttpAuthenticator.java
--- misc/commons-httpclient-3.1/src/java/org/apache/commons/httpclient/auth/HttpAuthenticator.java	2012-05-07 12:34:01.000000000 -0500
+++ misc/build/commons-httpclient-3.1/src/java/org/apache/commons/httpclient/auth/HttpAuthenticator.java	2012-05-07 12:37:50.000000000 -0500
@@ -62,7 +62,7 @@
  * @author <a href="mailto:remm@apache.org">Remy Maucherat</a>
  * @author Rodney Waldhoff
  * @author <a href="mailto:jsdever@apache.org">Jeff Dever</a>
- * @author Ortwin Glï¿½ck
+ * @author Ortwin Glueck
  * @author Sean C. Sullivan
  * @author <a href="mailto:adrian@ephox.com">Adrian Sutton</a>
  * @author <a href="mailto:mbowler@GargoyleSoftware.com">Mike Bowler</a>
diff -ru misc/commons-httpclient-3.1/src/java/org/apache/commons/httpclient/auth/NTLMScheme.java misc/build/commons-httpclient-3.1/src/java/org/apache/commons/httpclient/auth/NTLMScheme.java
--- misc/commons-httpclient-3.1/src/java/org/apache/commons/httpclient/auth/NTLMScheme.java	2012-05-07 12:34:01.000000000 -0500
+++ misc/build/commons-httpclient-3.1/src/java/org/apache/commons/httpclient/auth/NTLMScheme.java	2012-05-07 12:37:50.000000000 -0500
@@ -43,7 +43,7 @@
  * @author <a href="mailto:remm@apache.org">Remy Maucherat</a>
  * @author Rodney Waldhoff
  * @author <a href="mailto:jsdever@apache.org">Jeff Dever</a>
- * @author Ortwin Gl???ck
+ * @author Ortwin Glueck
  * @author Sean C. Sullivan
  * @author <a href="mailto:adrian@ephox.com">Adrian Sutton</a>
  * @author <a href="mailto:mbowler@GargoyleSoftware.com">Mike Bowler</a>
diff -ru misc/commons-httpclient-3.1/src/java/org/apache/commons/httpclient/methods/PostMethod.java misc/build/commons-httpclient-3.1/src/java/org/apache/commons/httpclient/methods/PostMethod.java
--- misc/commons-httpclient-3.1/src/java/org/apache/commons/httpclient/methods/PostMethod.java	2012-05-07 12:34:01.000000000 -0500
+++ misc/build/commons-httpclient-3.1/src/java/org/apache/commons/httpclient/methods/PostMethod.java	2012-05-07 12:37:50.000000000 -0500
@@ -62,7 +62,7 @@
  * @author <a href="mailto:remm@apache.org">Remy Maucherat</a>
  * @author <a href="mailto:dsale@us.britannica.com">Doug Sale</a>
  * @author <a href="mailto:jsdever@apache.org">Jeff Dever</a>
- * @author Ortwin Gl???ck
+ * @author Ortwin Glueck
  * @author <a href="mailto:mbowler@GargoyleSoftware.com">Mike Bowler</a>
  * @author <a href="mailto:oleg@ural.ru">Oleg Kalnichevski</a>
  *
