diff -ur misc/icu/source/data/Makefile.in misc/build/icu/source/data/Makefile.in
--- misc/icu/source/data/Makefile.in	Mi Jul 21 12:19:00 2010
+++ misc/build/icu/source/data/Makefile.in	Mi Jul 21 12:20:58 2010
@@ -77,7 +77,7 @@
 MISCSRCDIR=$(SRCDATADIR)/misc
 UCMSRCDIR=$(SRCDATADIR)/mappings
 COMINCDIR=$(top_srcdir)/common/unicode
-SRCLISTDEPS=Makefile $(srcdir)/Makefile.in
+SRCLISTDEPS=Makefile $(srcdir)/Makefile.in build-dir
 BUILD_DIRS=$(OUTDIR) $(MAINBUILDDIR) $(BUILDDIR) $(BRKBLDDIR) $(COLBLDDIR) $(RBNFBLDDIR) $(TRANSLITBLDDIR) $(OUTTMPDIR) $(OUTTMPDIR_390STUB) $(OUTTMPDIR)/$(COLLATION_TREE) $(OUTTMPDIR)/$(RBNF_TREE) $(OUTTMPDIR)/$(TRANSLIT_TREE) $(OUTTMPDIR)/$(BREAK_TREE)
 
 # relative lib links from pkgdata are the same as for tmp
diff -ru misc/icu/source/data/Makefile.in misc/build/icu//source/data/Makefile.in
--- misc/icu/source/data/Makefile.in	Do Jan 15 08:46:48 2009
+++ misc/build/icu//source/data/Makefile.in	Mi Jul 21 11:37:49 2010
@@ -132,7 +132,7 @@
 # and convert it to the current type.
 ifneq ($(ICUDATA_ARCHIVE),)
 ICUDATA_SOURCE_ARCHIVE = $(OUTDIR)/$(ICUDATA_PLATFORM_NAME).dat
-$(ICUDATA_SOURCE_ARCHIVE): $(ICUDATA_ARCHIVE) $(OUTDIR)
+$(ICUDATA_SOURCE_ARCHIVE): $(ICUDATA_ARCHIVE) build-dir
 	$(INVOKE) $(BINDIR)/icupkg -t$(ICUDATA_CHAR) $(ICUDATA_ARCHIVE) $(ICUDATA_SOURCE_ARCHIVE)
 endif
 else
