--- misc/icu/source/tools/toolutil/pkg_genc.h
+++ misc/build/icu/source/tools/toolutil/pkg_genc.h
@@ -58,7 +58,7 @@
 #endif
 
 #define LARGE_BUFFER_MAX_SIZE 2048
-#define SMALL_BUFFER_MAX_SIZE 512
+#define SMALL_BUFFER_MAX_SIZE 2048
 #define SMALL_BUFFER_FLAG_NAMES 32
 #define BUFFER_PADDING_SIZE 20
 
--- misc/icu/source/tools/toolutil/pkg_genc.c
+++ misc/build/icu/source/tools/toolutil/pkg_genc.c
@@ -145,6 +145,28 @@
 
         ".long ","",HEX_0X
     },
+    {"gcc-android-arm",
+	"\t.arch armv5te\n"
+	"\t.fpu softvfp\n"
+	"\t.eabi_attribute 20, 1\n"
+	"\t.eabi_attribute 21, 1\n"
+	"\t.eabi_attribute 23, 3\n"
+	"\t.eabi_attribute 24, 1\n"
+	"\t.eabi_attribute 25, 1\n"
+	"\t.eabi_attribute 26, 2\n"
+	"\t.eabi_attribute 30, 6\n"
+	"\t.eabi_attribute 18, 4\n"
+	"\t.file \"%s.s\"\n"
+	"\t.global %s\n"
+	"\t.section .rodata\n"
+	"\t.align 2\n"
+	"\t.type %s, %%object\n"
+	"%s:\n",
+
+	"\t.word ",
+	"\t.section .note.GNU-stack,\"\",%%progbits\n",
+	HEX_0X
+    },
     {"sun",
         "\t.section \".rodata\"\n"
         "\t.align   8\n"

--- misc/icu/source/config/mh-darwin	2010-09-29 20:37:36.000000000 +0200
+++ misc/build/icu/source/config/mh-darwin	2011-03-15 10:56:26.653056004 +0100
@@ -25,11 +25,7 @@
 SHLIB.cc=	$(CXX) -dynamiclib -dynamic $(CXXFLAGS) $(LDFLAGS) $(LD_SOOPTIONS)
 
 ## Compiler switches to embed a library name and version information
-ifeq ($(ENABLE_RPATH),YES)
-LD_SONAME = -Wl,-compatibility_version -Wl,$(SO_TARGET_VERSION_MAJOR) -Wl,-current_version -Wl,$(SO_TARGET_VERSION) -install_name $(libdir)/$(notdir $(MIDDLE_SO_TARGET))
-else
-LD_SONAME = -Wl,-compatibility_version -Wl,$(SO_TARGET_VERSION_MAJOR) -Wl,-current_version -Wl,$(SO_TARGET_VERSION) -install_name $(notdir $(MIDDLE_SO_TARGET))
-endif
+LD_SONAME = -Wl,-compatibility_version -Wl,$(SO_TARGET_VERSION_MAJOR) -Wl,-current_version -Wl,$(SO_TARGET_VERSION) -install_name @executable_path/$(notdir $(MIDDLE_SO_TARGET))
 
 ## Compiler switch to embed a runtime search path
 LD_RPATH=
@@ -45,10 +41,6 @@
 ## Non-shared intermediate object suffix
 STATIC_O = ao
 
-## Override Versioned target for a shared library.
-FINAL_SO_TARGET=  $(basename $(SO_TARGET)).$(SO_TARGET_VERSION).$(SO)
-MIDDLE_SO_TARGET= $(basename $(SO_TARGET)).$(SO_TARGET_VERSION_MAJOR).$(SO)
-
 ## Compilation and dependency rules
 %.$(STATIC_O): $(srcdir)/%.c
	$(call SILENT_COMPILE,$(strip $(COMPILE.c) $(STATICCPPFLAGS) $(STATICCFLAGS)) -MMD -MT "$*.d $*.o $*.$(STATIC_O)" -o $@ $<)
@@ -80,16 +72,10 @@
 
 ## Versioned libraries rules
 
-%.$(SO_TARGET_VERSION_MAJOR).$(SO): %.$(SO_TARGET_VERSION).$(SO)
+%.$(SO).$(SO_TARGET_VERSION_MAJOR): %.$(SO).$(SO_TARGET_VERSION)
 	$(RM) $@ && ln -s ${<F} $@
-%.$(SO): %.$(SO_TARGET_VERSION_MAJOR).$(SO)
-	$(RM) $@ && ln -s ${*F}.$(SO_TARGET_VERSION).$(SO) $@
-
-# tzcode option
-TZORIG_EXTRA_CFLAGS=-DSTD_INSPIRED
-
-# genren opts
-GENREN_PL_OPTS=-x Mach-O -n '-g' -p '| c++filt'
+%.$(SO): %.$(SO).$(SO_TARGET_VERSION_MAJOR)
+	$(RM) $@ && ln -s ${*F}.$(SO).$(SO_TARGET_VERSION) $@
 
 ## Remove shared library 's'
 STATIC_PREFIX_WHEN_USED = 
--- misc/icu/source/layout/ArabicShaping.cpp	2010-09-29 20:38:38.000000000 +0200
+++ misc/build/icu/source/layout/ArabicShaping.cpp	2011-03-15 10:56:26.655056238 +0100
@@ -79,7 +79,6 @@
 #define markFeatureMask 0x00040000UL
 #define mkmkFeatureMask 0x00020000UL
 
-#define NO_FEATURES   0
 #define ISOL_FEATURES (isolFeatureMask | ligaFeatureMask | msetFeatureMask | markFeatureMask | ccmpFeatureMask | rligFeatureMask | caltFeatureMask | dligFeatureMask | cswhFeatureMask | cursFeatureMask | kernFeatureMask | mkmkFeatureMask)
 
 #define SHAPE_MASK 0xF0000000UL
@@ -174,11 +173,7 @@
         LEUnicode c = chars[in];
         ShapeType t = getShapeType(c);
 
-        if (t == ST_NOSHAPE_NONE) {
-            glyphStorage.setAuxData(out, NO_FEATURES, success);
-        } else {
             glyphStorage.setAuxData(out, ISOL_FEATURES, success);
-        }
 
         if ((t & MASK_TRANSPARENT) != 0) {
             continue;
