--- misc/icu/source/icudefs.mk.in
+++ misc/build/icu/source/icudefs.mk.in
@@ -266,9 +266,9 @@
 include @platform_make_fragment@
 
 # Current full path directory.
-CURR_FULL_DIR=$(shell pwd | sed 's/ /\\ /g')
+CURR_FULL_DIR?=$(shell pwd | sed 's/ /\\ /g')
 # Current full path directory for use in source code in a -D compiler option.
-CURR_SRCCODE_FULL_DIR=$(shell pwd | sed 's/ /\\ /')
+CURR_SRCCODE_FULL_DIR?=$(shell pwd | sed 's/ /\\ /')
 
 # When shared libraries are disabled and static libraries are enabled,
 # the C++ compiler must be used to link in the libraries for the tools.
