diff -uriwb build/icu.old/source/common/uposixdefs.h misc/build/icu/source/common/uposixdefs.h
--- build/icu.old/source/common/uposixdefs.h	2012-04-05 21:46:18.000000000 +0100
+++ build/icu/source/common/uposixdefs.h	2012-06-21 15:45:17.613369477 +0100
@@ -52,7 +52,7 @@
  *
  * z/OS needs this definition for timeval and to get usleep.
  */
-#if !defined(_XOPEN_SOURCE_EXTENDED)
+#if !defined(_XOPEN_SOURCE_EXTENDED) && (defined(__IBMC__) || defined(__IBMCPP__))
 #   define _XOPEN_SOURCE_EXTENDED 1
 #endif
 
