--- misc/icu/source/common/unicode/utf16.h
+++ misc/build/icu/source/common/unicode/utf16.h
@@ -316,6 +316,7 @@
         (s)[(i)++]=(uint16_t)(((c)&0x3ff)|0xdc00); \
     } else /* c>0x10ffff or not enough space */ { \
         (isError)=TRUE; \
+        (void)(isError); \
     } \
 }
 
--- misc/icu/source/common/unicode/utypes.h
+++ misc/build/icu/source/common/unicode/utypes.h
@@ -399,7 +399,7 @@ typedef double UDate;
  * some Linux/Unix compilers have problems with defining global new/delete.
  * On Windows, it is _MSC_VER>=1200 for MSVC 6.0 and higher.
  */
-#if defined(__cplusplus) && U_DEBUG && U_OVERRIDE_CXX_ALLOCATION && (_MSC_VER>=1200) && !defined(U_STATIC_IMPLEMENTATION) && (defined(U_COMMON_IMPLEMENTATION) || defined(U_I18N_IMPLEMENTATION) || defined(U_IO_IMPLEMENTATION) || defined(U_LAYOUT_IMPLEMENTATION) || defined(U_LAYOUTEX_IMPLEMENTATION))
+#if defined(__cplusplus) && U_DEBUG && U_OVERRIDE_CXX_ALLOCATION && defined (_MSC_VER) && (_MSC_VER>=1200) && !defined(U_STATIC_IMPLEMENTATION) && (defined(U_COMMON_IMPLEMENTATION) || defined(U_I18N_IMPLEMENTATION) || defined(U_IO_IMPLEMENTATION) || defined(U_LAYOUT_IMPLEMENTATION) || defined(U_LAYOUTEX_IMPLEMENTATION))
 
 #ifndef U_HIDE_INTERNAL_API
 /**
