--- misc/jpeg-9a/jmorecfg.h	2009-11-20 09:01:24.000000000 +0100
+++ misc/build/jpeg-9a/jmorecfg.h	2011-03-24 21:29:02.073761750 +0100
@@ -210,6 +210,7 @@
 /* INT32 must hold at least signed 32-bit values. */
 
 #ifndef XMD_H			/* X11/xmd.h correctly defines INT32 */
+#ifndef INCLUDED_TOOLS_SOLAR_H
 #ifndef _BASETSD_H_		/* Microsoft defines it in basetsd.h */
 #ifndef _BASETSD_H		/* MinGW is slightly different */
 #ifndef QGLOBAL_H		/* Qt defines it in qglobal.h */
@@ -217,6 +218,7 @@
 #endif
 #endif
 #endif
+#endif
 #endif
 
 /* Datatype used for image dimensions.  The JPEG standard only supports
