diff -ur lang/cpp/src/context.cpp.old lang/cpp/src/context.cpp
--- lang/cpp/src/context.cpp.old	2019-04-20 16:03:09.575605300 +0300
+++ lang/cpp/src/context.cpp	2019-04-20 16:06:43.078419700 +0300
@@ -1656,6 +1656,11 @@
     return Error(gpgme_engine_check_version(p));
 }
 
+int GpgME::setGlobalFlag(const char *name, const char *value)
+{
+    return gpgme_set_global_flag(name, value);
+}
+
 static const unsigned long supported_features = 0
         | GpgME::ValidatingKeylistModeFeature
         | GpgME::CancelOperationFeature
diff -ur lang/cpp/src/global.h lang/cpp/src/global.h
--- lang/cpp/src/global.h.old	2016-09-07 09:48:44.000000000 +0300
+++ lang/cpp/src/global.h	2019-04-20 16:06:41.918912600 +0300
@@ -95,6 +95,8 @@
 GPGMEPP_EXPORT Error checkEngine(Protocol proto);
 GPGMEPP_EXPORT Error checkEngine(Engine engine);
 
+GPGMEPP_EXPORT int setGlobalFlag(const char *name, const char *value);
+
 GPGMEPP_EXPORT GIOChannel *getGIOChannel(int fd);
 GPGMEPP_EXPORT QIODevice   *getQIODevice(int fd);
 
