diff --git a/configure.ac b/configure.ac
index f28480b..92a3e85 100644
--- configure.ac
+++ configure.ac
@@ -719,8 +719,18 @@
 
 # Checking for libgpg-error.
 have_gpg_error=no
-AM_PATH_GPG_ERROR("$NEED_GPG_ERROR_VERSION",
-                  have_gpg_error=yes, have_gpg_error=no)
+if test "x${GPG_ERROR_CFLAGS}" = x; then
+  AM_PATH_GPG_ERROR("$NEED_GPG_ERROR_VERSION",
+                have_gpg_error=yes, have_gpg_error=no)
+else
+  have_gpg_error=yes
+  GPG_ERROR_CFLAGS="$GPG_ERROR_CFLAGS"
+  GPG_ERROR_LIBS="$GPG_ERROR_LIBS"
+  AC_SUBST(GPG_ERROR_CFLAGS)
+  AC_SUBST(GPG_ERROR_LIBS)
+
+fi
+
 AC_DEFINE(GPG_ERR_SOURCE_DEFAULT, GPG_ERR_SOURCE_GPGME,
           [The default error source for GPGME.])
 
@@ -729,8 +739,17 @@
 
 # And for libassuan.
 have_libassuan=no
-AM_PATH_LIBASSUAN("$NEED_LIBASSUAN_API:$NEED_LIBASSUAN_VERSION",
+if test "x${LIBASSUAN_CFLAGS}" = x; then
+  AM_PATH_LIBASSUAN("$NEED_LIBASSUAN_API:$NEED_LIBASSUAN_VERSION",
                   have_libassuan=yes, have_libassuan=no)
+else
+  have_libassuan=yes
+  LIBASSUAN_CFLAGS="$LIBASSUAN_CFLAGS"
+  LIBASSUAN_LIBS="$LIBASSUAN_LIBS"
+  AC_SUBST(LIBASSUAN_CFLAGS)
+  AC_SUBST(LIBASSUAN_LIBS)
+fi
+
 if test "$have_libassuan" = "yes"; then
   AC_DEFINE_UNQUOTED(GPGME_LIBASSUAN_VERSION, "$libassuan_version",
                      [version of the libassuan library])
--- lang/cpp/tests/Makefile.am	2020-06-20 00:43:49.213657887 +0200
+++ lang/cpp/tests/Makefile.am~	2019-01-25 13:27:34.000000000 +0100
@@ -21,7 +21,7 @@
 AM_LDFLAGS = -no-install
 
 LDADD = ../../cpp/src/libgpgmepp.la \
+        ../../../src/libgpgme.la @LIBASSUAN_LIBS@ @GPG_ERROR_LIBS@ \
-        ../../../src/libgpgme.la @GPG_ERROR_LIBS@ \
         @LDADD_FOR_TESTS_KLUDGE@ -lstdc++
 
 AM_CPPFLAGS = -I$(top_srcdir)/lang/cpp/src -I$(top_builddir)/src \
diff -ur gpgmepp.org/src/Makefile.am gpgmepp/src/Makefile.am
--- src/Makefile.am	2024-07-02 13:24:02.549609519 +0200
+++ src/Makefile.am	2024-07-02 13:24:29.573606613 +0200
@@ -111,7 +111,7 @@
 gpgme_tool_LDADD = libgpgme.la @LIBASSUAN_LIBS@ @GPG_ERROR_LIBS@
 
 gpgme_json_SOURCES = gpgme-json.c cJSON.c cJSON.h
-gpgme_json_LDADD = -lm libgpgme.la $(GPG_ERROR_LIBS)
+gpgme_json_LDADD = -lm libgpgme.la @LIBASSUAN_LIBS@ $(GPG_ERROR_LIBS)
 
 
 if HAVE_W32_SYSTEM
