--- src/assuan-support.c
+++ src/assuan-support.c
@@ -126,7 +126,7 @@
 }
 
 
-static gpgme_ssize_t
+static ssize_t
 my_read (assuan_context_t ctx, assuan_fd_t fd, void *buffer, size_t size)
 {
   (void)ctx;
@@ -134,7 +134,7 @@
 }
 
 
-static gpgme_ssize_t
+static ssize_t
 my_write (assuan_context_t ctx, assuan_fd_t fd, const void *buffer, size_t size)
 {
   (void)ctx;
--- src/gpgme-w32spawn.c
+++ src/gpgme-w32spawn.c
@@ -24,7 +249,7 @@
           handle = LoadLibraryA ("user32.dll");
           if (handle)
             {
-              func = GetProcAddress (handle, "AllowSetForegroundWindow");
+              func = (BOOL (WINAPI *)(DWORD)) GetProcAddress (handle, "AllowSetForegroundWindow");
               if (!func)
                 FreeLibrary (handle);
             }
