diff -ru nss.orig/nss/coreconf/WIN32.mk nss/nss/coreconf/WIN32.mk
--- foo/nss.orig/nss/coreconf/WIN32.mk	2015-06-17 20:26:43.000000000 +0200
+++ foo/nss/nss/coreconf/WIN32.mk	2015-08-31 20:40:40.146490300 +0200
@@ -192,7 +192,7 @@
 ifneq ($(_MSC_VER),$(_MSC_VER_6))
     # Convert certain deadly warnings to errors (see list at end of file)
     OS_CFLAGS += -we4002 -we4003 -we4004 -we4006 -we4009 -we4013 \
-     -we4015 -we4028 -we4033 -we4035 -we4045 -we4047 -we4053 -we4054 -we4063 \
+     -we4015 -we4028 -we4033 -we4035 -we4045 -we4047 -we4053 -we4063 \
      -we4064 -we4078 -we4087 -we4090 -we4098 -we4390 -we4551 -we4553 -we4715
 
     ifeq ($(_MSC_VER_GE_12),1)
