--- misc/cppunit-1.13.1/src/cppunit/cppunit_dll.vcxproj
+++ misc/build/cppunit-1.13.1/src/cppunit/cppunit_dll.vcxproj
@@ -107,7 +107,6 @@
       <SuppressStartupBanner>true</SuppressStartupBanner>
       <WarningLevel>Level3</WarningLevel>
       <RuntimeTypeInfo>true</RuntimeTypeInfo>
-      <DebugInformationFormat>OldStyle</DebugInformationFormat>
       <AdditionalIncludeDirectories>..\..\include;%(AdditionalIncludeDirectories)</AdditionalIncludeDirectories>
       <PreprocessorDefinitions>WIN32;NDEBUG;_WINDOWS;_USRDLL;CPPUNIT_BUILD_DLL;%(PreprocessorDefinitions)</PreprocessorDefinitions>
       <AssemblerListingLocation>.\ReleaseDll\</AssemblerListingLocation>
@@ -138,6 +137,7 @@
     <Link>
       <SuppressStartupBanner>true</SuppressStartupBanner>
       <LinkDLL>true</LinkDLL>
+      <GenerateDebugInformation>true</GenerateDebugInformation>
       <SubSystem>Console</SubSystem>
       <OutputFile>.\ReleaseDll\cppunit_dll.dll</OutputFile>
       <ImportLibrary>.\ReleaseDll\cppunit_dll.lib</ImportLibrary>
