--- misc/cppunit-1.13.1/src/cppunit/UnixDynamicLibraryManager.cpp
+++ misc/build/cppunit-1.13.1/src/cppunit/UnixDynamicLibraryManager.cpp
@@ -20,7 +20,6 @@
 void 
 DynamicLibraryManager::doReleaseLibrary()
 {
-  ::dlclose( m_libraryHandle);
 }
 
 
@@ -34,7 +33,11 @@
 std::string 
 DynamicLibraryManager::getLastErrorDetail() const
 {
-  return "";
+  const char *last_dlerror = ::dlerror();
+  if (last_dlerror != NULL)
+    return last_dlerror;
+  else
+    return "";
 }
 
 
