--- src/lib/Makefile.in
+++ src/lib/Makefile.in
@@ -79,7 +79,7 @@
 POST_UNINSTALL = :
 build_triplet = @build@
 host_triplet = @host@
-@HAVE_VISIBILITY_TRUE@am__append_1 = -fvisibility=hidden
+@HAVE_VISIBILITY_TRUE@am__append_1 = -fvisibility-ms-compat
 @HAVE_VISIBILITY_TRUE@am__append_2 = -DLIBREVENGE_VISIBILITY
 @BUILD_GENERATORS_TRUE@am__append_3 = librevenge-generators-@RVNG_MAJOR_VERSION@.@RVNG_MINOR_VERSION@.la
 @BUILD_GENERATORS_TRUE@@HAVE_VISIBILITY_TRUE@am__append_4 = -DLIBREVENGE_GENERATORS_VISIBILITY
