--- misc/glib-2.28.1/Makefile.in	2011-02-19 00:22:03.000000000 +0900
+++ misc/build/glib-2.28.1/Makefile.in	2011-04-06 20:49:38.648000000 +0900
@@ -422,7 +422,7 @@
 	gio-2.0-uninstalled.pc.in gio-unix-2.0-uninstalled.pc.in
 TEST_PROGS = 
 ACLOCAL_AMFLAGS = -I m4macros ${ACLOCAL_FLAGS}
-SUBDIRS = . m4macros glib gmodule gthread gobject gio tests po docs
+SUBDIRS = . m4macros glib gmodule gthread gobject gio
 DIST_SUBDIRS = $(SUBDIRS) build
 bin_SCRIPTS = glib-gettextize
 AM_CPPFLAGS = \
--- misc/glib-2.28.1/gobject/Makefile.in	2011-02-19 00:22:02.000000000 +0900
+++ misc/build/glib-2.28.1/gobject/Makefile.in	2011-04-06 06:37:37.274125000 +0900
@@ -723,6 +723,7 @@
 glib-genmarshal$(EXEEXT): $(glib_genmarshal_OBJECTS) $(glib_genmarshal_DEPENDENCIES) 
 	@rm -f glib-genmarshal$(EXEEXT)
 	$(AM_V_CCLD)$(LINK) $(glib_genmarshal_OBJECTS) $(glib_genmarshal_LDADD) $(LIBS)
+	@cp /dev/null @am__quote@./$(DEPDIR)/glib-genmarshal.Po@am__quote@
 gobject-query$(EXEEXT): $(gobject_query_OBJECTS) $(gobject_query_DEPENDENCIES) 
 	@rm -f gobject-query$(EXEEXT)
 	$(AM_V_CCLD)$(LINK) $(gobject_query_OBJECTS) $(gobject_query_LDADD) $(LIBS)
--- misc/glib-2.28.1/ltmain.sh	2011-02-08 21:25:07.000000000 +0900
+++ misc/build/glib-2.28.1/ltmain.sh	2011-04-05 20:46:38.586000000 +0900
@@ -6485,7 +6485,7 @@
 	case $ladir in
 	[\\/]* | [A-Za-z]:[\\/]*) abs_ladir="$ladir" ;;
 	*)
-	  abs_ladir=`cd "$ladir" && pwd`
+	  abs_ladir=`cd "$ladir" && cygpath -am .`
 	  if test -z "$abs_ladir"; then
 	    func_warning "cannot determine absolute directory name of \`$ladir'"
 	    func_warning "passing it literally to the linker, although it might fail"
--- misc/glib-2.28.1/configure	2011-02-19 00:22:04.000000000 +0900
+++ misc/build/glib-2.28.1/configure	2011-04-03 22:22:36.941750000 +0900
@@ -1443,6 +1443,7 @@
     cross_compiling=yes
   fi
 fi
+cross_compiling=no
 
 ac_tool_prefix=
 test -n "$host_alias" && ac_tool_prefix=$host_alias-
@@ -7931,13 +7932,13 @@
 done
 
 
-{ $as_echo "$as_me:${as_lineno-$LINENO}: checking for inflate in -lz" >&5
-$as_echo_n "checking for inflate in -lz... " >&6; }
+{ $as_echo "$as_me:${as_lineno-$LINENO}: checking for inflate in $ZLIB3RDLIB" >&5
+$as_echo_n "checking for inflate in $ZLIB3RDLIB... " >&6; }
 if ${ac_cv_lib_z_inflate+:} false; then :
   $as_echo_n "(cached) " >&6
 else
   ac_check_lib_save_LIBS=$LIBS
-LIBS="-lz  $LIBS"
+LIBS="$ZLIB3RDLIB  $LIBS"
 cat confdefs.h - <<_ACEOF >conftest.$ac_ext
 /* end confdefs.h.  */
 
@@ -7979,7 +7980,7 @@
   if test "x$found_zlib" = "xno" ; then
     as_fn_error $? "*** Working zlib library and headers not found ***" "$LINENO" 5
   fi
-  ZLIB_LIBS='-lz'
+  ZLIB_LIBS="$ZLIB3RDLIB"
 
 fi
 
@@ -8389,6 +8390,7 @@
 
 $as_echo "#define HAVE_GETTEXT 1" >>confdefs.h
 
+if test ! 1; then
 	# Extract the first word of "msgfmt", so it can be a program name with args.
 set dummy msgfmt; ac_word=$2
 { $as_echo "$as_me:${as_lineno-$LINENO}: checking for $ac_word" >&5
@@ -8605,8 +8607,10 @@
 $as_echo "found xgettext program is not GNU xgettext; ignore it" >&6; }
         XGETTEXT=":"
       fi
+fi
     fi
 
+if test ! 1; then
     # We need to process the po/ directory.
     POSUB=po
 
@@ -8690,6 +8694,7 @@
    rm -f po/POTFILES
    sed -e "/^#/d" -e "/^\$/d" -e "s,.*,	$posrcprefix& \\\\," -e "\$s/\(.*\) \\\\/\1/" \
 	< $srcdir/po/POTFILES.in > po/POTFILES
+fi
 
 
 if test "$gt_cv_have_gettext" != "yes" ; then
@@ -9797,7 +9802,7 @@
   # unless we find 'file', for example because we are cross-compiling.
   # func_win32_libid assumes BSD nm, so disallow it if using MS dumpbin.
   if ( test "$lt_cv_nm_interface" = "BSD nm" && file / ) >/dev/null 2>&1; then
-    lt_cv_deplibs_check_method='file_magic ^x86 archive import|^x86 DLL'
+    lt_cv_deplibs_check_method='file_magic ^x86 archive|^x86 DLL'
     lt_cv_file_magic_cmd='func_win32_libid'
   else
     # Keep this pattern in sync with the one in func_win32_libid.
