diff --git a/officecfg/registry/data/org/openoffice/Office/Math.xcu b/officecfg/registry/data/org/openoffice/Office/Math.xcu
index 96e6653..66c520d 100644
--- a/officecfg/registry/data/org/openoffice/Office/Math.xcu
+++ b/officecfg/registry/data/org/openoffice/Office/Math.xcu
@@ -76,7 +76,7 @@
         <value>Id2</value>
       </prop>
     </node>
-    <node oor:name="ALPHA" oor:op="replace">
+    <node oor:name="Alpha" oor:op="replace">
       <prop oor:name="Char">
         <value>913</value>
       </prop>
@@ -104,7 +104,7 @@
         <value>Id2</value>
       </prop>
     </node>
-    <node oor:name="BETA" oor:op="replace">
+    <node oor:name="Beta" oor:op="replace">
       <prop oor:name="Char">
         <value>914</value>
       </prop>
@@ -132,7 +132,7 @@
         <value>Id2</value>
       </prop>
     </node>
-    <node oor:name="GAMMA" oor:op="replace">
+    <node oor:name="Gamma" oor:op="replace">
       <prop oor:name="Char">
         <value>915</value>
       </prop>
@@ -160,7 +160,7 @@
         <value>Id2</value>
       </prop>
     </node>
-    <node oor:name="DELTA" oor:op="replace">
+    <node oor:name="Delta" oor:op="replace">
       <prop oor:name="Char">
         <value>916</value>
       </prop>
@@ -188,7 +188,7 @@
         <value>Id2</value>
       </prop>
     </node>
-    <node oor:name="EPSILON" oor:op="replace">
+    <node oor:name="Epsilon" oor:op="replace">
       <prop oor:name="Char">
         <value>917</value>
       </prop>
@@ -216,7 +216,7 @@
         <value>Id2</value>
       </prop>
     </node>
-    <node oor:name="ZETA" oor:op="replace">
+    <node oor:name="Zeta" oor:op="replace">
       <prop oor:name="Char">
         <value>918</value>
       </prop>
@@ -244,7 +244,7 @@
         <value>Id2</value>
       </prop>
     </node>
-    <node oor:name="ETA" oor:op="replace">
+    <node oor:name="Eta" oor:op="replace">
       <prop oor:name="Char">
         <value>919</value>
       </prop>
@@ -272,7 +272,7 @@
         <value>Id2</value>
       </prop>
     </node>
-    <node oor:name="THETA" oor:op="replace">
+    <node oor:name="Theta" oor:op="replace">
       <prop oor:name="Char">
         <value>920</value>
       </prop>
@@ -300,7 +300,7 @@
         <value>Id2</value>
       </prop>
     </node>
-    <node oor:name="IOTA" oor:op="replace">
+    <node oor:name="Iota" oor:op="replace">
       <prop oor:name="Char">
         <value>921</value>
       </prop>
@@ -328,7 +328,7 @@
         <value>Id2</value>
       </prop>
     </node>
-    <node oor:name="KAPPA" oor:op="replace">
+    <node oor:name="Kappa" oor:op="replace">
       <prop oor:name="Char">
         <value>922</value>
       </prop>
@@ -356,7 +356,7 @@
         <value>Id2</value>
       </prop>
     </node>
-    <node oor:name="LAMBDA" oor:op="replace">
+    <node oor:name="Lambda" oor:op="replace">
       <prop oor:name="Char">
         <value>923</value>
       </prop>
@@ -384,7 +384,7 @@
         <value>Id2</value>
       </prop>
     </node>
-    <node oor:name="MU" oor:op="replace">
+    <node oor:name="Mu" oor:op="replace">
       <prop oor:name="Char">
         <value>924</value>
       </prop>
@@ -412,7 +412,7 @@
         <value>Id2</value>
       </prop>
     </node>
-    <node oor:name="NU" oor:op="replace">
+    <node oor:name="Nu" oor:op="replace">
       <prop oor:name="Char">
         <value>925</value>
       </prop>
@@ -440,7 +440,7 @@
         <value>Id2</value>
       </prop>
     </node>
-    <node oor:name="XI" oor:op="replace">
+    <node oor:name="Xi" oor:op="replace">
       <prop oor:name="Char">
         <value>926</value>
       </prop>
@@ -468,7 +468,7 @@
         <value>Id2</value>
       </prop>
     </node>
-    <node oor:name="OMICRON" oor:op="replace">
+    <node oor:name="Omicron" oor:op="replace">
       <prop oor:name="Char">
         <value>927</value>
       </prop>
@@ -496,7 +496,7 @@
         <value>Id2</value>
       </prop>
     </node>
-    <node oor:name="PI" oor:op="replace">
+    <node oor:name="Pi" oor:op="replace">
       <prop oor:name="Char">
         <value>928</value>
       </prop>
@@ -524,7 +524,7 @@
         <value>Id2</value>
       </prop>
     </node>
-    <node oor:name="RHO" oor:op="replace">
+    <node oor:name="Rho" oor:op="replace">
       <prop oor:name="Char">
         <value>929</value>
       </prop>
@@ -552,7 +552,7 @@
         <value>Id2</value>
       </prop>
     </node>
-    <node oor:name="SIGMA" oor:op="replace">
+    <node oor:name="Sigma" oor:op="replace">
       <prop oor:name="Char">
         <value>931</value>
       </prop>
@@ -580,7 +580,7 @@
         <value>Id2</value>
       </prop>
     </node>
-    <node oor:name="TAU" oor:op="replace">
+    <node oor:name="Tau" oor:op="replace">
       <prop oor:name="Char">
         <value>932</value>
       </prop>
@@ -608,7 +608,7 @@
         <value>Id2</value>
       </prop>
     </node>
-    <node oor:name="UPSILON" oor:op="replace">
+    <node oor:name="Upsilon" oor:op="replace">
       <prop oor:name="Char">
         <value>933</value>
       </prop>
@@ -636,7 +636,7 @@
         <value>Id2</value>
       </prop>
     </node>
-    <node oor:name="PHI" oor:op="replace">
+    <node oor:name="Phi" oor:op="replace">
       <prop oor:name="Char">
         <value>934</value>
       </prop>
@@ -664,7 +664,7 @@
         <value>Id2</value>
       </prop>
     </node>
-    <node oor:name="CHI" oor:op="replace">
+    <node oor:name="Chi" oor:op="replace">
       <prop oor:name="Char">
         <value>935</value>
       </prop>
@@ -692,7 +692,7 @@
         <value>Id2</value>
       </prop>
     </node>
-    <node oor:name="PSI" oor:op="replace">
+    <node oor:name="Psi" oor:op="replace">
       <prop oor:name="Char">
         <value>936</value>
       </prop>
@@ -720,7 +720,7 @@
         <value>Id2</value>
       </prop>
     </node>
-    <node oor:name="OMEGA" oor:op="replace">
+    <node oor:name="Omega" oor:op="replace">
       <prop oor:name="Char">
         <value>937</value>
       </prop>
diff --git a/starmath/qa/cppunit/test_nodetotextvisitors.cxx b/starmath/qa/cppunit/test_nodetotextvisitors.cxx
index 62b37fc..801e1c9 100644
--- a/starmath/qa/cppunit/test_nodetotextvisitors.cxx
+++ b/starmath/qa/cppunit/test_nodetotextvisitors.cxx
@@ -346,29 +346,29 @@ void Test::SimpleFormats()
 
 void Test::SimpleGreekChars()
 {
-    parseandparseagain("%ALPHA", "Capital alpha");
-    parseandparseagain("%BETA", "Capital beta");
-    parseandparseagain("%CHI", "Capital chi");
-    parseandparseagain("%DELTA", "Capital delta");
-    parseandparseagain("%EPSILON", "Capital epsilon");
-    parseandparseagain("%ETA", "Capital eta");
-    parseandparseagain("%GAMMA", "Capital gamma");
-    parseandparseagain("%IOTA", "Capital iota");
-    parseandparseagain("%LAMBDA", "Capital lambda");
-    parseandparseagain("%MU", "Capital mu");
-    parseandparseagain("%NU", "Capital nu");
-    parseandparseagain("%OMEGA", "Capital omega");
-    parseandparseagain("%OMICRON", "Capital omicron");
-    parseandparseagain("%PHI", "Capital phi");
-    parseandparseagain("%PI", "Capital pi");
-    parseandparseagain("%PSI", "Capital psi");
-    parseandparseagain("%RHO", "Capital rho");
-    parseandparseagain("%SIGMA", "Capital sigma");
-    parseandparseagain("%TAU", "Capital tau");
-    parseandparseagain("%THETA", "Capital theta");
-    parseandparseagain("%UPSILON", "Capital upsilon");
-    parseandparseagain("%XI", "Capital xi");
-    parseandparseagain("%ZETA", "Capital zeta");
+    parseandparseagain("%Alpha", "Capital a");
+    parseandparseagain("%Beta", "Capital b");
+    parseandparseagain("%Chi", "Capital c");
+    parseandparseagain("%Delta", "Capital d");
+    parseandparseagain("%Epsilon", "Capital e");
+    parseandparseagain("%Eta", "Capital e");
+    parseandparseagain("%Gamma", "Capital g");
+    parseandparseagain("%Iota", "Capital i");
+    parseandparseagain("%Lambda", "Capital l");
+    parseandparseagain("%Mu", "Capital m");
+    parseandparseagain("%NU", "Capital n");
+    parseandparseagain("%Omega", "Capital o");
+    parseandparseagain("%Omicron", "Capital o");
+    parseandparseagain("%Phi", "Capital p");
+    parseandparseagain("%Pi", "Capital p");
+    parseandparseagain("%Psi", "Capital p");
+    parseandparseagain("%Rho", "Capital r");
+    parseandparseagain("%Sigma", "Capital s");
+    parseandparseagain("%Tau", "Capital t");
+    parseandparseagain("%Theta", "Capital t");
+    parseandparseagain("%Upsilon", "Capital u");
+    parseandparseagain("%Xi", "Capital x");
+    parseandparseagain("%Zeta", "Capital z");
     parseandparseagain("%alpha", "lowercase alpha");
     parseandparseagain("%beta", "lowercase beta");
     parseandparseagain("%chi", "lowercase chi");
