--- misc/build/libwpd-0.9.2/src/lib/libwpd_internal.cpp
+++ misc/build/libwpd-0.9.2/src/lib/libwpd_internal.cpp
@@ -1229,7 +1229,11 @@
 		tempString.sprintf("0.0000");
 	else
 		tempString.sprintf("%.4f", value);
+#ifndef __ANDROID__
 	std::string decimalPoint(localeconv()->decimal_point);
+#else
+	std::string decimalPoint(".");
+#endif
 	if ((decimalPoint.size() == 0) || (decimalPoint == "."))
 		return tempString;
 	std::string stringValue(tempString.cstr());
