--- foo/UnpackedTarball/wpd/src/lib/WP6BoxGroup.cpp	2012-09-12 23:00:03.552693060 +0100
+++ foo/UnpackedTarball/wpd/src/lib/WP6BoxGroup.cpp	2012-09-12 23:00:41.293089947 +0100
@@ -188,7 +188,7 @@
 		{
 			long tmpEndOfData = readU16(input, encryption) + input->tell();
 #ifdef DEBUG
-			uint16_t tmpOverrideFlags = readU16(input, encryption);
+			tmpOverrideFlags = readU16(input, encryption);
 #else
 			readU16(input, encryption);
 #endif
@@ -199,7 +199,7 @@
 		{
 			long tmpEndOfData = readU16(input, encryption) + input->tell();
 #ifdef DEBUG
-			uint16_t tmpOverrideFlags = readU16(input, encryption);
+			tmpOverrideFlags = readU16(input, encryption);
 #else
 			readU16(input, encryption);
 #endif
@@ -210,7 +210,7 @@
 		{
 			long tmpEndOfData = readU16(input, encryption) + input->tell();
 #ifdef DEBUG
-			uint16_t tmpOverrideFlags = readU16(input, encryption);
+			tmpOverrideFlags = readU16(input, encryption);
 #else
 			readU16(input, encryption);
 #endif
@@ -221,7 +221,7 @@
 		{
 			long tmpEndOfData = readU16(input, encryption) + input->tell();
 #ifdef DEBUG
-			uint16_t tmpOverrideFlags = readU16(input, encryption);
+			tmpOverrideFlags = readU16(input, encryption);
 #else
 			readU16(input, encryption);
 #endif
@@ -232,7 +232,7 @@
 		{
 			long tmpEndOfData = readU16(input, encryption) + input->tell();
 #ifdef DEBUG
-			uint16_t tmpOverrideFlags = readU16(input, encryption);
+			tmpOverrideFlags = readU16(input, encryption);
 #else
 			readU16(input, encryption);
 #endif
@@ -247,7 +247,7 @@
 		{
 			long tmpEndOfData = readU16(input, encryption) + input->tell();
 #ifdef DEBUG
-			uint16_t tmpOverrideFlags = readU16(input, encryption);
+			tmpOverrideFlags = readU16(input, encryption);
 #else
 			readU16(input, encryption);
 #endif
@@ -258,7 +258,7 @@
 		{
 			long tmpEndOfData = readU16(input, encryption) + input->tell();
 #ifdef DEBUG
-			uint16_t tmpOverrideFlags = readU16(input, encryption);
+			tmpOverrideFlags = readU16(input, encryption);
 #else
 			readU16(input, encryption);
 #endif
