--- mariadb/include/global.h	2013-03-06 17:06:42.545906061 +0100
+++ mariadb/include/global.h	2013-03-07 14:45:03.811128674 +0100
@@ -247,7 +247,9 @@
 #endif
 
 #define sgn(a)		(((a) < 0) ? -1 : ((a) > 0) ? 1 : 0)
+#ifndef __cplusplus
 #define swap(t,a,b)	{ register t dummy; dummy = a; a = b; b = dummy; }
+#endif
 #define test(a)		((a) ? 1 : 0)
 #define set_if_bigger(a,b)  { if ((a) < (b)) (a)=(b); }
 #define set_if_smaller(a,b) { if ((a) > (b)) (a)=(b); }
--- mariadb/include/my_global.h	2013-03-06 17:06:42.546906037 +0100
+++ mariadb/include/my_global.h	2013-03-07 14:44:47.032529753 +0100
@@ -278,7 +278,9 @@
 #endif
 
 #define sgn(a)		(((a) < 0) ? -1 : ((a) > 0) ? 1 : 0)
+#ifndef __cplusplus
 #define swap(t,a,b)	{ register t dummy; dummy = a; a = b; b = dummy; }
+#endif
 #define test(a)		((a) ? 1 : 0)
 #define set_if_bigger(a,b)  { if ((a) < (b)) (a)=(b); }
 #define set_if_smaller(a,b) { if ((a) > (b)) (a)=(b); }
