--- misc/mozilla/mailnews/extensions/smime/build/Makefile.in	2010-09-04 16:00:38.000000000 +0100
+++ misc/build/mozilla/mailnews/extensions/smime/build/Makefile.in	2010-09-04 16:02:24.000000000 +0100
@@ -79,10 +79,6 @@
 		$(DIST)/lib/$(LIB_PREFIX)msgsmime_s.$(LIB_SUFFIX) \
 		$(NULL)
 
-ifndef MOZ_STATIC_MAIL_BUILD
-SHARED_LIBRARY_LIBS + = $(DIST)/lib/$(LIB_PREFIX)msgbsutl_s.$(LIB_SUFFIX)
-endif
-
 EXTRA_DSO_LDOPTS = \
 		$(LIBS_DIR) \
 		$(EXTRA_DSO_LIBS) \
