--- mozilla/mailnews/addrbook/src/nsAbLDAPDirectoryQuery.cpp	2009-02-02 09:39:32.054968600 +0100
+++ mozilla.patched/mailnews/addrbook/src/nsAbLDAPDirectoryQuery.cpp	2009-02-02 09:35:35.633832119 +0100
@@ -833,6 +833,10 @@
       if (msgListener)
       {
         msgListener->mUrl = url;
+        msgListener->mQueryListener = listener;
+        msgListener->mResultLimit = resultLimit;
+        msgListener->mTimeOut = timeOut;
+        msgListener->mQueryArguments = arguments;
         return msgListener->DoSearch();
       }
     }
