--- mozilla.org/directory/c-sdk/ldap/include/Makefile.in	2006-02-03 15:44:33.000000000 +0100
+++ mozilla/directory/c-sdk/ldap/include/Makefile.in	2008-10-31 23:34:41.695625000 +0100
@@ -85,6 +85,9 @@
 
 ###########################################################################
 
+INCLUDEDIR:=$(shell cygpath -u $(INCLUDEDIR))
+GENHEADERS:=$(shell cygpath -u $(GENHEADERS))
+
 all export::	$(INCLUDEDIR) $(GENHEADERS)
 	$(NSINSTALL) -D $(PRIVATEINCDIR)
 	$(INSTALL) $(INSTALLFLAGS) -m 644 $(HEADERS) $(INCLUDEDIR)
