--- misc/mozilla/modules/libpr0n/build/Makefile.in	2006-05-20 19:20:48.000000000 +0200
+++ misc/build/mozilla/modules/libpr0n/build/Makefile.in	2011-01-19 05:18:01.318000024 +0100
@@ -106,9 +106,9 @@
 export::
 	@{ \
 	$(foreach d,$(filter-out icon,$(MOZ_IMG_DECODERS)), \
-		echo "#define IMG_BUILD_DECODER_${d}";) \
+		echo "#define IMG_BUILD_DECODER_${d}" ; ) \
 	$(foreach d,$(MOZ_IMG_ENCODERS), \
-		echo "#define IMG_BUILD_ENCODER_${d}";) \
+		echo "#define IMG_BUILD_ENCODER_${d}" ; ) \
 	} > nsImgBuildDefines.tmp
 	@if `cmp -s nsImgBuildDefines.h nsImgBuildDefines.tmp`; then \
 		rm -f nsImgBuildDefines.tmp; \
