--- misc/mozilla/directory/c-sdk/build.mk	2011-09-27 14:45:02.000000000 +0200
+++ misc/build/mozilla/directory/c-sdk/build.mk	2011-09-27 14:44:18.000000000 +0200
@@ -157,8 +157,8 @@
 PLDSBASE=plds$(NSPR_LIBVERSION)
 NSPRBASE=nspr$(NSPR_LIBVERSION)
 
-DYNAMICNSPR = -l$(PLCBASE) -l$(PLDSBASE) -l$(NSPRBASE)
+DYNAMICNSPR = $(NSPR_LIBS)
 
 PLC_LIBNAME=plc$(NSPR_LIBVERSION)
 PLDS_LIBNAME=plds$(NSPR_LIBVERSION)
 ifeq ($(OS_ARCH), WINNT)
--- misc/mozilla/xpcom/build/Makefile.in	2008-06-16 00:26:11.000000000 +0200
+++ misc/build/mozilla/xpcom/build/Makefile.in	2011-09-28 10:41:02.000000000 +0200
@@ -135,7 +135,7 @@
 # pull in MoreFiles for MacOSX
 ifneq (,$(filter mac cocoa,$(MOZ_WIDGET_TOOLKIT)))
 EXTRA_DSO_LIBS = macmorefiles_s
-EXTRA_DSO_LDOPTS += $(EXTRA_DSO_LIBS)
+EXTRA_DSO_LDOPTS += -L$(DIST)/lib $(EXTRA_DSO_LIBS)
 EXTRA_DEPS += $(DIST)/lib/$(LIB_PREFIX)macmorefiles_s.$(LIB_SUFFIX)
 endif
 
