--- mozilla.pure/security/nss/cmd/shlibsign/Makefile	2007-02-16 03:16:24.000000000 +0100
+++ mozilla/security/nss/cmd/shlibsign/Makefile	2008-11-05 14:04:54.798875000 +0100
@@ -86,18 +86,47 @@
 
 include ../platrules.mk
 
-SRCDIR = $(call core_abspath,.)
-
+ifeq ($(OS_TARGET), Darwin)
+	SRCDIR = .
+else
+ifeq ($(OS_TARGET), Linux)
+	SRCDIR = .
+else
+ifeq ($(OS_TARGET), WIN95)
+	SRCDIR = $(shell cygpath -d $(SRCDIR))
+else
+	SRCDIR = $(call core_abspath,.)
+endif
+endif
+endif
 %.chk: %.$(DLL_SUFFIX) 
 ifeq ($(OS_TARGET), OS2)
 	cd $(OBJDIR) ; cmd.exe /c $(SRCDIR)/sign.cmd $(DIST) \
 	$(call core_abspath,$(OBJDIR)) $(OS_TARGET) \
 	$(call core_abspath,$(NSPR_LIB_DIR)) $(call core_abspath,$<)
 else
+ifeq ($(OS_TARGET), WIN95)
+	sh $(CYGWIN_WRAPPER) ./sign.sh $(shell cygpath -d -a $(DIST)) \
+	$(shell cygpath -d -a $(OBJDIR)) $(OS_TARGET) \
+	$(shell cygpath -d -a $(NSPR_LIB_DIR)) $(shell cygpath -d -a $<)
+else
+ifeq ($(OS_TARGET), Darwin)
+	cd $(SRCDIR) ; sh ./sign.sh $(DIST) \
+	$(OBJDIR) $(OS_TARGET) \
+	$(NSPR_LIB_DIR) $<
+else
+ifeq ($(OS_TARGET), Linux)
+	cd $(SRCDIR) ; sh ./sign.sh $(DIST) \
+	$(OBJDIR) $(OS_TARGET) \
+	$(NSPR_LIB_DIR) $<
+else
 	cd $(OBJDIR) ; sh $(SRCDIR)/sign.sh $(call core_abspath,$(DIST)) \
 	$(call core_abspath,$(OBJDIR)) $(OS_TARGET) \
 	$(call core_abspath,$(NSPR_LIB_DIR)) $(call core_abspath,$<)
 endif
+endif
+endif
+endif
 
 libs install :: $(CHECKLOC)
 
