--- misc/mozilla/security/coreconf/arch.mk	2011-10-06 19:58:31.808695724 +0200
+++ misc/build/mozilla/security/coreconf/arch.mk	2011-10-06 19:58:00.074648943 +0200
@@ -152,6 +152,10 @@
 
 ifeq ($(OS_ARCH),Linux)
     OS_RELEASE := $(subst ., ,$(OS_RELEASE))
+    # force Linux 3.x to 2.6
+    ifeq ($(word 1,$(OS_RELEASE)),3)
+	OS_RELEASE := 2 6
+    endif
     ifneq ($(words $(OS_RELEASE)),1)
 	OS_RELEASE := $(word 1,$(OS_RELEASE)).$(word 2,$(OS_RELEASE))
     endif
