--- misc/mozilla/xpcom/base/nscore.h	2006-08-24 23:46:32.000000000 +0200
+++ misc/build/mozilla/xpcom/base/nscore.h	2012-02-21 09:22:51.136057997 +0100
@@ -211,9 +211,14 @@
  *  when http://gcc.gnu.org/bugzilla/show_bug.cgi?id=11893 is fixed.
  */
 
+// At least "g++ (GCC) 4.6.2 20111027 (Red Hat 4.6.2-1)" with -std=c++0x no
+// longer recognizes "typeof" as a (GCC extension) keyword (an alternative fix
+// might be to check the GCC version, as the bug mentioned above is reportedly
+// fixed in GCC 4.4.3, see
+// <http://gcc.gnu.org/bugzilla/show_bug.cgi?id=9381#c15>):
 #ifdef __GNUC__
 #define NS_STDCALL_FUNCPROTO(ret, name, class, func, args) \
-  typeof(&class::func) name
+  __typeof__(&class::func) name
 #else
 #define NS_STDCALL_FUNCPROTO(ret, name, class, func, args) \
   ret (NS_STDCALL class::*name) args
