--- misc/mozilla/widget/src/mac/Makefile.in	2006-08-04 20:52:25.000000000 +0200
+++ misc/build/mozilla/widget/src/mac/Makefile.in	2012-06-04 23:26:18.000000000 +0200
@@ -130,7 +130,7 @@
 RES_SRC = $(srcdir)/nsMacWidget.r
 
 $(RES_DEST): $(RES_SRC)
-	/Developer/Tools/Rez -i /Developer/Headers/FlatCarbon -useDF $(RES_SRC) -o $(RES_DEST)
+	Rez -i $(MACOS_SDK_DIR)/Developer/Headers/FlatCarbon -useDF $(RES_SRC) -o $(RES_DEST)
 
 libs:: $(RES_DEST)
 
--- misc/mozilla/widget/src/mac/nsMacWindow.cpp	2007-11-19 21:40:08.000000000 +0100
+++ misc/build/mozilla/widget/src/mac/nsMacWindow.cpp	2012-06-04 22:26:29.000000000 +0200
@@ -1209,7 +1209,7 @@
           if (!sChecked) {
             sChecked = PR_TRUE;
 #if MAC_OS_X_VERSION_MIN_REQUIRED >= MAC_OS_X_VERSION_10_3
-            transitionFunc = ::TransitionWindowWithOptions;
+            transitionFunc = (TransitionWindowWithOptions_type) ::TransitionWindowWithOptions; // last TransitionWindowOptions* param is const* in 10.6
 #else
             CFBundleRef carbonBundle =
              ::CFBundleGetBundleWithIdentifier(CFSTR("com.apple.Carbon"));
--- misc/mozilla/xpfe/bootstrap/appleevents/Makefile.in	2006-01-05 17:41:23.000000000 +0100
+++ misc/build/mozilla/xpfe/bootstrap/appleevents/Makefile.in	2012-06-04 23:35:22.000000000 +0200
@@ -112,7 +112,7 @@
 
 # convert AppleScript dictionary .sdef file into .r format, and thence to .rsrc
 SDP_TOOL = $(SDP)
-REZ_TOOL = /Developer/Tools/Rez
+REZ_TOOL = Rez
 SDEF_SRC = $(srcdir)/mozilla.sdef
 RES_DEST = mozillaSuite.r
 RSRC_DEST = mozillaSuite.rsrc
