--- mozilla.pure/configure.in	2008-06-16 00:25:32.000000000 +0200
+++ mozilla/configure.in	2008-11-10 16:15:36.859375000 +0100
@@ -425,6 +425,7 @@
             _CC_SUITE=7
         elif test "$_CC_MAJOR_VERSION" = "14"; then
             _CC_SUITE=8
+            CXXFLAGS="$CXXFLAGS -Zc:wchar_t-"
         else
             AC_MSG_ERROR([This version of the MSVC compiler, $CC_VERSION , is unsupported.])
         fi
--- mozilla.pure/configure	2008-06-16 00:25:31.000000000 +0200
+++ mozilla/configure	2008-11-10 16:15:30.437500000 +0100
@@ -3022,6 +3022,7 @@
             _CC_SUITE=7
         elif test "$_CC_MAJOR_VERSION" = "14"; then
             _CC_SUITE=8
+            CXXFLAGS="$CXXFLAGS -Zc:wchar_t-"
         else
             { echo "configure: error: This version of the MSVC compiler, $CC_VERSION , is unsupported." 1>&2; exit 1; }
         fi
