diff -ur misc/build/libcmis-0.2.2.old/src/libcmis/atom-session.cxx misc/build/libcmis-0.2.2/src/libcmis/atom-session.cxx
--- misc/build/libcmis-0.2.2.old/src/libcmis/atom-session.cxx	2012-06-04 20:35:46.400203393 +0200
+++ misc/build/libcmis-0.2.2/src/libcmis/atom-session.cxx	2012-06-04 20:36:16.255205010 +0200
@@ -382,7 +382,7 @@
         if ( pos != string::npos )
         {
             // Escape the URL by chunks
-#if LIBCURL_VERSION_VALUE >= 71504
+#if LIBCURL_VERSION_VALUE >= 0x071504
             char* escaped = curl_easy_escape( m_curlHandle, value.c_str(), value.length() );
 #else
             char* escaped = curl_escape( value.c_str(), value.length() );
@@ -550,7 +550,7 @@
     {
         curl_easy_setopt( m_curlHandle, CURLOPT_HTTPAUTH, CURLAUTH_ANY );
 
-#if LIBCURL_VERSION_VALUE >= 71901
+#if LIBCURL_VERSION_VALUE >= 0x071901
         curl_easy_setopt( m_curlHandle, CURLOPT_USERNAME, m_username.c_str() );
         curl_easy_setopt( m_curlHandle, CURLOPT_PASSWORD, m_password.c_str() );
 #else
