--- a/configure.ac
+++ b/configure.ac
@@ -42,6 +42,11 @@ AS_IF([test x"$GCC" = xyes], [
 	CXXFLAGS="$CXXFLAGS -Wall -pedantic"
 ])
 
+AC_ARG_WITH([libxml2],
+	[AS_HELP_STRING([--with-libxml2=PREFIX], [look for the xml2 library in
+	 PREFIX/lib and headers in PREFIX/include])]
+)
+
 LIBCMIS_API_VERSION=libcmis_api_version
 AC_SUBST(LIBCMIS_API_VERSION)
 
@@ -62,7 +67,7 @@ AC_SUBST(CURL_CFLAGS)
 AC_SUBST(LIBCURL)
 
 # Check for lixml2
-AC_PATH_PROGS([TOOL_XML2_CONFIG], [xml2-config])
+AC_PATH_PROGS([TOOL_XML2_CONFIG], [xml2-config],, [$with_libxml2/bin:$PATH])
 AS_IF([test -n "$TOOL_XML2_CONFIG"],
 	[XML2_CFLAGS=`$TOOL_XML2_CONFIG --cflags`
 	 XML2_LIBS=`$TOOL_XML2_CONFIG --libs`],
--- bleh/config.sub	2011-09-27 22:48:43.000000000 +0200
+++ blah/config.sub	2011-10-04 09:26:51.000000000 +0200
@@ -4,7 +4,7 @@
 #   2000, 2001, 2002, 2003, 2004, 2005, 2006, 2007, 2008, 2009
 #   Free Software Foundation, Inc.
 
-timestamp='2009-11-20'
+timestamp='2011-10-04'
 
 # This file is (in principle) common to ALL GNU software.
 # The presence of a machine in this file suggests that SOME GNU software
@@ -123,7 +123,7 @@
 # Here we must recognize all the valid KERNEL-OS combinations.
 maybe_os=`echo $1 | sed 's/^\(.*\)-\([^-]*-[^-]*\)$/\2/'`
 case $maybe_os in
-  nto-qnx* | linux-gnu* | linux-dietlibc | linux-newlib* | linux-uclibc* | \
+  nto-qnx* | linux-gnu* | linux-android* | linux-dietlibc | linux-newlib* | linux-uclibc* | \
   uclinux-uclibc* | uclinux-gnu* | kfreebsd*-gnu* | knetbsd*-gnu* | netbsd*-gnu* | \
   kopensolaris*-gnu* | \
   storm-chaos* | os2-emx* | rtmk-nova*)
@@ -1294,7 +1294,7 @@
 	      | -udi* | -eabi* | -lites* | -ieee* | -go32* | -aux* \
 	      | -chorusos* | -chorusrdb* | -cegcc* \
 	      | -cygwin* | -pe* | -psos* | -moss* | -proelf* | -rtems* \
-	      | -mingw32* | -linux-gnu* | -linux-newlib* | -linux-uclibc* \
+	      | -mingw32* | -linux-gnu* | -linux-androideabi* | -linux-newlib* | -linux-uclibc* \
 	      | -uxpv* | -beos* | -mpeix* | -udk* \
 	      | -interix* | -uwin* | -mks* | -rhapsody* | -darwin* | -opened* \
 	      | -openstep* | -oskit* | -conix* | -pw32* | -nonstopux* \
