--- misc/rasqal-0.9.16/config.guess	2010-09-16 03:13:19.000000000 -0500
+++ misc/build/rasqal-0.9.16/config.guess	2010-09-16 03:15:05.000000000 -0500
@@ -532,7 +532,7 @@
 		echo rs6000-ibm-aix3.2
 	fi
 	exit ;;
-    *:AIX:*:[456])
+    *:AIX:*:[4567])
 	IBM_CPU_ID=`/usr/sbin/lsdev -C -c processor -S available | sed 1q | awk '{ print $1 }'`
 	if /usr/sbin/lsattr -El ${IBM_CPU_ID} | grep ' POWER' >/dev/null 2>&1; then
 		IBM_ARCH=rs6000
@@ -544,6 +544,9 @@
 	else
 		IBM_REV=${UNAME_VERSION}.${UNAME_RELEASE}
 	fi
+	if [ "$IBM_REV" == "V7BETA" ]; then
+		IBM_REV=${UNAME_VERSION}.${UNAME_RELEASE}
+	fi
 	echo ${IBM_ARCH}-ibm-aix${IBM_REV}
 	exit ;;
     *:AIX:*:*)
--- misc/rasqal-0.9.16/src/rasqal_literal.c	2010-09-09 14:53:30.000000000 -0500
+++ misc/build/rasqal-0.9.16/src/rasqal_literal.c	2010-09-09 14:54:47.000000000 -0500
@@ -36,8 +36,20 @@
 #include <stdlib.h>
 #endif
 #include <stdarg.h>
+#ifdef _AIX
+#  ifndef isnan
+#   define isnan(x) \
+      (sizeof (x) == sizeof (long double) ? isnan_ld (x) \
+       : sizeof (x) == sizeof (double) ? isnan_d (x) \
+       : isnan_f (x))
+  static inline int isnan_f  (float       x) { return x != x; }
+  static inline int isnan_d  (double      x) { return x != x; }
+  static inline int isnan_ld (long double x) { return x != x; }
+#  endif
+#else
 /* for isnan() */
 #include <math.h>
+#endif
 
 #ifdef RASQAL_REGEX_PCRE
 #include <pcre.h>
