--- misc/liblangtag-0.4.0/liblangtag/lt-messages.h	2012-11-06 23:10:33.191940625 +0100
+++ misc/build/liblangtag-0.4.0/liblangtag/lt-messages.h	2012-11-06 22:31:12.372708079 +0100
@@ -16,7 +16,10 @@
 #ifndef __LT_MESSAGES_H__
 #define __LT_MESSAGES_H__
 
+#ifdef HAVE_CONFIG_H
 #include "config.h"
+#endif
+
 #include <stdarg.h>
 #include "lt-macros.h"
 
--- misc/liblangtag-0.4.0/liblangtag/lt-utils.c	2012-11-06 23:10:33.195940539 +0100
+++ misc/build/liblangtag-0.4.0/liblangtag/lt-utils.c	2012-11-06 22:43:39.266557764 +0100
@@ -17,6 +17,7 @@
 #include <ctype.h>
 #include <stdio.h>
 #include <string.h>
+#include <memory.h>
 #include "lt-messages.h"
 #include "lt-utils.h"
 
--- misc/liblangtag-0.4.0/msvs2008/liblangtag.sln	2012-11-06 23:10:33.138941754 +0100
+++ misc/build/liblangtag-0.4.0/msvs2008/liblangtag.sln	2012-11-06 23:14:12.516267264 +0100
@@ -1 +1,20 @@
-dummy
+
+Microsoft Visual Studio Solution File, Format Version 10.00
+# Visual Studio 2008
+Project("{8BC9CEB8-8B4A-11D0-8D11-00A0C91BC942}") = "liblangtag", "liblangtag.vcproj", "{8991ACE9-2B73-4C10-BBC8-F16787132D98}"
+EndProject
+Global
+	GlobalSection(SolutionConfigurationPlatforms) = preSolution
+		Debug|Win32 = Debug|Win32
+		Release|Win32 = Release|Win32
+	EndGlobalSection
+	GlobalSection(ProjectConfigurationPlatforms) = postSolution
+		{8991ACE9-2B73-4C10-BBC8-F16787132D98}.Debug|Win32.ActiveCfg = Debug|Win32
+		{8991ACE9-2B73-4C10-BBC8-F16787132D98}.Debug|Win32.Build.0 = Debug|Win32
+		{8991ACE9-2B73-4C10-BBC8-F16787132D98}.Release|Win32.ActiveCfg = Release|Win32
+		{8991ACE9-2B73-4C10-BBC8-F16787132D98}.Release|Win32.Build.0 = Release|Win32
+	EndGlobalSection
+	GlobalSection(SolutionProperties) = preSolution
+		HideSolutionNode = FALSE
+	EndGlobalSection
+EndGlobal
--- misc/liblangtag-0.4.0/msvs2008/liblangtag.vcproj	2012-11-06 23:10:33.148941542 +0100
+++ misc/build/liblangtag-0.4.0/msvs2008/liblangtag.vcproj	2012-11-06 23:14:12.516267264 +0100
@@ -1 +1,452 @@
-dummy
+<?xml version="1.0" encoding="Windows-1252"?>
+<VisualStudioProject
+	ProjectType="Visual C++"
+	Version="9.00"
+	Name="liblangtag"
+	ProjectGUID="{8991ACE9-2B73-4C10-BBC8-F16787132D98}"
+	RootNamespace="liblangtag"
+	TargetFrameworkVersion="0"
+	>
+	<Platforms>
+		<Platform
+			Name="Win32"
+		/>
+	</Platforms>
+	<ToolFiles>
+	</ToolFiles>
+	<Configurations>
+		<Configuration
+			Name="Release|Win32"
+			OutputDirectory=".\Release"
+			IntermediateDirectory=".\Release"
+			ConfigurationType="4"
+			UseOfMFC="0"
+			ATLMinimizesCRunTimeLibraryUsage="false"
+			>
+			<Tool
+				Name="VCPreBuildEventTool"
+			/>
+			<Tool
+				Name="VCCustomBuildTool"
+			/>
+			<Tool
+				Name="VCXMLDataGeneratorTool"
+			/>
+			<Tool
+				Name="VCWebServiceProxyGeneratorTool"
+			/>
+			<Tool
+				Name="VCMIDLTool"
+			/>
+			<Tool
+				Name="VCCLCompilerTool"
+				Optimization="2"
+				InlineFunctionExpansion="1"
+				AdditionalIncludeDirectories="..;$(LIBXML_INCLUDE_DIR)"
+				PreprocessorDefinitions="NDEBUG;WIN32;_LIB;_CRT_SECURE_NO_WARNINGS;__LANGTAG_COMPILATION;REGDATADIR=\&quot;.\&quot;ALIGNOF_VOID_P=8"
+				StringPooling="true"
+				RuntimeLibrary="2"
+				EnableFunctionLevelLinking="true"
+				TreatWChar_tAsBuiltInType="false"
+				RuntimeTypeInfo="true"
+				PrecompiledHeaderFile=".\Release/liblangtag.pch"
+				AssemblerListingLocation=".\Release/"
+				ObjectFile=".\Release/"
+				ProgramDataBaseFileName=".\Release/"
+				WarningLevel="3"
+				SuppressStartupBanner="true"
+			/>
+			<Tool
+				Name="VCManagedResourceCompilerTool"
+			/>
+			<Tool
+				Name="VCResourceCompilerTool"
+			/>
+			<Tool
+				Name="VCPreLinkEventTool"
+			/>
+			<Tool
+				Name="VCLibrarianTool"
+				OutputFile=".\Release\lib\liblangtag.lib"
+				SuppressStartupBanner="true"
+			/>
+			<Tool
+				Name="VCALinkTool"
+			/>
+			<Tool
+				Name="VCXDCMakeTool"
+			/>
+			<Tool
+				Name="VCBscMakeTool"
+				SuppressStartupBanner="true"
+				OutputFile=".\Release/liblangtag.bsc"
+			/>
+			<Tool
+				Name="VCFxCopTool"
+			/>
+			<Tool
+				Name="VCPostBuildEventTool"
+			/>
+		</Configuration>
+		<Configuration
+			Name="Debug|Win32"
+			OutputDirectory=".\Debug"
+			IntermediateDirectory=".\Debug"
+			ConfigurationType="4"
+			UseOfMFC="0"
+			ATLMinimizesCRunTimeLibraryUsage="false"
+			>
+			<Tool
+				Name="VCPreBuildEventTool"
+			/>
+			<Tool
+				Name="VCCustomBuildTool"
+			/>
+			<Tool
+				Name="VCXMLDataGeneratorTool"
+			/>
+			<Tool
+				Name="VCWebServiceProxyGeneratorTool"
+			/>
+			<Tool
+				Name="VCMIDLTool"
+			/>
+			<Tool
+				Name="VCCLCompilerTool"
+				Optimization="0"
+				AdditionalIncludeDirectories="..;$(LIBXML_INCLUDE_DIR)"
+				PreprocessorDefinitions="DEBUG;WIN32;_LIB;_CRT_SECURE_NO_WARNINGS;__LANGTAG_COMPILATION;REGDATADIR=\&quot;.\&quot;ALIGNOF_VOID_P=8"
+				BasicRuntimeChecks="3"
+				RuntimeLibrary="3"
+				TreatWChar_tAsBuiltInType="false"
+				RuntimeTypeInfo="true"
+				PrecompiledHeaderFile=".\Debug/liblangtag.pch"
+				AssemblerListingLocation=".\Debug/"
+				ObjectFile=".\Debug/"
+				ProgramDataBaseFileName=".\Debug/"
+				WarningLevel="3"
+				SuppressStartupBanner="true"
+				DebugInformationFormat="4"
+			/>
+			<Tool
+				Name="VCManagedResourceCompilerTool"
+			/>
+			<Tool
+				Name="VCResourceCompilerTool"
+			/>
+			<Tool
+				Name="VCPreLinkEventTool"
+			/>
+			<Tool
+				Name="VCLibrarianTool"
+				OutputFile=".\Debug\lib\liblangtag.lib"
+				SuppressStartupBanner="true"
+			/>
+			<Tool
+				Name="VCALinkTool"
+			/>
+			<Tool
+				Name="VCXDCMakeTool"
+			/>
+			<Tool
+				Name="VCBscMakeTool"
+				SuppressStartupBanner="true"
+				OutputFile=".\Debug/liblangtag.bsc"
+			/>
+			<Tool
+				Name="VCFxCopTool"
+			/>
+			<Tool
+				Name="VCPostBuildEventTool"
+			/>
+		</Configuration>
+	</Configurations>
+	<References>
+	</References>
+	<Files>
+		<Filter
+			Name="Source Files"
+			Filter="cpp;c;cxx;rc;def;r;odl;idl;hpj;bat"
+			>
+			<File
+				RelativePath="..\liblangtag\lt-database.c"
+				>
+			</File>
+			<File
+				RelativePath="..\liblangtag\lt-error.c"
+				>
+			</File>
+			<File
+				RelativePath="..\liblangtag\lt-ext-module-data.c"
+				>
+			</File>
+			<File
+				RelativePath="..\liblangtag\lt-ext-module.c"
+				>
+			</File>
+			<File
+				RelativePath="..\liblangtag\lt-extension.c"
+				>
+			</File>
+			<File
+				RelativePath="..\liblangtag\lt-extlang-db.c"
+				>
+			</File>
+			<File
+				RelativePath="..\liblangtag\lt-extlang.c"
+				>
+			</File>
+			<File
+				RelativePath="..\liblangtag\lt-grandfathered-db.c"
+				>
+			</File>
+			<File
+				RelativePath="..\liblangtag\lt-grandfathered.c"
+				>
+			</File>
+			<File
+				RelativePath="..\liblangtag\lt-lang-db.c"
+				>
+			</File>
+			<File
+				RelativePath="..\liblangtag\lt-lang.c"
+				>
+			</File>
+			<File
+				RelativePath="..\liblangtag\lt-list.c"
+				>
+			</File>
+			<File
+				RelativePath="..\liblangtag\lt-mem.c"
+				>
+			</File>
+			<File
+				RelativePath="..\liblangtag\lt-messages.c"
+				>
+			</File>
+			<File
+				RelativePath="..\liblangtag\lt-redundant-db.c"
+				>
+			</File>
+			<File
+				RelativePath="..\liblangtag\lt-redundant.c"
+				>
+			</File>
+			<File
+				RelativePath="..\liblangtag\lt-region-db.c"
+				>
+			</File>
+			<File
+				RelativePath="..\liblangtag\lt-region.c"
+				>
+			</File>
+			<File
+				RelativePath="..\liblangtag\lt-script-db.c"
+				>
+			</File>
+			<File
+				RelativePath="..\liblangtag\lt-script.c"
+				>
+			</File>
+			<File
+				RelativePath="..\liblangtag\lt-string.c"
+				>
+			</File>
+			<File
+				RelativePath="..\liblangtag\lt-tag.c"
+				>
+			</File>
+			<File
+				RelativePath="..\liblangtag\lt-trie.c"
+				>
+			</File>
+			<File
+				RelativePath="..\liblangtag\lt-utils.c"
+				>
+			</File>
+			<File
+				RelativePath="..\liblangtag\lt-variant-db.c"
+				>
+			</File>
+			<File
+				RelativePath="..\liblangtag\lt-variant.c"
+				>
+			</File>
+			<File
+				RelativePath="..\liblangtag\lt-xml.c"
+				>
+			</File>
+		</Filter>
+		<Filter
+			Name="Header Files"
+			Filter="h;hpp;hxx;hm;inl"
+			>
+			<File
+				RelativePath="..\liblangtag\langtag.h"
+				>
+			</File>
+			<File
+				RelativePath="..\liblangtag\lt-atomic.h"
+				>
+			</File>
+			<File
+				RelativePath="..\liblangtag\lt-database.h"
+				>
+			</File>
+			<File
+				RelativePath="..\liblangtag\lt-error.h"
+				>
+			</File>
+			<File
+				RelativePath="..\liblangtag\lt-ext-module-data.h"
+				>
+			</File>
+			<File
+				RelativePath="..\liblangtag\lt-ext-module-private.h"
+				>
+			</File>
+			<File
+				RelativePath="..\liblangtag\lt-ext-module.h"
+				>
+			</File>
+			<File
+				RelativePath="..\liblangtag\lt-extension-private.h"
+				>
+			</File>
+			<File
+				RelativePath="..\liblangtag\lt-extension.h"
+				>
+			</File>
+			<File
+				RelativePath="..\liblangtag\lt-extlang-db.h"
+				>
+			</File>
+			<File
+				RelativePath="..\liblangtag\lt-extlang-private.h"
+				>
+			</File>
+			<File
+				RelativePath="..\liblangtag\lt-extlang.h"
+				>
+			</File>
+			<File
+				RelativePath="..\liblangtag\lt-grandfathered-db.h"
+				>
+			</File>
+			<File
+				RelativePath="..\liblangtag\lt-grandfathered-private.h"
+				>
+			</File>
+			<File
+				RelativePath="..\liblangtag\lt-grandfathered.h"
+				>
+			</File>
+			<File
+				RelativePath="..\liblangtag\lt-lang-db.h"
+				>
+			</File>
+			<File
+				RelativePath="..\liblangtag\lt-lang-private.h"
+				>
+			</File>
+			<File
+				RelativePath="..\liblangtag\lt-lang.h"
+				>
+			</File>
+			<File
+				RelativePath="..\liblangtag\lt-list.h"
+				>
+			</File>
+			<File
+				RelativePath="..\liblangtag\lt-localealias.h"
+				>
+			</File>
+			<File
+				RelativePath="..\liblangtag\lt-macros.h"
+				>
+			</File>
+			<File
+				RelativePath="..\liblangtag\lt-mem.h"
+				>
+			</File>
+			<File
+				RelativePath="..\liblangtag\lt-messages.h"
+				>
+			</File>
+			<File
+				RelativePath="..\liblangtag\lt-redundant-db.h"
+				>
+			</File>
+			<File
+				RelativePath="..\liblangtag\lt-redundant-private.h"
+				>
+			</File>
+			<File
+				RelativePath="..\liblangtag\lt-redundant.h"
+				>
+			</File>
+			<File
+				RelativePath="..\liblangtag\lt-region-db.h"
+				>
+			</File>
+			<File
+				RelativePath="..\liblangtag\lt-region-private.h"
+				>
+			</File>
+			<File
+				RelativePath="..\liblangtag\lt-region.h"
+				>
+			</File>
+			<File
+				RelativePath="..\liblangtag\lt-script-db.h"
+				>
+			</File>
+			<File
+				RelativePath="..\liblangtag\lt-script-private.h"
+				>
+			</File>
+			<File
+				RelativePath="..\liblangtag\lt-script.h"
+				>
+			</File>
+			<File
+				RelativePath="..\liblangtag\lt-string.h"
+				>
+			</File>
+			<File
+				RelativePath="..\liblangtag\lt-tag-private.h"
+				>
+			</File>
+			<File
+				RelativePath="..\liblangtag\lt-tag.h"
+				>
+			</File>
+			<File
+				RelativePath="..\liblangtag\lt-trie.h"
+				>
+			</File>
+			<File
+				RelativePath="..\liblangtag\lt-utils.h"
+				>
+			</File>
+			<File
+				RelativePath="..\liblangtag\lt-variant-db.h"
+				>
+			</File>
+			<File
+				RelativePath="..\liblangtag\lt-variant-private.h"
+				>
+			</File>
+			<File
+				RelativePath="..\liblangtag\lt-variant.h"
+				>
+			</File>
+			<File
+				RelativePath="..\liblangtag\lt-xml.h"
+				>
+			</File>
+		</Filter>
+	</Files>
+	<Globals>
+	</Globals>
+</VisualStudioProject>
--- misc/liblangtag-0.4.0/msvs2010/liblangtag.sln	2012-11-06 23:10:33.162941243 +0100
+++ misc/build/liblangtag-0.4.0/msvs2010/liblangtag.sln	2012-11-06 23:14:20.905088613 +0100
@@ -1 +1,19 @@
-dummy
+Microsoft Visual Studio Solution File, Format Version 11.00
+# Visual Studio 2010
+Project("{8BC9CEB8-8B4A-11D0-8D11-00A0C91BC942}") = "liblangtag", "liblangtag.vcxproj", "{8991ACE9-2B73-4C10-BBC8-F16787132D98}"
+EndProject
+Global
+	GlobalSection(SolutionConfigurationPlatforms) = preSolution
+		Debug|Win32 = Debug|Win32
+		Release|Win32 = Release|Win32
+	EndGlobalSection
+	GlobalSection(ProjectConfigurationPlatforms) = postSolution
+		{8991ACE9-2B73-4C10-BBC8-F16787132D98}.Debug|Win32.ActiveCfg = Debug|Win32
+		{8991ACE9-2B73-4C10-BBC8-F16787132D98}.Debug|Win32.Build.0 = Debug|Win32
+		{8991ACE9-2B73-4C10-BBC8-F16787132D98}.Release|Win32.ActiveCfg = Release|Win32
+		{8991ACE9-2B73-4C10-BBC8-F16787132D98}.Release|Win32.Build.0 = Release|Win32
+	EndGlobalSection
+	GlobalSection(SolutionProperties) = preSolution
+		HideSolutionNode = FALSE
+	EndGlobalSection
+EndGlobal
--- misc/liblangtag-0.4.0/msvs2010/liblangtag.vcxproj	2012-11-06 23:10:33.176940944 +0100
+++ misc/build/liblangtag-0.4.0/msvs2010/liblangtag.vcxproj	2012-11-06 23:14:20.906088591 +0100
@@ -1 +1,171 @@
-dummy
+<?xml version="1.0" encoding="utf-8"?>
+<Project DefaultTargets="Build" ToolsVersion="4.0" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
+  <ItemGroup Label="ProjectConfigurations">
+    <ProjectConfiguration Include="Debug|Win32">
+      <Configuration>Debug</Configuration>
+      <Platform>Win32</Platform>
+    </ProjectConfiguration>
+    <ProjectConfiguration Include="Release|Win32">
+      <Configuration>Release</Configuration>
+      <Platform>Win32</Platform>
+    </ProjectConfiguration>
+  </ItemGroup>
+  <PropertyGroup Label="Globals">
+    <ProjectGuid>{8991ACE9-2B73-4C10-BBC8-F16787132D98}</ProjectGuid>
+    <RootNamespace>liblangtag</RootNamespace>
+  </PropertyGroup>
+  <Import Project="$(VCTargetsPath)\Microsoft.Cpp.Default.props" />
+  <PropertyGroup Condition="'$(Configuration)|$(Platform)'=='Debug|Win32'" Label="Configuration">
+    <ConfigurationType>StaticLibrary</ConfigurationType>
+    <UseOfMfc>false</UseOfMfc>
+  </PropertyGroup>
+  <PropertyGroup Condition="'$(Configuration)|$(Platform)'=='Release|Win32'" Label="Configuration">
+    <ConfigurationType>StaticLibrary</ConfigurationType>
+    <UseOfMfc>false</UseOfMfc>
+  </PropertyGroup>
+  <Import Project="$(VCTargetsPath)\Microsoft.Cpp.props" />
+  <ImportGroup Label="ExtensionSettings">
+  </ImportGroup>
+  <ImportGroup Condition="'$(Configuration)|$(Platform)'=='Debug|Win32'" Label="PropertySheets">
+    <Import Project="$(UserRootDir)\Microsoft.Cpp.$(Platform).user.props" Condition="exists('$(UserRootDir)\Microsoft.Cpp.$(Platform).user.props')" Label="LocalAppDataPlatform" />
+  </ImportGroup>
+  <ImportGroup Condition="'$(Configuration)|$(Platform)'=='Release|Win32'" Label="PropertySheets">
+    <Import Project="$(UserRootDir)\Microsoft.Cpp.$(Platform).user.props" Condition="exists('$(UserRootDir)\Microsoft.Cpp.$(Platform).user.props')" Label="LocalAppDataPlatform" />
+  </ImportGroup>
+  <PropertyGroup Label="UserMacros" />
+  <PropertyGroup>
+    <_ProjectFileVersion>10.0.40219.1</_ProjectFileVersion>
+    <OutDir Condition="'$(Configuration)|$(Platform)'=='Release|Win32'">.\Release\</OutDir>
+    <IntDir Condition="'$(Configuration)|$(Platform)'=='Release|Win32'">.\Release\</IntDir>
+    <OutDir Condition="'$(Configuration)|$(Platform)'=='Debug|Win32'">.\Debug\</OutDir>
+    <IntDir Condition="'$(Configuration)|$(Platform)'=='Debug|Win32'">.\Debug\</IntDir>
+  </PropertyGroup>
+  <ItemDefinitionGroup Condition="'$(Configuration)|$(Platform)'=='Release|Win32'">
+    <ClCompile>
+      <Optimization>MaxSpeed</Optimization>
+      <InlineFunctionExpansion>OnlyExplicitInline</InlineFunctionExpansion>
+      <AdditionalIncludeDirectories>..;$(LIBXML_INCLUDE_DIR);%(AdditionalIncludeDirectories)</AdditionalIncludeDirectories>
+      <PreprocessorDefinitions>NDEBUG;WIN32;_LIB;_CRT_SECURE_NO_WARNINGS;__LANGTAG_COMPILATION;REGDATADIR="."ALIGNOF_VOID_P=8;%(PreprocessorDefinitions)</PreprocessorDefinitions>
+      <StringPooling>true</StringPooling>
+      <RuntimeLibrary>MultiThreadedDLL</RuntimeLibrary>
+      <FunctionLevelLinking>true</FunctionLevelLinking>
+      <TreatWChar_tAsBuiltInType>false</TreatWChar_tAsBuiltInType>
+      <RuntimeTypeInfo>true</RuntimeTypeInfo>
+      <PrecompiledHeaderOutputFile>.\Release/liblangtag.pch</PrecompiledHeaderOutputFile>
+      <AssemblerListingLocation>.\Release/</AssemblerListingLocation>
+      <ObjectFileName>.\Release/</ObjectFileName>
+      <ProgramDataBaseFileName>.\Release/</ProgramDataBaseFileName>
+      <WarningLevel>Level3</WarningLevel>
+      <SuppressStartupBanner>true</SuppressStartupBanner>
+    </ClCompile>
+    <Lib>
+      <OutputFile>.\Release\lib\liblangtag.lib</OutputFile>
+      <SuppressStartupBanner>true</SuppressStartupBanner>
+    </Lib>
+    <Bscmake>
+      <SuppressStartupBanner>true</SuppressStartupBanner>
+      <OutputFile>.\Release/liblangtag.bsc</OutputFile>
+    </Bscmake>
+  </ItemDefinitionGroup>
+  <ItemDefinitionGroup Condition="'$(Configuration)|$(Platform)'=='Debug|Win32'">
+    <ClCompile>
+      <Optimization>Disabled</Optimization>
+      <AdditionalIncludeDirectories>..;$(LIBXML_INCLUDE_DIR);%(AdditionalIncludeDirectories)</AdditionalIncludeDirectories>
+      <PreprocessorDefinitions>DEBUG;WIN32;_LIB;_CRT_SECURE_NO_WARNINGS;__LANGTAG_COMPILATION;REGDATADIR="."ALIGNOF_VOID_P=8;%(PreprocessorDefinitions)</PreprocessorDefinitions>
+      <BasicRuntimeChecks>EnableFastChecks</BasicRuntimeChecks>
+      <RuntimeLibrary>MultiThreadedDebugDLL</RuntimeLibrary>
+      <TreatWChar_tAsBuiltInType>false</TreatWChar_tAsBuiltInType>
+      <RuntimeTypeInfo>true</RuntimeTypeInfo>
+      <PrecompiledHeaderOutputFile>.\Debug/liblangtag.pch</PrecompiledHeaderOutputFile>
+      <AssemblerListingLocation>.\Debug/</AssemblerListingLocation>
+      <ObjectFileName>.\Debug/</ObjectFileName>
+      <ProgramDataBaseFileName>.\Debug/</ProgramDataBaseFileName>
+      <WarningLevel>Level3</WarningLevel>
+      <SuppressStartupBanner>true</SuppressStartupBanner>
+      <DebugInformationFormat>EditAndContinue</DebugInformationFormat>
+    </ClCompile>
+    <Lib>
+      <OutputFile>.\Debug\lib\liblangtag.lib</OutputFile>
+      <SuppressStartupBanner>true</SuppressStartupBanner>
+    </Lib>
+    <Bscmake>
+      <SuppressStartupBanner>true</SuppressStartupBanner>
+      <OutputFile>.\Debug/liblangtag.bsc</OutputFile>
+    </Bscmake>
+  </ItemDefinitionGroup>
+  <ItemGroup>
+    <ClCompile Include="..\liblangtag\lt-database.c" />
+    <ClCompile Include="..\liblangtag\lt-error.c" />
+    <ClCompile Include="..\liblangtag\lt-ext-module-data.c" />
+    <ClCompile Include="..\liblangtag\lt-ext-module.c" />
+    <ClCompile Include="..\liblangtag\lt-extension.c" />
+    <ClCompile Include="..\liblangtag\lt-extlang-db.c" />
+    <ClCompile Include="..\liblangtag\lt-extlang.c" />
+    <ClCompile Include="..\liblangtag\lt-grandfathered-db.c" />
+    <ClCompile Include="..\liblangtag\lt-grandfathered.c" />
+    <ClCompile Include="..\liblangtag\lt-lang-db.c" />
+    <ClCompile Include="..\liblangtag\lt-lang.c" />
+    <ClCompile Include="..\liblangtag\lt-list.c" />
+    <ClCompile Include="..\liblangtag\lt-mem.c" />
+    <ClCompile Include="..\liblangtag\lt-messages.c" />
+    <ClCompile Include="..\liblangtag\lt-redundant-db.c" />
+    <ClCompile Include="..\liblangtag\lt-redundant.c" />
+    <ClCompile Include="..\liblangtag\lt-region-db.c" />
+    <ClCompile Include="..\liblangtag\lt-region.c" />
+    <ClCompile Include="..\liblangtag\lt-script-db.c" />
+    <ClCompile Include="..\liblangtag\lt-script.c" />
+    <ClCompile Include="..\liblangtag\lt-string.c" />
+    <ClCompile Include="..\liblangtag\lt-tag.c" />
+    <ClCompile Include="..\liblangtag\lt-trie.c" />
+    <ClCompile Include="..\liblangtag\lt-utils.c" />
+    <ClCompile Include="..\liblangtag\lt-variant-db.c" />
+    <ClCompile Include="..\liblangtag\lt-variant.c" />
+    <ClCompile Include="..\liblangtag\lt-xml.c" />
+  </ItemGroup>
+  <ItemGroup>
+    <ClInclude Include="..\liblangtag\langtag.h" />
+    <ClInclude Include="..\liblangtag\lt-atomic.h" />
+    <ClInclude Include="..\liblangtag\lt-database.h" />
+    <ClInclude Include="..\liblangtag\lt-error.h" />
+    <ClInclude Include="..\liblangtag\lt-ext-module-data.h" />
+    <ClInclude Include="..\liblangtag\lt-ext-module-private.h" />
+    <ClInclude Include="..\liblangtag\lt-ext-module.h" />
+    <ClInclude Include="..\liblangtag\lt-extension-private.h" />
+    <ClInclude Include="..\liblangtag\lt-extension.h" />
+    <ClInclude Include="..\liblangtag\lt-extlang-db.h" />
+    <ClInclude Include="..\liblangtag\lt-extlang-private.h" />
+    <ClInclude Include="..\liblangtag\lt-extlang.h" />
+    <ClInclude Include="..\liblangtag\lt-grandfathered-db.h" />
+    <ClInclude Include="..\liblangtag\lt-grandfathered-private.h" />
+    <ClInclude Include="..\liblangtag\lt-grandfathered.h" />
+    <ClInclude Include="..\liblangtag\lt-lang-db.h" />
+    <ClInclude Include="..\liblangtag\lt-lang-private.h" />
+    <ClInclude Include="..\liblangtag\lt-lang.h" />
+    <ClInclude Include="..\liblangtag\lt-list.h" />
+    <ClInclude Include="..\liblangtag\lt-localealias.h" />
+    <ClInclude Include="..\liblangtag\lt-macros.h" />
+    <ClInclude Include="..\liblangtag\lt-mem.h" />
+    <ClInclude Include="..\liblangtag\lt-messages.h" />
+    <ClInclude Include="..\liblangtag\lt-redundant-db.h" />
+    <ClInclude Include="..\liblangtag\lt-redundant-private.h" />
+    <ClInclude Include="..\liblangtag\lt-redundant.h" />
+    <ClInclude Include="..\liblangtag\lt-region-db.h" />
+    <ClInclude Include="..\liblangtag\lt-region-private.h" />
+    <ClInclude Include="..\liblangtag\lt-region.h" />
+    <ClInclude Include="..\liblangtag\lt-script-db.h" />
+    <ClInclude Include="..\liblangtag\lt-script-private.h" />
+    <ClInclude Include="..\liblangtag\lt-script.h" />
+    <ClInclude Include="..\liblangtag\lt-string.h" />
+    <ClInclude Include="..\liblangtag\lt-tag-private.h" />
+    <ClInclude Include="..\liblangtag\lt-tag.h" />
+    <ClInclude Include="..\liblangtag\lt-trie.h" />
+    <ClInclude Include="..\liblangtag\lt-utils.h" />
+    <ClInclude Include="..\liblangtag\lt-variant-db.h" />
+    <ClInclude Include="..\liblangtag\lt-variant-private.h" />
+    <ClInclude Include="..\liblangtag\lt-variant.h" />
+    <ClInclude Include="..\liblangtag\lt-xml.h" />
+  </ItemGroup>
+  <Import Project="$(VCTargetsPath)\Microsoft.Cpp.targets" />
+  <ImportGroup Label="ExtensionTargets">
+  </ImportGroup>
+</Project>
\ No newline at end of file
