--- misc/liblangtag-0.4.0/configure	2012-10-22 05:47:24.000000000 +0200
+++ misc/build/liblangtag-0.4.0/configure	2012-11-06 14:40:01.653623759 +0100
@@ -3871,6 +3871,8 @@
 rm -f core conftest.err conftest.$ac_objext conftest.$ac_ext
    ac_c_werror_flag=$ac_save_c_werror_flag
 fi
+# MSVC displays only " Command line warning D9002 : ignoring unknown option '-g'", explicitly set to no.
+ac_cv_prog_cc_g=no
 { $as_echo "$as_me:${as_lineno-$LINENO}: result: $ac_cv_prog_cc_g" >&5
 $as_echo "$ac_cv_prog_cc_g" >&6; }
 if test "$ac_test_CFLAGS" = set; then
--- misc/liblangtag-0.4.0/Makefile.in	2012-10-22 04:47:24.000000000 +0100
+++ misc/build/liblangtag-0.4.0/Makefile.in	2012-11-06 20:03:08.906250000 +0000
@@ -136,8 +136,7 @@
 	distdir dist dist-all distcheck
 ETAGS = etags
 CTAGS = ctags
-DIST_SUBDIRS = liblangtag extensions data docs liblangtag-gobject \
-	tests
+DIST_SUBDIRS = liblangtag extensions data docs liblangtag-gobject
 DISTFILES = $(DIST_COMMON) $(DIST_SOURCES) $(TEXINFOS) $(EXTRA_DIST)
 distdir = $(PACKAGE)-$(VERSION)
 top_distdir = $(distdir)
@@ -339,7 +338,7 @@
 NULL = 
 AUTOMAKE_OPTIONS = dist-bzip2
 #SUBDIRS += docs tests
-SUBDIRS = liblangtag extensions data docs $(am__append_1) tests
+SUBDIRS = liblangtag extensions data docs $(am__append_1)
 CONFIGURE_DEPENDENCIES = \
 	requires			\
 	$(NULL)
--- misc/liblangtag-0.4.0/data/Makefile.in	2012-10-22 05:47:23.000000000 +0200
+++ misc/build/liblangtag-0.4.0/data/Makefile.in	2012-11-06 14:47:23.173704887 +0100
@@ -52,7 +52,7 @@
 POST_UNINSTALL = :
 build_triplet = @build@
 host_triplet = @host@
-noinst_PROGRAMS = reg2xml$(EXEEXT) $(am__EXEEXT_1)
+noinst_PROGRAMS = $(am__EXEEXT_1)
 subdir = data
 DIST_COMMON = $(srcdir)/Makefile.am $(srcdir)/Makefile.in
 ACLOCAL_M4 = $(top_srcdir)/aclocal.m4
@@ -80,8 +80,8 @@
 am__v_lt_ = $(am__v_lt_@AM_DEFAULT_V@)
 am__v_lt_0 = --silent
 reg2xml_LINK = $(LIBTOOL) $(AM_V_lt) --tag=CC $(AM_LIBTOOLFLAGS) \
-	$(LIBTOOLFLAGS) --mode=link $(CCLD) $(reg2xml_CFLAGS) \
-	$(CFLAGS) $(reg2xml_LDFLAGS) $(LDFLAGS) -o $@
+	$(LIBTOOLFLAGS) --mode=link $(CCLD) \
+	$(reg2xml_LDFLAGS) $(LDFLAGS) -o $@
 DEFAULT_INCLUDES = -I.@am__isrc@ -I$(top_builddir)
 depcomp = $(SHELL) $(top_srcdir)/depcomp
 am__depfiles_maybe = depfiles
@@ -98,9 +98,9 @@
 AM_V_at = $(am__v_at_@AM_V@)
 am__v_at_ = $(am__v_at_@AM_DEFAULT_V@)
 am__v_at_0 = @
-CCLD = $(CC)
+CCLD = link
 LINK = $(LIBTOOL) $(AM_V_lt) --tag=CC $(AM_LIBTOOLFLAGS) \
-	$(LIBTOOLFLAGS) --mode=link $(CCLD) $(AM_CFLAGS) $(CFLAGS) \
+	$(LIBTOOLFLAGS) --mode=link $(CCLD) \
 	$(AM_LDFLAGS) $(LDFLAGS) -o $@
 AM_V_CCLD = $(am__v_CCLD_@AM_V@)
 am__v_CCLD_ = $(am__v_CCLD_@AM_DEFAULT_V@)
@@ -485,14 +485,14 @@
 @am__fastdepCC_TRUE@	$(AM_V_at)$(am__mv) $(DEPDIR)/reg2xml-reg2xml.Tpo $(DEPDIR)/reg2xml-reg2xml.Po
 @AMDEP_TRUE@@am__fastdepCC_FALSE@	$(AM_V_CC)source='reg2xml.c' object='reg2xml-reg2xml.o' libtool=no @AMDEPBACKSLASH@
 @AMDEP_TRUE@@am__fastdepCC_FALSE@	DEPDIR=$(DEPDIR) $(CCDEPMODE) $(depcomp) @AMDEPBACKSLASH@
-@am__fastdepCC_FALSE@	$(AM_V_CC@am__nodep@)$(CC) $(DEFS) $(DEFAULT_INCLUDES) $(INCLUDES) $(AM_CPPFLAGS) $(CPPFLAGS) $(reg2xml_CFLAGS) $(CFLAGS) -c -o reg2xml-reg2xml.o `test -f 'reg2xml.c' || echo '$(srcdir)/'`reg2xml.c
+@am__fastdepCC_FALSE@	$(AM_V_CC@am__nodep@)$(CC) $(DEFS) $(DEFAULT_INCLUDES) $(INCLUDES) $(AM_CPPFLAGS) $(CPPFLAGS) $(reg2xml_CFLAGS) $(CFLAGS) -c -Foreg2xml-reg2xml.o `test -f 'reg2xml.c' || echo '$(srcdir)/'`reg2xml.c
 
 reg2xml-reg2xml.obj: reg2xml.c
-@am__fastdepCC_TRUE@	$(AM_V_CC)$(CC) $(DEFS) $(DEFAULT_INCLUDES) $(INCLUDES) $(AM_CPPFLAGS) $(CPPFLAGS) $(reg2xml_CFLAGS) $(CFLAGS) -MT reg2xml-reg2xml.obj -MD -MP -MF $(DEPDIR)/reg2xml-reg2xml.Tpo -c -o reg2xml-reg2xml.obj `if test -f 'reg2xml.c'; then $(CYGPATH_W) 'reg2xml.c'; else $(CYGPATH_W) '$(srcdir)/reg2xml.c'; fi`
+@am__fastdepCC_TRUE@	$(AM_V_CC)$(CC) $(DEFS) $(DEFAULT_INCLUDES) $(INCLUDES) $(AM_CPPFLAGS) $(CPPFLAGS) $(reg2xml_CFLAGS) $(CFLAGS) -MT reg2xml-reg2xml.obj -MD -MP -MF $(DEPDIR)/reg2xml-reg2xml.Tpo -c -Foreg2xml-reg2xml.obj `if test -f 'reg2xml.c'; then $(CYGPATH_W) 'reg2xml.c'; else $(CYGPATH_W) '$(srcdir)/reg2xml.c'; fi`
 @am__fastdepCC_TRUE@	$(AM_V_at)$(am__mv) $(DEPDIR)/reg2xml-reg2xml.Tpo $(DEPDIR)/reg2xml-reg2xml.Po
 @AMDEP_TRUE@@am__fastdepCC_FALSE@	$(AM_V_CC)source='reg2xml.c' object='reg2xml-reg2xml.obj' libtool=no @AMDEPBACKSLASH@
 @AMDEP_TRUE@@am__fastdepCC_FALSE@	DEPDIR=$(DEPDIR) $(CCDEPMODE) $(depcomp) @AMDEPBACKSLASH@
-@am__fastdepCC_FALSE@	$(AM_V_CC@am__nodep@)$(CC) $(DEFS) $(DEFAULT_INCLUDES) $(INCLUDES) $(AM_CPPFLAGS) $(CPPFLAGS) $(reg2xml_CFLAGS) $(CFLAGS) -c -o reg2xml-reg2xml.obj `if test -f 'reg2xml.c'; then $(CYGPATH_W) 'reg2xml.c'; else $(CYGPATH_W) '$(srcdir)/reg2xml.c'; fi`
+@am__fastdepCC_FALSE@	$(AM_V_CC@am__nodep@)$(CC) $(DEFS) $(DEFAULT_INCLUDES) $(INCLUDES) $(AM_CPPFLAGS) $(CPPFLAGS) $(reg2xml_CFLAGS) $(CFLAGS) -c -Foreg2xml-reg2xml.obj `if test -f 'reg2xml.c'; then $(CYGPATH_W) 'reg2xml.c'; else $(CYGPATH_W) '$(srcdir)/reg2xml.c'; fi`
 
 mostlyclean-libtool:
 	-rm -f *.lo
--- misc/liblangtag-0.4.0/liblangtag/Makefile.in	2012-10-22 05:47:23.000000000 +0200
+++ misc/build/liblangtag-0.4.0/liblangtag/Makefile.in	2012-11-06 14:51:04.551628998 +0100
@@ -136,8 +136,8 @@
 am__v_lt_ = $(am__v_lt_@AM_DEFAULT_V@)
 am__v_lt_0 = --silent
 liblangtag_la_LINK = $(LIBTOOL) $(AM_V_lt) --tag=CC $(AM_LIBTOOLFLAGS) \
-	$(LIBTOOLFLAGS) --mode=link $(CCLD) $(liblangtag_la_CFLAGS) \
-	$(CFLAGS) $(liblangtag_la_LDFLAGS) $(LDFLAGS) -o $@
+	$(LIBTOOLFLAGS) --mode=link $(CCLD) \
+	$(liblangtag_la_LDFLAGS) $(LDFLAGS) -o $@
 DEFAULT_INCLUDES = -I.@am__isrc@ -I$(top_builddir)
 depcomp = $(SHELL) $(top_srcdir)/depcomp
 am__depfiles_maybe = depfiles
@@ -154,9 +154,9 @@
 AM_V_at = $(am__v_at_@AM_V@)
 am__v_at_ = $(am__v_at_@AM_DEFAULT_V@)
 am__v_at_0 = @
-CCLD = $(CC)
+CCLD = link
 LINK = $(LIBTOOL) $(AM_V_lt) --tag=CC $(AM_LIBTOOLFLAGS) \
-	$(LIBTOOLFLAGS) --mode=link $(CCLD) $(AM_CFLAGS) $(CFLAGS) \
+	$(LIBTOOLFLAGS) --mode=link $(CCLD) \
 	$(AM_LDFLAGS) $(LDFLAGS) -o $@
 AM_V_CCLD = $(am__v_CCLD_@AM_V@)
 am__v_CCLD_ = $(am__v_CCLD_@AM_DEFAULT_V@)
--- misc/liblangtag-0.4.0/ltmain.sh	2012-10-09 11:00:50.000000000 +0200
+++ misc/build/liblangtag-0.4.0/ltmain.sh	2012-11-06 14:40:01.656623672 +0100
@@ -8985,7 +8985,7 @@
       fi
 
       # Replace the output file specification.
-      link_command=`$ECHO "$link_command" | $SED 's%@OUTPUT@%'"$output_objdir/$outputname"'%g'`
+      link_command=`$ECHO "$link_command" | $SED 's%-o \+@OUTPUT@%/OUT:'"$output_objdir/$outputname"'%g'`
 
       # Delete the old output files.
       $opt_dry_run || $RM $output $output_objdir/$outputname $output_objdir/lt-$outputname
--- misc/liblangtag-0.4.0/tests/Makefile.in	2012-10-22 05:47:23.000000000 +0200
+++ misc/build/liblangtag-0.4.0/tests/Makefile.in	2012-11-06 14:49:08.108344563 +0100
@@ -187,9 +187,9 @@
 AM_V_at = $(am__v_at_@AM_V@)
 am__v_at_ = $(am__v_at_@AM_DEFAULT_V@)
 am__v_at_0 = @
-CCLD = $(CC)
+CCLD = link
 LINK = $(LIBTOOL) $(AM_V_lt) --tag=CC $(AM_LIBTOOLFLAGS) \
-	$(LIBTOOLFLAGS) --mode=link $(CCLD) $(AM_CFLAGS) $(CFLAGS) \
+	$(LIBTOOLFLAGS) --mode=link $(CCLD) \
 	$(AM_LDFLAGS) $(LDFLAGS) -o $@
 AM_V_CCLD = $(am__v_CCLD_@AM_V@)
 am__v_CCLD_ = $(am__v_CCLD_@AM_DEFAULT_V@)
@@ -324,6 +324,7 @@
 	@LDFLAGS@					\
 	$(top_builddir)/liblangtag/liblangtag.la	\
 	$(CHECK_LIBS)					\
+	$(LIBXML2_LIBS) \
 	$(NULL)
 
 LIBTOOL = @LIBTOOL@
