--- misc/liblangtag-0.4.0/data/Makefile.am	2012-10-16 07:39:20.000000000 +0200
+++ misc/build/liblangtag-0.4.0/data/Makefile.am	2012-11-06 14:25:29.489175992 +0100
@@ -59,7 +59,7 @@
 language-subtag-registry:
 	$(AM_V_DL) wget -O $@ http://www.iana.org/assignments/language-subtag-registry || \
 	curl -o $@ http://www.iana.org/assignments/language-subtag-registry
-language-subtag-registry.xml: language-subtag-registry reg2xml Makefile
+language-subtag-registry.xml: language-subtag-registry
 	$(AM_V_GEN) $(builddir)/reg2xml $(srcdir)/language-subtag-registry $@.tmp;	\
 	head -1 $@.tmp | grep -E '^<\?xml version'>/dev/null 2>&1 && mv $@.tmp $@ || (echo "E: $@ isn't an expected result"; rm $@.tmp)
 stamp-core-zip:
--- misc/liblangtag-0.4.0/data/Makefile.in	2012-10-22 05:47:23.000000000 +0200
+++ misc/build/liblangtag-0.4.0/data/Makefile.in	2012-11-06 14:26:06.487085669 +0100
@@ -780,7 +780,7 @@
 language-subtag-registry:
 	$(AM_V_DL) wget -O $@ http://www.iana.org/assignments/language-subtag-registry || \
 	curl -o $@ http://www.iana.org/assignments/language-subtag-registry
-language-subtag-registry.xml: language-subtag-registry reg2xml Makefile
+language-subtag-registry.xml: language-subtag-registry
 	$(AM_V_GEN) $(builddir)/reg2xml $(srcdir)/language-subtag-registry $@.tmp;	\
 	head -1 $@.tmp | grep -E '^<\?xml version'>/dev/null 2>&1 && mv $@.tmp $@ || (echo "E: $@ isn't an expected result"; rm $@.tmp)
 stamp-core-zip:
