--- misc/build/liblangtag-0.2/Makefile.in.sav	2012-09-13 17:01:57.000000000 +0200
+++ misc/build/liblangtag-0.2/Makefile.in	2012-09-13 17:02:27.000000000 +0200
@@ -315,7 +315,7 @@ NULL =
 AUTOMAKE_OPTIONS = dist-bzip2
 
 #SUBDIRS += docs tests
-SUBDIRS = liblangtag data docs $(am__append_1) tests
+SUBDIRS = liblangtag data docs $(am__append_1)
 CONFIGURE_DEPENDENCIES = \
 	requires			\
 	$(NULL)
--- misc/build/liblangtag-0.2/liblangtag-gobject/Makefile.in.sav	2012-02-02 13:31:17.000000000 +0100
+++ misc/build/liblangtag-0.2/liblangtag-gobject/Makefile.in	2012-09-13 17:05:22.000000000 +0200
@@ -425,6 +425,7 @@ liblangtag_gobject_la_CFLAGS = \
 liblangtag_gobject_la_LDFLAGS = \
 	$(LDFLAGS)							\
 	-version-info $(LT_G_CURRENT):$(LT_G_REVISION):$(LT_G_AGE)	\
+	-no-undefined							\
 	$(NULL)
 
 #
--- misc/build/liblangtag-0.2/liblangtag-gobject/Makefile.am.sav	2012-02-02 13:31:15.000000000 +0100
+++ misc/build/liblangtag-0.2/liblangtag-gobject/Makefile.am	2012-09-13 17:05:05.000000000 +0200
@@ -137,6 +137,7 @@ liblangtag_gobject_la_CFLAGS =				\
 liblangtag_gobject_la_LDFLAGS =						\
 	$(LDFLAGS)							\
 	-version-info $(LT_G_CURRENT):$(LT_G_REVISION):$(LT_G_AGE)	\
+	-no-undefined							\
 	$(NULL)
 #
 if HAVE_INTROSPECTION
--- misc/build/liblangtag-0.2/Makefile.am.sav	2012-09-13 17:01:57.000000000 +0200
+++ misc/build/liblangtag-0.2/Makefile.am	2012-09-13 17:02:27.000000000 +0200
@@ -7,7 +7,7 @@ SUBDIRS += liblangtag-gobject
 endif
 
 #SUBDIRS += docs tests
-SUBDIRS += tests
+#SUBDIRS += tests
 
 ACLOCAL_AMFLAGS =			\
 	-I m4macros			\
--- misc/build/liblangtag-0.2/liblangtag/Makefile.in.sav	2012-02-02 11:04:10.000000000 +0100
+++ misc/build/liblangtag-0.2/liblangtag/Makefile.in	2012-09-13 17:04:35.000000000 +0200
@@ -487,6 +487,7 @@ liblangtag_la_CFLAGS = \
 liblangtag_la_LDFLAGS = \
 	$(LDFLAGS)						\
 	-version-info $(LT_CURRENT):$(LT_REVISION):$(LT_AGE)	\
+	-no-undefined						\
 	$(NULL)
 
 all: all-recursive
--- misc/build/liblangtag-0.2/liblangtag/Makefile.am.sav	2012-02-02 09:54:40.000000000 +0100
+++ misc/build/liblangtag-0.2/liblangtag/Makefile.am	2012-09-13 17:04:19.000000000 +0200
@@ -149,4 +149,5 @@ liblangtag_la_CFLAGS =				\
 liblangtag_la_LDFLAGS =						\
 	$(LDFLAGS)						\
 	-version-info $(LT_CURRENT):$(LT_REVISION):$(LT_AGE)	\
+	-no-undefined						\
 	$(NULL)
--- misc/build/liblangtag-0.2/liblangtag/lt-ext-module.c.sav	2012-02-02 07:57:44.000000000 +0100
+++ misc/build/liblangtag-0.2/liblangtag/lt-ext-module.c	2012-09-13 17:33:24.000000000 +0200
@@ -584,8 +584,8 @@ lt_ext_modules_load(void)
 			gsize len;
 
 			while (1) {
-				if (readdir_r(dir, &dent, &dresult) || dresult == NULL)
-					break;
+/*				if (readdir_r(dir, &dent, &dresult) || dresult == NULL)
+					break;*/ abort(); /* we do not use liblangtag-gobject anyway */
 
 				len = strlen(dent.d_name);
 				if (len > suffix_len &&
