--- misc/liblangtag-0.4.0/liblangtag/lt-messages.h	2012-11-06 23:10:33.191940625 +0100
+++ misc/build/liblangtag-0.4.0/liblangtag/lt-messages.h	2012-11-06 22:31:12.372708079 +0100
@@ -16,7 +16,10 @@
 #ifndef __LT_MESSAGES_H__
 #define __LT_MESSAGES_H__
 
+#ifdef HAVE_CONFIG_H
 #include "config.h"
+#endif
+
 #include <stdarg.h>
 #include "lt-macros.h"
 
--- misc/liblangtag-0.4.0/liblangtag/lt-utils.c	2012-11-06 23:10:33.195940539 +0100
+++ misc/build/liblangtag-0.4.0/liblangtag/lt-utils.c	2012-11-06 22:43:39.266557764 +0100
@@ -17,6 +17,7 @@
 #include <ctype.h>
 #include <stdio.h>
 #include <string.h>
+#include <memory.h>
 #include "lt-messages.h"
 #include "lt-utils.h"
 
