--- misc/liblangtag-0.2/configure	2012-02-02 11:04:11.000000000 +0100
+++ misc/build/liblangtag-0.2/configure	2012-05-24 22:50:18.991459556 +0200
@@ -13514,7 +13514,9 @@
   no_glib=""
 
   if test "x$PKG_CONFIG" = x ; then
-    no_glib=yes
+      if test "x$LIBO_GLIB_CHEAT" = x ; then
+          no_glib=yes
+      fi
     PKG_CONFIG=no
   fi
 
@@ -13537,18 +13539,20 @@
   fi
 
   if test x"$no_glib" = x ; then
-    GLIB_GENMARSHAL=`$PKG_CONFIG --variable=glib_genmarshal glib-2.0`
-    GOBJECT_QUERY=`$PKG_CONFIG --variable=gobject_query glib-2.0`
-    GLIB_MKENUMS=`$PKG_CONFIG --variable=glib_mkenums glib-2.0`
-
-    GLIB_CFLAGS=`$PKG_CONFIG --cflags $pkg_config_args`
-    GLIB_LIBS=`$PKG_CONFIG --libs $pkg_config_args`
-    glib_config_major_version=`$PKG_CONFIG --modversion glib-2.0 | \
-           sed 's/\([0-9]*\).\([0-9]*\).\([0-9]*\)/\1/'`
-    glib_config_minor_version=`$PKG_CONFIG --modversion glib-2.0 | \
-           sed 's/\([0-9]*\).\([0-9]*\).\([0-9]*\)/\2/'`
-    glib_config_micro_version=`$PKG_CONFIG --modversion glib-2.0 | \
-           sed 's/\([0-9]*\).\([0-9]*\).\([0-9]*\)/\3/'`
+      if test "x$LIBO_GLIB_CHEAT" = x ; then
+          GLIB_GENMARSHAL=`$PKG_CONFIG --variable=glib_genmarshal glib-2.0`
+          GOBJECT_QUERY=`$PKG_CONFIG --variable=gobject_query glib-2.0`
+          GLIB_MKENUMS=`$PKG_CONFIG --variable=glib_mkenums glib-2.0`
+
+          GLIB_CFLAGS=`$PKG_CONFIG --cflags $pkg_config_args`
+          GLIB_LIBS=`$PKG_CONFIG --libs $pkg_config_args`
+          glib_config_major_version=`$PKG_CONFIG --modversion glib-2.0 | \
+              sed 's/\([0-9]*\).\([0-9]*\).\([0-9]*\)/\1/'`
+          glib_config_minor_version=`$PKG_CONFIG --modversion glib-2.0 | \
+              sed 's/\([0-9]*\).\([0-9]*\).\([0-9]*\)/\2/'`
+          glib_config_micro_version=`$PKG_CONFIG --modversion glib-2.0 | \
+              sed 's/\([0-9]*\).\([0-9]*\).\([0-9]*\)/\3/'`
+      fi
     if test "x$enable_glibtest" = "xyes" ; then
       ac_save_CFLAGS="$CFLAGS"
       ac_save_LIBS="$LIBS"
