--- misc/liblangtag-0.2/liblangtag/lt-xml.c
+++ misc/build/liblangtag-0.2/liblangtag/lt-xml.c
@@ -101,8 +101,6 @@ lt_xml_read_subtag_registry(lt_xml_t  *x
 	if (xmlparser)
 		xmlFreeParserCtxt(xmlparser);
 
-	xmlCleanupParser();
-
 	if (err) {
 		if (error)
 			*error = g_error_copy(err);
@@ -158,8 +156,6 @@ lt_xml_read_cldr_bcp47(lt_xml_t     *xml
 	if (xmlparser)
 		xmlFreeParserCtxt(xmlparser);
 
-	xmlCleanupParser();
-
 	if (err) {
 		if (error)
 			*error = g_error_copy(err);
