--- misc/liblangtag-0.2/configure	2012-06-03 03:21:20.513755000 +0200
+++ misc/build/liblangtag-0.2/configure	2012-06-02 23:04:17.742605700 +0200
@@ -3751,6 +3751,8 @@
 rm -f core conftest.err conftest.$ac_objext conftest.$ac_ext
    ac_c_werror_flag=$ac_save_c_werror_flag
 fi
+# MSVC displays only " Command line warning D9002 : ignoring unknown option '-g'", explicitly set to no.
+ac_cv_prog_cc_g=no
 { $as_echo "$as_me:${as_lineno-$LINENO}: result: $ac_cv_prog_cc_g" >&5
 $as_echo "$ac_cv_prog_cc_g" >&6; }
 if test "$ac_test_CFLAGS" = set; then
--- misc/liblangtag-0.2/data/Makefile.in	2012-02-02 11:04:10.000000000 +0100
+++ misc/build/liblangtag-0.2/data/Makefile.in	2012-06-03 03:20:28.045005000 +0200
@@ -59,7 +59,7 @@
 reg2xml_OBJECTS = $(am_reg2xml_OBJECTS)
 reg2xml_LDADD = $(LDADD)
 reg2xml_LINK = $(LIBTOOL) --tag=CC $(AM_LIBTOOLFLAGS) $(LIBTOOLFLAGS) \
-	--mode=link $(CCLD) $(reg2xml_CFLAGS) $(CFLAGS) \
+	--mode=link $(CCLD) \
 	$(reg2xml_LDFLAGS) $(LDFLAGS) -o $@
 DEFAULT_INCLUDES = -I.@am__isrc@ -I$(top_builddir)
 depcomp = $(SHELL) $(top_srcdir)/depcomp
@@ -70,9 +70,9 @@
 LTCOMPILE = $(LIBTOOL) --tag=CC $(AM_LIBTOOLFLAGS) $(LIBTOOLFLAGS) \
 	--mode=compile $(CC) $(DEFS) $(DEFAULT_INCLUDES) $(INCLUDES) \
 	$(AM_CPPFLAGS) $(CPPFLAGS) $(AM_CFLAGS) $(CFLAGS)
-CCLD = $(CC)
+CCLD = link
 LINK = $(LIBTOOL) --tag=CC $(AM_LIBTOOLFLAGS) $(LIBTOOLFLAGS) \
-	--mode=link $(CCLD) $(AM_CFLAGS) $(CFLAGS) $(AM_LDFLAGS) \
+	--mode=link $(CCLD) $(AM_LDFLAGS) \
 	$(LDFLAGS) -o $@
 SOURCES = $(reg2xml_SOURCES)
 DIST_SOURCES = $(reg2xml_SOURCES)
@@ -422,18 +422,18 @@
 @am__fastdepCC_FALSE@	$(LTCOMPILE) -c -o $@ $<
 
 reg2xml-reg2xml.o: reg2xml.c
-@am__fastdepCC_TRUE@	$(CC) $(DEFS) $(DEFAULT_INCLUDES) $(INCLUDES) $(AM_CPPFLAGS) $(CPPFLAGS) $(reg2xml_CFLAGS) $(CFLAGS) -MT reg2xml-reg2xml.o -MD -MP -MF $(DEPDIR)/reg2xml-reg2xml.Tpo -c -o reg2xml-reg2xml.o `test -f 'reg2xml.c' || echo '$(srcdir)/'`reg2xml.c
+@am__fastdepCC_TRUE@	$(CC) $(DEFS) $(DEFAULT_INCLUDES) $(INCLUDES) $(AM_CPPFLAGS) $(CPPFLAGS) $(reg2xml_CFLAGS) $(CFLAGS) -MT reg2xml-reg2xml.o -MD -MP -MF $(DEPDIR)/reg2xml-reg2xml.Tpo -c -Foreg2xml-reg2xml.o `test -f 'reg2xml.c' || echo '$(srcdir)/'`reg2xml.c
 @am__fastdepCC_TRUE@	$(am__mv) $(DEPDIR)/reg2xml-reg2xml.Tpo $(DEPDIR)/reg2xml-reg2xml.Po
 @AMDEP_TRUE@@am__fastdepCC_FALSE@	source='reg2xml.c' object='reg2xml-reg2xml.o' libtool=no @AMDEPBACKSLASH@
 @AMDEP_TRUE@@am__fastdepCC_FALSE@	DEPDIR=$(DEPDIR) $(CCDEPMODE) $(depcomp) @AMDEPBACKSLASH@
-@am__fastdepCC_FALSE@	$(CC) $(DEFS) $(DEFAULT_INCLUDES) $(INCLUDES) $(AM_CPPFLAGS) $(CPPFLAGS) $(reg2xml_CFLAGS) $(CFLAGS) -c -o reg2xml-reg2xml.o `test -f 'reg2xml.c' || echo '$(srcdir)/'`reg2xml.c
+@am__fastdepCC_FALSE@	$(CC) $(DEFS) $(DEFAULT_INCLUDES) $(INCLUDES) $(AM_CPPFLAGS) $(CPPFLAGS) $(reg2xml_CFLAGS) $(CFLAGS) -c -Foreg2xml-reg2xml.o `test -f 'reg2xml.c' || echo '$(srcdir)/'`reg2xml.c
 
 reg2xml-reg2xml.obj: reg2xml.c
-@am__fastdepCC_TRUE@	$(CC) $(DEFS) $(DEFAULT_INCLUDES) $(INCLUDES) $(AM_CPPFLAGS) $(CPPFLAGS) $(reg2xml_CFLAGS) $(CFLAGS) -MT reg2xml-reg2xml.obj -MD -MP -MF $(DEPDIR)/reg2xml-reg2xml.Tpo -c -o reg2xml-reg2xml.obj `if test -f 'reg2xml.c'; then $(CYGPATH_W) 'reg2xml.c'; else $(CYGPATH_W) '$(srcdir)/reg2xml.c'; fi`
+@am__fastdepCC_TRUE@	$(CC) $(DEFS) $(DEFAULT_INCLUDES) $(INCLUDES) $(AM_CPPFLAGS) $(CPPFLAGS) $(reg2xml_CFLAGS) $(CFLAGS) -MT reg2xml-reg2xml.obj -MD -MP -MF $(DEPDIR)/reg2xml-reg2xml.Tpo -c -Foreg2xml-reg2xml.obj `if test -f 'reg2xml.c'; then $(CYGPATH_W) 'reg2xml.c'; else $(CYGPATH_W) '$(srcdir)/reg2xml.c'; fi`
 @am__fastdepCC_TRUE@	$(am__mv) $(DEPDIR)/reg2xml-reg2xml.Tpo $(DEPDIR)/reg2xml-reg2xml.Po
 @AMDEP_TRUE@@am__fastdepCC_FALSE@	source='reg2xml.c' object='reg2xml-reg2xml.obj' libtool=no @AMDEPBACKSLASH@
 @AMDEP_TRUE@@am__fastdepCC_FALSE@	DEPDIR=$(DEPDIR) $(CCDEPMODE) $(depcomp) @AMDEPBACKSLASH@
-@am__fastdepCC_FALSE@	$(CC) $(DEFS) $(DEFAULT_INCLUDES) $(INCLUDES) $(AM_CPPFLAGS) $(CPPFLAGS) $(reg2xml_CFLAGS) $(CFLAGS) -c -o reg2xml-reg2xml.obj `if test -f 'reg2xml.c'; then $(CYGPATH_W) 'reg2xml.c'; else $(CYGPATH_W) '$(srcdir)/reg2xml.c'; fi`
+@am__fastdepCC_FALSE@	$(CC) $(DEFS) $(DEFAULT_INCLUDES) $(INCLUDES) $(AM_CPPFLAGS) $(CPPFLAGS) $(reg2xml_CFLAGS) $(CFLAGS) -c -Foreg2xml-reg2xml.obj `if test -f 'reg2xml.c'; then $(CYGPATH_W) 'reg2xml.c'; else $(CYGPATH_W) '$(srcdir)/reg2xml.c'; fi`
 
 mostlyclean-libtool:
 	-rm -f *.lo
--- misc/liblangtag-0.2/liblangtag/Makefile.in	2012-02-02 11:04:10.000000000 +0100
+++ misc/build/liblangtag-0.2/liblangtag/Makefile.in	2012-06-02 23:04:17.757766800 +0200
@@ -97,8 +97,8 @@
 am_liblangtag_la_OBJECTS = $(am__objects_3) $(am__objects_1)
 liblangtag_la_OBJECTS = $(am_liblangtag_la_OBJECTS)
 liblangtag_la_LINK = $(LIBTOOL) --tag=CC $(AM_LIBTOOLFLAGS) \
-	$(LIBTOOLFLAGS) --mode=link $(CCLD) $(liblangtag_la_CFLAGS) \
-	$(CFLAGS) $(liblangtag_la_LDFLAGS) $(LDFLAGS) -o $@
+	$(LIBTOOLFLAGS) --mode=link $(CCLD) \
+	$(liblangtag_la_LDFLAGS) $(LDFLAGS) -o $@
 DEFAULT_INCLUDES = -I.@am__isrc@ -I$(top_builddir)
 depcomp = $(SHELL) $(top_srcdir)/depcomp
 am__depfiles_maybe = depfiles
@@ -108,9 +108,9 @@
 LTCOMPILE = $(LIBTOOL) --tag=CC $(AM_LIBTOOLFLAGS) $(LIBTOOLFLAGS) \
 	--mode=compile $(CC) $(DEFS) $(DEFAULT_INCLUDES) $(INCLUDES) \
 	$(AM_CPPFLAGS) $(CPPFLAGS) $(AM_CFLAGS) $(CFLAGS)
-CCLD = $(CC)
+CCLD = link
 LINK = $(LIBTOOL) --tag=CC $(AM_LIBTOOLFLAGS) $(LIBTOOLFLAGS) \
-	--mode=link $(CCLD) $(AM_CFLAGS) $(CFLAGS) $(AM_LDFLAGS) \
+	--mode=link $(CCLD) $(AM_LDFLAGS) \
 	$(LDFLAGS) -o $@
 SOURCES = $(liblangtag_la_SOURCES)
 DIST_SOURCES = $(liblangtag_la_SOURCES)
--- misc/liblangtag-0.2/ltmain.sh	2011-12-27 11:43:05.000000000 +0100
+++ misc/build/liblangtag-0.2/ltmain.sh	2012-06-02 23:04:17.772927900 +0200
@@ -8969,7 +8969,7 @@
       fi
 
       # Replace the output file specification.
-      link_command=`$ECHO "$link_command" | $SED 's%@OUTPUT@%'"$output_objdir/$outputname"'%g'`
+      link_command=`$ECHO "$link_command" | $SED 's%-o \+@OUTPUT@%/OUT:'"$output_objdir/$outputname"'%g'`
 
       # Delete the old output files.
       $opt_dry_run || $RM $output $output_objdir/$outputname $output_objdir/lt-$outputname
--- misc/liblangtag-0.2/tests/Makefile.in	2012-02-02 11:04:10.000000000 +0100
+++ misc/build/liblangtag-0.2/tests/Makefile.in	2012-06-03 03:21:02.982505000 +0200
@@ -149,9 +149,9 @@
 LTCOMPILE = $(LIBTOOL) --tag=CC $(AM_LIBTOOLFLAGS) $(LIBTOOLFLAGS) \
 	--mode=compile $(CC) $(DEFS) $(DEFAULT_INCLUDES) $(INCLUDES) \
 	$(AM_CPPFLAGS) $(CPPFLAGS) $(AM_CFLAGS) $(CFLAGS)
-CCLD = $(CC)
+CCLD = link
 LINK = $(LIBTOOL) --tag=CC $(AM_LIBTOOLFLAGS) $(LIBTOOLFLAGS) \
-	--mode=link $(CCLD) $(AM_CFLAGS) $(CFLAGS) $(AM_LDFLAGS) \
+	--mode=link $(CCLD) $(AM_LDFLAGS) \
 	$(LDFLAGS) -o $@
 SOURCES = $(check_extlang_SOURCES) $(check_grandfathered_SOURCES) \
 	$(check_lang_SOURCES) $(check_region_SOURCES) \
@@ -286,6 +286,8 @@
 	@LDFLAGS@					\
 	$(top_builddir)/liblangtag/liblangtag.la	\
 	$(CHECK_LIBS)					\
+	$(LIBXML2_LIBS) \
+	$(GLIB_LIBS) \
 	$(NULL)
 
 LIBTOOL = @LIBTOOL@
