--- misc/BeanShell-2.0b5/build.xml	Fri Dec 19 17:14:27 2003
+++ misc/build/BeanShell-2.0b5/build.xml	Fri Mar 28 15:55:04 2008
@@ -10,7 +10,7 @@
 		- Why can't I nest filesets?  This seems like it would be so easy and
 		useful...
 -->
-<project name="beanshell" default="compile" basedir=".">
+<project name="beanshell" default="jarall" basedir=".">
 
 <!-- Project Configuration -->
 
@@ -65,6 +65,9 @@
 	<property name="exclude-engine" value="bsh/engine/**"/>
 	-->
 
+	<property name="exclude-bsf" 
+		value="bsh/util/BeanShellBSFEngine.java,TestBshBSF.java"/>
+
 	<!-- Uncomment to build without the ASM class generator code.
 	<property name="exclude-classgen" 
 		value="bsh/org/objectweb/asm/**,bsh/ClassGeneratorImpl.java,bsh/ClassGeneratorUtil.java,bsh/DelayedEvalBshMethod.java"/>
@@ -75,6 +78,9 @@
 		value="bsh/servlet/*"/>
 	-->
 
+	<property name="exclude-servlet" 
+		value="bsh/servlet/*"/>
+
 	<!-- Legacy excludes.  Comment this *out* to build these legacy items -->
 	<property name="excludes-legacy" 
 		value="bsh/JThis.java"/>
--- misc/BeanShell-2.0b5/makefile.mk	Fri Mar 28 15:56:40 2008
+++ misc/build/BeanShell-2.0b5/makefile.mk	Fri Mar 28 15:56:06 2008
@@ -1 +1,29 @@
-dummy
+#**************************************************************
+#  
+#  Licensed to the Apache Software Foundation (ASF) under one
+#  or more contributor license agreements.  See the NOTICE file
+#  distributed with this work for additional information
+#  regarding copyright ownership.  The ASF licenses this file
+#  to you under the Apache License, Version 2.0 (the
+#  "License"); you may not use this file except in compliance
+#  with the License.  You may obtain a copy of the License at
+#  
+#    http://www.apache.org/licenses/LICENSE-2.0
+#  
+#  Unless required by applicable law or agreed to in writing,
+#  software distributed under the License is distributed on an
+#  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+#  KIND, either express or implied.  See the License for the
+#  specific language governing permissions and limitations
+#  under the License.
+#  
+#**************************************************************
+
+PRJ=..$/..$/..$/..
+PRJNAME=ooo_beanshell
+TARGET=jarall
+
+.INCLUDE : ant.mk
+
+ALLTAR : ANTBUILD
+
