--- build/ltmain.sh
+++ build/ltmain.sh
@@ -5301,6 +5301,12 @@
 	  prev=
 	  continue
 	  ;;
+	mllvm)
+	  # Clang does not use LLVM to link, so we can simply discard any
+	  # '-mllvm $arg' options when doing the link step.
+	  prev=
+	  continue
+	  ;;
 	objectlist)
 	  if test -f "$arg"; then
 	    save_arg=$arg
@@ -5639,6 +5645,11 @@
 	continue
 	;;
 
+      -mllvm)
+	prev=mllvm
+	continue
+	;;
+
       -module)
 	module=yes
 	continue
