From de94f4f83973df18a0cea257656c9cdc0fdd214b Mon Sep 17 00:00:00 2001
From: David Tardon <dtardon@redhat.com>
Date: Mon, 31 Aug 2015 12:19:53 +0200
Subject: [PATCH] error: 'atoi' was not declared in this scope

---
 src/lib/WPSOLEParser.cpp | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/src/lib/WPSOLEParser.cpp b/src/lib/WPSOLEParser.cpp
index 9a57341..ac564b4 100644
--- a/src/lib/WPSOLEParser.cpp
+++ b/src/lib/WPSOLEParser.cpp
@@ -64,6 +64,7 @@
  * ------------------------------------------------------------
  */
 
+#include <cstdlib>
 #include <cstring>
 #include <map>
 #include <sstream>
@@ -296,7 +297,7 @@ bool WPSOLEParser::parse(RVNGInputStreamPtr file)
 				std::string::size_type idP = pos-1;
 				while (idP >=1 && dir[idP-1] >= '0' && dir[idP-1] <= '9')
 					idP--;
-				int val = atoi(dir.substr(idP, idP-pos).c_str());
+				int val = std::atoi(dir.substr(idP, idP-pos).c_str());
 				if (id[0] == -1) id[0] = val;
 				else
 				{
-- 
2.4.3

