From 53c03b23520005aac0fb9b3036a200d63c7a8f06 Mon Sep 17 00:00:00 2001
From: Sean Young <sean@mess.org>
Date: Tue, 6 Oct 2015 09:16:11 +0100
Subject: [PATCH] add missing include

---
 src/lib/MSWrite.cpp | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/lib/MSWrite.cpp b/src/lib/MSWrite.cpp
index 2bc2511..ab5fdd6 100644
--- a/src/lib/MSWrite.cpp
+++ b/src/lib/MSWrite.cpp
@@ -32,6 +32,7 @@
 
 #include "MSWrite.h"
 
+#include <algorithm>
 #include <cstring>
 
 namespace MSWriteParserInternal
-- 
2.1.4

