--- tommath.h
+++ tommath.h
@@ -15,6 +15,13 @@
 #ifndef BN_H_
 #define BN_H_
 
+// Work around clang-cl issue when mp_word is a typedef for unsigned __int128, see
+// <https://bugs.llvm.org/show_bug.cgi?id=25305> "Clang-cl generates a call to an undefined symbol
+// _udivti3":
+#if defined _WIN32 && defined __clang__
+#define MP_8BIT
+#endif
+
 #include <stdio.h>
 #include <stdlib.h>
 #include <stdint.h>
