diff -ur misc/expat-2.2.1/lib/expat_external.h misc/build/expat-2.2.0/lib/expat_external.h
--- misc/expat-2.2.1/lib/expat_external.h	2016-06-21 05:58:38.000000000 -0700
+++ misc/build/expat-2.2.1/lib/expat_external.h	2016-07-31 17:10:53.551556000 -0700
@@ -7,10 +7,6 @@
 
 /* External API definitions */
 
-#if defined(_MSC_EXTENSIONS) && !defined(__BEOS__) && !defined(__CYGWIN__)
-#define XML_USE_MSC_EXTENSIONS 1
-#endif
-
 /* Expat tries very hard to make the API boundary very specifically
    defined.  There are two macros defined to control this boundary;
    each of these can be defined before including this header to
