--- misc/libcroco-0.6.2/src/makefile.mk	Wed Aug 20 14:33:55 2008
+++ misc/build/libcroco-0.6.2/src/makefile.mk	Wed Aug 20 14:26:42 2008
@@ -1 +1,107 @@
-dummy
+#*************************************************************************
+#
+# DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+# 
+# Copyright 2000, 2010 Oracle and/or its affiliates.
+#
+# OpenOffice.org - a multi-platform office productivity suite
+#
+# This file is part of OpenOffice.org.
+#
+# OpenOffice.org is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Lesser General Public License version 3
+# only, as published by the Free Software Foundation.
+#
+# OpenOffice.org is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Lesser General Public License version 3 for more details
+# (a copy is included in the LICENSE file that accompanied this code).
+#
+# You should have received a copy of the GNU Lesser General Public License
+# version 3 along with OpenOffice.org.  If not, see
+# <http://www.openoffice.org/license.html>
+# for a copy of the LGPLv3 License.
+#
+#*************************************************************************
+
+PRJ=..$/..$/..$/..$/..
+#PRJINC=..
+PRJNAME=libcroco
+TARGET=libcroco-0.6-3
+
+#ENABLE_EXCEPTIONS=TRUE
+VISIBILITY_HIDDEN=TRUE
+EXTERNAL_WARNINGS_NOT_ERRORS=TRUE
+UWINAPILIB=
+
+# --- Settings ----------------------------------
+
+.INCLUDE : settings.mk
+
+CFLAGS+=-I$(SOLARINCDIR)$/external -I$(SOLARINCDIR)$/external$/glib-2.0
+
+# --- Files -------------------------------------
+
+SLOFILES=\
+	$(SLO)$/cr-utils.obj \
+	$(SLO)$/cr-input.obj \
+	$(SLO)$/cr-enc-handler.obj \
+	$(SLO)$/cr-num.obj \
+	$(SLO)$/cr-rgb.obj \
+	$(SLO)$/cr-token.obj \
+	$(SLO)$/cr-tknzr.obj \
+	$(SLO)$/cr-term.obj \
+	$(SLO)$/cr-attr-sel.obj \
+	$(SLO)$/cr-pseudo.obj \
+	$(SLO)$/cr-additional-sel.obj \
+	$(SLO)$/cr-simple-sel.obj \
+	$(SLO)$/cr-selector.obj \
+	$(SLO)$/cr-doc-handler.obj \
+	$(SLO)$/cr-parser.obj \
+	$(SLO)$/cr-declaration.obj \
+	$(SLO)$/cr-statement.obj \
+	$(SLO)$/cr-stylesheet.obj \
+	$(SLO)$/cr-cascade.obj \
+	$(SLO)$/cr-om-parser.obj \
+	$(SLO)$/cr-style.obj \
+	$(SLO)$/cr-sel-eng.obj \
+	$(SLO)$/cr-fonts.obj \
+	$(SLO)$/cr-prop-list.obj \
+	$(SLO)$/cr-parsing-location.obj \
+	$(SLO)$/cr-string.obj
+
+
+# --- Library -----------------------------------
+
+SHL1TARGET=	$(TARGET)
+SHL1OBJS=$(SLOFILES)
+SHL1STDLIBS=\
+	gio-2.0.lib \
+	gobject-2.0.lib \
+	gthread-2.0.lib \
+	glib-2.0.lib \
+	intl.lib \
+	$(LIBXML2LIB)
+
+
+SHL1DEPN=
+#SHL1IMPLIB=$(TARGET)
+#SHL1USE_EXPORTS=name
+
+SHL1DEF=	$(MISC)$/$(SHL1TARGET).def
+
+DEF1NAME=	$(SHL1TARGET)
+DEF1DEPN=	$(MISC)$/$(SHL1TARGET).flt \
+			$(SLB)$/$(TARGET).lib
+DEFLIB1NAME=$(TARGET)
+
+# --- Targets ----------------------------------
+
+.INCLUDE : target.mk
+
+# --- filter file ------------------------------
+
+$(MISC)$/$(SHL1TARGET).flt: makefile.mk
+    @echo CLEAR_THE_FILE	> $@
+
