--- misc/libformula/source/org/pentaho/reporting/libraries/formula/function/datetime/DateDifFunction.java	2008-12-12 13:04:46.000000000 +0100
+++ misc/build/libformula/source/org/pentaho/reporting/libraries/formula/function/datetime/DateDifFunction.java	2008-12-15 08:15:18.293125000 +0100
@@ -198,12 +198,6 @@
           LibFormulaErrorValue.ERROR_INVALID_ARGUMENT_VALUE);
     }
 
-    if (res < 0)
-    {
-      throw new EvaluationException(
-          LibFormulaErrorValue.ERROR_INVALID_ARGUMENT_VALUE);
-    }
-
     //noinspection UnpredictableBigDecimalConstructorCall
     return new TypeValuePair(NumberType.GENERIC_NUMBER, new BigDecimal((double) res));
   }
--- misc/libformula/source/org/pentaho/reporting/libraries/formula/lvalues/FormulaFunction.java	2008-12-12 13:04:46.000000000 +0100
+++ misc/build/libformula/source/org/pentaho/reporting/libraries/formula/lvalues/FormulaFunction.java	2008-12-15 08:13:14.308750000 +0100
@@ -161,7 +161,8 @@
 
     for (int i = 0; i < parameters.length; i++)
     {
-      parameters[i].initialize(context);
+        if ( parameters[i] != null )
+            parameters[i].initialize(context);
     }
   }
 
