*** misc/liblayout/build.properties	Wed May 30 19:06:41 2007
--- misc/build/liblayout/build.properties	Fri Jul 27 09:05:42 2007
***************
*** 14,21 ****
  
  #
  # Make sure that JDK 1.2 compatible class files are generated
! build.target=1.2
! build.source=1.2
  build.retro.jdk=/opt/jdk1.2.2
  
  #
--- 14,21 ----
  
  #
  # Make sure that JDK 1.2 compatible class files are generated
! build.target=1.3
! build.source=1.3
  build.retro.jdk=/opt/jdk1.2.2
  
  #
***************
*** 39,51 ****
  # JFree.org/JFreeReport libraries
  jcommon-jar-file=jcommon-1.0.10.jar
  flute-jar-file=flute-1.3-jfree-20061107.jar
! libloader-jar-file=libloader-0.3.5.jar
! librepository-jar-file=librepository-0.1.3.jar
  #
  # One of these is required.
  libfonts-core-jar-file=libfonts-core-0.3.3.jar
! libfonts-jar-file=libfonts-0.3.2.jar
! libxml-jar-file=libxml-0.9.8.jar

  # 3rd party libraries ..

--- 39,51 ----
  # JFree.org/JFreeReport libraries
  jcommon-jar-file=jcommon-1.0.10.jar
  flute-jar-file=flute-1.3-jfree-20061107.jar
! libloader-jar-file=libloader-0.3.6.jar
! librepository-jar-file=librepository-0.1.4.jar
  #
  # One of these is required.
  libfonts-core-jar-file=libfonts-core-0.3.3.jar
! libfonts-jar-file=libfonts-0.3.3.jar
! libxml-jar-file=libxml-0.9.9.jar

  # 3rd party libraries ..

***************
*** 66,69 ****
  liblayout-bundle-file=${liblayout.name}-${liblayout.version}-bundle.jar
  liblayout-jar-file=${liblayout.name}-${liblayout.version}.jar
  liblayout-core-jar-file=${liblayout.name}-core-${liblayout.version}.jar
! module-output-pdf-jar-file=${liblayout.name}-module-output-pdf-${liblayout.version}.jar
--- 66,69 ----
  liblayout-bundle-file=${liblayout.name}-${liblayout.version}-bundle.jar
  liblayout-jar-file=${liblayout.name}-${liblayout.version}.jar
  liblayout-core-jar-file=${liblayout.name}-core-${liblayout.version}.jar
! module-output-pdf-jar-file=${liblayout.name}-module-output-pdf-${liblayout.version}.jar
*** misc/liblayout/build.xml    Thu Oct 18 08:21:18 2007
--- misc/build/liblayout/build.xml      Mon Oct 22 12:42:45 2007
***************
*** 55,67 ****
      <isClassAvailableEx class-name="org.jfree.xmlns.LibXmlInfo"
                        property-name="lib.xmlns.present">
        <test-classpath>
-         <pathelement location="${libdir}/${jaxp-jar-file}"/>
          <pathelement location="${libdir}/${libxml-jar-file}"/>
          <pathelement location="${libdir}/${jcommon-jar-file}"/>
          <pathelement location="${libdir}/${libloader-jar-file}"/>
        </test-classpath>
        <additional-tests>
-         <available file="${libdir}/${jaxp-jar-file}"/>
          <available file="${libdir}/${libxml-jar-file}"/>
          <available file="${libdir}/${jcommon-jar-file}"/>
        </additional-tests>
--- 55,65 ----
