--- vsprojects/liborcus-static-nozip/liborcus-static-nozip.vcproj	2012-09-07 06:00:35.000000000 +0200
+++ vsprojects/liborcus-static-nozip/liborcus-static-nozip.vcproj	2012-09-10 18:10:23.821600000 +0200
@@ -42,7 +42,7 @@
 				Name="VCCLCompilerTool"
 				Optimization="0"
 				AdditionalIncludeDirectories=""
-				PreprocessorDefinitions="__ORCUS_STATIC_LIB"
+				PreprocessorDefinitions="__ORCUS_STATIC_LIB;BOOST_ALL_NO_LIB"
 				MinimalRebuild="true"
 				BasicRuntimeChecks="3"
 				RuntimeLibrary="3"
@@ -107,7 +107,7 @@
 				Optimization="2"
 				EnableIntrinsicFunctions="true"
 				AdditionalIncludeDirectories=""
-				PreprocessorDefinitions="__ORCUS_STATIC_LIB"
+				PreprocessorDefinitions="__ORCUS_STATIC_LIB;BOOST_ALL_NO_LIB"
 				RuntimeLibrary="2"
 				EnableFunctionLevelLinking="true"
 				WarningLevel="3"
