--- source/common/ucasemap.cpp
+++ source/common/ucasemap.cpp
@@ -687,13 +687,13 @@
             if (change) {
                 ByteSinkUtil::appendTwoBytes(upper, sink);
                 if ((data & HAS_EITHER_DIALYTIKA) != 0) {
-                    sink.Append(u8"\u0308", 2);  // restore or add a dialytika
+                    sink.Append(reinterpret_cast<char const *>(u8"\u0308"), 2);  // restore or add a dialytika
                 }
                 if (addTonos) {
-                    sink.Append(u8"\u0301", 2);
+                    sink.Append(reinterpret_cast<char const *>(u8"\u0301"), 2);
                 }
                 while (numYpogegrammeni > 0) {
-                    sink.Append(u8"\u0399", 2);
+                    sink.Append(reinterpret_cast<char const *>(u8"\u0399"), 2);
                     --numYpogegrammeni;
                 }
             }
