--- misc/icu/source/io/ufile.c	2015-07-04 19:08:35.889168902 +0200
+++ misc/build/icu/source/io/ufile.c	2015-07-04 19:12:43.040185494 +0200
@@ -66,7 +66,9 @@
 #if U_PLATFORM_USES_ONLY_WIN32_API
     if (0 <= result->fFileno && result->fFileno <= 2) {
         /* stdin, stdout and stderr need to be special cased for Windows 98 */
-#if _MSC_VER >= 1400
+#if _MSC_VER >= 1900
+        result->fFile = __acrt_iob_func(_fileno(f));
+#elif _MSC_VER >= 1400
         result->fFile = &__iob_func()[_fileno(f)];
 #else
         result->fFile = &_iob[_fileno(f)];
