--- misc/icu/source/config/mh-cygwin-msvc
+++ misc/build/icu/source/config/mh-cygwin-msvc
@@ -43,10 +43,9 @@
 
 # /GF pools strings and places them into read-only memory
 # /EHsc enables exception handling
-# /Zc:wchar_t makes wchar_t a native type. Required for C++ ABI compatibility.
 # -D_CRT_SECURE_NO_DEPRECATE is needed to quiet warnings about using standard C functions.
-CFLAGS+=-GF -nologo
-CXXFLAGS+=-GF -nologo -EHsc -Zc:wchar_t
+CFLAGS+=-GF -nologo -Zc:wchar_t
+CXXFLAGS+=-GF -nologo -EHsc -Zc:wchar_t
 CPPFLAGS+=-D_CRT_SECURE_NO_DEPRECATE
 DEFS+=-DWIN32 -DCYGWINMSVC
 LDFLAGS+=-nologo

