--- misc/icu/source/common/putil.cpp
+++ misc/build/icu/source/common/putil.cpp
@@ -1111,8 +1111,16 @@
         static const time_t decemberSolstice=1198332540; /*2007-12-22 06:09 UT*/
 
         /* This probing will tell us when daylight savings occurs.  */
+#if U_PLATFORM_IS_DARWIN_BASED
+        struct tm *tmp;
+        tmp = localtime(&juneSolstice);
+        juneSol = *tmp;
+        tmp = localtime(&decemberSolstice);
+        decemberSol = *tmp;
+#else
         localtime_r(&juneSolstice, &juneSol);
         localtime_r(&decemberSolstice, &decemberSol);
+#endif
         if(decemberSol.tm_isdst > 0) {
           daylightType = U_DAYLIGHT_DECEMBER;
         } else if(juneSol.tm_isdst > 0) {

