--- source/io/ufile.cpp
+++ source/io/ufile.cpp
@@ -21,12 +21,6 @@
 */
 
 #include "unicode/platform.h"
-#if defined(__GNUC__) && !defined(__clang__) && defined(__STRICT_ANSI__)
-// g++, fileno isn't defined                  if     __STRICT_ANSI__ is defined.
-// clang fails to compile the <string> header unless __STRICT_ANSI__ is defined.
-// __GNUC__ is set by both gcc and clang.
-#undef __STRICT_ANSI__
-#endif
 
 #include "locmap.h"
 #include "unicode/ustdio.h"
