--- misc/vigra1.4.0/include/vigra/sized_int.hxx	2012-03-28 13:36:02.743956393 +0200
+++ misc/build/vigra1.4.0/include/vigra/sized_int.hxx	2012-03-28 13:35:44.956057143 +0200
@@ -80,8 +80,8 @@
 template<class LIST>
 struct SelectBiggestIntegerType
 {
-    enum { cursize = LIST::size, 
-           nextsize = SelectBiggestIntegerType<typename LIST::next>::size,
+    enum { cursize = static_cast< int >(LIST::size), 
+           nextsize = static_cast< int >(SelectBiggestIntegerType<typename LIST::next>::size),
            size = (cursize < nextsize) ? nextsize : cursize };
     typedef typename 
        IfBool<(cursize < nextsize), 
