--- src/lib/VSDXTheme.cpp
+++ src/lib/VSDXTheme.cpp
@@ -341,18 +341,25 @@ boost::optional<libvisio::Colour> libvisio::VSDXTheme::getThemeColour(unsigned v
     switch (value)
     {
     case 100:
+    case 200:
       return m_clrScheme.m_variationClrSchemeLst[variationIndex].m_varColor1;
     case 101:
+    case 201:
       return m_clrScheme.m_variationClrSchemeLst[variationIndex].m_varColor2;
     case 102:
+    case 202:
       return m_clrScheme.m_variationClrSchemeLst[variationIndex].m_varColor3;
     case 103:
+    case 203:
       return m_clrScheme.m_variationClrSchemeLst[variationIndex].m_varColor4;
     case 104:
+    case 204:
       return m_clrScheme.m_variationClrSchemeLst[variationIndex].m_varColor5;
     case 105:
+    case 205:
       return m_clrScheme.m_variationClrSchemeLst[variationIndex].m_varColor6;
     case 106:
+    case 206:
       return m_clrScheme.m_variationClrSchemeLst[variationIndex].m_varColor7;
     default:
       break;
