From 0250d3067f6d233263a9217e22c16daacf5cace3 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Caol=C3=A1n=20McNamara?= <caolanm@redhat.com>
Date: Mon, 23 Jan 2017 15:13:42 +0000
Subject: [PATCH] autoreconfonmacproblems

---
 configure.ac | 1 +
 1 file changed, 1 insertion(+)

diff --git a/configure.ac b/configure.ac
index 71360e6..5a8acb8 100644
--- a/configure.ac
+++ b/configure.ac
@@ -9,6 +9,7 @@ AC_INIT([hunspell],[1.6.0],[nemeth@numbertext.org])
 AC_CANONICAL_HOST
 AC_SUBST(XFAILED)
 
+AC_CONFIG_MACRO_DIR(m4)
 AM_INIT_AUTOMAKE(hunspell, 1.6.0)
 HUNSPELL_VERSION_MAJOR=`echo $VERSION | cut -d"." -f1`
 HUNSPELL_VERSION_MINOR=`echo $VERSION | cut -d"." -f2`
-- 
2.9.3

