--- misc/libexttextcat-3.0.1/config.sub	2011-09-20 17:14:22.000000000 +0200
+++ misc/build/libexttextcat-3.0.1/config.sub	2011-09-26 16:05:16.000000000 +0200
@@ -320,7 +320,7 @@
 	  basic_machine=$basic_machine-pc
 	  ;;
 	# Object if more than one company name word.
-	*-*-*)
+	*-*-*-*)
 		echo Invalid configuration \`$1\': machine \`$basic_machine\' not recognized 1>&2
 		exit 1
 		;;
@@ -1292,6 +1292,7 @@
 	      | -bosx* | -nextstep* | -cxux* | -aout* | -elf* | -oabi* \
 	      | -ptx* | -coff* | -ecoff* | -winnt* | -domain* | -vsta* \
 	      | -udi* | -eabi* | -lites* | -ieee* | -go32* | -aux* \
+	      | -androideabi \
 	      | -chorusos* | -chorusrdb* | -cegcc* \
 	      | -cygwin* | -pe* | -psos* | -moss* | -proelf* | -rtems* \
 	      | -mingw32* | -linux-gnu* | -linux-newlib* | -linux-uclibc* \
