--- misc/build/fontconfig-2.8.0/config.sub
+++ misc/build/fontconfig-2.8.0/config.sub
@@ -120,7 +120,7 @@
 # Here we must recognize all the valid KERNEL-OS combinations.
 maybe_os=`echo $1 | sed 's/^\(.*\)-\([^-]*-[^-]*\)$/\2/'`
 case $maybe_os in
-  nto-qnx* | linux-gnu* | linux-dietlibc | linux-newlib* | linux-uclibc* | \
+  nto-qnx* | linux-android* | linux-gnu* | linux-dietlibc | linux-newlib* | linux-uclibc* | \
   uclinux-uclibc* | uclinux-gnu* | kfreebsd*-gnu* | knetbsd*-gnu* | netbsd*-gnu* | \
   storm-chaos* | os2-emx* | rtmk-nova*)
     os=-$maybe_os
@@ -1272,7 +1272,7 @@
 	      | -ekkobsd* | -kfreebsd* | -freebsd* | -riscix* | -lynxos* \
 	      | -bosx* | -nextstep* | -cxux* | -aout* | -elf* | -oabi* \
 	      | -ptx* | -coff* | -ecoff* | -winnt* | -domain* | -vsta* \
-	      | -udi* | -eabi* | -lites* | -ieee* | -go32* | -aux* \
+	      | -udi* | -android* | -eabi* | -lites* | -ieee* | -go32* | -aux* \
 	      | -chorusos* | -chorusrdb* \
 	      | -cygwin* | -pe* | -psos* | -moss* | -proelf* | -rtems* \
 	      | -mingw32* | -linux-gnu* | -linux-newlib* | -linux-uclibc* \
--- misc/build/fontconfig-2.8.0/configure
+++ misc/build/fontconfig-2.8.0/configure
@@ -14368,7 +14368,7 @@
 			;;
 		yes)
 
-for ac_func in XML_SetDoctypeDeclHandler
+for ac_func in XML_SetDoctypeDeclHandler_UTF8
 do
 as_ac_var=`$as_echo "ac_cv_func_$ac_func" | $as_tr_sh`
 { $as_echo "$as_me:$LINENO: checking for $ac_func" >&5
@@ -14468,7 +14468,7 @@
 fi
 done
 
-			case "$ac_cv_func_XML_SetDoctypeDeclHandler" in
+			case "$ac_cv_func_XML_SetDoctypeDeclHandler_UTF8" in
 			yes)
 				HAVE_EXPAT=1
 
--- misc/build/fontconfig-2.8.0/configure
+++ misc/build/fontconfig-2.8.0/configure
@@ -8718,7 +8718,7 @@
 	*Sun\ F*)			# Sun Fortran 8.3
 	  tmp_sharedflag='-G' ;;
 	esac
-	archive_cmds='$CC '"$tmp_sharedflag""$tmp_addflag"' $libobjs $deplibs $compiler_flags ${wl}-soname $wl$soname -o $lib'
+	archive_cmds='$CC '"$tmp_sharedflag""$tmp_addflag"' $libobjs $deplibs $compiler_flags -o $lib'
 
         if test "x$supports_anon_versioning" = xyes; then
           archive_expsym_cmds='echo "{ global:" > $output_objdir/$libname.ver~
@@ -14073,7 +14073,7 @@
 		no)
 			;;
 		*)
-			EXPAT_LIBS="-L$expat_lib -lexpat"
+			EXPAT_LIBS="-L$expat_lib -lascii_expat_xmlparse -lexpat_xmltok"
 			;;
 		esac
 
--- misc/build/fontconfig-2.8.0/src/fcxml.c
+++ misc/build/fontconfig-2.8.0/src/fcxml.c
@@ -1123,6 +1123,7 @@
      * Have to swap the decimal point to match the current locale
      * if that locale doesn't use 0x2e
      */
+#ifndef __ANDROID__
     if ((dot = strchr (s, 0x2e)) &&
 	(locale_data = localeconv ()) &&
 	(locale_data->decimal_point[0] != 0x2e ||
@@ -1158,6 +1158,7 @@
 	}
     }
     else
+#endif
 	v = strtod (s, end);
     return v;
 }
