--- misc/xmlsec1-1.2.14/configure.in	Wed Jun 30 11:55:37 2010
+++ misc/build/xmlsec1-1.2.14/configure.in	Wed Jun 30 11:53:55 2010
@@ -231,7 +231,7 @@
 if test "z$LIBXML_FOUND" = "zno" ; then
     if test "z$with_libxml" != "zyes" ; then
 	AC_PATH_PROG([LIBXML_CONFIG], [$LIBXML_CONFIG], [],
-		     [$with_libxml/bin:$PATH])
+		     [$with_libxml${with_libxml:+/bin:}$PATH])
     fi
     AC_MSG_CHECKING([libxml2 $LIBXML_CONFIG ])
     if ! LIBXML_VERSION=`$LIBXML_CONFIG --version 2>/dev/null`; then
@@ -296,7 +296,7 @@
 if test "z$LIBXSLT_FOUND" = "zno" ; then
     if test "z$with_libxslt" != "zyes" ; then
 	AC_PATH_PROG([LIBXSLT_CONFIG], [$LIBXSLT_CONFIG], [],
-		     [$with_libxslt/bin:$PATH])
+		     [$with_libxslt${with_libxslt:+/bin:}:$PATH])
     fi
     AC_MSG_CHECKING(for libxslt libraries >= $LIBXSLT_MIN_VERSION) 
     if ! LIBXSLT_VERSION=`$LIBXSLT_CONFIG --version 2>/dev/null`; then
