--- misc/xmlsec1-1.2.6/configure	2009-09-18 17:19:00.000000000 +0200
+++ misc/build/xmlsec1-1.2.6/configure	2009-09-18 17:18:43.000000000 +0200
@@ -21749,6 +21749,10 @@
 ac_compiler_gnu=$ac_cv_c_compiler_gnu
 
 
+case $host_os in
+mingw*)
+;;
+*)
 echo "$as_me:$LINENO: checking for shl_load" >&5
 echo $ECHO_N "checking for shl_load... $ECHO_C" >&6
 if test "${ac_cv_func_shl_load+set}" = set; then
@@ -22299,7 +22303,8 @@
 
 
 fi
-
+;;
+esac
 
 if test x"$libltdl_cv_func_dlopen" = xyes || test x"$libltdl_cv_lib_dl_dlopen" = xyes
 then
