--- misc/xmlsec1-1.2.14/configure	2010-09-29 22:51:25.000000000 +0200
+++ misc/build/xmlsec1-1.2.14/configure	2010-09-29 22:50:42.000000000 +0200
@@ -13325,7 +13325,7 @@
 
     if test "z$NSS_INCLUDES_FOUND" = "zyes" -a "z$NSS_LIBS_FOUND" = "zyes" ; then
        OLD_CPPFLAGS=$CPPFLAGS
-       CPPFLAGS="$NSS_CFLAGS"
+       CPPFLAGS="$NSS_CFLAGS $NSPR_CFLAGS"
        cat confdefs.h - <<_ACEOF >conftest.$ac_ext
 /* end confdefs.h.  */
 
