--- misc/xmlsec1-1.2.14.orig/src/nss/crypto.c	2009-09-10 07:06:17.000000000 -0400
+++ misc/build/xmlsec1-1.2.14/src/nss/crypto.c	2009-09-10 07:08:24.000000000 -0400
@@ -136,6 +136,7 @@
     /**
      * High level routines form xmlsec command line utility
      */ 
+#if 0
     gXmlSecNssFunctions->cryptoAppInit 			= xmlSecNssAppInit;
     gXmlSecNssFunctions->cryptoAppShutdown 		= xmlSecNssAppShutdown;
     gXmlSecNssFunctions->cryptoAppDefaultKeysMngrInit 	= xmlSecNssAppDefaultKeysMngrInit;
@@ -153,6 +154,25 @@
     gXmlSecNssFunctions->cryptoAppKeyLoad 		= xmlSecNssAppKeyLoad; 
     gXmlSecNssFunctions->cryptoAppKeyLoadMemory		= xmlSecNssAppKeyLoadMemory; 
     gXmlSecNssFunctions->cryptoAppDefaultPwdCallback	= (void*)xmlSecNssAppGetDefaultPwdCallback();
+#else
+    gXmlSecNssFunctions->cryptoAppInit                         = NULL ;
+    gXmlSecNssFunctions->cryptoAppShutdown             = NULL ;
+    gXmlSecNssFunctions->cryptoAppDefaultKeysMngrInit  = NULL ;
+    gXmlSecNssFunctions->cryptoAppDefaultKeysMngrAdoptKey      = NULL ;
+    gXmlSecNssFunctions->cryptoAppDefaultKeysMngrLoad  = NULL ;
+    gXmlSecNssFunctions->cryptoAppDefaultKeysMngrSave  = NULL ;
+#ifndef XMLSEC_NO_X509
+    gXmlSecNssFunctions->cryptoAppKeysMngrCertLoad     = NULL ;
+    gXmlSecNssFunctions->cryptoAppKeysMngrCertLoadMemory= NULL ;
+    gXmlSecNssFunctions->cryptoAppPkcs12Load           = NULL ; 
+    gXmlSecNssFunctions->cryptoAppPkcs12LoadMemory     = NULL ; 
+    gXmlSecNssFunctions->cryptoAppKeyCertLoad          = NULL ;
+    gXmlSecNssFunctions->cryptoAppKeyCertLoadMemory    = NULL ;
+#endif /* XMLSEC_NO_X509 */
+    gXmlSecNssFunctions->cryptoAppKeyLoad              = NULL ; 
+    gXmlSecNssFunctions->cryptoAppKeyLoadMemory                = NULL ; 
+    gXmlSecNssFunctions->cryptoAppDefaultPwdCallback   = (void*)NULL ;
+#endif
 
     return(gXmlSecNssFunctions);
 }
