--- misc/libgsf-1.14.19/gsf/gsf-config.h	2011-03-23 16:18:27.163747600 +0100
+++ misc/build/libgsf-1.14.19/gsf/gsf-config.h	2011-03-23 16:01:26.598374600 +0100
@@ -1 +1,10 @@
-dummy
+#ifndef GSF_CONFIG_H
+#define GSF_CONFIG_H
+
+#define GETTEXT_PACKAGE "gsf"
+
+typedef unsigned short mode_t;
+
+#define S_ISREG(x) ((x & _S_IFREG)!=0)
+
+#endif
--- misc/libgsf-1.14.19/gsf/makefile.mk	2011-03-23 16:18:27.070742300 +0100
+++ misc/build/libgsf-1.14.19/gsf/makefile.mk	2011-03-23 16:17:29.477448100 +0100
@@ -1 +1,121 @@
-dummy
+#*************************************************************************
+#
+# DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+# 
+# Copyright 2000, 2010 Oracle and/or its affiliates.
+#
+# OpenOffice.org - a multi-platform office productivity suite
+#
+# This file is part of OpenOffice.org.
+#
+# OpenOffice.org is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Lesser General Public License version 3
+# only, as published by the Free Software Foundation.
+#
+# OpenOffice.org is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Lesser General Public License version 3 for more details
+# (a copy is included in the LICENSE file that accompanied this code).
+#
+# You should have received a copy of the GNU Lesser General Public License
+# version 3 along with OpenOffice.org.  If not, see
+# <http://www.openoffice.org/license.html>
+# for a copy of the LGPLv3 License.
+#
+#*************************************************************************
+
+PRJ=..$/..$/..$/..$/..
+PRJINC=..
+PRJNAME=libgsf
+TARGET=gsf-1
+
+VISIBILITY_HIDDEN=TRUE
+EXTERNAL_WARNINGS_NOT_ERRORS=TRUE
+UWINAPILIB=
+
+# --- Settings ----------------------------------
+
+.INCLUDE : settings.mk
+
+CFLAGS+= -I.. -I$(SOLARINCDIR)$/external$/glib-2.0 -I$(SOLARINCDIR)$/external$/zlib -I$(SOLARINCDIR)$/external
+
+# --- Files -------------------------------------
+
+SLOFILES=\
+    $(SLO)$/gsf-blob.obj \
+    $(SLO)$/gsf-input-memory.obj \
+    $(SLO)$/gsf-output-gio.obj \
+    $(SLO)$/gsf-clip-data.obj \
+    $(SLO)$/gsf-input-proxy.obj \
+    $(SLO)$/gsf-output-gzip.obj \
+    $(SLO)$/gsf-doc-meta-data.obj \
+    $(SLO)$/gsf-input-stdio.obj \
+    $(SLO)$/gsf-output-iconv.obj \
+    $(SLO)$/gsf-docprop-vector.obj \
+    $(SLO)$/gsf-input-textline.obj \
+    $(SLO)$/gsf-output-iochannel.obj \
+    $(SLO)$/gsf-infile-msole.obj \
+    $(SLO)$/gsf-input.obj \
+    $(SLO)$/gsf-output-memory.obj \
+    $(SLO)$/gsf-infile-msvba.obj \
+    $(SLO)$/gsf-libxml.obj \
+    $(SLO)$/gsf-output-stdio.obj \
+    $(SLO)$/gsf-infile-stdio.obj \
+    $(SLO)$/gsf-msole-utils.obj \
+    $(SLO)$/gsf-output.obj \
+    $(SLO)$/gsf-infile-tar.obj \
+    $(SLO)$/gsf-open-pkg-utils.obj \
+    $(SLO)$/gsf-shared-memory.obj \
+    $(SLO)$/gsf-infile-zip.obj \
+    $(SLO)$/gsf-opendoc-utils.obj \
+    $(SLO)$/gsf-structured-blob.obj \
+    $(SLO)$/gsf-infile.obj \
+    $(SLO)$/gsf-outfile-msole.obj \
+    $(SLO)$/gsf-timestamp.obj \
+    $(SLO)$/gsf-input-bzip.obj \
+    $(SLO)$/gsf-outfile-stdio.obj \
+    $(SLO)$/gsf-utils.obj \
+    $(SLO)$/gsf-input-gio.obj \
+    $(SLO)$/gsf-outfile-zip.obj \
+    $(SLO)$/gsf-zip-utils.obj \
+    $(SLO)$/gsf-input-gzip.obj \
+    $(SLO)$/gsf-outfile.obj \
+    $(SLO)$/version.obj \
+    $(SLO)$/gsf-input-http.obj \
+    $(SLO)$/gsf-output-bzip.obj \
+    $(SLO)$/gsf-input-iochannel.obj \
+    $(SLO)$/gsf-output-csv.obj
+
+# --- Library -----------------------------------
+
+SHL1TARGET=	$(TARGET)
+SHL1OBJS=$(SLOFILES)
+SHL1STDLIBS=\
+    libxml2.lib \
+    gio-2.0.lib \
+    gmodule-2.0.lib \
+    gobject-2.0.lib \
+    glib-2.0.lib \
+    intl.lib \
+    zlib.lib \
+	Advapi32.lib
+
+
+SHL1IMPLIB=	i$(TARGET)
+SHL1DEF=	$(MISC)$/$(SHL1TARGET).def
+
+DEF1NAME=	$(SHL1TARGET)
+DEF1DEPN=	$(MISC)$/$(SHL1TARGET).flt \
+			$(SLB)$/$(TARGET).lib
+DEFLIB1NAME=$(TARGET)
+
+# --- Targets ----------------------------------
+
+.INCLUDE : target.mk
+
+# --- filter file ------------------------------
+
+$(MISC)$/$(SHL1TARGET).flt: makefile.mk
+    @echo CLEAR_THE_FILE	> $@
+
--- misc/libgsf-1.14.19/gsf/gsf-infile-zip.c
+++ misc/build/libgsf-1.14.19/gsf/gsf-infile-zip.c
@@ -30,6 +30,10 @@
 
 #include <string.h>
 #include <zlib.h>
+
+#if defined(Z_PREFIX) && defined(crc32)
+#undef crc32
+#endif
 
 #undef G_LOG_DOMAIN
 #define G_LOG_DOMAIN "libgsf:zip"

--- misc/libgsf-1.14.19/gsf/gsf-outfile-zip.c
+++ misc/build/libgsf-1.14.19/gsf/gsf-outfile-zip.c
@@ -29,6 +29,11 @@
 #include <string.h>
 #include <time.h>
 #include <zlib.h>
+
+#if defined(Z_PREFIX) && defined(crc32)
+#define CRC32_WAS_DEFINED_AS_Z_CRC32
+#undef crc32
+#endif
 
 #undef G_LOG_DOMAIN
 #define G_LOG_DOMAIN "libgsf:zip"
@@ -345,7 +345,11 @@
 	zip_header_write (zip);
 	zip->writing = TRUE;
 	zip->root->writing = TRUE;
+#if defined(CRC32_WAS_DEFINED_AS_Z_CRC32)
+	dirent->crc32 = z_crc32 (0L, Z_NULL, 0);
+#else
 	dirent->crc32 = crc32 (0L, Z_NULL, 0);
+#endif
 	if (zip->compression_method == GSF_ZIP_DEFLATED) {
 		if (!zip->stream) {
 			zip->stream = g_new0 (z_stream, 1);
@@ -544,7 +544,11 @@
 			return FALSE;
 		dirent->csize += num_bytes;
 	}
+#if defined(CRC32_WAS_DEFINED_AS_Z_CRC32)
+	dirent->crc32 = z_crc32 (dirent->crc32, data, num_bytes);
+#else
 	dirent->crc32 = crc32 (dirent->crc32, data, num_bytes);
+#endif
 	dirent->usize += num_bytes;
 
 	return TRUE;
