diff -ur boost.orig/boost/detail/workaround.hpp boost/boost/detail/workaround.hpp
--- a/b/boost.orig/boost/detail/workaround.hpp	2015-12-13 13:49:42.000000000 +0100
+++ a/b/boost/boost/detail/workaround.hpp	2016-01-29 15:07:43.427781248 +0100
@@ -5,6 +5,12 @@
 #ifndef WORKAROUND_DWA2002126_HPP
 #define WORKAROUND_DWA2002126_HPP
 
+#ifndef BOOST_GCC_VERSION
+#define BOOST_GCC_VERSION_WORKAROUND_GUARD 1
+#else
+#define BOOST_GCC_VERSION_WORKAROUND_GUARD 0
+#endif
+
 #include <boost/config/workaround.hpp>
 
 #endif // WORKAROUND_DWA2002126_HPP
