diff -ru boost.orig/boost/optional/detail/optional_aligned_storage.hpp boost/boost/optional/detail/optional_aligned_storage.hpp
--- foo/misc/boost.orig/boost/optional/detail/optional_aligned_storage.hpp	2015-07-18 11:27:36.168127029 +0200
+++ foo/misc/boost/boost/optional/detail/optional_aligned_storage.hpp	2015-07-18 20:36:13.777997833 +0200
@@ -14,6 +14,8 @@
 #ifndef BOOST_OPTIONAL_OPTIONAL_DETAIL_OPTIONAL_ALIGNED_STORAGE_AJK_12FEB2016_HPP
 #define BOOST_OPTIONAL_OPTIONAL_DETAIL_OPTIONAL_ALIGNED_STORAGE_AJK_12FEB2016_HPP
 
+#include <string.h>
+
 namespace boost {
 
 namespace optional_detail {
@@ -39,6 +41,11 @@ class aligned_storage
 
   public:
 
+    aligned_storage()
+    {
+      memset(&dummy_, 0, sizeof(dummy_));
+    }
+
 #if defined(BOOST_OPTIONAL_DETAIL_USE_ATTRIBUTE_MAY_ALIAS)
     void const* address() const { return &dummy_; }
     void      * address()       { return &dummy_; }
diff -ru boost.orig/boost/spirit/home/classic/core/primitives/impl/numerics.ipp boost/boost/spirit/home/classic/core/primitives/impl/numerics.ipp
--- foo/misc/boost.orig/boost/spirit/home/classic/core/primitives/impl/numerics.ipp	2015-07-18 11:27:36.169127029 +0200
+++ foo/misc/boost/boost/spirit/home/classic/core/primitives/impl/numerics.ipp	2015-07-18 20:34:32.110998976 +0200
@@ -218,6 +218,19 @@
                 }
             }
         };
+        template <int Radix>
+        struct negative_accumulate<unsigned char, Radix>
+        {
+            //  Use this accumulator if number is negative
+            static bool add(unsigned char& n, unsigned digit)
+            {
+                n *= Radix;
+                if (n < digit)
+                    return false;
+                n -= digit;
+                return true;
+            }
+        };
 
         template <int MaxDigits>
         inline bool allow_more_digits(std::size_t i)
diff -ru boost.orig/boost/utility/compare_pointees.hpp boost/boost/utility/compare_pointees.hpp
--- foo/misc/boost.orig/boost/utility/compare_pointees.hpp	2015-02-27 07:26:11.000000000 +0100
+++ foo/misc/boost/boost/utility/compare_pointees.hpp	2015-07-18 20:20:41.388988458 +0200
@@ -29,7 +29,11 @@
 inline
 bool equal_pointees ( OptionalPointee const& x, OptionalPointee const& y )
 {
-  return (!x) != (!y) ? false : ( !x ? true : (*x) == (*y) ) ;
+  if (!x && !y)
+    return true;
+  if (!x || !y)
+    return false;
+  return (*x) == (*y);
 }
 
 template<class OptionalPointee>
