diff -ru boost.orig/boost/type_index/type_index_facade.hpp boost/boost/type_index/type_index_facade.hpp
--- foo/misc/boost.orig/boost/type_index/type_index_facade.hpp	2015-07-19 11:39:38.476942225 +0200
+++ foo/misc/boost/boost/type_index/type_index_facade.hpp	2015-07-19 11:41:13.032941163 +0200
@@ -105,8 +105,8 @@
     /// \return Hash code of a type. By default hashes types by raw_name().
     /// \note <boost/functional/hash.hpp> has to be included if this function is used.
     inline std::size_t hash_code() const BOOST_NOEXCEPT {
-        const char* const name = derived().raw_name();
-        return boost::hash_range(name, name + std::strlen(name));
+        const char* const name_ = derived().raw_name();
+        return boost::hash_range(name_, name_ + std::strlen(name_));
     }
 
 #if defined(BOOST_TYPE_INDEX_DOXYGEN_INVOKED)

