diff -ru boost.orig/boost/random/generate_canonical.hpp boost/boost/random/generate_canonical.hpp
--- a/b/boost.orig/boost/random/generate_canonical.hpp	2015-02-27 18:44:45.000000000 +0100
+++ a/b/boost/boost/random/generate_canonical.hpp	2015-07-18 21:55:44.152972440 +0200
@@ -32,7 +32,6 @@
 RealType generate_canonical_impl(URNG& g, boost::mpl::true_ /*is_integral*/)
 {
     using std::pow;
-    typedef typename URNG::result_type base_result;
     std::size_t digits = std::numeric_limits<RealType>::digits;
     RealType R = RealType((g.max)()) - RealType((g.min)()) + 1;
     RealType mult = R;
