diff -ru boost.org/boost/property_tree/detail/json_parser/standard_callbacks.hpp boost/boost/property_tree/detail/json_parser/standard_callbacks.hpp
--- foo/misc/boost.org/boost/property_tree/detail/json_parser/standard_callbacks.hpp	2015-07-07 14:20:48.000000000 +0200
+++ foo/misc/boost/boost/property_tree/detail/json_parser/standard_callbacks.hpp	2015-07-22 08:35:07.764263463 +0200
@@ -128,7 +128,7 @@
                 stack.pop_back();
                 return new_tree();
             }
-            assert(false);
+	    std::abort();
         }
         string& new_value() {
             if (stack.empty()) return new_tree().data();

