--- a/b/boost/boost/config/workaround.hpp
+++ a/b/boost/boost/config/workaround.hpp
@@ -237,6 +237,13 @@
 #else
 #define BOOST_INTEL_WORKAROUND_GUARD 0
 #endif
+
+#ifndef __clang_major__
+#define __clang_major___WORKAROUND_GUARD 1
+#else
+#define __clang_major___WORKAROUND_GUARD 0
+#endif
+
 // Always define to zero, if it's used it'll be defined my MPL:
 #define BOOST_MPL_CFG_GCC_WORKAROUND_GUARD 0
 
