--- examples/empbuild/empbuild.e
+++ examples/empbuild/empbuild.e
@@ -65,6 +65,7 @@
  *
  **************************************/
 TEXT	cmd [140];
+int status;
 
 if (argc > 1)
     strcpy (Db_name, argv[1]);
@@ -95,9 +96,9 @@
 
 printf ("Turning forced writes off\n");
 sprintf (cmd, "gfix -write async %s", Db_name);
-if (system (cmd))
+if ((status = system (cmd)))
     {
-    printf ("Couldn't turn forced writes off\n");
+    printf ("Couldn't turn forced writes off (%d)\n", status);
     exit (FINI_ERROR);
     }
 
