Index: hyphen.c
===================================================================
RCS file: /cvs/whiteboard/lingucomponent/source/hyphenator/altlinuxhyph/libhnj/hyphen.c,v
retrieving revision 1.8
retrieving revision 1.8.32.1
diff -u -u -r1.8 -r1.8.32.1
--- hyphen.c	3 Aug 2007 12:30:49 -0000	1.8
+++ misc/build/hyphen-2.3/hyphen.c	2 Jan 2008 16:08:20 -0000	1.8.32.1
@@ -300,7 +300,7 @@
             } else {
                 hnj_strchomp(repl + 1);
                 replindex = 0;
-                replcut = strlen(buf);
+                replcut = (signed char) strlen(buf);
             }
             repl = hnj_strdup(repl + 1);
           }
@@ -333,10 +333,10 @@
                     if ((((unsigned char) word[pc]) >> 6) != 2) pu++;
                     if ((ps < 0) && (replindex == pu)) {
                         ps = replindex;
-                        replindex = pc;
+                        replindex = (signed char) pc;
                     }
                     if ((ps >= 0) && ((pu - ps) == replcut)) {
-                        replcut = (pc - replindex);
+                        replcut = (signed char) (pc - replindex);
                         break;
                     }
                 }
@@ -353,7 +353,7 @@
 	  dict->states[state_num].repl = repl;
 	  dict->states[state_num].replindex = replindex;
           if (!replcut) {
-            dict->states[state_num].replcut = strlen(word);
+            dict->states[state_num].replcut = (signed char) strlen(word);
           } else {
             dict->states[state_num].replcut = replcut;
           }
*** misc/build/hyphen-2.3/makefile.mk	2008-02-18 10:30:28.000000000 +0100
--- misc/build/hyphen-2.3/makefile.mk	2008-02-18 10:27:19.000000000 +0100
***************
*** 1 ****
! dummy
--- 1,58 ----
! #*************************************************************************
! #
! #   $RCSfile: hyphen-2.3.patch,v $
! #
! #   $Revision: 1.2 $
! #
! #   last change: $Author: rene $ $Date: 2008-02-18 09:42:59 $
! #
! #*  The Contents of this file are made available subject to
! #*  the terms of GNU Lesser General Public License Version 2.1.
! #*
! #*
! #*    GNU Lesser General Public License Version 2.1
! #*    =============================================
! #*    Copyright 2005 by Sun Microsystems, Inc.
! #*    901 San Antonio Road, Palo Alto, CA 94303, USA
! #*
! #*    This library is free software; you can redistribute it and/or
! #*    modify it under the terms of the GNU Lesser General Public
! #*    License version 2.1, as published by the Free Software Foundation.
! #*
! #*    This library is distributed in the hope that it will be useful,
! #*    but WITHOUT ANY WARRANTY; without even the implied warranty of
! #*    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
! #*    Lesser General Public License for more details.
! #*
! #*    You should have received a copy of the GNU Lesser General Public
! #*    License along with this library; if not, write to the Free Software
! #*    Foundation, Inc., 59 Temple Place, Suite 330, Boston,
! #*    MA  02111-1307  USA
! #*
! #*************************************************************************
! 
! PRJ = ..$/..$/..$/..
! 
! PRJNAME = hyphen
! TARGET  = hyphen
! CFLAGSCALL=gsd
! 
! USE_DEFFILE=TRUE
! EXTERNAL_WARNINGS_NOT_ERRORS := TRUE
! 
! .INCLUDE : settings.mk
! 
! # --- Files --------------------------------------------------------
! 
! # !! not to be compiled because those belong to a stand alone programs: !!
! #        $(SLO)$/createfp.obj\
! #        $(SLO)$/testtextcat.obj
! 
! OBJFILES=   \
!         $(OBJ)$/hyphen.obj\
!         $(OBJ)$/hnjalloc.obj\
!         $(OBJ)$/csutil.obj
! 
! # --- Targets ------------------------------------------------------
! 
! .INCLUDE : target.mk
