--- misc/hyphen-2.3.1/Makefile.am	Tue Feb 19 01:58:34 2008
+++ misc/build/hyphen-2.3.1/Makefile.am	Wed May  7 10:25:03 2008
@@ -24,7 +24,7 @@
 
 hyphen.us:
 	cp -f $(srcdir)/hyphen.tex hyphen.us
-	patch < $(srcdir)/hyphen.patch
+	$(GNUPATCH) < $(srcdir)/hyphen.patch
 
 hyph_en_US.dic: hyphen.us
 	perl $(srcdir)/substrings.pl hyphen.us hyph_en_US.dic ISO8859-1 > /dev/null
--- misc/hyphen-2.3.1/Makefile.in	Tue Feb 19 01:58:38 2008
+++ misc/build/hyphen-2.3.1/Makefile.in	Wed May  7 10:25:26 2008
@@ -795,7 +795,7 @@
 
 hyphen.us:
 	cp -f $(srcdir)/hyphen.tex hyphen.us
-	patch < $(srcdir)/hyphen.patch
+	$(GNUPATCH) < $(srcdir)/hyphen.patch
 
 hyph_en_US.dic: hyphen.us
 	perl $(srcdir)/substrings.pl hyphen.us hyph_en_US.dic ISO8859-1 > /dev/null
--- misc/hyphen-2.3.1/csutil.c	Sun Jun  1 02:04:00 2003
+++ misc/build/hyphen-2.3.1/csutil.c	Wed May  7 10:23:59 2008
@@ -3493,7 +3493,7 @@
 };
 
 
-struct enc_entry encds[] = {
+static struct enc_entry encds[] = {
 {"ISO8859-1",iso1_tbl},
 {"ISO8859-2",iso2_tbl},
 {"ISO8859-3",iso3_tbl},
--- misc/hyphen-2.3.1/hyphen.c	Tue Feb 19 00:58:09 2008
+++ misc/build/hyphen-2.3.1/hyphen.c	Wed May  7 10:23:59 2008
@@ -298,7 +298,7 @@
             } else {
                 hnj_strchomp(repl + 1);
                 replindex = 0;
-                replcut = strlen(buf);
+                replcut = (signed char) strlen(buf);
             }
             repl = hnj_strdup(repl + 1);
           }
@@ -331,10 +331,10 @@
                     if ((((unsigned char) word[pc]) >> 6) != 2) pu++;
                     if ((ps < 0) && (replindex == pu)) {
                         ps = replindex;
-                        replindex = pc;
+                        replindex = (signed char) pc;
                     }
                     if ((ps >= 0) && ((pu - ps) == replcut)) {
-                        replcut = (pc - replindex);
+                        replcut = (signed char) (pc - replindex);
                         break;
                     }
                 }
@@ -351,7 +351,7 @@
 	  dict->states[state_num].repl = repl;
 	  dict->states[state_num].replindex = replindex;
           if (!replcut) {
-            dict->states[state_num].replcut = strlen(word);
+            dict->states[state_num].replcut = (signed char) strlen(word);
           } else {
             dict->states[state_num].replcut = replcut;
           }
--- misc/hyphen-2.3.1/makefile.mk	Wed May  7 10:27:43 2008
+++ misc/build/hyphen-2.3.1/makefile.mk	Wed May  7 10:27:33 2008
@@ -1,1 +1,63 @@
-dummy
+#*************************************************************************
+#
+# DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+# 
+# Copyright 2008 by Sun Microsystems, Inc.
+#
+# OpenOffice.org - a multi-platform office productivity suite
+#
+# $RCSfile: hyphen-2.3.patch,v $
+#
+# $Revision: 1.8 $
+#
+# This file is part of OpenOffice.org.
+#
+# OpenOffice.org is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Lesser General Public License version 3
+# only, as published by the Free Software Foundation.
+#
+# OpenOffice.org is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Lesser General Public License version 3 for more details
+# (a copy is included in the LICENSE file that accompanied this code).
+#
+# You should have received a copy of the GNU Lesser General Public License
+# version 3 along with OpenOffice.org.  If not, see
+# <http://www.openoffice.org/license.html>
+# for a copy of the LGPLv3 License.
+#
+#*************************************************************************
+
+PRJ = ..$/..$/..$/..
+
+PRJNAME = hyphen
+TARGET  = hyphen
+CFLAGSCALL=gsd
+
+USE_DEFFILE=TRUE
+EXTERNAL_WARNINGS_NOT_ERRORS := TRUE
+
+.INCLUDE : settings.mk
+
+# --- Files --------------------------------------------------------
+
+# !! not to be compiled because those belong to a stand alone programs: !!
+#        $(SLO)$/createfp.obj\
+#        $(SLO)$/testtextcat.obj
+
+SLOFILES=   \
+        $(SLO)$/hyphen.obj\
+        $(SLO)$/hnjalloc.obj\
+        $(SLO)$/csutil.obj
+
+# --- Targets ------------------------------------------------------
+
+ALL: hyph_en_US.dic ALLTAR
+
+.INCLUDE : target.mk
+
+hyph_en_US.dic : hyphen.tex
+	$(COPY) hyphen.tex hyphen.us
+	$(GNUPATCH) < hyphen.patch
+	$(PERL) substrings.pl hyphen.us hyph_en_US.dic ISO8859.-1 
