*** misc/hyphen-2.3.1/hyphen.c	Thu Nov 22 09:51:20 2007
--- misc/build/hyphen-2.3.1/hyphen.c	Mon Feb 18 10:46:20 2008
***************
*** 298,304 ****
              } else {
                  hnj_strchomp(repl + 1);
                  replindex = 0;
!                 replcut = strlen(buf);
              }
              repl = hnj_strdup(repl + 1);
            }
--- 298,304 ----
              } else {
                  hnj_strchomp(repl + 1);
                  replindex = 0;
!                 replcut = (signed char) strlen(buf);
              }
              repl = hnj_strdup(repl + 1);
            }
***************
*** 331,340 ****
                      if ((((unsigned char) word[pc]) >> 6) != 2) pu++;
                      if ((ps < 0) && (replindex == pu)) {
                          ps = replindex;
!                         replindex = pc;
                      }
                      if ((ps >= 0) && ((pu - ps) == replcut)) {
!                         replcut = (pc - replindex);
                          break;
                      }
                  }
--- 331,340 ----
                      if ((((unsigned char) word[pc]) >> 6) != 2) pu++;
                      if ((ps < 0) && (replindex == pu)) {
                          ps = replindex;
!                         replindex = (signed char) pc;
                      }
                      if ((ps >= 0) && ((pu - ps) == replcut)) {
!                         replcut = (signed char) (pc - replindex);
                          break;
                      }
                  }
***************
*** 351,357 ****
  	  dict->states[state_num].repl = repl;
  	  dict->states[state_num].replindex = replindex;
            if (!replcut) {
!             dict->states[state_num].replcut = strlen(word);
            } else {
              dict->states[state_num].replcut = replcut;
            }
--- 351,357 ----
  	  dict->states[state_num].repl = repl;
  	  dict->states[state_num].replindex = replindex;
            if (!replcut) {
!             dict->states[state_num].replcut = (signed char) strlen(word);
            } else {
              dict->states[state_num].replcut = replcut;
            }
*** misc/hyphen-2.3.1/makefile.mk	Mon Feb 18 11:06:40 2008
--- misc/build/hyphen-2.3.1/makefile.mk	Mon Feb 18 11:05:49 2008
***************
*** 1 ****
! dummy
--- 1,65 ----
! #*************************************************************************
! #
! #   $RCSfile: hyphen-2.3.patch,v $
! #
! #   $Revision: 1.5 $
! #
! #   last change: $Author: rene $ $Date: 2008-04-11 08:55:46 $
! #
! #*  The Contents of this file are made available subject to
! #*  the terms of GNU Lesser General Public License Version 2.1.
! #*
! #*
! #*    GNU Lesser General Public License Version 2.1
! #*    =============================================
! #*    Copyright 2005 by Sun Microsystems, Inc.
! #*    901 San Antonio Road, Palo Alto, CA 94303, USA
! #*
! #*    This library is free software; you can redistribute it and/or
! #*    modify it under the terms of the GNU Lesser General Public
! #*    License version 2.1, as published by the Free Software Foundation.
! #*
! #*    This library is distributed in the hope that it will be useful,
! #*    but WITHOUT ANY WARRANTY; without even the implied warranty of
! #*    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
! #*    Lesser General Public License for more details.
! #*
! #*    You should have received a copy of the GNU Lesser General Public
! #*    License along with this library; if not, write to the Free Software
! #*    Foundation, Inc., 59 Temple Place, Suite 330, Boston,
! #*    MA  02111-1307  USA
! #*
! #*************************************************************************
! 
! PRJ = ..$/..$/..$/..
! 
! PRJNAME = hyphen
! TARGET  = hyphen
! CFLAGSCALL=gsd
! 
! USE_DEFFILE=TRUE
! EXTERNAL_WARNINGS_NOT_ERRORS := TRUE
! 
! .INCLUDE : settings.mk
! 
! # --- Files --------------------------------------------------------
! 
! # !! not to be compiled because those belong to a stand alone programs: !!
! #        $(SLO)$/createfp.obj\
! #        $(SLO)$/testtextcat.obj
! 
! OBJFILES=   \
!         $(OBJ)$/hyphen.obj\
!         $(OBJ)$/hnjalloc.obj\
!         $(OBJ)$/csutil.obj
! 
! # --- Targets ------------------------------------------------------
! 
! ALL: hyph_en_US.dic ALLTAR
! 
! .INCLUDE : target.mk
! 
! hyph_en_US.dic : hyphen.tex
! 	cp hyphen.tex hyphen.us
! 	patch < hyphen.patch
! 	perl substrings.pl hyphen.us hyph_en_US.dic ISO8859.-1 
diff -urNad hyphen-2.3.1~/csutil.c hyphen-2.3.1/csutil.c
--- misc/build/hyphen-2.3.1~/csutil.c	2003-06-01 02:04:00.000000000 +0200
+++ misc/build/hyphen-2.3.1/csutil.c	2008-04-06 23:43:46.000000000 +0200
@@ -3493,7 +3493,7 @@
 };
 
 
-struct enc_entry encds[] = {
+static struct enc_entry encds[] = {
 {"ISO8859-1",iso1_tbl},
 {"ISO8859-2",iso2_tbl},
 {"ISO8859-3",iso3_tbl},
