--- a/core/src/CreateBarcode.cpp	2026-03-03 14:41:48.255770172 +0100
+++ b/core/src/CreateBarcode.cpp	2026-03-03 14:42:12.911576522 +0100
@@ -45,6 +45,10 @@
 	// structured_append (idx, cnt, ID)

 	mutable unique_zint_symbol zint;
+
+#ifndef __cpp_aggregate_paren_init // make XCode 14.x happy
+	Data(BarcodeFormat format, std::string options) : format(format), options(std::move(options)) {}
+#endif
 };

 // TODO: check return type
