--- misc/hsqldb/src/org/hsqldb/Token.java	2013-06-13 10:04:28.975934479 +0200
+++ misc/build/hsqldb/src/org/hsqldb/Token.java	2013-06-13 10:06:09.915931966 +0200
@@ -819,7 +819,7 @@
             Token.T_EXTRACT, Token.T_FOR, Token.T_FROM, Token.T_GROUP,
             Token.T_HAVING, Token.T_IF, Token.T_INTO, Token.T_IFNULL,
             Token.T_IS, Token.T_IN, Token.T_INTERSECT, Token.T_JOIN,
-            Token.T_INNER, Token.T_LEADING, Token.T_LIKE, Token.T_MAX,
+            Token.T_INNER, Token.T_LEADING, Token.T_LIKE, Token.T_LIMIT, Token.T_MAX,
             Token.T_MIN, Token.T_NEXT, Token.T_NULLIF, Token.T_NOT,
             Token.T_NVL, Token.T_MINUS, Token.T_ON, Token.T_ORDER, Token.T_OR,
             Token.T_OUTER, Token.T_POSITION, Token.T_PRIMARY, Token.T_SELECT,
