--- src/shared/CLucene/util/Misc.cpp	2012-02-17 15:33:09.217891100 +0000
+++ src/shared/CLucene/util/Misc.cpp	2012-02-17 15:38:10.836597500 +0000
@@ -34,7 +34,11 @@
 #include "CLucene/util/dirent.h" //if we have dirent, then the native one will be used
 
 //for zlib...
-#include "zlib.h"
+#ifdef SYSTEM_ZLIB
+#include <zlib.h>
+#else
+#include <external/zlib/zlib.h>
+#endif
 #if defined(MSDOS) || defined(OS2) || defined(WIN32) || defined(__CYGWIN__)
 #  include <fcntl.h>
 #  include <io.h>
