--- src/shared/CLucene/LuceneThreads.h
+++ src/shared/CLucene/LuceneThreads.h
@@ -7,6 +7,7 @@
 #ifndef _LuceneThreads_h
 #define  _LuceneThreads_h
 
+#include "CLucene/config/_threads.h"
 
 CL_NS_DEF(util)
 class CLuceneThreadIdCompare;
--- src/core/CLucene/util/VoidMap.h
+++ src/core/CLucene/util/VoidMap.h
@@ -11,8 +11,13 @@
 #include "CLucene/LuceneThreads.h"
 
 #if defined(_CL_HAVE_TR1_UNORDERED_MAP) && defined(_CL_HAVE_TR1_UNORDERED_SET)
+#ifdef _LIBCPP_VERSION
+	#include <unordered_map>
+	#include <unordered_set>
+#else
 	#include <tr1/unordered_map>
 	#include <tr1/unordered_set>
+#endif
 #elif defined(_CL_HAVE_HASH_MAP) && defined(_CL_HAVE_HASH_SET)
 	//hashing is all or nothing!
 	#include <hash_map>
--- src/shared/CLucene/config/repl_tchar.h
+++ src/shared/CLucene/config/repl_tchar.h
@@ -36,8 +36,13 @@
         #define _tcsncpy wcsncpy //copy a specified amount of one string to another string.
         #define _tcscat wcscat //copy a string onto the end of the other string
     		#define _tcsncat wcsncat
+#if defined(_LIBCPP_VERSION) && defined(__APPLE__)
+        #define _tcschr ::wcschr //find location of one character
+        #define _tcsstr ::wcsstr //find location of a string
+#else
         #define _tcschr wcschr //find location of one character
         #define _tcsstr wcsstr //find location of a string
+#endif
         #define _tcslen wcslen //get length of a string
         #define _tcscmp wcscmp //case sensitive compare two strings
         #define _tcsncmp wcsncmp //case sensitive compare two strings
