--- core/fxge/linux/fx_linux_impl.cpp.orig	2024-04-16 00:00:38 UTC
+++ core/fxge/linux/fx_linux_impl.cpp
@@ -17,7 +17,7 @@
 #include "core/fxge/fx_font.h"
 #include "core/fxge/systemfontinfo_iface.h"
 
-#if !BUILDFLAG(IS_LINUX) && !BUILDFLAG(IS_CHROMEOS) && !defined(OS_ASMJS)
+#if !BUILDFLAG(IS_LINUX) && !BUILDFLAG(IS_CHROMEOS) && !defined(OS_ASMJS) && !defined(OS_FREEBSD) && !defined(OS_OPENBSD)
 #error "Included on the wrong platform"
 #endif
 
--- core/fxge/cfx_fontmapper.cpp.orig	2024-04-16 00:00:38 UTC
+++ core/fxge/cfx_fontmapper.cpp
@@ -157,7 +157,7 @@ constexpr AltFontFamily kAltFontFamilies[] = {
     {"ForteMT", "Forte"},
 };
 
-#if BUILDFLAG(IS_LINUX) || BUILDFLAG(IS_CHROMEOS) || defined(OS_ASMJS)
+#if BUILDFLAG(IS_LINUX) || BUILDFLAG(IS_CHROMEOS) || defined(OS_ASMJS) || defined(OS_FREEBSD) || defined(OS_OPENBSD)
 const char kNarrowFamily[] = "LiberationSansNarrow";
 #elif BUILDFLAG(IS_ANDROID)
 const char kNarrowFamily[] = "RobotoCondensed";

