diff -urN before/core/fpdfapi/page/cpdf_sampledfunc.cpp after/core/fpdfapi/page/cpdf_sampledfunc.cpp
--- core/fpdfapi/page/cpdf_sampledfunc.cpp	2026-09-24 15:11:27.153556875 +0000
+++ core/fpdfapi/page/cpdf_sampledfunc.cpp	2026-09-24 15:11:41.206965278 +0000
@@ -18,7 +18,11 @@
 #include "core/fxcrt/fx_ceil_div.h"
 #include "core/fxcrt/fx_memory_wrappers.h"
 #include "core/fxcrt/fx_safe_types.h"
+#ifdef USE_SYSTEM_ABSEIL
+#include <absl/container/inlined_vector.h>
+#else
 #include "third_party/abseil-cpp/absl/container/inlined_vector.h"
+#endif
 
 namespace {
 
diff -urN before/core/fpdfdoc/cpdf_nametree.cpp after/core/fpdfdoc/cpdf_nametree.cpp
--- core/fpdfdoc/cpdf_nametree.cpp	2026-09-24 15:11:27.137556410 +0000
+++ core/fpdfdoc/cpdf_nametree.cpp	2026-09-24 15:11:41.169964202 +0000
@@ -19,7 +19,11 @@
 #include "core/fxcrt/check.h"
 #include "core/fxcrt/ptr_util.h"
 #include "core/fxcrt/stl_util.h"
+#ifdef USE_SYSTEM_ABSEIL
+#include <absl/container/flat_hash_set.h>
+#else
 #include "third_party/abseil-cpp/absl/container/flat_hash_set.h"
+#endif
 
 namespace {
 
diff -urN before/core/fpdfdoc/cpvt_variabletext.cpp after/core/fpdfdoc/cpvt_variabletext.cpp
--- core/fpdfdoc/cpvt_variabletext.cpp	2026-09-24 15:11:27.145556642 +0000
+++ core/fpdfdoc/cpvt_variabletext.cpp	2026-09-24 15:11:41.189964784 +0000
@@ -21,7 +21,11 @@
 #include "core/fxcrt/fx_safe_types.h"
 #include "core/fxcrt/span.h"
 #include "core/fxcrt/stl_util.h"
+#ifdef USE_SYSTEM_ABSEIL
+#include <absl/cleanup/cleanup.h>
+#else
 #include "third_party/abseil-cpp/absl/cleanup/cleanup.h"
+#endif
 
 namespace {
 
diff -urN before/core/fxge/cfx_face.cpp after/core/fxge/cfx_face.cpp
--- core/fxge/cfx_face.cpp	2026-09-24 15:11:27.161557107 +0000
+++ core/fxge/cfx_face.cpp	2026-09-24 15:11:41.224965801 +0000
@@ -46,7 +46,11 @@
 
 #if defined(PDF_ENABLE_FONTATIONS)
 #include "core/fxge/skrifa/src/main.rs.h"
+#ifdef USE_SYSTEM_ABSEIL
+#include <absl/cleanup/cleanup.h>
+#else
 #include "third_party/abseil-cpp/absl/cleanup/cleanup.h"
+#endif
 #include "third_party/rust/cxx/v1/cxx.h"
 #endif
 
