--- cdr/build/win32/cdr2raw.vcproj	2012-10-06 06:45:36.436443100 +0200
+++ cdr/build/win32/cdr2raw.vcproj	2012-10-06 06:46:42.153510700 +0200
@@ -69,7 +69,7 @@
 			/>
 			<Tool
 				Name="VCLinkerTool"
-				AdditionalDependencies="libcdr-0.2.lib libwpd-stream-0.9.lib libwpd-0.9.lib"
+				AdditionalDependencies="libcdr-0.0.lib libwpd-stream-0.9.lib libwpd-0.9.lib"
 				OutputFile="Debug\bin\cdr2raw.exe"
 				LinkIncremental="2"
 				SuppressStartupBanner="true"
@@ -157,7 +157,7 @@
 			/>
 			<Tool
 				Name="VCLinkerTool"
-				AdditionalDependencies="libcdr-0.2.lib libwpd-stream-0.9.lib libwpd-0.9.lib"
+				AdditionalDependencies="libcdr-0.0.lib libwpd-stream-0.9.lib libwpd-0.9.lib"
 				OutputFile="Release\bin\cdr2raw.exe"
 				LinkIncremental="1"
 				SuppressStartupBanner="true"
--- cdr/build/win32/cdr2raw.vcxproj	2012-10-06 06:45:36.436443100 +0200
+++ cdr/build/win32/cdr2raw.vcxproj	2012-10-06 06:46:42.153510700 +0200
@@ -68,7 +68,7 @@
       <Culture>0x0409</Culture>
     </ResourceCompile>
     <Link>
-      <AdditionalDependencies>libcdr-0.2.lib;libwpd-stream-0.9.lib;libwpd-0.9.lib;%(AdditionalDependencies)</AdditionalDependencies>
+      <AdditionalDependencies>libcdr-0.0.lib;libwpd-stream-0.9.lib;libwpd-0.9.lib;%(AdditionalDependencies)</AdditionalDependencies>
       <OutputFile>Debug\bin\cdr2raw.exe</OutputFile>
       <SuppressStartupBanner>true</SuppressStartupBanner>
       <AdditionalLibraryDirectories>Debug\lib;$(LIBWPD_LIB_DIR);$(LCMS2_LIB_DIR);$(ZLIB_LIB_DIR);%(AdditionalLibraryDirectories)</AdditionalLibraryDirectories>
@@ -109,7 +109,7 @@
       <Culture>0x0409</Culture>
     </ResourceCompile>
     <Link>
-      <AdditionalDependencies>libcdr-0.2.lib;libwpd-stream-0.9.lib;libwpd-0.9.lib;%(AdditionalDependencies)</AdditionalDependencies>
+      <AdditionalDependencies>libcdr-0.0.lib;libwpd-stream-0.9.lib;libwpd-0.9.lib;%(AdditionalDependencies)</AdditionalDependencies>
       <OutputFile>Release\bin\cdr2raw.exe</OutputFile>
       <SuppressStartupBanner>true</SuppressStartupBanner>
       <AdditionalLibraryDirectories>Release\lib;$(LIBWPD_LIB_DIR);$(LCMS2_LIB_DIR);$(ZLIB_LIB_DIR);%(AdditionalLibraryDirectories)</AdditionalLibraryDirectories>
--- cdr/build/win32/cdr2xhtml.vcproj	2012-10-06 06:45:36.436443100 +0200
+++ cdr/build/win32/cdr2xhtml.vcproj	2012-10-06 06:46:42.169135300 +0200
@@ -67,7 +67,7 @@
 			/>
 			<Tool
 				Name="VCLinkerTool"
-				AdditionalDependencies="libcdr-0.2.lib libwpd-stream-0.9.lib libwpd-0.9.lib"
+				AdditionalDependencies="libcdr-0.0.lib libwpd-stream-0.9.lib libwpd-0.9.lib"
 				OutputFile="Release\bin\cdr2xhtml.exe"
 				LinkIncremental="1"
 				SuppressStartupBanner="true"
@@ -156,7 +156,7 @@
 			/>
 			<Tool
 				Name="VCLinkerTool"
-				AdditionalDependencies="libcdr-0.2.lib libwpd-stream-0.9.lib libwpd-0.9.lib"
+				AdditionalDependencies="libcdr-0.0.lib libwpd-stream-0.9.lib libwpd-0.9.lib"
 				OutputFile="Debug\bin\cdr2xhtml.exe"
 				LinkIncremental="2"
 				SuppressStartupBanner="true"
--- cdr/build/win32/cdr2xhtml.vcxproj	2012-10-06 06:45:36.452067700 +0200
+++ cdr/build/win32/cdr2xhtml.vcxproj	2012-10-06 06:46:42.169135300 +0200
@@ -66,7 +66,7 @@
       <Culture>0x0409</Culture>
     </ResourceCompile>
     <Link>
-      <AdditionalDependencies>libcdr-0.2.lib;libwpd-stream-0.9.lib;libwpd-0.9.lib;%(AdditionalDependencies)</AdditionalDependencies>
+      <AdditionalDependencies>libcdr-0.0.lib;libwpd-stream-0.9.lib;libwpd-0.9.lib;%(AdditionalDependencies)</AdditionalDependencies>
       <OutputFile>Release\bin\cdr2xhtml.exe</OutputFile>
       <SuppressStartupBanner>true</SuppressStartupBanner>
       <AdditionalLibraryDirectories>Release\lib;$(LIBWPD_LIB_DIR);$(LCMS2_LIB_DIR);$(ZLIB_LIB_DIR);%(AdditionalLibraryDirectories)</AdditionalLibraryDirectories>
@@ -108,7 +108,7 @@
       <Culture>0x0409</Culture>
     </ResourceCompile>
     <Link>
-      <AdditionalDependencies>libcdr-0.2.lib;libwpd-stream-0.9.lib;libwpd-0.9.lib;%(AdditionalDependencies)</AdditionalDependencies>
+      <AdditionalDependencies>libcdr-0.0.lib;libwpd-stream-0.9.lib;libwpd-0.9.lib;%(AdditionalDependencies)</AdditionalDependencies>
       <OutputFile>Debug\bin\cdr2xhtml.exe</OutputFile>
       <SuppressStartupBanner>true</SuppressStartupBanner>
       <AdditionalLibraryDirectories>Debug\lib;$(LIBWPD_LIB_DIR);$(LCMS2_LIB_DIR);%(AdditionalLibraryDirectories)</AdditionalLibraryDirectories>
--- cdr/build/win32/cmx2raw.vcproj	2012-10-06 06:45:36.452067700 +0200
+++ cdr/build/win32/cmx2raw.vcproj	2012-10-06 06:46:42.169135300 +0200
@@ -69,7 +69,7 @@
 			/>
 			<Tool
 				Name="VCLinkerTool"
-				AdditionalDependencies="libcdr-0.2.lib libwpd-stream-0.9.lib libwpd-0.9.lib"
+				AdditionalDependencies="libcdr-0.0.lib libwpd-stream-0.9.lib libwpd-0.9.lib"
 				OutputFile="Debug\bin\cmx2raw.exe"
 				LinkIncremental="2"
 				SuppressStartupBanner="true"
@@ -157,7 +157,7 @@
 			/>
 			<Tool
 				Name="VCLinkerTool"
-				AdditionalDependencies="libcdr-0.2.lib libwpd-stream-0.9.lib libwpd-0.9.lib"
+				AdditionalDependencies="libcdr-0.0.lib libwpd-stream-0.9.lib libwpd-0.9.lib"
 				OutputFile="Release\bin\cmx2raw.exe"
 				LinkIncremental="1"
 				SuppressStartupBanner="true"
--- cdr/build/win32/cmx2raw.vcxproj	2012-10-06 06:45:36.452067700 +0200
+++ cdr/build/win32/cmx2raw.vcxproj	2012-10-06 06:46:42.184759900 +0200
@@ -68,7 +68,7 @@
       <Culture>0x0409</Culture>
     </ResourceCompile>
     <Link>
-      <AdditionalDependencies>libcdr-0.2.lib;libwpd-stream-0.9.lib;libwpd-0.9.lib;%(AdditionalDependencies)</AdditionalDependencies>
+      <AdditionalDependencies>libcdr-0.0.lib;libwpd-stream-0.9.lib;libwpd-0.9.lib;%(AdditionalDependencies)</AdditionalDependencies>
       <OutputFile>Debug\bin\cmx2raw.exe</OutputFile>
       <SuppressStartupBanner>true</SuppressStartupBanner>
       <AdditionalLibraryDirectories>Debug\lib;$(LIBWPD_LIB_DIR);$(LCMS2_LIB_DIR);$(ZLIB_LIB_DIR);%(AdditionalLibraryDirectories)</AdditionalLibraryDirectories>
@@ -109,7 +109,7 @@
       <Culture>0x0409</Culture>
     </ResourceCompile>
     <Link>
-      <AdditionalDependencies>libcdr-0.2.lib;libwpd-stream-0.9.lib;libwpd-0.9.lib;%(AdditionalDependencies)</AdditionalDependencies>
+      <AdditionalDependencies>libcdr-0.0.lib;libwpd-stream-0.9.lib;libwpd-0.9.lib;%(AdditionalDependencies)</AdditionalDependencies>
       <OutputFile>Release\bin\cmx2raw.exe</OutputFile>
       <SuppressStartupBanner>true</SuppressStartupBanner>
       <AdditionalLibraryDirectories>Release\lib;$(LIBWPD_LIB_DIR);$(LCMS2_LIB_DIR);$(ZLIB_LIB_DIR);%(AdditionalLibraryDirectories)</AdditionalLibraryDirectories>
--- cdr/build/win32/cmx2xhtml.vcproj	2012-10-06 06:45:36.452067700 +0200
+++ cdr/build/win32/cmx2xhtml.vcproj	2012-10-06 06:46:42.184759900 +0200
@@ -69,7 +69,7 @@
 			/>
 			<Tool
 				Name="VCLinkerTool"
-				AdditionalDependencies="libcdr-0.2.lib libwpd-stream-0.9.lib libwpd-0.9.lib"
+				AdditionalDependencies="libcdr-0.0.lib libwpd-stream-0.9.lib libwpd-0.9.lib"
 				OutputFile="Debug\bin\cmx2xhtml.exe"
 				LinkIncremental="2"
 				SuppressStartupBanner="true"
@@ -157,7 +157,7 @@
 			/>
 			<Tool
 				Name="VCLinkerTool"
-				AdditionalDependencies="libcdr-0.2.lib libwpd-stream-0.9.lib libwpd-0.9.lib"
+				AdditionalDependencies="libcdr-0.0.lib libwpd-stream-0.9.lib libwpd-0.9.lib"
 				OutputFile="Release\bin\cmx2xhtml.exe"
 				LinkIncremental="1"
 				SuppressStartupBanner="true"
--- cdr/build/win32/cmx2xhtml.vcxproj	2012-10-06 06:45:36.467692300 +0200
+++ cdr/build/win32/cmx2xhtml.vcxproj	2012-10-06 06:46:42.184759900 +0200
@@ -68,7 +68,7 @@
       <Culture>0x0409</Culture>
     </ResourceCompile>
     <Link>
-      <AdditionalDependencies>libcdr-0.2.lib;libwpd-stream-0.9.lib;libwpd-0.9.lib;%(AdditionalDependencies)</AdditionalDependencies>
+      <AdditionalDependencies>libcdr-0.0.lib;libwpd-stream-0.9.lib;libwpd-0.9.lib;%(AdditionalDependencies)</AdditionalDependencies>
       <OutputFile>Debug\bin\cmx2xhtml.exe</OutputFile>
       <SuppressStartupBanner>true</SuppressStartupBanner>
       <AdditionalLibraryDirectories>Debug\lib;$(LIBWPD_LIB_DIR);$(LCMS2_LIB_DIR);$(ZLIB_LIB_DIR);%(AdditionalLibraryDirectories)</AdditionalLibraryDirectories>
@@ -109,7 +109,7 @@
       <Culture>0x0409</Culture>
     </ResourceCompile>
     <Link>
-      <AdditionalDependencies>libcdr-0.2.lib;libwpd-stream-0.9.lib;libwpd-0.9.lib;%(AdditionalDependencies)</AdditionalDependencies>
+      <AdditionalDependencies>libcdr-0.0.lib;libwpd-stream-0.9.lib;libwpd-0.9.lib;%(AdditionalDependencies)</AdditionalDependencies>
       <OutputFile>Release\bin\cmx2xhtml.exe</OutputFile>
       <SuppressStartupBanner>true</SuppressStartupBanner>
       <AdditionalLibraryDirectories>Release\lib;$(LIBWPD_LIB_DIR);$(LCMS2_LIB_DIR);$(ZLIB_LIB_DIR);%(AdditionalLibraryDirectories)</AdditionalLibraryDirectories>
--- cdr/build/win32/libcdr.vcproj	2012-10-06 06:45:36.483316900 +0200
+++ cdr/build/win32/libcdr.vcproj	2012-10-06 06:46:42.216009100 +0200
@@ -67,7 +67,7 @@
 			/>
 			<Tool
 				Name="VCLibrarianTool"
-				OutputFile="Debug\lib\libcdr-0.2.lib"
+				OutputFile="Debug\lib\libcdr-0.0.lib"
 				SuppressStartupBanner="true"
 			/>
 			<Tool
@@ -141,7 +141,7 @@
 			/>
 			<Tool
 				Name="VCLibrarianTool"
-				OutputFile="Release\lib\libcdr-0.2.lib"
+				OutputFile="Release\lib\libcdr-0.0.lib"
 				SuppressStartupBanner="true"
 			/>
 			<Tool
--- cdr/build/win32/libcdr.vcxproj	2012-10-06 06:45:36.483316900 +0200
+++ cdr/build/win32/libcdr.vcxproj	2012-10-06 06:46:42.231633700 +0200
@@ -61,7 +61,7 @@
       <Culture>0x0409</Culture>
     </ResourceCompile>
     <Lib>
-      <OutputFile>Debug\lib\libcdr-0.2.lib</OutputFile>
+      <OutputFile>Debug\lib\libcdr-0.0.lib</OutputFile>
       <SuppressStartupBanner>true</SuppressStartupBanner>
     </Lib>
     <Bscmake>
@@ -91,7 +91,7 @@
       <Culture>0x0409</Culture>
     </ResourceCompile>
     <Lib>
-      <OutputFile>Release\lib\libcdr-0.2.lib</OutputFile>
+      <OutputFile>Release\lib\libcdr-0.0.lib</OutputFile>
       <SuppressStartupBanner>true</SuppressStartupBanner>
     </Lib>
     <Bscmake>
