--- misc/libcdr-0.0.3/src/lib/CDRParser.cpp	2012-02-23 07:42:39.000000000 +0100
+++ misc/build/libcdr-0.0.3/src/lib/CDRParser.cpp	2012-02-23 13:44:27.970033236 +0100
@@ -215,7 +215,7 @@
   else if (fourCC == "mcfg")
     readMcfg(input);
   else if (fourCC == "bmp ")
-    readBmp(input, length);
+    readBmp(input);
   input->seek(recordStart + length, WPX_SEEK_CUR);
 }
 
@@ -770,7 +770,7 @@
   m_collector->collectPolygonTransform(numAngles, nextPoint, rx, ry, cx, cy);
 }
 
-void libcdr::CDRParser::readBmp(WPXInputStream *input, unsigned length)
+void libcdr::CDRParser::readBmp(WPXInputStream *input)
 {
   unsigned imageId = readU32(input);
   input->seek(50, WPX_SEEK_CUR);
--- misc/libcdr-0.0.3/src/lib/CDRParser.h	2012-02-09 08:36:51.000000000 +0100
+++ misc/build/libcdr-0.0.3/src/lib/CDRParser.h	2012-02-23 13:45:50.667974573 +0100
@@ -70,7 +70,7 @@
   void readPath(WPXInputStream *input);
   void readPolygonCoords(WPXInputStream *input);
   void readPolygonTransform(WPXInputStream *input);
-  void readBmp(WPXInputStream *input, unsigned length);
+  void readBmp(WPXInputStream *input);
 
   WPXInputStream *m_input;
   CDRCollector *m_collector;
