--- misc/silgraphite-2.3.1.orig/engine/include/graphite/Segment.h
+++ misc/build/silgraphite-2.3.1/engine/include/graphite/Segment.h
@@ -142,7 +142,6 @@
 		LineBrk lbStart, LineBrk lbEnd, SegEnd est,
 		bool fStartLine, bool fEndLine, bool fWsRtl);
 	void DestroyContents();
-	void SwapWith(Segment * pgrseg);
 	void ClearAltEndLineSeg();
 
 	ITextSource * GetString(); // client is responsible to Release()
--- misc/silgraphite-2.3.1.orig/engine/src/segment/Segment.cpp
+++ misc/build/silgraphite-2.3.1/engine/src/segment/Segment.cpp
@@ -569,21 +569,6 @@
 }
 
 /*----------------------------------------------------------------------------------------------
-	Swap the guts of the two segments.
-	OBSOLETE
-----------------------------------------------------------------------------------------------*/
-void Segment::SwapWith(Segment * pgrseg)
-{
-	int crefThis = m_cref;
-	int crefOther = pgrseg->m_cref;
-
-	std::swap(*this, *pgrseg);
-
-	m_cref = crefThis;
-	pgrseg->m_cref = crefOther;
-}
-
-/*----------------------------------------------------------------------------------------------
 	Delete the pointer to the alternate-end-of-line segment.
 ----------------------------------------------------------------------------------------------*/
 void Segment::ClearAltEndLineSeg()
