--- misc/silgraphite-2.3.1.orig/engine/configure
+++ misc/build/silgraphite-2.3.1/engine/configure
@@ -16877,7 +16877,7 @@
 # (note: the flags here are gcc-specific and may fail with other compilers)
 build_flags=""
 if test "$enable_debug" = yes; then
-  build_flags="$build_flags -O0 -g -Wall -Wno-unknown-pragmas -Wparentheses -Werror"
+  build_flags="$build_flags -O0 -g -Wall -Wno-unknown-pragmas -Wparentheses"
 elif test "$enable_strict" = yes; then
   build_flags="$build_flags -DNDEBUG -Wall -Wno-unknown-pragmas -Wparentheses -Werror"
 else
