--- misc/silgraphite-2.3.1/engine/src/segment/SegmentAux.cpp	2008-08-21 16:24:32.000000000 +0200
+++ misc/build/silgraphite-2.3.1/engine/src/segment/SegmentAux.cpp	2011-02-21 13:22:23.159834368 +0100
@@ -247,7 +247,9 @@
 GlyphSetIterator::reference GlyphSetIterator::operator*() const
 {
 	assert(m_pseg != 0);
-	assert(m_vit != std::vector<int>::const_iterator());
+        // #i116227# the check below is well meant but breaks itself when using stlport with debug facilities
+        // a valid iterator cannot be compared to an empty one in that case
+	// assert(m_vit != std::vector<int>::const_iterator());
 														// in the case of a non-contiguous list
 	return m_pseg->m_prgginf[(*m_vit) - m_pseg->m_isloutGinf0];
 }
