--- misc/mdds_0.6.0/include/mdds/multi_type_vector_types.hpp	2012-07-19 05:09:49.000000000 +0200
+++ misc/build/mdds_0.6.0/include/mdds/multi_type_vector_types.hpp	2012-07-23 09:31:51.636388620 +0200
@@ -302,7 +302,7 @@
     noncopyable_element_block(size_t n, const _Data& val) : base_type(n, val) {}
 
 public:
-    static _Self* clone_block(const base_element_block& blk)
+    static _Self* clone_block(const base_element_block&)
     {
         throw element_block_error("attempted to clone a noncopyable element block.");
     }
