--- misc/mdds_0.6.0/include/mdds/mixed_type_matrix_def.inl	2012-07-23 16:20:34.101608964 +0200
+++ misc/build/mdds_0.6.0/include/mdds/mixed_type_matrix_def.inl	2012-07-23 16:19:13.022858991 +0200
@@ -44,7 +44,6 @@
         default:
             throw matrix_error("unknown density type");
     }
-    return NULL;
 }
 
 template<typename _String, typename _Flag>
--- misc/mdds_0.6.0/include/mdds/multi_type_vector_trait.hpp	2012-07-19 05:09:49.000000000 +0200
+++ misc/build/mdds_0.6.0/include/mdds/multi_type_vector_trait.hpp	2012-07-24 09:22:04.396955159 +0200
@@ -95,7 +95,6 @@
         default:
             throw general_error("create_new_block: failed to create a new block of unknown type.");
     }
-    return NULL;
 }
 
 base_element_block* element_block_func_base::clone_block(const base_element_block& block)
@@ -123,7 +122,6 @@
         default:
             throw general_error("clone_block: failed to clone a block of unknown type.");
     }
-    return NULL;
 }
 
 void element_block_func_base::delete_block(base_element_block* p)
