--- misc/libxml2-2.7.6/runtest.c	2009-09-24 23:32:00.000000000 +0800
+++ misc/build/libxml2-2.7.6/runtest.c	2012-05-29 14:17:16.852600200 +0800
@@ -2728,7 +2728,7 @@
     "file:///path/to/a%20b.html",
     "/path/to/a b.html",
     "/path/to/a%20b.html",
-    "urip://example.com/résumé.html",
+    "urip://example.com/r\xE9sum\xE9.html",   /* i93433: replaced 'e' encoded in ISO-8859-1 as in 'resume' with \xe9 */
     "urip://example.com/test?a=1&b=2%263&c=4#foo",
     NULL
 };
