--- misc/libxml2-2.7.6/relaxng.c	2009-12-17 11:53:12.000000000 +0000
+++ misc/build/libxml2-2.7.6/relaxng.c	2009-12-17 16:30:43.000000000 +0000
@@ -5369,7 +5369,7 @@
     } else {
         xmlRngPErr(ctxt, node, XML_RNGP_CHOICE_CONTENT,
                    "expecting name, anyName, nsName or choice : got %s\n",
-                   (node == NULL ? "nothing" : node->name), NULL);
+                   (node == NULL ? BAD_CAST "nothing" : node->name), NULL);
         return (NULL);
     }
     if (ret != def) {
@@ -9459,7 +9459,7 @@
             ctxt->states = NULL;
             if (found == 0) {
                 if (cur == NULL) {
-                    VALID_ERR2(XML_RELAXNG_ERR_INTEREXTRA, "noname");
+                    VALID_ERR2(XML_RELAXNG_ERR_INTEREXTRA, BAD_CAST "noname");
                 } else {
                     VALID_ERR2(XML_RELAXNG_ERR_INTEREXTRA, cur->name);
                 }
