--- misc/libxml2-2.7.8/testapi.c	2009-09-24 23:32:00.000000000 +0800
+++ misc/build/libxml2-2.7.8/testapi.c	2012-05-29 14:17:00.868020600 +0800
@@ -291,7 +291,7 @@
 static xmlChar gen_xmlChar(int no, int nr ATTRIBUTE_UNUSED) {
     if (no == 0) return('a');
     if (no == 1) return(' ');
-    if (no == 2) return((xmlChar) 'ø');
+    if (no == 2) return((xmlChar) '\xF8'); /* i93433: replaced 'ø' encoded in ISO-8859-1 with '\xf8' */
     return(0);
 }
 
@@ -399,7 +399,7 @@
 static xmlChar *gen_const_xmlChar_ptr(int no, int nr ATTRIBUTE_UNUSED) {
     if (no == 0) return((xmlChar *) "foo");
     if (no == 1) return((xmlChar *) "<foo/>");
-    if (no == 2) return((xmlChar *) "nøne");
+    if (no == 2) return((xmlChar *) "n\xF8ne"); /* i93433: replaced 'ø' encoded in ISO-8859-1 with '\xf8' */
     if (no == 3) return((xmlChar *) " 2ab ");
     return(NULL);
 }
