--- misc/libxml2-2.7.6/ltmain.sh	2009-10-06 17:39:54.000000000 +0100
+++ misc/build/libxml2-2.7.6/ltmain.sh	2009-12-17 11:43:56.000000000 +0000
@@ -6271,8 +6271,8 @@
 	  ;;
 
 	freebsd-elf)
-	  major=".$current"
-	  versuffix=".$current"
+	  major=.`expr $current - $age`
+	  versuffix=".$major.$age.$revision";
 	  ;;
 
 	irix | nonstopux)
--- misc/libxml2-2.7.6/include/libxml/xmlversion.h	2009-12-17 11:45:19.000000000 +0000
+++ misc/build/libxml2-2.7.6/include/libxml/xmlversion.h	2009-12-17 11:45:36.000000000 +0000
@@ -264,7 +264,7 @@
  *
  * Whether iconv support is available
  */
-#if 1
+#if 0
 #define LIBXML_ICONV_ENABLED
 #endif
 
@@ -282,7 +282,7 @@
  *
  * Whether Debugging module is configured in
  */
-#if 1
+#if 0
 #define LIBXML_DEBUG_ENABLED
 #endif
 
@@ -291,7 +291,7 @@
  *
  * Whether the memory debugging is configured in
  */
-#if 1
+#if 0
 #define DEBUG_MEMORY_LOCATION
 #endif
 
@@ -300,7 +300,7 @@
  *
  * Whether the runtime debugging is configured in
  */
-#if 1
+#if 0
 #define LIBXML_DEBUG_RUNTIME
 #endif
 
--- misc/libxml2-2.7.6/xml2-config.in	2009-12-17 11:45:20.000000000 +0000
+++ misc/build/libxml2-2.7.6/xml2-config.in	2009-12-17 11:45:36.000000000 +0000
@@ -1,9 +1,14 @@
 #! /bin/sh
 
-prefix=@prefix@
-exec_prefix=@exec_prefix@
-includedir=@includedir@
-libdir=@libdir@
+#prefix=@prefix@
+#exec_prefix=@exec_prefix@
+#includedir=@includedir@
+#libdir=@libdir@
+
+prefix=${SOLARVERSION}/${INPATH}
+exec_prefix=${SOLARVERSION}/${INPATH}
+includedir=${SOLARVERSION}/${INPATH}/inc${UPDMINOREXT}/external
+libdir=${SOLARVERSION}/${INPATH}/lib${UPDMINOREXT}
 
 usage()
 {
@@ -67,7 +72,8 @@
 	;;
 
     --cflags)
-       	echo @XML_INCLUDEDIR@ @XML_CFLAGS@
+        echo -I${includedir}
+#       	echo @XML_INCLUDEDIR@ @XML_CFLAGS@
        	;;
 
     --libtool-libs)
@@ -82,19 +88,24 @@
        	;;
 
     --libs)
-        if [ "`uname`" = "Linux" ]
-	then
-	    if [ "@XML_LIBDIR@" = "-L/usr/lib" -o "@XML_LIBDIR@" = "-L/usr/lib64" ]
-	    then
-		echo @XML_LIBS@ 
-	    else
-		echo @XML_LIBDIR@ @XML_LIBS@ 
-	    fi
-	else
-	    echo @XML_LIBDIR@ @XML_LIBS@ @WIN32_EXTRA_LIBADD@
-	fi
+        echo -L${libdir} ${LIBXML2LIB} -lm
+#        if [ "`uname`" = "Linux" ]
+#	then
+#	    if [ "@XML_LIBDIR@" = "-L/usr/lib" -o "@XML_LIBDIR@" = "-L/usr/lib64" ]
+#	    then
+#		echo @XML_LIBS@ 
+#	    else
+#		echo @XML_LIBDIR@ @XML_LIBS@ 
+#	    fi
+#	else
+#	    echo @XML_LIBDIR@ @XML_LIBS@ @WIN32_EXTRA_LIBADD@
+#	fi
        	;;
 
+    print) # ugly configure hack
+        exit 0
+        ;;
+ 
     *)
 	usage
 	exit 1
