--- misc/libxml2-2.7.6/configure	2008-01-11 17:01:56.000000000 +0900
+++ misc/build/libxml2-2.7.6/configure	2009-09-07 20:48:47.656250000 +0900
@@ -19914,6 +19914,8 @@
 
 if test "$with_modules" != "no" ; then
  case "$host" in
+  *-*-mingw*)
+  ;;
   *-*-cygwin*)
   MODULE_EXTENSION=".dll"
   { $as_echo "$as_me:$LINENO: checking for dlopen in -lcygwin" >&5
--- misc/libxml2-2.7.6/libxml.h	2007-11-23 19:47:23.000000000 +0900
+++ misc/build/libxml2-2.7.6/libxml.h	2009-07-10 14:37:34.988250000 +0900
@@ -30,6 +30,10 @@
 #include <libxml/xmlversion.h>
 #else
 #include "config.h"
+#ifdef __MINGW32__
+#undef HAVE_LIBPTHREAD 
+#undef HAVE_PTHREAD_H 
+#endif
 #include <libxml/xmlversion.h>
 #endif
 
