--- misc/jpeg-6b/jconfig.h	2009-03-19 11:30:46.000000000 +0100
+++ misc/build/jpeg-6b/jconfig.h	2009-03-19 11:30:30.000000000 +0100
@@ -1 +1,16 @@
-dummy
+#define HAVE_PROTOTYPES
+#define HAVE_UNSIGNED_CHAR
+#define HAVE_UNSIGNED_SHORT
+#undef CHAR_IS_UNSIGNED
+#define HAVE_STDDEF_H
+#define HAVE_STDLIB_H
+#undef NEED_BSD_STRINGS
+#undef NEED_SYS_TYPES_H
+#undef NEED_FAR_POINTERS
+#undef NEED_SHORT_EXTERNAL_NAMES
+#undef INCOMPLETE_TYPES_BROKEN
+
+#ifdef JPEG_INTERNALS
+#undef RIGHT_SHIFT_IS_UNSIGNED
+#define NO_GETENV
+#endif
--- misc/jpeg-6b/jmorecfg.h	1997-08-10 01:58:56.000000000 +0200
+++ misc/build/jpeg-6b/jmorecfg.h	2009-03-19 11:30:30.000000000 +0100
@@ -157,8 +157,8 @@
 
 /* INT32 must hold at least signed 32-bit values. */
 
-#ifndef XMD_H			/* X11/xmd.h correctly defines INT32 */
-typedef long INT32;
+#if !defined XMD_H && !defined _SOLAR_H		/* X11/xmd.h correctly defines INT32 */
+typedef int INT32;
 #endif
 
 /* Datatype used for image dimensions.  The JPEG standard only supports
--- misc/jpeg-6b/makefile.mk	2009-03-19 11:30:46.000000000 +0100
+++ misc/build/jpeg-6b/makefile.mk	2009-03-19 11:30:30.000000000 +0100
@@ -1 +1,76 @@
-dummy
+#*************************************************************************
+#
+#   Copyright according the GNU Public License.
+#
+#*************************************************************************
+
+PRJ=..$/..$/..$/..
+
+PRJNAME=jpeg
+TARGET=jpeglib
+LIBTARGET=NO
+EXTERNAL_WARNINGS_NOT_ERRORS=TRUE
+VISIBILITY_HIDDEN=TRUE
+
+# --- Settings -----------------------------------------------------------
+
+.INCLUDE :  settings.mk
+
+
+.IF "$(GUI)$(CPU)"=="WNTP"
+CFLAGS=$(CFLAGS) -Od
+.ENDIF
+
+# --- Files --------------------------------------------------------
+
+SLOFILES=	$(SLO)$/jcapimin.obj \
+			$(SLO)$/jcapistd.obj \
+			$(SLO)$/jccoefct.obj \
+			$(SLO)$/jccolor.obj  \
+			$(SLO)$/jcdctmgr.obj \
+			$(SLO)$/jcinit.obj   \
+			$(SLO)$/jchuff.obj   \
+			$(SLO)$/jcphuff.obj  \
+			$(SLO)$/jcmainct.obj \
+			$(SLO)$/jcmarker.obj \
+			$(SLO)$/jcmaster.obj \
+			$(SLO)$/jcomapi.obj  \
+			$(SLO)$/jcparam.obj  \
+			$(SLO)$/jcprepct.obj \
+			$(SLO)$/jcsample.obj \
+			$(SLO)$/jdhuff.obj   \
+			$(SLO)$/jdinput.obj  \
+			$(SLO)$/jdtrans.obj  \
+			$(SLO)$/jdapimin.obj \
+			$(SLO)$/jdapistd.obj \
+			$(SLO)$/jdatadst.obj \
+			$(SLO)$/jdatasrc.obj \
+			$(SLO)$/jdcoefct.obj \
+			$(SLO)$/jdcolor.obj  \
+			$(SLO)$/jddctmgr.obj \
+			$(SLO)$/jdphuff.obj  \
+			$(SLO)$/jdmainct.obj \
+			$(SLO)$/jdmarker.obj \
+			$(SLO)$/jdmaster.obj \
+			$(SLO)$/jdmerge.obj  \
+			$(SLO)$/jdpostct.obj \
+			$(SLO)$/jdsample.obj \
+			$(SLO)$/jerror.obj   \
+			$(SLO)$/jfdctflt.obj \
+			$(SLO)$/jfdctfst.obj \
+			$(SLO)$/jfdctint.obj \
+			$(SLO)$/jidctflt.obj \
+			$(SLO)$/jidctfst.obj \
+			$(SLO)$/jidctint.obj \
+			$(SLO)$/jidctred.obj \
+			$(SLO)$/jmemmgr.obj  \
+			$(SLO)$/jmemnobs.obj \
+			$(SLO)$/jquant1.obj  \
+			$(SLO)$/jquant2.obj  \
+			$(SLO)$/jutils.obj
+
+LIB1TARGET=$(SLB)$/$(TARGET).lib
+LIB1ARCHIV=$(LB)$/lib$(TARGET).a
+LIB1OBJFILES=$(SLOFILES)
+
+.INCLUDE :  target.mk
