--- build/openssl-0.9.8v/Configure-old	2010-04-17 13:51:42.000000000 +0200
+++ build/openssl-0.9.8v/Configure	2010-04-17 13:52:03.000000000 +0200
@@ -388,6 +388,9 @@
 
 "bsdi-elf-gcc",     "gcc:-DPERL5 -DL_ENDIAN -fomit-frame-pointer -O3 -march=i486 -Wall::(unknown)::-ldl:BN_LLONG ${x86_gcc_des} ${x86_gcc_opts}:${x86_elf_asm}:dlfcn:bsd-gcc-shared:-fPIC::.so.\$(SHLIB_MAJOR).\$(SHLIB_MINOR)",
 
+"debian-kfreebsd-amd64","gcc:-m64 -DL_ENDIAN -DTERMIOS -O3 -Wa,--noexecstack -Wall -DMD32_REG_T=int::-D_REENTRANT::-ldl:SIXTY_FOUR_BIT_LONG RC4_CHUNK BF_PTR2 DES_INT DES_UNROLL:${x86_64_asm}:dlfcn:linux-shared:-fPIC:-m64:.so.\$(SHLIB_MAJOR).\$(SHLIB_MINOR)",
+"debian-kfreebsd-i386","gcc:-DL_ENDIAN -DTERMIOS -O3 -Wa,--noexecstack -g -march=i486 -Wall::-D_REENTRANT::-ldl:BN_LLONG ${x86_gcc_des} ${x86_gcc_opts}:${x86_elf_asm}:dlfcn:linux-shared:-fPIC::.so.\$(SHLIB_MAJOR).\$(SHLIB_MINOR)",
+
 "nextstep",	"cc:-O -Wall:<libc.h>:(unknown):::BN_LLONG ${x86_gcc_des} ${x86_gcc_opts}:::",
 "nextstep3.3",	"cc:-O3 -Wall:<libc.h>:(unknown):::BN_LLONG ${x86_gcc_des} ${x86_gcc_opts}:::",
 
--- build/openssl-0.9.8v/Makefile.org	2010-01-27 17:06:36.000000000 +0100
+++ build/openssl-0.9.8v/Makefile.org	2010-09-20 09:24:00.000000000 +0100
@@ -199,7 +199,7 @@
 
 BUILDENV=	PLATFORM='${PLATFORM}' PROCESSOR='${PROCESSOR}' \
 		CC='${CC}' CFLAG='${CFLAG}' 			\
-		AS='${CC}' ASFLAG='${CFLAG} -c'			\
+		AS='${CC}' ASFLAG='${CFLAG} -c -Wa,--noexecstack'	\
 		AR='${AR}' PERL='${PERL}' RANLIB='${RANLIB}'	\
 		SDIRS='${SDIRS}' LIBRPATH='${INSTALLTOP}/$(LIBDIR)'	\
 		INSTALL_PREFIX='${INSTALL_PREFIX}'		\
--- misc/openssl-0.9.8v/config	2009-02-16 09:43:41.000000000 +0100
+++ build/openssl-0.9.8v/config	2009-03-30 08:59:09.588448000 +0200
@@ -399,27 +399,21 @@
 # this is where the translation occurs into SSLeay terms
 # ---------------------------------------------------------------------------
 
-GCCVER=`(gcc -dumpversion) 2>/dev/null`
-if [ "$GCCVER" != "" ]; then
-  # then strip off whatever prefix egcs prepends the number with...
-  # Hopefully, this will work for any future prefixes as well.
-  GCCVER=`echo $GCCVER | LC_ALL=C sed 's/^[a-zA-Z]*\-//'`
-  # Since gcc 3.1 gcc --version behaviour has changed.  gcc -dumpversion
-  # does give us what we want though, so we use that.  We just just the
-  # major and minor version numbers.
-  # peak single digit before and after first dot, e.g. 2.95.1 gives 29
-  GCCVER=`echo $GCCVER | sed 's/\([0-9]\)\.\([0-9]\).*/\1\2/'`
-fi
-
-# Only set CC if not supplied already
-if [ -z "$CC" ]; then
-# figure out if gcc is available and if so we use it otherwise
-# we fallback to whatever cc does on the system
-  if [ "$GCCVER" != "" ]; then
-    CC=gcc
-  else
-    CC=cc
-  fi
+if [ -z "$CC" ];then
+   GCCVER=`(gcc -dumpversion) 2>/dev/null`
+   if [ "$GCCVER" != "" ]; then
+     CC=gcc
+     # then strip off whatever prefix egcs prepends the number with...
+     # Hopefully, this will work for any future prefixes as well.
+     GCCVER=`echo $GCCVER | LC_ALL=C sed 's/^[a-zA-Z]*\-//'`
+     # Since gcc 3.1 gcc --version behaviour has changed.  gcc -dumpversion
+     # does give us what we want though, so we use that.  We just just the
+     # major and minor version numbers.
+     # peak single digit before and after first dot, e.g. 2.95.1 gives 29
+     GCCVER=`echo $GCCVER | sed 's/\([0-9]\)\.\([0-9]\).*/\1\2/'`
+   else
+     CC=cc
+   fi
 fi
 GCCVER=${GCCVER:-0}
 if [ "$SYSTEM" = "HP-UX" ];then
