--- misc/build/JLanguageTool-1.7.0/build.xml	2012-02-29 22:18:05.118789546 +0100
+++ misc/build/JLanguageTool-1.7.0/build.xml	2012-02-29 22:17:02.385235706 +0100
@@ -82,7 +82,6 @@
                                 <include name="openoffice/Linguistic.xcu"/>
 				<include name="openoffice/description.xml"/>
 				<include name="openoffice/images/dialog.png"/>
-				<include name="openoffice/images/dialogh.png"/>
 				<include name="openoffice/images/LanguageToolBig.png"/>
 				<include name="openoffice/images/LanguageToolSmall.png"/>
 				<include name="openoffice/images/dialog.png"/>
--- misc/build/JLanguageTool-1.7.0/openoffice/Linguistic.xcu	2011-06-26 20:29:43.000000000 +0200
+++ misc/build/JLanguageTool-1.7.0/openoffice/Linguistic.xcu	2012-02-29 22:17:25.551071099 +0100
@@ -27,9 +27,6 @@
                 <prop oor:name="SpellAndGrammarDialogImage" oor:type="xs:string">
                     <value>%origin%/images/dialog.png</value>
                 </prop>
-                <prop oor:name="SpellAndGrammarDialogImage_HC" oor:type="xs:string">
-                    <value>%origin%/images/dialogh.png</value>
-                </prop>
             </node>
         </node>
     </node>
