--- src/libcmis/base-session.cxx
+++ src/libcmis/base-session.cxx
@@ -281,7 +281,7 @@
         httpRunRequest( url );
         response->getData( )->finish( );
     }
-    catch ( const CurlException& e )
+    catch ( const CurlException& )
     {
         // If the access token is expired, we get 401 error,
         // Need to use the refresh token to get a new one.
@@ -351,7 +351,7 @@
         httpRunRequest( url, headers );
         response->getData( )->finish();
     }
-    catch ( const CurlException& e )
+    catch ( const CurlException& )
     {
         long status = getHttpStatus( );
         /** If we had a HTTP 417 response, this is likely to be due to some 
@@ -438,7 +438,7 @@
         httpRunRequest( url, headers, redirect );
         response->getData( )->finish();    
     }
-    catch ( const CurlException& e )
+    catch ( const CurlException& )
     {
         
         long status = getHttpStatus( );
@@ -495,7 +495,7 @@
     {
         httpRunRequest( url );
     }
-    catch ( const CurlException& e )
+    catch ( const CurlException& )
     {
         // If the access token is expired, we get 401 error,
         // Need to use the refresh token to get a new one.
--- src/libcmis/oauth2-handler.cxx
+++ src/libcmis/oauth2-handler.cxx
@@ -104,7 +104,7 @@
         resp = m_session->httpPostRequest ( m_data->getTokenUrl(), is, 
                                         "application/x-www-form-urlencoded" );
     }
-    catch ( const CurlException& e )
+    catch ( const CurlException& )
     {
         throw libcmis::Exception(
                 "Couldn't get tokens from the authorization code ");
@@ -131,7 +131,7 @@
         resp = m_session->httpPostRequest( m_data->getTokenUrl( ), is,
                                            "application/x-www-form-urlencoded" );
     }
-    catch (const CurlException& e )
+    catch (const CurlException& )
     {
         throw libcmis::Exception( "Couldn't refresh token ");
     }
--- src/libcmis/oauth2-providers.cxx
+++ src/libcmis/oauth2-providers.cxx
@@ -44,7 +44,7 @@
     {
         res = session->httpGetRequest( authUrl )->getStream( )->str( );
     }
-    catch ( const CurlException& e )
+    catch ( const CurlException& )
     {
         return string( );
     }
@@ -65,7 +65,7 @@
         loginRes = session->httpPostRequest ( loginLink, loginIs, CONTENT_TYPE )
                         ->getStream( )->str( );
     }
-    catch ( const CurlException& e )
+    catch ( const CurlException& )
     {
         return string( );
     }
@@ -105,7 +105,7 @@
     {
         res = session->httpGetRequest( authUrl )->getStream( )->str( );
     }
-    catch ( const CurlException& e )
+    catch ( const CurlException& )
     {
         return string( );
     }
@@ -131,7 +131,7 @@
         // Alfresco code is in the redirect link
         resp = session->httpPostRequest( loginLink, loginIs, CONTENT_TYPE, false ); 
     }
-    catch ( const CurlException& e )
+    catch ( const CurlException& )
     {
         return string( );
     }
