--- misc/libxml2-2.7.6/trionan.c
+++ misc/build/libxml2-2.7.6/trionan.c
@@ -327,7 +327,7 @@
 
   if (result == 0.0) {
 
-#if defined(TRIO_COMPILER_SUPPORTS_C99)
+#if defined(TRIO_COMPILER_SUPPORTS_C99) && !(defined(__ANDROID__) && defined(__clang__))
     result = nan("");
 
 #elif defined(NAN) && defined(__STDC_IEC_559__)

--- misc/libxml2-2.7.6/Makefile.in
+++ misc/build/libxml2-2.7.6/Makefile.in
@@ -1635,7 +1635,7 @@
 check-am: all-am
 check: $(BUILT_SOURCES)
 	$(MAKE) $(AM_MAKEFLAGS) check-recursive
-all-am: Makefile $(LTLIBRARIES) $(PROGRAMS) $(SCRIPTS) $(MANS) $(DATA) \
+all-am: Makefile $(LTLIBRARIES) \
 		config.h
 install-binPROGRAMS: install-libLTLIBRARIES
 
