--- misc/build/libxml2-2.7.6/include/libxml/xmlexports.h.orig	2011-08-23 10:26:54.000000000 +0200
+++ misc/build/libxml2-2.7.6/include/libxml/xmlexports.h	2011-08-23 10:27:21.000000000 +0200
@@ -56,7 +56,7 @@
   #undef XMLPUBVAR
   #undef XMLCALL
   #undef XMLCDECL
-  #if defined(IN_LIBXML) && !defined(LIBXML_STATIC)
+  #if 0
     #define XMLPUBFUN __declspec(dllexport)
     #define XMLPUBVAR __declspec(dllexport)
   #else
