diff -ur xml2.org/include/win32config.h xml2/include/win32config.h
--- misc/xml2/include/win32config.h	2015-07-04 21:35:50.600013534 +0200
+++ misc/xml2/include/win32config.h	2015-07-04 21:36:12.455015001 +0200
@@ -95,7 +95,9 @@
 
 #if defined(_MSC_VER)
 #define mkdir(p,m) _mkdir(p)
+#if _MSC_VER < 1900
 #define snprintf _snprintf
+#endif
 #if _MSC_VER < 1500
 #define vsnprintf(b,c,f,a) _vsnprintf(b,c,f,a)
 #endif
