--- misc/build/libwpg-0.2.0/src/lib/WPGSVGGenerator.cpp
+++ misc/build/libwpg-0.2.0/src/lib/WPGSVGGenerator.cpp
@@ -30,7 +30,11 @@
 {
   std::ostringstream tempStream;
   tempStream << value;
+#ifndef __ANDROID__
   std::string decimalPoint(localeconv()->decimal_point);
+#else
+  std::string decimalPoint(".");
+#endif
   if ((decimalPoint.size() == 0) || (decimalPoint == "."))
     return tempStream.str();
   std::string stringValue(tempStream.str());
