--- misc/raptor-1.4.18/configure	2011-06-05 11:57:41.000000000 -0400
+++ misc/raptor-1.4.18/configure	2011-06-05 11:58:11.000000000 -0400
@@ -23470,73 +23470,6 @@
 LIBS="$oLIBS"
 
 
-# Check whether --with-xslt-config was given.
-if test "${with_xslt_config+set}" = set; then
-  withval=$with_xslt_config; xslt_config="$withval"
-else
-  xslt_config=""
-fi
-
-
-if test "X$xslt_config" != "X" ; then
-  { echo "$as_me:$LINENO: checking for $xslt_config" >&5
-echo $ECHO_N "checking for $xslt_config... $ECHO_C" >&6; }
-
-  if test -f $xslt_config ; then
-    XSLT_CONFIG=$xslt_config
-    { echo "$as_me:$LINENO: result: yes" >&5
-echo "${ECHO_T}yes" >&6; }
-  else
-    { echo "$as_me:$LINENO: result: no - searching PATH" >&5
-echo "${ECHO_T}no - searching PATH" >&6; }
-  fi
-fi
-if test "X$XSLT_CONFIG" = "X"; then
-  for ac_prog in xslt-config
-do
-  # Extract the first word of "$ac_prog", so it can be a program name with args.
-set dummy $ac_prog; ac_word=$2
-{ echo "$as_me:$LINENO: checking for $ac_word" >&5
-echo $ECHO_N "checking for $ac_word... $ECHO_C" >&6; }
-if test "${ac_cv_prog_XSLT_CONFIG+set}" = set; then
-  echo $ECHO_N "(cached) $ECHO_C" >&6
-else
-  if test -n "$XSLT_CONFIG"; then
-  ac_cv_prog_XSLT_CONFIG="$XSLT_CONFIG" # Let the user override the test.
-else
-as_save_IFS=$IFS; IFS=$PATH_SEPARATOR
-for as_dir in $PATH
-do
-  IFS=$as_save_IFS
-  test -z "$as_dir" && as_dir=.
-  for ac_exec_ext in '' $ac_executable_extensions; do
-  if { test -f "$as_dir/$ac_word$ac_exec_ext" && $as_test_x "$as_dir/$ac_word$ac_exec_ext"; }; then
-    ac_cv_prog_XSLT_CONFIG="$ac_prog"
-    echo "$as_me:$LINENO: found $as_dir/$ac_word$ac_exec_ext" >&5
-    break 2
-  fi
-done
-done
-IFS=$as_save_IFS
-
-fi
-fi
-XSLT_CONFIG=$ac_cv_prog_XSLT_CONFIG
-if test -n "$XSLT_CONFIG"; then
-  { echo "$as_me:$LINENO: result: $XSLT_CONFIG" >&5
-echo "${ECHO_T}$XSLT_CONFIG" >&6; }
-else
-  { echo "$as_me:$LINENO: result: no" >&5
-echo "${ECHO_T}no" >&6; }
-fi
-
-
-  test -n "$XSLT_CONFIG" && break
-done
-
-fi
-
-
 # Check whether --with-curl-config was given.
 if test "${with_curl_config+set}" = set; then
   withval=$with_curl_config; curl_config="$withval"
@@ -24702,8 +24635,8 @@
 
 oCPPFLAGS="$CPPFLAGS"
 oLIBS="$LIBS"
-if test "X$XSLT_CONFIG" != X; then
-  LIBS="$LIBS `$XSLT_CONFIG --libs`"
+if test 0; then
+  LIBS="$LIBS $LIBXSLT_LIBS"
   { echo "$as_me:$LINENO: checking for xsltSaveResultToString" >&5
 echo $ECHO_N "checking for xsltSaveResultToString... $ECHO_C" >&6; }
 if test "${ac_cv_func_xsltSaveResultToString+set}" = set; then
@@ -24795,17 +24728,6 @@
 echo $ECHO_N "checking for system libxslt library... $ECHO_C" >&6; }
   if test $have_xsltSaveResultToString = yes; then
     have_libxslt=1
-    CPPFLAGS="`$XSLT_CONFIG --cflags` $CPPFLAGS"
-    LIBXSLT_VERSION=`$XSLT_CONFIG --version`
-    libxslt_version_dec=`echo $LIBXSLT_VERSION | awk -F. '{printf("%d\n", 10000*$1 + 100*$2 + $3)};'`
-    libxslt_min_version_dec=`echo $libxslt_min_version | awk -F. '{printf("%d\n", 10000*$1 + 100*$2 + $3)};'`
-    { echo "$as_me:$LINENO: result: yes - version $LIBXSLT_VERSION" >&5
-echo "${ECHO_T}yes - version $LIBXSLT_VERSION" >&6; }
-    if test $libxslt_version_dec -lt $libxslt_min_version_dec; then
-       { echo "$as_me:$LINENO: WARNING: Using libxslt $LIBXSLT_VERSION is unsupported - $libxslt_min_version or newer required." >&5
-echo "$as_me: WARNING: Using libxslt $LIBXSLT_VERSION is unsupported - $libxslt_min_version or newer required." >&2;}
-       have_libxslt=0
-    fi
   else
     { echo "$as_me:$LINENO: result: no" >&5
 echo "${ECHO_T}no" >&6; }
@@ -26395,8 +26317,8 @@
 fi
 
 if test $need_libxslt = 1; then
-  LIBS="$LIBS `$XSLT_CONFIG --libs`"
-  CPPFLAGS="`$XSLT_CONFIG --cflags` $CPPFLAGS"
+  LIBS="$LIBS $LIBXSLT_LIBS"
+  CPPFLAGS="$LIBXSLT_CFLAGS $CPPFLAGS"
 fi
 
 RAPTOR_LIBTOOLLIBS=libraptor.la
