--- misc/raptor-1.4.18/configure	2008-06-15 17:25:55.000000000 +0900
+++ misc/raptor-1.4.18/configure	2009-06-04 21:10:22.447125000 +0900
@@ -25238,6 +25238,12 @@
 rdfa_parser=yes
 
 rdf_parsers_available="rdfxml ntriples turtle trig guess rss-tag-soup rdfa"
+case $host_os in
+mingw*)
+  rdfa_parser=no
+  rdf_parsers_available="rdfxml ntriples turtle trig guess rss-tag-soup"
+  ;;
+esac
 if test "$USE_MAINTAINER_MODE" = yes; then
   rdf_parsers_available="$rdf_parsers_available n3"
 fi
--- misc/raptor-1.4.18/configure.ac	2008-06-15 17:25:20.000000000 +0900
+++ misc/build/raptor-1.4.18/configure.ac	2009-06-04 21:10:31.025250000 +0900
@@ -758,6 +758,12 @@
 rdfa_parser=yes
 
 rdf_parsers_available="rdfxml ntriples turtle trig guess rss-tag-soup rdfa"
+case $host_os in
+mingw*)
+  rdfa_parser=no
+  rdf_parsers_available="rdfxml ntriples turtle trig guess rss-tag-soup"
+  ;;
+esac
 if test "$USE_MAINTAINER_MODE" = yes; then
   rdf_parsers_available="$rdf_parsers_available n3"
 fi
--- misc/raptor-1.4.18/src/Makefile.in
+++ misc/build/raptor-1.4.18/src/Makefile.in
@@ -404,7 +404,7 @@
 	$(am__append_20) $(am__append_21) $(am__append_22) \
 	$(am__append_23) $(am__append_24) $(am__append_25) \
 	$(am__append_26)
-libraptor_la_LDFLAGS = -version-info @RAPTOR_LIBTOOL_VERSION@
+libraptor_la_LDFLAGS = -version-info @RAPTOR_LIBTOOL_VERSION@ -no-undefined
 libraptor_la_LIBADD = @LTLIBOBJS@
 EXTRA_DIST = \
 raptor-config.in \
