--- misc/raptor-1.4.18/configure
+++ misc/raptor-1.4.18/configure
@@ -22643,11 +22643,13 @@
   { echo "$as_me:$LINENO: checking vsnprintf has C99 compatible return value" >&5
 echo $ECHO_N "checking vsnprintf has C99 compatible return value... $ECHO_C" >&6; }
   if test "$cross_compiling" = yes; then
-  { { echo "$as_me:$LINENO: error: cannot run test program while cross compiling
-See \`config.log' for more details." >&5
-echo "$as_me: error: cannot run test program while cross compiling
-See \`config.log' for more details." >&2;}
-   { (exit 1); exit 1; }; }
+
+cat >>confdefs.h <<\_ACEOF
+#define HAVE_C99_VSNPRINTF 1
+_ACEOF
+
+    { echo "$as_me:$LINENO: result: guessing yes" >&5
+echo "${ECHO_T}guessing yes" >&6; }
 else
   cat >conftest.$ac_ext <<_ACEOF
 /* confdefs.h.  */
@@ -23367,11 +23369,8 @@
     { echo "$as_me:$LINENO: checking for expat support of UTF-8 BOM" >&5
 echo $ECHO_N "checking for expat support of UTF-8 BOM... $ECHO_C" >&6; }
     if test "$cross_compiling" = yes; then
-  { { echo "$as_me:$LINENO: error: cannot run test program while cross compiling
-See \`config.log' for more details." >&5
-echo "$as_me: error: cannot run test program while cross compiling
-See \`config.log' for more details." >&2;}
-   { (exit 1); exit 1; }; }
+  { echo "$as_me:$LINENO: result: guessing yes" >&5
+echo "${ECHO_T}guessing yes" >&6; }
 else
   cat >conftest.$ac_ext <<_ACEOF
 /* confdefs.h.  */
