--- misc/raptor-1.4.18/src/raptor_sax2.c.old	2013-06-26 22:03:48.493875315 +0200
+++ misc/build/raptor-1.4.18/src/raptor_sax2.c	2013-06-26 22:03:09.494878340 +0200
@@ -102,7 +102,8 @@
   
   sax2->error_handlers=error_handlers;
 
-#ifdef RAPTOR_XML_LIBXML
+  // fdo#64672 disable setting globals
+#if 0
   /* sets xmlGenericErrorContext and xmlStructuredError */
   xmlSetStructuredErrorFunc(&sax2->error_handlers, 
                             raptor_libxml_xmlStructuredErrorFunc);
