--- misc/raptor-1.4.18/configure	2011-06-05 11:55:18.000000000 -0400
+++ misc/raptor-1.4.18/configure	2011-06-05 11:55:46.000000000 -0400
@@ -23470,73 +23470,6 @@
 LIBS="$oLIBS"
 
 
-# Check whether --with-xml2-config was given.
-if test "${with_xml2_config+set}" = set; then
-  withval=$with_xml2_config; xml2_config="$withval"
-else
-  xml2_config=""
-fi
-
-
-if test "X$xml2_config" != "X" ; then
-  { echo "$as_me:$LINENO: checking for $xml2_config" >&5
-echo $ECHO_N "checking for $xml2_config... $ECHO_C" >&6; }
-
-  if test -f $xml2_config ; then
-    XML_CONFIG=$xml2_config
-    { echo "$as_me:$LINENO: result: yes" >&5
-echo "${ECHO_T}yes" >&6; }
-  else
-    { echo "$as_me:$LINENO: result: no - searching PATH" >&5
-echo "${ECHO_T}no - searching PATH" >&6; }
-  fi
-fi
-if test "X$XML_CONFIG" = "X"; then
-  for ac_prog in xml2-config
-do
-  # Extract the first word of "$ac_prog", so it can be a program name with args.
-set dummy $ac_prog; ac_word=$2
-{ echo "$as_me:$LINENO: checking for $ac_word" >&5
-echo $ECHO_N "checking for $ac_word... $ECHO_C" >&6; }
-if test "${ac_cv_prog_XML_CONFIG+set}" = set; then
-  echo $ECHO_N "(cached) $ECHO_C" >&6
-else
-  if test -n "$XML_CONFIG"; then
-  ac_cv_prog_XML_CONFIG="$XML_CONFIG" # Let the user override the test.
-else
-as_save_IFS=$IFS; IFS=$PATH_SEPARATOR
-for as_dir in $PATH
-do
-  IFS=$as_save_IFS
-  test -z "$as_dir" && as_dir=.
-  for ac_exec_ext in '' $ac_executable_extensions; do
-  if { test -f "$as_dir/$ac_word$ac_exec_ext" && $as_test_x "$as_dir/$ac_word$ac_exec_ext"; }; then
-    ac_cv_prog_XML_CONFIG="$ac_prog"
-    echo "$as_me:$LINENO: found $as_dir/$ac_word$ac_exec_ext" >&5
-    break 2
-  fi
-done
-done
-IFS=$as_save_IFS
-
-fi
-fi
-XML_CONFIG=$ac_cv_prog_XML_CONFIG
-if test -n "$XML_CONFIG"; then
-  { echo "$as_me:$LINENO: result: $XML_CONFIG" >&5
-echo "${ECHO_T}$XML_CONFIG" >&6; }
-else
-  { echo "$as_me:$LINENO: result: no" >&5
-echo "${ECHO_T}no" >&6; }
-fi
-
-
-  test -n "$XML_CONFIG" && break
-done
-
-fi
-
-
 # Check whether --with-xslt-config was given.
 if test "${with_xslt_config+set}" = set; then
   withval=$with_xslt_config; xslt_config="$withval"
@@ -23719,8 +23652,8 @@
 
 oCPPFLAGS="$CPPFLAGS"
 oLIBS="$LIBS"
-if test "X$XML_CONFIG" != X; then
-  LIBS="$LIBS `$XML_CONFIG --libs`"
+if test 0; then
+  LIBS="$LIBS $LIBXML_LIBS"
   { echo "$as_me:$LINENO: checking for xmlCreatePushParserCtxt" >&5
 echo $ECHO_N "checking for xmlCreatePushParserCtxt... $ECHO_C" >&6; }
 if test "${ac_cv_func_xmlCreatePushParserCtxt+set}" = set; then
@@ -23813,18 +23746,6 @@
   if test $have_xmlCreatePushParserCtxt = yes; then
     have_libxml_lib=1
     have_libxml=1
-    CPPFLAGS="`$XML_CONFIG --cflags` $CPPFLAGS"
-    LIBXML_VERSION=`$XML_CONFIG --version`
-    libxml_version_dec=`echo $LIBXML_VERSION | awk -F. '{printf("%d\n", 10000*$1 + 100*$2 + $3)};'`
-    libxml_min_version_dec=`echo $libxml_min_version | awk -F. '{printf("%d\n", 10000*$1 + 100*$2 + $3)};'`
-    { echo "$as_me:$LINENO: result: yes - version $LIBXML_VERSION" >&5
-echo "${ECHO_T}yes - version $LIBXML_VERSION" >&6; }
-    if test $libxml_version_dec -lt $libxml_min_version_dec; then
-       { echo "$as_me:$LINENO: WARNING: Using libxml $LIBXML_VERSION is unsupported - $libxml_min_version or newer required." >&5
-echo "$as_me: WARNING: Using libxml $LIBXML_VERSION is unsupported - $libxml_min_version or newer required." >&2;}
-       have_libxml_lib=0
-       have_libxml=0
-    fi
   else
     { echo "$as_me:$LINENO: result: no" >&5
 echo "${ECHO_T}no" >&6; }
@@ -26447,8 +26368,8 @@
     CPPFLAGS="-I$srcdir/libxml $CPPFLAGS"
     LIBS="$LIBS -Llibxml -llibxml"
   else
-    LIBS="$LIBS `$XML_CONFIG --libs`"
-    CPPFLAGS="`$XML_CONFIG --cflags` $CPPFLAGS"
+    LIBS="$LIBS $LIBXML_LIBS"
+    CPPFLAGS="$LIBXML_CFLAGS $CPPFLAGS"
   fi
   RAPTOR_XML_PARSER=libxml
 fi
