--- lzma/dos/config.h	2025-11-11 18:04:24.929259033 +0100
+++ lzma/dos/config.h	2025-11-11 18:04:45.193130245 +0100
@@ -12,9 +12,6 @@
 /* Define to 1 if sha256 integrity check is enabled. */
 #define HAVE_CHECK_SHA256 1
 
-/* Define to 1 if the 32-bit x86 CRC assembly files are used. */
-#define HAVE_CRC_X86_ASM 1
-
 /* Define to 1 if any of HAVE_DECODER_foo have been defined. */
 #define HAVE_DECODERS 1
 
@@ -84,9 +81,6 @@
 /* Define to 1 if you have the <inttypes.h> header file. */
 #define HAVE_INTTYPES_H 1
 
-/* Define to 1 if .lz (lzip) decompression support is enabled. */
-#define HAVE_LZIP_DECODER 1
-
 /* Define to 1 to enable bt2 match finder. */
 #define HAVE_MF_BT2 1
 
@@ -122,9 +116,6 @@
    */
 #define HAVE___BUILTIN_ASSUME_ALIGNED 1
 
-/* Define to 1 if the GNU C extensions __builtin_bswap16/32/64 are supported.
-   */
-#define HAVE___BUILTIN_BSWAPXX 1
 
 /* Define to 1 to disable debugging code. */
 #define NDEBUG 1
