diff -ur misc/libxslt-1.1.32/configure misc/build/libxslt-1.1.32/configure
--- misc/libxslt-1.1.32/configure	2017-11-02 13:34:22.000000000 -0700
+++ misc/build/libxslt-1.1.32/configure	2018-08-22 23:00:50.689433000 -0700
@@ -6698,7 +6698,7 @@
 
 cygwin*)
   # func_win32_libid is a shell function defined in ltmain.sh
-  lt_cv_deplibs_check_method='file_magic ^x86 archive import|^x86 DLL'
+  lt_cv_deplibs_check_method='file_magic ^x86 archive|^x86 DLL'
   lt_cv_file_magic_cmd='func_win32_libid'
   ;;
 
diff -ur misc/libxslt-1.1.32/xslt-config.in misc/build/libxslt-1.1.32/xslt-config.in
--- misc/libxslt-1.1.32/xslt-config.in	2015-05-10 07:11:30.000000000 -0700
+++ misc/build/libxslt-1.1.32/xslt-config.in	2018-08-22 23:00:50.689988000 -0700
@@ -1,10 +1,15 @@
 #! /bin/sh
 
-prefix=@prefix@
-exec_prefix=@exec_prefix@
+#prefix=@prefix@
+#exec_prefix=@exec_prefix@
+#exec_prefix_set=no
+#includedir=@includedir@
+#libdir=@libdir@
+prefix=${SOLARVERSION}/${INPATH}
+exec_prefix=${SOLARVERSION}/${INPATH}
 exec_prefix_set=no
-includedir=@includedir@
-libdir=@libdir@
+includedir=${SOLARVERSION}/${INPATH}/inc${UPDMINOREXT}/external
+libdir=${SOLARVERSION}/${INPATH}/lib${UPDMINOREXT}
 
 usage()
 {
@@ -89,7 +94,8 @@
     shift
 done
 
-the_libs="@XSLT_LIBDIR@ @XSLT_LIBS@ @EXTRA_LIBS@"
+#the_libs="@XSLT_LIBDIR@ @XSLT_LIBS@ @EXTRA_LIBS@"
+the_libs="-L${libdir} ${XSLTLIB} -lm"
 if test "$includedir" != "/usr/include"; then
     the_flags="$the_flags -I$includedir `@XML_CONFIG@ --cflags`"
 else
