--- misc/libxslt-1.1.26/libxslt/libxslt.syms	Thu Sep 24 16:28:46 2009
+++ misc/build/libxslt-1.1.26/libxslt/libxslt.syms	Thu Jul  8 12:22:14 2010
@@ -107,7 +107,7 @@
   xsltFreeCompMatchList;
   xsltFreeTemplateHashes;
   xsltGetTemplate;
-  xsltMatchPattern;
+#NOT_IMPLEMENTED  xsltMatchPattern;
   xsltTestCompMatchList;
 
 # preproc
@@ -406,7 +406,7 @@
     global:
 
 # xsltInternals
-  xsltConstNamespaceNameXSLT; # variable
+#XSLT_REFACTORED_XSLT_NSCOMP  xsltConstNamespaceNameXSLT; # variable
   xsltExtensionInstructionResultFinalize;
   xsltExtensionInstructionResultRegister;
   xsltInitCtxtKey;
@@ -415,24 +415,24 @@
   xsltInit;
 
 # xsltInternals
-  xsltParseAnyXSLTElem;
-  xsltParseSequenceConstructor;
-  xsltPointerListAddSize;
-  xsltPointerListClear;
-  xsltPointerListCreate;
-  xsltPointerListFree;
+#XSLT_REFACTORED  xsltParseAnyXSLTElem;
+#XSLT_REFACTORED  xsltParseSequenceConstructor;
+#XSLT_REFACTORED  xsltPointerListAddSize;
+#XSLT_REFACTORED  xsltPointerListClear;
+#XSLT_REFACTORED  xsltPointerListCreate;
+#XSLT_REFACTORED  xsltPointerListFree;
   xsltRegisterLocalRVT;
   xsltReleaseRVT;
-  xsltRestoreDocumentNamespaces;
+#XSLT_REFACTORED_XSLT_NSCOMP  xsltRestoreDocumentNamespaces;
 
 # extensions
-  xsltStyleStylesheetLevelGetExtData;
+#XSLT_REFACTORED  xsltStyleStylesheetLevelGetExtData;
 
 # xsltInternals
-  xsltTransStorageAdd;
-  xsltTransStorageRemove;
+#NOT_IMPLEMENTED  xsltTransStorageAdd;
+#NOT_IMPLEMENTED  xsltTransStorageRemove;
   xsltUninit;
-  xsltXSLTAttrMarker; # variable
+#XSLT_REFACTORED  xsltXSLTAttrMarker; # variable
 } LIBXML2_1.1.9;
 
 LIBXML2_1.1.20 {
@@ -475,5 +475,9 @@
 
 # transform
   xsltProcessOneNode;
+
+# Solaris ld needs explicit auto-reduction (or, alternatively, "-B local")
+    local:
+  *;
 } LIBXML2_1.1.25;
 
