--- misc/xpdf-3.02/xpdf/GfxState.cc	2007-02-27 23:05:52.000000000 +0100
+++ misc/build/xpdf-3.02/xpdf/GfxState.cc	2011-05-18 20:08:45.003807128 +0200
@@ -2678,7 +2678,7 @@
   double x[16], y[16];
   Guint xi, yi;
   GfxColorComp c[4][gfxColorMaxComps];
-  Guint ci[4];
+  Guint ci[gfxColorMaxComps];
   GfxShadingBitBuf *bitBuf;
   Object obj1, obj2;
   int i, j;
--- misc/xpdf-3.02/xpdf/GfxState.h	2007-02-27 23:05:52.000000000 +0100
+++ misc/build/xpdf-3.02/xpdf/GfxState.h	2011-05-18 20:13:26.973936566 +0200
@@ -393,8 +393,8 @@
 
   int nComps;			// number of color components (1, 3, or 4)
   GfxColorSpace *alt;		// alternate color space
-  double rangeMin[4];		// min values for each component
-  double rangeMax[4];		// max values for each component
+  double rangeMin[gfxColorMaxComps];	// min values for each component
+  double rangeMax[gfxColorMaxComps];	// max values for each component
   Ref iccProfileStream;		// the ICC profile
 };
 
