--- misc/build/freetype-2.4.8/builds/unix/configure
+++ misc/build/freetype-2.4.8/builds/unix/configure
@@ -3619,7 +3619,7 @@
 
 if test "x$GCC" = xyes; then
   XX_CFLAGS="-Wall"
-  XX_ANSIFLAGS="-pedantic -ansi"
+  XX_ANSIFLAGS=
 else
   case "$host" in
   *-dec-osf*)
@@ -10039,7 +10039,7 @@
 	*Sun\ F*)			# Sun Fortran 8.3
 	  tmp_sharedflag='-G' ;;
 	esac
-	archive_cmds='$CC '"$tmp_sharedflag""$tmp_addflag"' $libobjs $deplibs $compiler_flags ${wl}-soname $wl$soname -o $lib'
+	archive_cmds='$CC '"$tmp_sharedflag""$tmp_addflag"' $libobjs $deplibs $compiler_flags -o $lib'
 
         if test "x$supports_anon_versioning" = xyes; then
           archive_expsym_cmds='echo "{ global:" > $output_objdir/$libname.ver~
