--- misc/mysql-connector-c++-1.1.0/driver/nativeapi/binding_config.h	2009-08-14 10:17:23.640625000 +0200
+++ misc/build/mysql-connector-c++-1.1.0/driver/nativeapi/binding_config.h	2009-08-14 10:13:50.359375000 +0200
@@ -1 +1,2 @@
-dummy
+#define HAVE_DLFCN_H 1
+//#define MYSQLCLIENT_STATIC_BINDING 1
--- misc/mysql-connector-c++-1.1.0/driver/nativeapi/mysql_client_api.h	2010-09-10 11:47:47.000000000 +0200
+++ misc/build/mysql-connector-c++-1.1.0/driver/nativeapi/mysql_client_api.h	2011-12-08 20:22:29.000000000 +0100
@@ -107,6 +107,8 @@
 
 typedef MYSQL_RES * (STDCALL *ptr2mysql_use_result)(MYSQL *);
 
+typedef unsigned long (STDCALL *ptr2mysql_realescapestring)(MYSQL *, char *, const char *, unsigned long);
+
 /* Prepared Statement stmt_* functions */
 typedef my_ulonglong (STDCALL *ptr2mysql_stmt_affected_rows)(MYSQL_STMT *);
 
