--- misc/libvisio-0.0.3/src/lib/VSDXCollector.h	2011-07-22 14:35:49.000000000 +0200
+++ misc/build/libvisio-0.0.3/src/lib/VSDXCollector.h	2011-07-25 12:18:22.334328001 +0200
@@ -22,7 +22,6 @@
 #define VSDXCOLLECTOR_H
 
 #include <vector>
-#include <stdint.h>
 #include "VSDXParser.h"
 
 namespace libvisio {
@@ -63,7 +62,7 @@
   virtual void collectColours(const std::vector<Colour> &colours) = 0;
 
   virtual void collectCharList(unsigned id, unsigned level) = 0;
-  virtual void collectText(unsigned id, unsigned level, const std::vector<uint8_t> &textStream, TextFormat format) = 0;
+  virtual void collectText(unsigned id, unsigned level, const std::vector<unsigned char> &textStream, TextFormat format) = 0;
   virtual void collectCharFormat(unsigned id , unsigned level, unsigned charCount, unsigned langId, double fontSize, bool bold, bool italic, bool underline, WPXString fontFace) = 0;
 
   // Temporary hack
--- misc/libvisio-0.0.3/src/lib/VSDXContentCollector.cpp	2011-07-22 14:35:49.000000000 +0200
+++ misc/build/libvisio-0.0.3/src/lib/VSDXContentCollector.cpp	2011-07-25 12:18:30.722328001 +0200
@@ -987,7 +987,7 @@
     m_colours.push_back(colours[i]);
 }
 
-void libvisio::VSDXContentCollector::collectText(unsigned /*id*/, unsigned level, const std::vector<uint8_t> &textStream, TextFormat format)
+void libvisio::VSDXContentCollector::collectText(unsigned /*id*/, unsigned level, const std::vector<unsigned char> &textStream, TextFormat format)
 {
   _handleLevelChange(level);
 
@@ -1156,7 +1156,7 @@
   if (fail)
     throw GenericException();
 
-  uint8_t first;
+  unsigned char first;
   int len;
   if (ucs4Character < 0x80)
   {
@@ -1189,7 +1189,7 @@
     len = 6;
   }
 
-  uint8_t outbuf[6] = { 0, 0, 0, 0, 0, 0};
+  unsigned char outbuf[6] = { 0, 0, 0, 0, 0, 0};
   int i;
   for (i = len - 1; i > 0; --i)
   {
--- misc/libvisio-0.0.3/src/lib/VSDXContentCollector.h	2011-07-22 14:35:49.000000000 +0200
+++ misc/build/libvisio-0.0.3/src/lib/VSDXContentCollector.h	2011-07-25 12:18:13.810328000 +0200
@@ -77,7 +77,7 @@
   void collectColours(const std::vector<Colour> &colours);
 
   void collectCharList(unsigned id, unsigned level);
-  void collectText(unsigned id, unsigned level, const std::vector<uint8_t> &textStream, TextFormat format);
+  void collectText(unsigned id, unsigned level, const std::vector<unsigned char> &textStream, TextFormat format);
   void collectCharFormat(unsigned id , unsigned level, unsigned charCount, unsigned langId, double fontSize, bool bold, bool italic, bool underline, WPXString fontFace);
 
   void startPage();
@@ -148,7 +148,7 @@
 
   std::map<unsigned, NURBSData> m_NURBSData;
   std::map<unsigned, PolylineData> m_polylineData;
-  std::vector<uint8_t> m_textStream;
+  std::vector<unsigned char> m_textStream;
   TextFormat m_textFormat;
   bool m_outputTextStart;
 };
--- misc/libvisio-0.0.3/src/lib/VSDXStylesCollector.cpp	2011-07-22 14:35:49.000000000 +0200
+++ misc/build/libvisio-0.0.3/src/lib/VSDXStylesCollector.cpp	2011-07-25 12:18:26.586328031 +0200
@@ -180,7 +180,7 @@
 {
 }
 
-void libvisio::VSDXStylesCollector::collectText(unsigned /*id*/, unsigned level, const std::vector<uint8_t> & /*textStream*/, TextFormat /*format*/)
+void libvisio::VSDXStylesCollector::collectText(unsigned /*id*/, unsigned level, const std::vector<unsigned char> & /*textStream*/, TextFormat /*format*/)
 {
  _handleLevelChange(level);
 }
--- misc/libvisio-0.0.3/src/lib/VSDXStylesCollector.h	2011-07-22 14:35:49.000000000 +0200
+++ misc/build/libvisio-0.0.3/src/lib/VSDXStylesCollector.h	2011-07-25 12:18:06.970328002 +0200
@@ -68,7 +68,7 @@
   void collectColours(const std::vector<Colour> &colours);
 
   void collectCharList(unsigned id, unsigned level);
-  void collectText(unsigned id, unsigned level, const std::vector<uint8_t> &textStream, TextFormat format);
+  void collectText(unsigned id, unsigned level, const std::vector<unsigned char> &textStream, TextFormat format);
   void collectCharFormat(unsigned id , unsigned level, unsigned charCount, unsigned langId, double fontSize, bool bold, bool italic, bool underline, WPXString fontFace);
   // Temporary hack
   void startPage();
