--- misc/libvisio-0.0.8/src/lib/makefile.mk	2011-10-05 14:44:34.615230761 +0200
+++ misc/build/libvisio-0.0.8/src/lib/makefile.mk	2011-10-05 14:44:28.456175697 +0200
@@ -17,13 +17,13 @@
 .ENDIF
 
 .IF "$(SYSTEM_LIBWPD)" == "YES"
-INCPRE+=$(LIBWPD_CFLAGS) -I..
+INCPRE+=$(WPD_CFLAGS) -I..
 .ELSE
 INCPRE+=$(SOLARVER)$/$(UPD)$/$(INPATH)$/inc$/libwpd
 .ENDIF
 
 .IF "$(SYSTEM_LIBWPG)" == "YES"
-INCPRE+=$(LIBWPG_CFLAGS) -I..
+INCPRE+=$(WPG_CFLAGS) -I..
 .ELSE
 INCPRE+=$(SOLARVER)$/$(UPD)$/$(INPATH)$/inc$/libwpg
 .ENDIF
