--- misc/build/libvisio-0.0.1/src/lib/VSDSVGGenerator.cpp
+++ misc/build/libvisio-0.0.1/src/lib/VSDSVGGenerator.cpp
@@ -33,7 +33,11 @@
 {
   std::ostringstream tempStream;
   tempStream << value;
+#ifndef __ANDROID__
   std::string decimalPoint(localeconv()->decimal_point);
+#else
+  std::string decimalPoint(".");
+#endif
   if ((decimalPoint.size() == 0) || (decimalPoint == "."))
     return tempStream.str();
   std::string stringValue(tempStream.str());
