--- misc/build/cppunit-1.13.1/configure
+++ misc/build/cppunit-1.13.1/configure
@@ -23273,7 +23273,7 @@
   { $as_echo "$as_me:${as_lineno-$LINENO}: result: creating $ac_prefix_conf_OUT - prefix $ac_prefix_conf_UPP for $ac_prefix_conf_INP defines" >&5
 $as_echo "creating $ac_prefix_conf_OUT - prefix $ac_prefix_conf_UPP for $ac_prefix_conf_INP defines" >&6; }
   if test -f $ac_prefix_conf_INP ; then
-    $as_dirname -- /* automatically generated */ ||
+    $as_dirname -- '/* automatically generated */' ||
 $as_expr X/* automatically generated */ : 'X\(.*[^/]\)//*[^/][^/]*/*$' \| \
 	 X/* automatically generated */ : 'X\(//\)[^/]' \| \
 	 X/* automatically generated */ : 'X\(//\)$' \| \
@@ -23318,5 +23318,5 @@
   else
     as_fn_error $? "input file $ac_prefix_conf_IN does not exist,     skip generating $ac_prefix_conf_OUT" "$LINENO" 5
   fi
-  rm -f conftest.*
+  rm -f -r conftest.*
 fi
