--- misc/cppunit-1.12.1/include/cppunit/plugin/TestPlugIn.h	2007-01-12 05:54:34.000000000 +0100
+++ misc/build/cppunit-1.12.1/include/cppunit/plugin/TestPlugIn.h	2009-12-02 15:33:29.639857272 +0100
@@ -151,9 +151,7 @@
 #include <windows.h>
 #endif
 #define CPPUNIT_PLUGIN_IMPLEMENT_MAIN()               \
-  BOOL APIENTRY DllMain( HANDLE hModule,              \
-                         DWORD  ul_reason_for_call,   \
-                         LPVOID lpReserved )          \
+  BOOL APIENTRY DllMain( HANDLE, DWORD, LPVOID )      \
   {                                                   \
       return TRUE;                                    \
   }                                                   \
@@ -162,7 +160,7 @@
 // Unix
 #elif defined(CPPUNIT_HAVE_UNIX_DLL_LOADER) || defined(CPPUNIT_HAVE_UNIX_SHL_LOADER)
 #define CPPUNIT_PLUGIN_IMPLEMENT_MAIN()               \
-  int main( int argc, char *argv[] )                  \
+  int main( int, char *[] )                           \
   {                                                   \
     return 0;                                         \
   }                                                   \
