--- misc/librepository-1.1.6/source/org/pentaho/reporting/libraries/repository/ContentIOException.java	2010-04-27 16:04:50.000000000 +0200
+++ misc/build/librepository-1.1.6/source/org/pentaho/reporting/libraries/repository/ContentIOException.java	2011-06-13 01:44:35.000000000 +0200
@@ -43,7 +43,7 @@
    */
   public ContentIOException(final String message, final Exception ex)
   {
-    super(message, ex);
+    super(message, (Throwable) ex);
   }
 
   /**
--- misc/librepository-1.1.6/source/org/pentaho/reporting/libraries/repository/file/FileRepository.java	2010-04-27 16:04:50.000000000 +0200
+++ misc/build/librepository-1.1.6/source/org/pentaho/reporting/libraries/repository/file/FileRepository.java	2014-07-25 11:21:51.000000000 +0200
@@ -19,7 +19,9 @@
 
 import java.io.File;
 import java.io.Serializable;
+import java.lang.SecurityException;
 import java.net.MalformedURLException;
+import java.net.URI;
 import java.net.URL;
 
 import org.pentaho.reporting.libraries.repository.ContentIOException;
@@ -104,6 +106,11 @@
    */
   public URL getURL() throws MalformedURLException
   {
-    return root.getBackend().toURL();
+    try {
+      URI uri = root.getBackend().toURI();
+      return uri.toURL();
+    } catch (SecurityException e) {
+      throw new MalformedURLException("impossible" + e);
+    }
   }
 }

