From 9008d84be1eb0b3def0df39b069a309d54aa73ab Mon Sep 17 00:00:00 2001
From: David Tardon <dtardon@redhat.com>
Date: Wed, 30 Oct 2013 19:30:30 +0100
Subject: [PATCH] use correct header for std::swap

---
 src/GraphicFunctions.cxx | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/GraphicFunctions.cxx b/src/GraphicFunctions.cxx
index d8809f9..606c640 100644
--- a/src/GraphicFunctions.cxx
+++ b/src/GraphicFunctions.cxx
@@ -24,8 +24,8 @@
  * Corel Corporation or Corel Corporation Limited."
  */
 
+#include <algorithm>
 #include <math.h>
-#include <utility>
 
 #include "GraphicFunctions.hxx"
 
-- 
1.8.3.1

