diff -r -u configure configure
--- configure	2017-11-02 19:06:52.000000000 +0900
+++ configure	2017-11-02 19:07:16.000000000 +0900
@@ -11713,7 +11713,7 @@
 
   dynamic_linker='Android linker'
   # Don't embed -rpath directories since the linker doesn't support them.
-  hardcode_libdir_flag_spec='-L$libdir'
+  # hardcode_libdir_flag_spec='-L$libdir'
   ;;
 
 # This must be glibc/ELF.
@@ -15679,7 +15679,7 @@
 
   dynamic_linker='Android linker'
   # Don't embed -rpath directories since the linker doesn't support them.
-  hardcode_libdir_flag_spec_CXX='-L$libdir'
+  # hardcode_libdir_flag_spec_CXX='-L$libdir'
   ;;
 
 # This must be glibc/ELF.
