From b74fc4236cbc58719cc5f6c7fbb3a2213fefa41a Mon Sep 17 00:00:00 2001
From: Miklos Vajna <vmiklos@collabora.co.uk>
Date: Thu, 1 Jan 2015 20:29:36 +0100
Subject: [PATCH] error C1083: Cannot open include file: 'config.h': No such
 file or directory

---
 src/OdfGenerator.cxx | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/src/OdfGenerator.cxx b/src/OdfGenerator.cxx
index 0534528..c9ace1d 100644
--- src/OdfGenerator.cxx
+++ src/OdfGenerator.cxx
@@ -24,7 +24,9 @@
  * Corel Corporation or Corel Corporation Limited."
  */
 
+#ifdef HAVE_CONFIG_H
 #include "config.h"
+#endif
 
 #include <math.h>
 
-- 
2.1.2

