--- inc/libodfgen/OdfDocumentHandler.hxx
+++ inc/libodfgen/OdfDocumentHandler.hxx
@@ -32,7 +32,7 @@
   *
   * @sa OdgGenerator, OdpGenerator, OdtGenerator
   */
-enum OdfStreamType { ODF_FLAT_XML, ODF_CONTENT_XML, ODF_STYLES_XML, ODF_SETTINGS_XML, ODF_META_XML, ODF_MANIFEST_XML };
+enum ODFGENAPI OdfStreamType { ODF_FLAT_XML, ODF_CONTENT_XML, ODF_STYLES_XML, ODF_SETTINGS_XML, ODF_META_XML, ODF_MANIFEST_XML };
 
 class OdfDocumentHandler;
 
