--- src/ne_auth.c~	2019-12-05 15:38:50.246997951 +0100
+++ src/ne_auth.c	2019-12-20 06:54:31.555836285 +0100
@@ -300,8 +300,6 @@
     sess->sspi_token = NULL;
     ne_sspi_destroy_context(sess->sspi_context);
     sess->sspi_context = NULL;
-    if (sess->sspi_host) ne_free(sess->sspi_host);
-    sess->sspi_host = NULL;
 #endif
 #ifdef HAVE_NTLM
     if (sess->ntlm_context) {
@@ -1599,6 +1597,10 @@
     }
 
     clean_session(sess);
+#ifdef HAVE_SSPI
+    if (sess->sspi_host) ne_free(sess->sspi_host);
+    sess->sspi_host = NULL;
+#endif
     ne_free(sess);
 }
 
