diff -ur src/ne_uri.c
--- src/ne_uri.c	2020-02-07 10:49:58.764417840 +0000
+++ src/ne_uri.c	2020-02-07 10:51:33.675627141 +0000
@@ -87,7 +87,8 @@
 #define URI_PCHAR (URI_UNRESERVED | PC | URI_SUBDELIM | CL | AT)
 /* invented: segchar = pchar / "/" */
 /* (TKR) WS added */
-#define URI_SEGCHAR (URI_PCHAR | FS | WS)
+/* also allow OT characters to parse SharePoint 2016 href URIs with unescaped UTF8 */
+#define URI_SEGCHAR (URI_PCHAR | FS | WS | OT)
 /* query = *( pchar / "/" / "?" ) */
 #define URI_QUERY (URI_PCHAR | FS | QU)
 /* fragment == query */
@@ -237,7 +238,7 @@
 
     p = s;
 
-    while (uri_lookup(*p) & URI_SEGCHAR)
+    while (uri_lookup(*p) & URI_SEGCHAR && *p != '\0')
         p++;
 
     /* => p = [ "?" query ] [ "#" fragment ] */
