--- src/dispatch_common.h
+++ src/dispatch_common.h
@@ -24,20 +24,19 @@
 #include "config.h"
 
 #ifdef _WIN32
-#define PLATFORM_HAS_EGL ENABLE_EGL
-#define PLATFORM_HAS_GLX ENABLE_GLX
+#define PLATFORM_HAS_EGL 0
+#define PLATFORM_HAS_GLX 0
 #define PLATFORM_HAS_WGL 1
 #elif defined(__APPLE__)
 #define PLATFORM_HAS_EGL 0 
-#define PLATFORM_HAS_GLX ENABLE_GLX
+#define PLATFORM_HAS_GLX 0
 #define PLATFORM_HAS_WGL 0
 #elif defined(ANDROID)
-#define PLATFORM_HAS_EGL ENABLE_EGL
+#define PLATFORM_HAS_EGL 1
 #define PLATFORM_HAS_GLX 0
 #define PLATFORM_HAS_WGL 0
 #else
-#define PLATFORM_HAS_EGL ENABLE_EGL
-#define PLATFORM_HAS_GLX ENABLE_GLX
+#define PLATFORM_HAS_GLX 1
 #define PLATFORM_HAS_WGL 0
 #endif
 
