--- misc/cairo-1.10.2/build/ltmain.sh
+++ misc/build/cairo-1.10.2/build/ltmain.sh
@@ -3228,6 +3228,10 @@
 	  func_warning "\`-release' is ignored for convenience libraries"
       else
 
+	# Force no versioning suffix for Android
+
+	version_type=none
+
 	# Parse the version information argument.
 	save_ifs="$IFS"; IFS=':'
 	set dummy $vinfo 0 0 0
diff -ur cairo.org/src/cairo-compiler-private.h cairo/src/cairo-compiler-private.h
--- a/cairo.org/src/cairo-compiler-private.h	2016-01-16 00:47:17.392420675 +0100
+++ b/cairo/src/cairo-compiler-private.h	2016-01-16 00:51:55.601110129 +0100
@@ -258,4 +258,7 @@
 #define inline __inline__
 #endif
 
+/* no ffs in libc on android x86 (at least not in v15 that we compile against) */
+#define ffs __builtin_ffs
+
 #endif
