--- src/raptor_rfc2396.c
+++ src/raptor_rfc2396.c
@@ -386,7 +386,7 @@
   } 
 
   
-  if(prev && s == (cur+2) && cur[0] == '.' && cur[1] == '.') {
+  if(prev && cur && s == (cur+2) && cur[0] == '.' && cur[1] == '.') {
     /* Remove <component>/.. at the end of the path */
     *prev = '\0';
     path_len -= (s-prev);
--- utils/Makefile.in
+++ utils/Makefile.in
@@ -174,7 +17,7 @@
 am__v_CC_ = $(am__v_CC_@AM_DEFAULT_V@)
 am__v_CC_0 = @echo "  CC      " $@;
 am__v_CC_1 = 
-CCLD = $(CC)
+CCLD = $(CXX)
 LINK = $(LIBTOOL) $(AM_V_lt) --tag=CC $(AM_LIBTOOLFLAGS) \
 	$(LIBTOOLFLAGS) --mode=link $(CCLD) $(AM_CFLAGS) $(CFLAGS) \
 	$(AM_LDFLAGS) $(LDFLAGS) -o $@
