--- configure
+++ configure
@@ -14197,6 +14197,11 @@
   test -n "$XML_CONFIG" && break
 done
 
+    if test -n "$XML_CONFIG"; then
+      if ! "$XML_CONFIG" --version; then
+        XML_CONFIG=
+      fi
+    fi
   fi
 fi
 
@@ -14481,6 +14481,7 @@
 
     LIBXML_VERSION=`$PKG_CONFIG libxml-2.0 --modversion`
     libxml_source="pkg-config"
+    XML_CONFIG="$PKG_CONFIG libxml-2.0"
 
 fi
   { $as_echo "$as_me:${as_lineno-$LINENO}: checking for libxml via pkg-config" >&5
