diff -ur langtag.org/configure.ac langtag/configure.ac
--- UnpackedTarball/langtag.org/configure.ac	2015-07-05 10:51:30.470991734 +0200
+++ UnpackedTarball/langtag/configure.ac	2015-07-05 10:51:56.439991442 +0200
@@ -209,10 +209,12 @@
 AC_CHECK_FUNCS([backtrace getegid geteuid getgid getuid __secure_getenv secure_getenv strndup vasprintf vsnprintf])
 AC_CHECK_VA_COPY
 
-if test "x$ac_cv_func_vsnprintf" = xyes; then
-	AC_CACHE_CHECK([Whether vsnprintf is C99-compliant], [lt_cv_c99_vsnprintf],
-		[AC_RUN_IFELSE([AC_LANG_SOURCE([[
+AC_CACHE_CHECK([Whether vsnprintf is C99-compliant], [lt_cv_c99_vsnprintf],
+	[AC_RUN_IFELSE([AC_LANG_SOURCE([[
+#include <sys/types.h>
+#include <stdio.h>
 #include <stdarg.h>
+#include <stdlib.h>
 int
 foo(char *format, ...)
 {
@@ -236,9 +238,8 @@
 	return foo("foo: %s", "bar");
 }]])],
 		[lt_cv_c99_vsnprintf=yes],[lt_cv_c99_vsnprintf=no],[lt_cv_c99_vsnprintf=no])])
-	if test "x$lt_cv_c99_vsnprintf" = xyes; then
-		AC_DEFINE(LT_HAVE_C99_VSNPRINTF, 1, [Have C99-compliant vsnprintf])
-	fi
+if test "x$lt_cv_c99_vsnprintf" = xyes; then
+	AC_DEFINE(LT_HAVE_C99_VSNPRINTF, 1, [Have C99-compliant vsnprintf])
 fi
 
 dnl ======================================================================
