diff -ru UnpackedTarball/orig.langtag/liblangtag/lt-utils.c UnpackedTarball/langtag/liblangtag/lt-utils.c
--- UnpackedTarball/orig.langtag/liblangtag/lt-utils.c	2013-07-19 16:47:20.310734230 +0200
+++ UnpackedTarball/langtag/liblangtag/lt-utils.c	2013-07-19 16:53:10.839914271 +0200
@@ -173,9 +173,11 @@
 
 		va_end(ap);
 
-		retval = malloc(sizeof (char) * size);
-		if (retval) {
-			vsprintf(retval, format, args);
+		if (size > 0) {
+			retval = malloc(sizeof (char) * size);
+			if (retval) {
+				vsprintf(retval, format, args);
+			}
 		}
 	} LT_STMT_END;
 #elif HAVE_VSNPRINTF
