--- foo/langtag.orig/configure	2015-06-26 10:53:36.000000000 +0200
+++ foo/langtag/configure	2015-09-03 08:16:58.672240500 +0200
@@ -14786,8 +14786,7 @@
   fi
 
 
-if test "x$ac_cv_func_vsnprintf" = xyes; then
-	{ $as_echo "$as_me:${as_lineno-$LINENO}: checking Whether vsnprintf is C99-compliant" >&5
+{ $as_echo "$as_me:${as_lineno-$LINENO}: checking Whether vsnprintf is C99-compliant" >&5
 $as_echo_n "checking Whether vsnprintf is C99-compliant... " >&6; }
 if ${lt_cv_c99_vsnprintf+:} false; then :
   $as_echo_n "(cached) " >&6
@@ -14798,7 +14797,10 @@
   cat confdefs.h - <<_ACEOF >conftest.$ac_ext
 /* end confdefs.h.  */
 
+#include <sys/types.h>
+#include <stdio.h>
 #include <stdarg.h>
+#include <stdlib.h>
 int
 foo(char *format, ...)
 {
@@ -14834,11 +14836,10 @@
 fi
 { $as_echo "$as_me:${as_lineno-$LINENO}: result: $lt_cv_c99_vsnprintf" >&5
 $as_echo "$lt_cv_c99_vsnprintf" >&6; }
-	if test "x$lt_cv_c99_vsnprintf" = xyes; then
+if test "x$lt_cv_c99_vsnprintf" = xyes; then
 
 $as_echo "#define LT_HAVE_C99_VSNPRINTF 1" >>confdefs.h
 
-	fi
 fi
 
 
