--- misc/raptor-1.4.17/LICENSE.txt	Sun Mar 30 06:52:22 2008
+++ misc/build/raptor-1.4.17/LICENSE.txt	Mon Jun 16 15:55:38 2008
@@ -60,3 +60,15 @@
 
    Copyright (C) 2000-2008 David Beckett
    Copyright (C) 2000-2005 University of Bristol
+
+
+LGPL Disclaimer
+
+For the avoidance of doubt, except that if any license choice other 
+than GPL or LGPL is available it will apply instead, Sun elects to 
+use only the Lesser General Public License version 2.1 (LGPLv2) at 
+this time for any software where a choice of LGPL license versions 
+is made available with the language indicating that LGPLv2.1 or any 
+later version may be used, or where a choice of which version of the 
+LGPL is applied is otherwise unspecified.
+
--- misc/raptor-1.4.17/src/makefile.mk	Mon Jun 16 17:24:23 2008
+++ misc/build/raptor-1.4.17/src/makefile.mk	Mon Jun 16 19:25:14 2008
@@ -1,1 +1,129 @@
-dummy
+#*************************************************************************
+#
+# DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+#
+# Copyright 2008 by Sun Microsystems, Inc.
+#
+# OpenOffice.org - a multi-platform office productivity suite
+#
+# $RCSfile: raptor-1.4.17.patch,v $
+#
+# $Revision: 1.2.6.2 $
+#
+# This file is part of OpenOffice.org.
+#
+# OpenOffice.org is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Lesser General Public License version 3
+# only, as published by the Free Software Foundation.
+#
+# OpenOffice.org is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Lesser General Public License version 3 for more details
+# (a copy is included in the LICENSE file that accompanied this code).
+#
+# You should have received a copy of the GNU Lesser General Public License
+# version 3 along with OpenOffice.org.  If not, see
+# <http://www.openoffice.org/license.html>
+# for a copy of the LGPLv3 License.
+#
+#*************************************************************************
+
+PRJ=..$/..$/..$/..$/..
+
+PRJNAME=redland
+TARGET=raptor
+LIBTARGET=NO
+UWINAPILIB=
+
+EXTERNAL_WARNINGS_NOT_ERRORS=TRUE
+
+
+# --- Settings -----------------------------------------------------
+
+.INCLUDE :  settings.mk
+
+.IF "$(GUI)"=="OS2"
+CDEFS+=-DRAPTOR_INTERNAL -DHAVE_CONFIG_H -DNDEBUG
+.ELSE
+CDEFS+=-DRAPTOR_INTERNAL -DWIN32 -DNDEBUG -D_WINDOWS -D_USRDLL -DWIN32_EXPORTS
+.ENDIF
+
+# --- Files --------------------------------------------------------
+
+SLOFILES =	\
+   $(SLO)$/n3_lexer.obj \
+   $(SLO)$/n3_parser.obj \
+   $(SLO)$/ntriples_parse.obj \
+   $(SLO)$/parsedate.obj \
+   $(SLO)$/raptor_feature.obj \
+   $(SLO)$/raptor_general.obj \
+   $(SLO)$/raptor_guess.obj \
+   $(SLO)$/raptor_identifier.obj \
+   $(SLO)$/raptor_iostream.obj \
+   $(SLO)$/raptor_libxml.obj \
+   $(SLO)$/raptor_locator.obj \
+   $(SLO)$/raptor_namespace.obj \
+   $(SLO)$/raptor_nfc.obj \
+   $(SLO)$/raptor_nfc_data.obj \
+   $(SLO)$/raptor_parse.obj \
+   $(SLO)$/raptor_qname.obj \
+   $(SLO)$/raptor_rdfxml.obj \
+   $(SLO)$/raptor_rfc2396.obj \
+   $(SLO)$/raptor_rss.obj \
+   $(SLO)$/raptor_rss_common.obj \
+   $(SLO)$/raptor_sax2.obj \
+   $(SLO)$/raptor_sequence.obj \
+   $(SLO)$/raptor_serialize.obj \
+   $(SLO)$/raptor_serialize_ntriples.obj \
+   $(SLO)$/raptor_serialize_rdfxml.obj \
+   $(SLO)$/raptor_serialize_rdfxmla.obj \
+   $(SLO)$/raptor_serialize_rss.obj \
+   $(SLO)$/raptor_serialize_simple.obj \
+   $(SLO)$/raptor_set.obj \
+   $(SLO)$/raptor_stringbuffer.obj \
+   $(SLO)$/raptor_uri.obj \
+   $(SLO)$/raptor_utf8.obj \
+   $(SLO)$/raptor_win32.obj \
+   $(SLO)$/raptor_www.obj \
+   $(SLO)$/raptor_www_libxml.obj \
+   $(SLO)$/raptor_xml.obj \
+   $(SLO)$/raptor_xml_writer.obj \
+   $(SLO)$/raptor_xsd.obj \
+   $(SLO)$/raptor_strcasecmp.obj \
+   $(SLO)$/turtle_common.obj \
+   $(SLO)$/turtle_lexer.obj \
+   $(SLO)$/turtle_parser.obj \
+   $(SLO)$/raptor_grddl.obj \
+   $(SLO)$/raptor_abbrev.obj \
+   $(SLO)$/raptor_avltree.obj \
+   $(SLO)$/raptor_statement.obj \
+   $(SLO)$/raptor_turtle_writer.obj \
+
+
+SHL1DEPN=   makefile.mk
+SHL1OBJS=   $(SLOFILES)
+SHL1USE_EXPORTS=name
+
+#SHL1TARGET=	$(TARGET)$(DLLPOSTFIX)
+SHL1TARGET=	$(TARGET)
+SHL1IMPLIB= lib$(TARGET)
+
+#SHL1VERSIONMAP=../service/exports.map
+.IF "$(GUI)"=="OS2"
+SHL1VERSIONMAP=$(PRJ)$/raptor$/raptor2.map
+.ENDIF
+SHL1DEF=$(MISC)$/$(SHL1TARGET).def
+DEF1NAME=$(SHL1TARGET)
+
+SHL1STDLIBS= \
+    $(LIBXML2LIB) \
+    $(XSLTLIB) \
+
+# --- Targets ------------------------------------------------------
+
+.INCLUDE :  target.mk
+
+$(MISC)$/raptor_strcasecmp.c : strcasecmp.c
+	$(COPY) $< $@
+
--- misc/raptor-1.4.17/src/raptor-config.in	Sat Jan  6 05:02:00 2007
+++ misc/build/raptor-1.4.17/src/raptor-config.in	Mon Jun 16 15:55:38 2008
@@ -21,14 +21,23 @@
 # 
 #
 
-prefix=@prefix@
-exec_prefix=@exec_prefix@
+prefix=${SOLARVERSION}/${INPATH}
+exec_prefix=${SOLARVERSION}/${INPATH}
+#includedir=${SOLARVERSION}/${INPATH}/inc${UPDMINOREXT}/external
+#libdir=${SOLARVERSION}/${INPATH}/lib${UPDMINOREXT}
+# HACK: map to build directory
+includedir=../../../../inc
+libdir=../../../../lib
+includes=-I${includedir}
 
-if test "@includedir@" != /usr/include ; then
-  includes="-I@includedir@"
-else
-  includes=
-fi
+#prefix=@prefix@
+#exec_prefix=@exec_prefix@
+#
+#if test "@includedir@" != /usr/include ; then
+#  includes="-I@includedir@"
+#else
+#  includes=
+#fi
 
 
 usage()
@@ -130,7 +139,8 @@
   done
 fi
 if test "$echo_libs" = "yes"; then
-  echo -L@libdir@ -lraptor
+#  echo -L@libdir@ -lraptor
+  echo -L${libdir} -lraptor
 fi
 if test "$echo_libtool_libs" = "yes"; then
   echo @libdir@/@RAPTOR_LIBTOOLLIBS@
--- misc/raptor-1.4.17/src/win32_raptor_config.h	Tue Oct  2 06:12:55 2007
+++ misc/build/raptor-1.4.17/src/win32_raptor_config.h	2008-07-06 10:27:03.553625000 +0900
@@ -38,6 +38,8 @@
 
 #define HAVE_STDLIB_H 1
 
+#define HAVE_ERRNO_H 1
+
 #if 0
 /* For using expat on win32 */
 #define RAPTOR_XML_EXPAT 1
@@ -47,8 +49,17 @@
 /* For using libxml2 on win32 */
 #define RAPTOR_XML_LIBXML
 #define HAVE_LIBXML_XMLREADER_H
+#define HAVE_LIBXML_HASH_H 1
+#define HAVE_LIBXML_HTMLPARSER_H 1
+#define HAVE_LIBXML_NANOHTTP_H 1
+#define HAVE_LIBXML_PARSER_H 1
+#define HAVE_LIBXSLT_XSLT_H 1
+
+#define RAPTOR_LIBXML_HTML_PARSE_NONET 1
+#define RAPTOR_LIBXML_XML_PARSE_NONET 1
+
 /* does libxml struct xmlEntity have a field etype */
-/*#define RAPTOR_LIBXML_ENTITY_ETYPE*/
+#define RAPTOR_LIBXML_ENTITY_ETYPE
 
 /* does libxml struct xmlEntity have a field name_length */
 /*#define RAPTOR_LIBXML_ENTITY_NAME_LENGTH*/
@@ -60,16 +71,17 @@
 #define HAVE_XMLSAX2INTERNALSUBSET 1
 
 /* does libxml xmlSAXHandler have externalSubset field */
-/*#define RAPTOR_LIBXML_XMLSAXHANDLER_EXTERNALSUBSET*/
+#define RAPTOR_LIBXML_XMLSAXHANDLER_EXTERNALSUBSET 1
 
 /* does libxml xmlSAXHandler have initialized field */
-/*#define RAPTOR_LIBXML_XMLSAXHANDLER_INITIALIZED*/
+#define RAPTOR_LIBXML_XMLSAXHANDLER_INITIALIZED 1
 #endif
 
 #define HAVE_STRICMP 1
 
 /* MS names for these functions */
-#define vsnprintf _vsnprintf
+// next line breaks build on wntmsci12
+//#define vsnprintf _vsnprintf
 #define snprintf _snprintf
 #define access _access
 #define stricmp _stricmp
@@ -114,7 +126,9 @@
 
 #define RAPTOR_PARSER_GUESS 1
 #define RAPTOR_PARSER_GRDDL 1
+#ifndef __MINGW32__ /* conflicts with normal configure result */
 #define RAPTOR_PARSER_N3 1
+#endif
 #define RAPTOR_PARSER_TURTLE 1
 #define RAPTOR_PARSER_NTRIPLES 1
 #define RAPTOR_PARSER_RDFXML 1
@@ -125,7 +139,8 @@
 #define RAPTOR_SERIALIZER_RDFXML_ABBREV 1
 #define RAPTOR_SERIALIZER_NTRIPLES 1
 
-#define RAPTOR_WWW_LIBCURL 1
+//#define RAPTOR_WWW_LIBCURL 1
+#define RAPTOR_WWW_LIBXML 1
 
 
 #include <windows.h>
--- misc/raptor-1.4.17/ltmain.sh	2008-02-02 22:28:24.000000000 +0900
+++ misc/build/raptor-1.4.17/ltmain.sh	2008-07-08 11:58:42.000000000 +0900
@@ -3262,9 +3262,9 @@
 	    revision="$number_revision"
 	    ;;
 	  freebsd-aout|freebsd-elf|sunos)
-	    current="$number_major"
-	    revision="$number_minor"
-	    age="0"
+	    current=`expr $number_major + $number_minor`
+	    age="$number_minor"
+	    revision="$number_revision"
 	    ;;
 	  irix|nonstopux)
 	    current=`expr $number_major + $number_minor`
@@ -3344,8 +3344,8 @@
 	  ;;
 
 	freebsd-elf)
-	  major=".$current"
-	  versuffix=".$current";
+	  major=.`expr $current - $age`
+	  versuffix="$major.$age.$revision"
 	  ;;
 
 	irix | nonstopux)
--- misc/raptor-1.4.17/configure	2008-03-17 16:37:20.000000000 +0900
+++ misc/build/raptor-1.4.17/configure	2008-07-06 09:05:29.974875000 +0900
@@ -4317,7 +4317,7 @@
 
 cygwin*)
   # func_win32_libid is a shell function defined in ltmain.sh
-  lt_cv_deplibs_check_method='file_magic ^x86 archive import|^x86 DLL'
+  lt_cv_deplibs_check_method='file_magic ^x86 archive|^x86 DLL'
   lt_cv_file_magic_cmd='func_win32_libid'
   ;;
 
@@ -4326,7 +4326,7 @@
   # func_win32_libid shell function, so use a weaker test based on 'objdump',
   # unless we find 'file', for example because we are cross-compiling.
   if ( file / ) >/dev/null 2>&1; then
-    lt_cv_deplibs_check_method='file_magic ^x86 archive import|^x86 DLL'
+    lt_cv_deplibs_check_method='file_magic ^x86 archive|^x86 DLL'
     lt_cv_file_magic_cmd='func_win32_libid'
   else
     lt_cv_deplibs_check_method='file_magic file format pei*-i386(.*architecture: i386)?'
--- misc/raptor-1.4.17/src/raptor_config.h	2008-12-26 11:49:30.000000000 +0100
+++ misc/build/raptor-1.4.17/src/raptor_config.h	2008-12-26 01:10:34.000000000 +0100
@@ -1 +1,310 @@
-dummy
+/* src/raptor_config.h.  Generated from raptor_config.h.in by configure.  */
+/* src/raptor_config.h.in.  Generated from configure.ac by autoheader.  */
+
+/* does expat crash when it sees an initial UTF8 BOM? */
+/* #undef EXPAT_UTF8_BOM_CRASH */
+
+/* vsnprint has C99 compatible return value */
+#define HAVE_C99_VSNPRINTF 1
+
+/* Have curl/curl.h */
+/* #undef HAVE_CURL_CURL_H */
+
+/* Define to 1 if you have the <dlfcn.h> header file. */
+#define HAVE_DLFCN_H 1
+
+/* Define to 1 if you have the <dmalloc.h> header file. */
+/* #undef HAVE_DMALLOC_H */
+
+/* Define to 1 if you have the <errno.h> header file. */
+#define HAVE_ERRNO_H 1
+
+/* Define to 1 if you have the <expat.h> header file. */
+//#define HAVE_EXPAT_H 1
+
+/* Define to 1 if you have the <fetch.h> header file. */
+/* #undef HAVE_FETCH_H */
+
+/* Define to 1 if you have the `getopt' function. */
+#define HAVE_GETOPT 1
+
+/* Define to 1 if you have the <getopt.h> header file. */
+#define HAVE_GETOPT_H 1
+
+/* Define to 1 if you have the `getopt_long' function. */
+#define HAVE_GETOPT_LONG 1
+
+/* INN parsedate function present */
+/* #undef HAVE_INN_PARSEDATE */
+
+/* Define to 1 if you have the <inttypes.h> header file. */
+#define HAVE_INTTYPES_H 1
+
+/* Define to 1 if you have the `isascii' function. */
+#define HAVE_ISASCII 1
+
+/* Define to 1 if you have the <libxml/hash.h> header file. */
+/* #undef HAVE_LIBXML_HASH_H */
+
+/* Define to 1 if you have the <libxml/HTMLparser.h> header file. */
+/* #undef HAVE_LIBXML_HTMLPARSER_H */
+
+/* Define to 1 if you have the <libxml/nanohttp.h> header file. */
+/* #undef HAVE_LIBXML_NANOHTTP_H */
+
+/* Define to 1 if you have the <libxml/parser.h> header file. */
+/* #undef HAVE_LIBXML_PARSER_H */
+
+/* Define to 1 if you have the <libxml/SAX2.h> header file. */
+/* #undef HAVE_LIBXML_SAX2_H */
+
+/* Define to 1 if you have the <libxslt/xslt.h> header file. */
+/* #undef HAVE_LIBXSLT_XSLT_H */
+
+/* Define to 1 if you have the <limits.h> header file. */
+#define HAVE_LIMITS_H 1
+
+/* Define to 1 if you have the <math.h> header file. */
+#define HAVE_MATH_H 1
+
+/* Define to 1 if you have the <memory.h> header file. */
+#define HAVE_MEMORY_H 1
+
+/* Raptor raptor_parse_date available */
+#define HAVE_RAPTOR_PARSE_DATE 1
+
+/* have round() in libm */
+#define HAVE_ROUND 1
+
+/* Define to 1 if you have the <stdint.h> header file. */
+#define HAVE_STDINT_H 1
+
+/* Define to 1 if you have the <stdlib.h> header file. */
+#define HAVE_STDLIB_H 1
+
+/* Define to 1 if you have the `strcasecmp' function. */
+#define HAVE_STRCASECMP 1
+
+/* Define to 1 if you have the `stricmp' function. */
+#define HAVE_STRICMP 1
+
+/* Define to 1 if you have the <strings.h> header file. */
+#define HAVE_STRINGS_H 1
+
+/* Define to 1 if you have the <string.h> header file. */
+#define HAVE_STRING_H 1
+
+/* Define to 1 if you have the <sys/param.h> header file. */
+#define HAVE_SYS_PARAM_H 1
+
+/* Define to 1 if you have the <sys/stat.h> header file. */
+#define HAVE_SYS_STAT_H 1
+
+/* Define to 1 if you have the <sys/time.h> header file. */
+#define HAVE_SYS_TIME_H 1
+
+/* Define to 1 if you have the <sys/types.h> header file. */
+#define HAVE_SYS_TYPES_H 1
+
+/* have trunc() in libm */
+#define HAVE_TRUNC 1
+
+/* Define to 1 if you have the <unistd.h> header file. */
+#define HAVE_UNISTD_H 1
+
+/* Define to 1 if you have the `vsnprintf' function. */
+#define HAVE_VSNPRINTF 1
+
+/* Define to 1 if you have the `xmlCtxtUseOptions' function. */
+/* #undef HAVE_XMLCTXTUSEOPTIONS */
+
+/* Define to 1 if you have the <xmlparse.h> header file. */
+/* #undef HAVE_XMLPARSE_H */
+
+/* Define to 1 if you have the `xmlSAX2InternalSubset' function. */
+/* #undef HAVE_XMLSAX2INTERNALSUBSET */
+
+/* need 'extern int optind' declaration? */
+/* #undef NEED_OPTIND_DECLARATION */
+
+/* Name of package */
+#define PACKAGE "raptor"
+
+/* Define to the address where bug reports for this package should be sent. */
+#define PACKAGE_BUGREPORT "http://bugs.librdf.org/"
+
+/* Define to the full name of this package. */
+#define PACKAGE_NAME "Raptor RDF Parser"
+
+/* Define to the full name and version of this package. */
+#define PACKAGE_STRING "Raptor RDF Parser 1.4.17"
+
+/* Define to the one symbol short name of this package. */
+#define PACKAGE_TARNAME "raptor"
+
+/* Define to the version of this package. */
+#define PACKAGE_VERSION "1.4.17"
+
+/* does libxml struct xmlEntity have a field etype */
+/* #undef RAPTOR_LIBXML_ENTITY_ETYPE */
+
+/* does libxml struct xmlEntity have a field name_length */
+/* #undef RAPTOR_LIBXML_ENTITY_NAME_LENGTH */
+
+/* does libxml have HTML_PARSE_NONET */
+/* #undef RAPTOR_LIBXML_HTML_PARSE_NONET */
+
+/* does libxml xmlSAXHandler have externalSubset field */
+/* #undef RAPTOR_LIBXML_XMLSAXHANDLER_EXTERNALSUBSET */
+
+/* does libxml xmlSAXHandler have initialized field */
+/* #undef RAPTOR_LIBXML_XMLSAXHANDLER_INITIALIZED */
+
+/* does libxml have XML_PARSE_NONET */
+/* #undef RAPTOR_LIBXML_XML_PARSE_NONET */
+
+/* Provide a Unicode NFC check */
+#define RAPTOR_NFC_CHECK 1
+
+/* Building GRDDL parser */
+/* #undef RAPTOR_PARSER_GRDDL */
+
+/* Building guess parser */
+#define RAPTOR_PARSER_GUESS 1
+
+/* Building Notation 3 parser */
+/* #undef RAPTOR_PARSER_N3 */
+
+/* Building N-Triples parser */
+#define RAPTOR_PARSER_NTRIPLES 1
+
+/* Building RDF/XML parser */
+#define RAPTOR_PARSER_RDFXML 1
+
+/* Building RSS Tag Soup parser */
+#define RAPTOR_PARSER_RSS 1
+
+/* Building TRiG parser */
+#define RAPTOR_PARSER_TRIG 1
+
+/* Building Turtle parser */
+#define RAPTOR_PARSER_TURTLE 1
+
+/* Building Atom 1.0 serializer */
+/* #undef RAPTOR_SERIALIZER_ATOM */
+
+/* Building GraphViz DOT serializer */
+//#define RAPTOR_SERIALIZER_DOT 1
+
+/* Building JSON serializer */
+//#define RAPTOR_SERIALIZER_JSON 1
+
+/* Building N-Triples serializer */
+#define RAPTOR_SERIALIZER_NTRIPLES 1
+
+/* Building RDF/XML serializer */
+#define RAPTOR_SERIALIZER_RDFXML 1
+
+/* Building RDF/XML-abbreviated serializer */
+#define RAPTOR_SERIALIZER_RDFXML_ABBREV 1
+
+/* Building RSS 1.0 serializer */
+#define RAPTOR_SERIALIZER_RSS_1_0 1
+
+/* Building Turtle serializer */
+//#define RAPTOR_SERIALIZER_TURTLE 1
+
+/* Release version as a decimal */
+#define RAPTOR_VERSION_DECIMAL 10417
+
+/* Major version number */
+#define RAPTOR_VERSION_MAJOR 1
+
+/* Minor version number */
+#define RAPTOR_VERSION_MINOR 4
+
+/* Release version number */
+#define RAPTOR_VERSION_RELEASE 17
+
+/* Have libcurl WWW library */
+/* #undef RAPTOR_WWW_LIBCURL */
+
+/* Have libfetch WWW library */
+/* #undef RAPTOR_WWW_LIBFETCH */
+
+/* Have libxml available as a WWW library */
+/* #undef RAPTOR_WWW_LIBXML */
+
+/* No WWW library */
+#define RAPTOR_WWW_NONE 1
+
+/* Check XML 1.1 Names */
+/* #undef RAPTOR_XML_1_1 */
+
+/* Use expat XML parser */
+//#define RAPTOR_XML_EXPAT 1
+
+/* Use libxml XML parser */
+/* #undef RAPTOR_XML_LIBXML */
+
+/* The size of `unsigned char', as computed by sizeof. */
+#define SIZEOF_UNSIGNED_CHAR 1
+
+/* The size of `unsigned int', as computed by sizeof. */
+#define SIZEOF_UNSIGNED_INT 4
+
+/* The size of `unsigned long', as computed by sizeof. */
+#define SIZEOF_UNSIGNED_LONG 4
+
+/* The size of `unsigned short', as computed by sizeof. */
+#define SIZEOF_UNSIGNED_SHORT 2
+
+/* Define to 1 if you have the ANSI C header files. */
+#define STDC_HEADERS 1
+
+/* Define to 1 if you can safely include both <sys/time.h> and <time.h>. */
+#define TIME_WITH_SYS_TIME 1
+
+/* Version number of package */
+#define VERSION "1.4.17"
+
+/* Define to 1 if your processor stores words with the most significant byte
+   first (like Motorola and SPARC, unlike Intel and VAX). */
+/* #undef WORDS_BIGENDIAN */
+
+/* Define to 1 if `lex' declares `yytext' as a `char *' by default, not a
+   `char[]'. */
+/* #undef YYTEXT_POINTER */
+
+/* Define to empty if `const' does not conform to ANSI C. */
+/* #undef const */
+
+/* For using libxml2 on win32 */
+#define RAPTOR_XML_LIBXML
+#define HAVE_LIBXML_XMLREADER_H
+#define HAVE_LIBXML_HASH_H 1
+#define HAVE_LIBXML_HTMLPARSER_H 1
+#define HAVE_LIBXML_NANOHTTP_H 1
+#define HAVE_LIBXML_PARSER_H 1
+#define HAVE_LIBXSLT_XSLT_H 1
+
+#define RAPTOR_LIBXML_HTML_PARSE_NONET 1
+#define RAPTOR_LIBXML_XML_PARSE_NONET 1
+
+/* does libxml struct xmlEntity have a field etype */
+#define RAPTOR_LIBXML_ENTITY_ETYPE
+
+/* does libxml struct xmlEntity have a field name_length */
+/*#define RAPTOR_LIBXML_ENTITY_NAME_LENGTH*/
+
+/* Define to 1 if you have the `xmlCtxtUseOptions' function. */
+#define HAVE_XMLCTXTUSEOPTIONS 1
+
+/* Define to 1 if you have the `xmlSAX2InternalSubset' function. */
+#define HAVE_XMLSAX2INTERNALSUBSET 1
+
+/* does libxml xmlSAXHandler have externalSubset field */
+#define RAPTOR_LIBXML_XMLSAXHANDLER_EXTERNALSUBSET 1
+
+/* does libxml xmlSAXHandler have initialized field */
+#define RAPTOR_LIBXML_XMLSAXHANDLER_INITIALIZED 1
