--- misc/rasqal-0.9.15/LICENSE.txt	Sun Apr 22 23:09:08 2007
+++ misc/build/rasqal-0.9.15/LICENSE.txt	Mon Jun 16 15:55:45 2008
@@ -61,3 +61,15 @@
 
    Copyright (C) 2003-2007 Dave Beckett
    Copyright (C) 2003-2005 University of Bristol
+
+
+LGPL Disclaimer
+
+For the avoidance of doubt, except that if any license choice other 
+than GPL or LGPL is available it will apply instead, Sun elects to 
+use only the Lesser General Public License version 2.1 (LGPLv2) at 
+this time for any software where a choice of LGPL license versions 
+is made available with the language indicating that LGPLv2.1 or any 
+later version may be used, or where a choice of which version of the 
+LGPL is applied is otherwise unspecified.
+
--- misc/rasqal-0.9.15/src/makefile.mk	Mon Jun 16 19:30:20 2008
+++ misc/build/rasqal-0.9.15/src/makefile.mk	Mon Jun 16 19:25:48 2008
@@ -1,1 +1,103 @@
-dummy
+#*************************************************************************
+#
+# DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+#
+# Copyright 2008 by Sun Microsystems, Inc.
+#
+# OpenOffice.org - a multi-platform office productivity suite
+#
+# $RCSfile: rasqal-0.9.15.patch,v $
+#
+# $Revision: 1.2.6.2 $
+#
+# This file is part of OpenOffice.org.
+#
+# OpenOffice.org is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Lesser General Public License version 3
+# only, as published by the Free Software Foundation.
+#
+# OpenOffice.org is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Lesser General Public License version 3 for more details
+# (a copy is included in the LICENSE file that accompanied this code).
+#
+# You should have received a copy of the GNU Lesser General Public License
+# version 3 along with OpenOffice.org.  If not, see
+# <http://www.openoffice.org/license.html>
+# for a copy of the LGPLv3 License.
+#
+#*************************************************************************
+
+PRJ=..$/..$/..$/..$/..
+
+PRJNAME=redland
+TARGET=rasqal
+LIBTARGET=NO
+UWINAPILIB=
+
+EXTERNAL_WARNINGS_NOT_ERRORS=TRUE
+
+
+# --- Settings -----------------------------------------------------
+
+.INCLUDE :  settings.mk
+
+.IF "$(GUI)"=="OS2"
+CDEFS+=-DRASQAL_INTERNAL -DHAVE_CONFIG_H -DNDEBUG -D_MT
+.ELSE
+CDEFS+=-DRASQAL_INTERNAL -DWIN32 -DNDEBUG -D_WINDOWS -D_USRDLL -DWIN32_EXPORTS -D_MT
+.ENDIF
+
+# --- Files --------------------------------------------------------
+
+SLOFILES =	\
+   $(SLO)$/rasqal_datetime.obj \
+   $(SLO)$/rasqal_decimal.obj \
+   $(SLO)$/rasqal_engine.obj \
+   $(SLO)$/rasqal_expr.obj \
+   $(SLO)$/rasqal_feature.obj \
+   $(SLO)$/rasqal_general.obj \
+   $(SLO)$/rasqal_graph_pattern.obj \
+   $(SLO)$/rasqal_literal.obj \
+   $(SLO)$/rasqal_map.obj \
+   $(SLO)$/rasqal_query_results.obj \
+   $(SLO)$/rasqal_query.obj \
+   $(SLO)$/rasqal_raptor.obj \
+   $(SLO)$/rasqal_result_formats.obj \
+   $(SLO)$/rasqal_xsd_datatypes.obj \
+   $(SLO)$/rdql_lexer.obj \
+   $(SLO)$/rdql_parser.obj \
+   $(SLO)$/sparql_lexer.obj \
+   $(SLO)$/sparql_parser.obj \
+   $(SLO)$/strcasecmp.obj \
+
+
+#   $(SLO)$/rasqal_redland.obj \
+
+SHL1DEPN=   makefile.mk
+SHL1OBJS=   $(SLOFILES)
+SHL1USE_EXPORTS=name
+
+#SHL1TARGET=	$(TARGET)$(DLLPOSTFIX)
+SHL1TARGET=	$(TARGET)
+SHL1IMPLIB= lib$(TARGET)
+
+#SHL1VERSIONMAP=../service/exports.map
+.IF "$(GUI)"=="OS2"
+SHL1VERSIONMAP=$(PRJ)$/rasqal$/rasqal2.map
+.ENDIF
+SHL1DEF=$(MISC)$/$(SHL1TARGET).def
+DEF1NAME=$(SHL1TARGET)
+
+SHL1STDLIBS= \
+    $(PRJ)$/$(INPATH)$/lib$/libraptor.lib \
+
+
+#$(LIBXML2LIB) \
+#    $(XSLTLIB) \
+
+# --- Targets ------------------------------------------------------
+
+.INCLUDE :  target.mk
+
--- misc/rasqal-0.9.15/src/rasqal-config.in	Sun Apr 22 22:58:51 2007
+++ misc/build/rasqal-0.9.15/src/rasqal-config.in	Mon Jun 16 15:55:45 2008
@@ -22,13 +22,22 @@
 # 
 #
 
-prefix=@prefix@
+prefix=${SOLARVERSION}/${INPATH}
+exec_prefix=${SOLARVERSION}/${INPATH}
+#includedir=${SOLARVERSION}/${INPATH}/inc${UPDMINOREXT}/external
+#libdir=${SOLARVERSION}/${INPATH}/lib${UPDMINOREXT}
+# HACK: map to build directory
+includedir=../../../../inc
+libdir=../../../../lib
+includes=-I${includedir}
 
-if test "@includedir@" != /usr/include ; then
-  includes="-I@includedir@"
-else
-  includes=
-fi
+#prefix=@prefix@
+#
+#if test "@includedir@" != /usr/include ; then
+#  includes="-I@includedir@"
+#else
+#  includes=
+#fi
 
 
 usage()
@@ -109,7 +118,8 @@
   echo $includes
 fi
 if test "$echo_libs" = "yes"; then
-  echo -L@libdir@ -lrasqal
+#  echo -L@libdir@ -lrasqal
+  echo -L${libdir} -lrasqal
 fi
 if test "$echo_libtool_libs" = "yes"; then
   echo @libdir@/@RASQAL_LIBTOOLLIBS@
--- misc/rasqal-0.9.15/src/win32_rasqal_config.h	Mon Apr 23 09:18:27 2007
+++ misc/build/rasqal-0.9.15/src/win32_rasqal_config.h	Mon Jun 16 15:55:45 2008
@@ -39,10 +39,13 @@
 
 #define HAVE_STDLIB_H 1
 
+#define HAVE_ERRNO_H 1
+
 #define HAVE_STRICMP 1
 
 /* MS names for these functions */
-#define vsnprintf _vsnprintf
+// next line breaks build on wntmsci12
+//#define vsnprintf _vsnprintf
 #define snprintf _snprintf
 #define access _access
 #define stricmp _stricmp
@@ -87,6 +90,9 @@
 #include <io.h>
 #include <memory.h>
 
+// argh! stupid winnt.h...
+#undef DELETE
+
 /* This is a SPARQL token define */
 #ifdef OPTIONAL
 #undef OPTIONAL
@@ -118,8 +124,14 @@
 /* #define RAPTOR_TRIPLES_SOURCE_REDLAND 1 */
 
 /* Use PCRE regex library */
-#define RASQAL_REGEX_PCRE 1
+//#define RASQAL_REGEX_PCRE 1
 
+#define HAVE_LIBXML_SCHEMASINTERNALS_H 1
+#define HAVE_LIBXML_XMLSCHEMASTYPES_H 1
+#define HAVE_LIBXML_XMLSCHEMAS_H 1
+
+#define RASQAL_DECIMAL_NONE 1
+
 #ifdef _DEBUG
 #define RASQAL_DEBUG 1
 #endif
--- misc/rasqal-0.9.15/ltmain.sh	2006-03-12 03:49:04.000000000 +0900
+++ misc/build/rasqal-0.9.15/ltmain.sh	2008-07-06 14:30:56.975375000 +0900
@@ -1518,6 +1518,11 @@
 	[\\/]* | [A-Za-z]:[\\/]*) ;;
 	*)
 	  absdir=`cd "$dir" && pwd`
+	  case $host_os in
+	    mingw*)
+	      absdir=`cygpath -m $absdir`
+	      ;;
+	    esac
 	  if test -z "$absdir"; then
 	    $echo "$modename: cannot determine absolute directory name of \`$dir'" 1>&2
 	    absdir="$dir"
@@ -3195,9 +3200,9 @@
 	    revision="$number_revision"
 	    ;;
 	  freebsd-aout|freebsd-elf|sunos)
-	    current="$number_major"
-	    revision="$number_minor"
-	    age="0"
+	    current=`expr $number_major + $number_minor`
+	    age="$number_minor"
+	    revision="$number_revision"
 	    ;;
 	  irix|nonstopux)
 	    current=`expr $number_major + $number_minor - 1`
@@ -3275,8 +3280,8 @@
 	  ;;
 
 	freebsd-elf)
-	  major=".$current"
-	  versuffix=".$current";
+	  major=.`expr $current - $age`
+	  versuffix="$major.$age.$revision"
 	  ;;
 
 	irix | nonstopux)
--- misc/rasqal-0.9.15/src/rasqal_config.h	Fri Dec 26 00:18:44 2008
+++ misc/build/rasqal-0.9.15/src/rasqal_config.h	Wed Nov 26 23:35:40 2008
@@ -1 +1,169 @@
-dummy
+/* src/rasqal_config.h.  Generated from rasqal_config.h.in by configure.  */
+/* src/rasqal_config.h.in.  Generated from configure.ac by autoheader.  */
+
+/* Define to 1 if you have the <dlfcn.h> header file. */
+#define HAVE_DLFCN_H 1
+
+/* Define to 1 if you have the <dmalloc.h> header file. */
+/* #undef HAVE_DMALLOC_H */
+
+/* Define to 1 if you have the <errno.h> header file. */
+#define HAVE_ERRNO_H 1
+
+/* Define to 1 if you have the `getopt' function. */
+#define HAVE_GETOPT 1
+
+/* Define to 1 if you have the <getopt.h> header file. */
+#define HAVE_GETOPT_H 1
+
+/* Define to 1 if you have the `getopt_long' function. */
+#define HAVE_GETOPT_LONG 1
+
+/* Define to 1 if you have the <gmp.h> header file. */
+#define HAVE_GMP_H 1
+
+/* Define to 1 if you have the <inttypes.h> header file. */
+#define HAVE_INTTYPES_H 1
+
+/* Define to 1 if you have the <libxml/schemasInternals.h> header file. */
+/* #undef HAVE_LIBXML_SCHEMASINTERNALS_H */
+
+/* Define to 1 if you have the <libxml/xmlschemastypes.h> header file. */
+/* #undef HAVE_LIBXML_XMLSCHEMASTYPES_H */
+
+/* Define to 1 if you have the <libxml/xmlschemas.h> header file. */
+/* #undef HAVE_LIBXML_XMLSCHEMAS_H */
+
+/* Define to 1 if you have the <memory.h> header file. */
+#define HAVE_MEMORY_H 1
+
+/* Define to 1 if you have the <mpfr.h> header file. */
+/* #undef HAVE_MPFR_H */
+
+/* Define to 1 if you have the <regex.h> header file. */
+#define HAVE_REGEX_H 1
+
+/* have PCRE regex - Perl Compatible Regular Expressions */
+#define HAVE_REGEX_PCRE 1
+
+/* Define to 1 if you have the `srandomdev' function. */
+#define HAVE_SRANDOMDEV 1
+
+/* Define to 1 if you have the <stdint.h> header file. */
+#define HAVE_STDINT_H 1
+
+/* Define to 1 if you have the <stdlib.h> header file. */
+#define HAVE_STDLIB_H 1
+
+/* Define to 1 if you have the `strcasecmp' function. */
+#define HAVE_STRCASECMP 1
+
+/* Define to 1 if you have the `stricmp' function. */
+#define HAVE_STRICMP 1
+
+/* Define to 1 if you have the <strings.h> header file. */
+#define HAVE_STRINGS_H 1
+
+/* Define to 1 if you have the <string.h> header file. */
+#define HAVE_STRING_H 1
+
+/* Define to 1 if you have the <sys/stat.h> header file. */
+#define HAVE_SYS_STAT_H 1
+
+/* Define to 1 if you have the <sys/time.h> header file. */
+#define HAVE_SYS_TIME_H 1
+
+/* Define to 1 if you have the <sys/types.h> header file. */
+#define HAVE_SYS_TYPES_H 1
+
+/* Define to 1 if you have the <unistd.h> header file. */
+#define HAVE_UNISTD_H 1
+
+/* Define to 1 if you have the `vsnprintf' function. */
+#define HAVE_VSNPRINTF 1
+
+/* need 'extern int optind' declaration? */
+/* #undef NEED_OPTIND_DECLARATION */
+
+/* Name of package */
+#define PACKAGE "rasqal"
+
+/* Define to the address where bug reports for this package should be sent. */
+#define PACKAGE_BUGREPORT "http://bugs.librdf.org/"
+
+/* Define to the full name of this package. */
+#define PACKAGE_NAME "Rasqal RDF Query Library"
+
+/* Define to the full name and version of this package. */
+#define PACKAGE_STRING "Rasqal RDF Query Library 0.9.15"
+
+/* Define to the one symbol short name of this package. */
+#define PACKAGE_TARNAME "rasqal"
+
+/* Define to the version of this package. */
+#define PACKAGE_VERSION "0.9.15"
+
+/* Use raptor to provide triples */
+#define RAPTOR_TRIPLES_SOURCE_RAPTOR 1
+
+/* Use redland to provide triples */
+/* #undef RAPTOR_TRIPLES_SOURCE_REDLAND */
+
+/* Decimal from GNU MP library */
+/* #undef RASQAL_DECIMAL_GMP */
+
+/* Decimal from MPFR library */
+/* #undef RASQAL_DECIMAL_MPFR */
+
+/* Decimal without a library */
+#define RASQAL_DECIMAL_NONE 1
+
+/* Building LAQRS query */
+/* #undef RASQAL_QUERY_LAQRS */
+
+/* Building RDQL query */
+#define RASQAL_QUERY_RDQL 1
+
+/* Building SPARQL query */
+#define RASQAL_QUERY_SPARQL 1
+
+/* Use no regex library */
+/* #undef RASQAL_REGEX_NONE */
+
+/* Use PCRE regex library */
+/* #undef RASQAL_REGEX_PCRE */
+
+/* Use posix regex library */
+#define RASQAL_REGEX_POSIX 1
+
+/* Release version as a decimal */
+#define RASQAL_VERSION_DECIMAL 915
+
+/* Major version number */
+#define RASQAL_VERSION_MAJOR 0
+
+/* Minor version number */
+#define RASQAL_VERSION_MINOR 9
+
+/* Release version number */
+#define RASQAL_VERSION_RELEASE 15
+
+/* Define to 1 if you have the ANSI C header files. */
+#define STDC_HEADERS 1
+
+/* Define to 1 if you can safely include both <sys/time.h> and <time.h>. */
+#define TIME_WITH_SYS_TIME 1
+
+/* Version number of package */
+#define VERSION "0.9.15"
+
+/* Define to 1 if your processor stores words with the most significant byte
+   first (like Motorola and SPARC, unlike Intel and VAX). */
+/* #undef WORDS_BIGENDIAN */
+
+/* Define to 1 if `lex' declares `yytext' as a `char *' by default, not a
+   `char[]'. */
+/* #undef YYTEXT_POINTER */
+
+/* Define to empty if `const' does not conform to ANSI C. */
+/* #undef const */
