--- misc/freetype-2.4.4/config.mk	Wed Aug 20 14:33:55 2008
+++ misc/build/freetype-2.4.4/config.mk	Wed Aug 20 14:26:42 2008
@@ -1 +1,42 @@
-dummy
+#
+# FreeType 2 configuration rules for a `normal' pseudo ANSI compiler/system
+#
+
+
+# Copyright 1996-2000 by
+# David Turner, Robert Wilhelm, and Werner Lemberg.
+#
+# This file is part of the FreeType project, and may only be used, modified,
+# and distributed under the terms of the FreeType project license,
+# LICENSE.TXT.  By continuing to use, modify, or distribute this file you
+# indicate that you have read the license and understand and accept it
+# fully.
+all modules: setup
+
+CPPFLAGS ?= /DNDEBUG /DWIN32 /D_LIB /D_CRT_SECURE_NO_WARNINGS $(SOLARINC)
+
+include $(TOP_DIR)/builds/win32/win32-def.mk
+
+DELETE    := rm -f
+CAT       := cat
+SEP       := /
+BUILD_PROJECT := yes
+COPY := $(GNUCOPY)
+visualc: setup
+
+include $(TOP_DIR)/builds/compiler/visualc.mk
+include $(TOP_DIR)/builds/$(PROJECT).mk
+LINK_LIBRARY  = link /MAP /OPT:NOREF -safeseh -nxcompat -dynamicbase /DLL /SUBSYSTEM:CONSOLE -LIBPATH:$(OUTDIR)/lib -NODEFAULTLIB -RELEASE -INCREMENTAL:NO /nologo -map:../../../misc/$(PROJECT).map -implib:../../../lib/i$(PROJECT).lib -out:$(subst /,$(COMPILER_SEP),../../../bin/$(PROJECT).dll $(OBJECTS_LIST)) kernel32.lib msvcrt.lib
+LIB_LIBRARY = lib -machine:IX86 /IGNORE:4102 /IGNORE:4197 -out:../../../lib/i$(PROJECT).lib -def:$(PROJECT).def
+exported_symbols:
+	
+
+$(PROJECT_LIBRARY): $(OBJECTS_LIST)
+	  $(LINK_LIBRARY)
+	  $(LIB_LIBRARY)
+
+# include $(TOP_DIR)/builds/link_dos.mk
+
+
+# EOF
+
--- misc/freetype-2.4.4/objs/ftmodule.h	Wed Aug 20 14:33:55 2008
+++ misc/build/freetype-2.4.4/objs/ftmodule.h	Wed Aug 20 14:26:42 2008
@@ -1 +1,21 @@
-dummy
+/* This is a generated file. */
+FT_USE_MODULE( FT_Driver_ClassRec, tt_driver_class )
+FT_USE_MODULE( FT_Driver_ClassRec, t1_driver_class )
+FT_USE_MODULE( FT_Driver_ClassRec, cff_driver_class )
+FT_USE_MODULE( FT_Driver_ClassRec, t1cid_driver_class )
+FT_USE_MODULE( FT_Driver_ClassRec, pfr_driver_class )
+FT_USE_MODULE( FT_Driver_ClassRec, t42_driver_class )
+FT_USE_MODULE( FT_Driver_ClassRec, winfnt_driver_class )
+FT_USE_MODULE( FT_Driver_ClassRec, pcf_driver_class )
+FT_USE_MODULE( FT_Driver_ClassRec, bdf_driver_class )
+FT_USE_MODULE( FT_Module_Class, sfnt_module_class )
+FT_USE_MODULE( FT_Module_Class, autofit_module_class )
+FT_USE_MODULE( FT_Module_Class, pshinter_module_class )
+FT_USE_MODULE( FT_Renderer_Class, ft_raster1_renderer_class )
+FT_USE_MODULE( FT_Renderer_Class, ft_smooth_renderer_class )
+FT_USE_MODULE( FT_Renderer_Class, ft_smooth_lcd_renderer_class )
+FT_USE_MODULE( FT_Renderer_Class, ft_smooth_lcdv_renderer_class )
+FT_USE_MODULE( FT_Module_Class, psaux_module_class )
+FT_USE_MODULE( FT_Module_Class, psnames_module_class )
+/* EOF */
+
--- misc/freetype-2.4.4/freetype.def	Wed Aug 20 14:33:55 2008
+++ misc/build/freetype-2.4.4/freetype.def	Wed Aug 20 14:26:42 2008
@@ -1 +1,58 @@
-dummy
+LIBRARY		"freetype"
+
+VERSION		2.4.4
+
+EXPORTS
+    FT_Init_FreeType						@1
+	FT_Done_FreeType						@2
+	FT_New_Face								@3
+	FT_New_Memory_Face						@4
+	FT_Open_Face							@5
+	FT_Attach_File							@6
+	FT_Attach_Stream						@7
+	FT_Reference_Face						@8
+	FT_Done_Face							@9
+	FT_Select_Size							@10
+	FT_Request_Size							@11
+	FT_Set_Char_Size						@12
+	FT_Set_Pixel_Sizes						@13
+	FT_Load_Glyph							@14
+	FT_Load_Char							@15
+	FT_Set_Transform						@16
+	FT_Render_Glyph							@17
+	FT_Get_Kerning							@18
+	FT_Get_Track_Kerning					@19
+	FT_Get_Glyph_Name						@20
+	FT_Get_Postscript_Name					@21
+	FT_Select_Charmap						@22
+	FT_Set_Charmap							@23
+	FT_Get_Charmap_Index					@24
+	FT_Get_Char_Index						@25
+	FT_Get_First_Char						@26
+	FT_Get_Next_Char						@27
+	FT_Get_Name_Index						@28
+	FT_Get_SubGlyph_Info					@29
+	FT_Get_FSType_Flags						@30
+	FT_Face_GetCharVariantIndex				@31
+	FT_Face_GetCharVariantIsDefault			@32
+	FT_Face_GetVariantSelectors				@33
+	FT_Face_GetVariantsOfChar				@34
+	FT_Face_GetCharsOfVariant				@35
+	FT_MulDiv								@36
+	FT_DivFix								@37
+	FT_RoundFix								@38
+	FT_CeilFix								@39
+	FT_FloorFix								@40
+	FT_Vector_Transform						@41
+	FT_Library_Version						@42
+	FT_Face_CheckTrueTypePatents			@43
+	FT_Face_SetUnpatentedHinting			@44
+	FT_Get_X11_Font_Format					@45
+	FT_Has_PS_Glyph_Names					@46
+	FT_Get_Sfnt_Name_Count					@47
+	FT_Get_PS_Font_Info						@48
+	FT_Get_Sfnt_Name						@49
+	FT_Get_Sfnt_Table						@50
+	FT_Load_Sfnt_Table						@51
+	FT_Get_BDF_Property						@52
+
