--- misc/expat-2.1.0/lib/expat_external.h	2009-11-16 08:53:17.375000000 +0000
+++ misc/build/expat-2.1.0/lib/expat_external.h	2009-11-16 08:53:34.703125000 +0000
@@ -81,10 +81,6 @@
 #  ifndef XML_BUILDING_EXPAT
 /* using Expat from an application */
 
-#    if defined(_MSC_EXTENSIONS) && ! defined(__BEOS__) && ! defined(__CYGWIN__)
-#      define XMLIMPORT __declspec(dllimport)
-#    endif
-
 #  endif
 #endif /* not defined XML_STATIC */
 
--- misc/expat-2.1.0/lib/xmlparse.c	2021-05-23 16:56:25.000000000 +0100
+++ misc/build/expat-2.1.0/lib/xmlparse.c	2021-05-25 12:42:11.997173600 +0100
@@ -92,6 +92,11 @@
 
 #include <expat_config.h>
 
+#ifdef _WIN32
+#  undef HAVE_GETRANDOM
+#  undef HAVE_SYSCALL_GETRANDOM
+#endif
+
 #include "ascii.h"
 #include "expat.h"
 #include "siphash.h"
