--- misc/expat-2.5.0/lib/expat_external.h	2022-10-25 01:32:54.000000000 +0900
+++ misc/build/expat-2.5.0/lib/expat_external.h	2022-10-30 23:09:47.339459134 +0900
@@ -88,10 +88,6 @@
 #  ifndef XML_BUILDING_EXPAT
 /* using Expat from an application */
 
-#    if defined(_MSC_EXTENSIONS) && ! defined(__BEOS__) && ! defined(__CYGWIN__)
-#      define XMLIMPORT __declspec(dllimport)
-#    endif
-
 #  endif
 #endif /* not defined XML_STATIC */
 
--- misc/expat-2.5.0/lib/xmlparse.c	2022-10-26 00:09:08.000000000 +0900
+++ misc/build/expat-2.5.0/lib/xmlparse.c	2022-10-30 23:09:01.843006341 +0900
@@ -67,6 +67,9 @@
 #endif
 
 #ifdef _WIN32
+#  undef HAVE_ARC4RANDOM_BUF
+#  undef HAVE_GETRANDOM
+#  undef HAVE_SYSCALL_GETRANDOM
 /* force stdlib to define rand_s() */
 #  if ! defined(_CRT_RAND_S)
 #    define _CRT_RAND_S
