--- misc/expat-2.1.0/lib/xmlparse.c	2017-06-22 15:44:06.000000000 +0200
+++ misc/build/expat-2.1.0/lib/xmlparse.c	2017-06-22 15:50:36.000000000 +0200
@@ -696,6 +696,8 @@
 
 
 #if defined(HAVE_GETRANDOM) || defined(HAVE_SYSCALL_GETRANDOM)
+/* Patch buffer to undefined */
+#undef buffer
 # include <errno.h>
 
 # if defined(HAVE_GETRANDOM)
@@ -704,6 +706,9 @@
 #  include <unistd.h>        /* syscall */
 #  include <sys/syscall.h>   /* SYS_getrandom */
 # endif
+/* Set buffer back */
+#define buffer (parser->m_buffer)
+
 
 /* Obtain entropy on Linux 3.17+ */
 static int
