--- src/hb-coretext.cc
+++ src/hb-coretext.cc
@@ -161,7 +161,19 @@
       return ct_font;
   }
 
-  CFURLRef original_url = (CFURLRef)CTFontCopyAttribute(ct_font, kCTFontURLAttribute);
+  CFURLRef original_url = nullptr;
+  //original_url = (CFURLRef)CTFontCopyAttribute(ct_font, kCTFontURLAttribute);
+  /* CTFontDescriptor doesn't support kCTFontURLAttribute prior to 10.6,
+     so let’s go CTFont -> ATSFont -> FSRef -> CFURL */
+  {
+      ATSFontRef atsFont = CTFontGetPlatformFont( ct_font, nullptr );
+      if ( atsFont ) {
+        FSRef fsref;
+        OSStatus code = ATSFontGetFileReference( atsFont, &fsref );
+        if ( code == noErr )
+          original_url = CFURLCreateFromFSRef( nullptr, &fsref );
+      }
+  }
 
   /* Create font copy with cascade list that has LastResort first; this speeds up CoreText
    * font fallback which we don't need anyway. */
@@ -180,7 +192,20 @@
        * system locations that we cannot access from the sandboxed renderer
        * process in Blink. This can be detected by the new file URL location
        * that the newly found font points to. */
-      CFURLRef new_url = (CFURLRef) CTFontCopyAttribute (new_ct_font, kCTFontURLAttribute);
+
+      CFURLRef new_url = nullptr;
+      //new_url = (CFURLRef) CTFontCopyAttribute (new_ct_font, kCTFontURLAttribute);
+      /* CTFontDescriptor doesn't support kCTFontURLAttribute prior to 10.6 */
+      {
+          ATSFontRef atsFont = CTFontGetPlatformFont( new_ct_font, nullptr );
+          if ( atsFont ) {
+            FSRef fsref;
+            OSStatus code = ATSFontGetFileReference( atsFont, &fsref );
+            if ( code == noErr )
+              new_url = CFURLCreateFromFSRef( nullptr, &fsref );
+          }
+      }
+
       // Keep reconfigured font if URL cannot be retrieved (seems to be the case
       // on Mac OS 10.12 Sierra), speculative fix for crbug.com/625606
       if (!original_url || !new_url || CFEqual (original_url, new_url)) {
