--- src/hb-coretext.cc
+++ src/hb-coretext.cc
@@ -749,6 +749,8 @@
 
       int level = HB_DIRECTION_IS_FORWARD (buffer->props.direction) ? 0 : 1;
       CFNumberRef level_number = CFNumberCreate (kCFAllocatorDefault, kCFNumberIntType, &level);
+/* Marked as available in version 10.5 and later, but not in the header until SDK 10.6 */
+      extern const CFStringRef kCTTypesetterOptionForcedEmbeddingLevel;
       CFDictionaryRef options = CFDictionaryCreate (kCFAllocatorDefault,
 						    (const void **) &kCTTypesetterOptionForcedEmbeddingLevel,
 						    (const void **) &level_number,

Diff finished.  Mon Dec 15 16:56:50 2014
