--- src/Makefile.in
+++ src/Makefile.in
@@ -149,9 +149,7 @@
 @HAVE_GOBJECT_TRUE@	$(HB_GOBJECT_ENUM_headers) \
 @HAVE_GOBJECT_TRUE@	$(NULL)
 
-noinst_PROGRAMS = main$(EXEEXT) test$(EXEEXT) \
-	test-buffer-serialize$(EXEEXT) test-size-params$(EXEEXT) \
-	test-would-substitute$(EXEEXT) $(am__EXEEXT_1)
+noinst_PROGRAMS = main$(EXEEXT) $(am__EXEEXT_1)
 bin_PROGRAMS =
 check_PROGRAMS = test-ot-tag$(EXEEXT) $(am__EXEEXT_1)
 TESTS = $(am__EXEEXT_2) $(check_PROGRAMS)
