--- misc/Python-2.7.6/configure	2012-04-09 18:07:36.000000000 -0500
+++ misc/build/Python-2.7.6/configure	2012-07-22 15:01:20.000000000 -0500
@@ -2982,7 +2982,9 @@
 if test -z "$MACHDEP"
 then
 	ac_sys_system=`uname -s`
-	if test "$ac_sys_system" = "AIX" \
+	if test -n "$OOO_SYSBASE_SYS_RELEASE"; then
+		ac_sys_release=$OOO_SYSBASE_SYS_RELEASE
+	elif test "$ac_sys_system" = "AIX" \
 	-o "$ac_sys_system" = "UnixWare" -o "$ac_sys_system" = "OpenUNIX"; then
 		ac_sys_release=`uname -v`
 	else
