--- misc/Python-2.7.6/Include/pyport.h	2012-04-09 18:07:29.000000000 -0500
+++ misc/build/Python-2.7.6/Include/pyport.h	2012-07-24 18:19:52.000000000 -0500
@@ -603,11 +603,6 @@
 in platform-specific #ifdefs.
 **************************************************************************/
 
-#ifdef SOLARIS
-/* Unchecked */
-extern int gethostname(char *, int);
-#endif
-
 #ifdef __BEOS__
 /* Unchecked */
 /* It's in the libs, but not the headers... - [cjh] */
