diff -ruN misc/Python-2.7.8/PCbuild/build_ssl.py misc/build/Python-2.7.8/PCbuild/build_ssl.py
--- misc/Python-2.7.8/PCbuild/build_ssl.py	2014-06-29 21:05:46.000000000 -0500
+++ misc/build/Python-2.7.8/PCbuild/build_ssl.py	2016-01-01 15:15:32.256357072 -0500
@@ -229,4 +229,4 @@
     sys.exit(rc)
 
 if __name__=='__main__':
-    main()
+    sys.exit(0)
diff -ruN misc/Python-2.7.8/setup.py misc/build/Python-2.7.8/setup.py
--- misc/Python-2.7.8/setup.py	2014-06-29 21:05:48.000000000 -0500
+++ misc/build/Python-2.7.8/setup.py	2016-01-01 15:15:32.258356209 -0500
@@ -786,7 +786,15 @@
                                depends=['socketmodule.h'],
                                libraries=math_libs) )
         # Detect SSL support for the socket module (via _ssl)
+        UPDMINOR = os.environ.get('UPDMINOREXT')
+        ooosslinc = os.environ.get('SOLARVERSION') + '/' + \
+            os.environ.get('INPATH') + '/' + \
+            'inc'
+        if UPDMINOR != None:
+            ooosslinc = ooosslinc + UPDMINOR
+        ooosslinc = ooosslinc + '/external/'
         search_for_ssl_incs_in = [
+                              ooosslinc,
                               '/usr/local/ssl/include',
                               '/usr/contrib/ssl/include/'
                              ]
@@ -798,8 +806,15 @@
                                ['/usr/kerberos/include'])
             if krb5_h:
                 ssl_incs += krb5_h
+        ooossllib = os.environ.get('SOLARVER') + '/' + \
+            os.environ.get('INPATH') + '/' + \
+            'lib'
+        if UPDMINOR != None:
+            ooosslinc = ooosslinc + UPDMINOR
+        ooosslinc = ooosslinc + '/'
         ssl_libs = find_library_file(self.compiler, 'ssl',lib_dirs,
-                                     ['/usr/local/ssl/lib',
+                                     [ooossllib,
+                                      '/usr/local/ssl/lib',
                                       '/usr/contrib/ssl/lib/'
                                      ] )
 
