diff -uNrp misc/Python-2.6.1/configure misc/build/Python-2.6.1/configure
--- misc/Python-2.6.1/configure	2012-06-09 16:56:26.914428547 -0300
+++ misc/build/Python-2.6.1/configure	2012-06-09 16:58:30.606432414 -0300
@@ -4332,7 +4332,7 @@ if test $SVNVERSION = found
 then
 	SVNVERSION="svnversion \$(srcdir)"
 else
-	SVNVERSION="echo exported"
+	SVNVERSION="echo Unversioned directory"
 fi
 
 case $MACHDEP in
diff -uNrp misc/Python-2.6.1/configure.in misc/build/Python-2.6.1/configure.in
--- misc/Python-2.6.1/configure.in	2012-06-09 16:56:26.917428547 -0300
+++ misc/build/Python-2.6.1/configure.in	2012-06-09 16:58:30.612432414 -0300
@@ -760,7 +760,7 @@ if test $SVNVERSION = found
 then
 	SVNVERSION="svnversion \$(srcdir)"
 else
-	SVNVERSION="echo exported"
+	SVNVERSION="echo Unversioned directory"
 fi
 
 case $MACHDEP in
diff -uNrp misc/Python-2.6.1/Makefile.pre.in misc/build/Python-2.6.1/Makefile.pre.in
--- misc/Python-2.6.1/Makefile.pre.in	2012-06-09 16:56:26.907428547 -0300
+++ misc/build/Python-2.6.1/Makefile.pre.in	2012-06-09 16:58:30.587432414 -0300
@@ -498,7 +498,7 @@ Modules/getbuildinfo.o: $(PARSER_OBJS) \
 		$(SIGNAL_OBJS) \
 		$(MODOBJS) \
 		$(srcdir)/Modules/getbuildinfo.c
-	$(CC) -c $(PY_CFLAGS) -DSVNVERSION=\"`LC_ALL=C $(SVNVERSION)`\" -o $@ $(srcdir)/Modules/getbuildinfo.c
+	$(CC) -c $(PY_CFLAGS) -DSVNVERSION="\"`LC_ALL=C $(SVNVERSION)`\"" -o $@ $(srcdir)/Modules/getbuildinfo.c
 
 Modules/getpath.o: $(srcdir)/Modules/getpath.c Makefile
 	$(CC) -c $(PY_CFLAGS) -DPYTHONPATH='"$(PYTHONPATH)"' \
diff -uNrp misc/Python-2.6.1/Misc/ACKS misc/build/Python-2.6.1/Misc/ACKS
--- misc/Python-2.6.1/Misc/ACKS	2008-11-03 13:18:30.000000000 -0200
+++ misc/build/Python-2.6.1/Misc/ACKS	2012-06-09 16:58:30.588432414 -0300
@@ -21,6 +21,7 @@ John Anderson
 Erik Andersén
 Oliver Andrich
 Ross Andrus
+Arfrever Frehtes Taifersar Arahesis
 Jason Asbahr
 David Ascher
 Chris AtLee
diff -uNrp misc/Python-2.6.1/Modules/getbuildinfo.c misc/build/Python-2.6.1/Modules/getbuildinfo.c
--- misc/Python-2.6.1/Modules/getbuildinfo.c	2007-06-07 20:53:49.000000000 -0300
+++ misc/build/Python-2.6.1/Modules/getbuildinfo.c	2012-06-09 16:58:30.593432414 -0300
@@ -48,5 +48,5 @@ _Py_svnversion(void)
 	static const char svnversion[] = SVNVERSION;
 	if (svnversion[0] != '$')
 		return svnversion; /* it was interpolated, or passed on command line */
-	return "exported";
+	return "Unversioned directory";
 }
diff -uNrp misc/Python-2.6.1/Python/sysmodule.c misc/build/Python-2.6.1/Python/sysmodule.c
--- misc/Python-2.6.1/Python/sysmodule.c	2008-07-10 14:13:55.000000000 -0300
+++ misc/build/Python-2.6.1/Python/sysmodule.c	2012-06-09 16:58:30.595432414 -0300
@@ -1161,7 +1161,7 @@ svnversion_init(void)
 
 
 	svnversion = _Py_svnversion();
-	if (strcmp(svnversion, "exported") != 0)
+	if (strcmp(svnversion, "Unversioned directory") != 0 && strcmp(svnversion, "exported") != 0)
 		svn_revision = svnversion;
 	else if (istag) {
 		len = strlen(_patchlevel_revision);
