diff -uNrp misc/build/Python-2.6.1/configure misc/Python-2.6.1/configure
--- misc/build/Python-2.6.1/configure	2008-11-16 15:57:10.000000000 -0200
+++ misc/Python-2.6.1/configure	2011-11-19 21:10:49.259397448 -0300
@@ -2055,6 +2055,7 @@ then
 	MACHDEP="$ac_md_system$ac_md_release"
 
 	case $MACHDEP in
+    linux*) MACHDEP="linux2";;
 	cygwin*) MACHDEP="cygwin";;
 	darwin*) MACHDEP="darwin";;
 	atheos*) MACHDEP="atheos";;
diff -uNrp misc/build/Python-2.6.1/configure.in misc/Python-2.6.1/configure.in
--- misc/build/Python-2.6.1/configure.in	2008-11-16 06:02:56.000000000 -0200
+++ misc/Python-2.6.1/configure.in	2011-11-19 21:10:55.388397133 -0300
@@ -229,6 +229,7 @@ then
 	MACHDEP="$ac_md_system$ac_md_release"
 
 	case $MACHDEP in
+    linux*) MACHDEP="linux2";;
 	cygwin*) MACHDEP="cygwin";;
 	darwin*) MACHDEP="darwin";;
 	atheos*) MACHDEP="atheos";;
