diff -ur libgltf.org/src/RenderScene.cpp libgltf/src/RenderScene.cpp
--- libgltf.org/src/RenderScene.cpp	2014-05-05 14:45:21.447395883 +0200
+++ libgltf/src/RenderScene.cpp	2014-05-05 14:47:14.819400718 +0200
@@ -776,7 +776,7 @@
     return;
 }
 
-void RenderScene::updateBoneMatrix(Bone* pBone, glm::mat4& matrix)
+void RenderScene::updateBoneMatrix(Bone* pBone, const glm::mat4& matrix)
 {
     if (0 == pBone)
     {
diff -ur libgltf.org/src/RenderScene.h libgltf/src/RenderScene.h
--- libgltf.org/src/RenderScene.h	2014-05-05 14:45:21.447395883 +0200
+++ libgltf/src/RenderScene.h	2014-05-05 14:47:48.111402138 +0200
@@ -170,7 +170,7 @@
     void renderPrimitive(class RenderPrimitive* pPrimitive);
 
     void updateJointInfo(Bone* pBone);
-    void updateBoneMatrix(Bone* pBone, glm::mat4& matrix);
+    void updateBoneMatrix(Bone* pBone, const glm::mat4& matrix);
 
     glm::mat4* calcProjectionMatrix();
     glm::vec3* calcLightColor();
