diff --git libgltf/src/FPSCounter.h libgltf/src/FPSCounter.h
index c6c4279..66f7e2b 100644
--- libgltf/src/FPSCounter.h
+++ libgltf/src/FPSCounter.h
@@ -7,6 +7,11 @@
  */
 #ifndef FPSCOUNTER_H
 #define FPSCOUNTER_H
+
+#define ENABLE_FPS    0
+
+#if ENABLE_FPS
+
 #include "time.h"
 #include "Font.h"
 #include "Shaders.h"
@@ -17,7 +22,6 @@
 
 #define WRITEFPS2FILE 1
 #define TIMETHRESHOLD 30
-#define ENABLE_FPS    1
 #define FPS_FILE      "./FPSCounter.txt"
 #if WRITEFPS2FILE
 #include <iostream>
@@ -136,4 +140,6 @@ private:
 #endif
 };
 
+#endif // ENABLE_FPS
+
 #endif
