diff -ur libgltf.org/inc/libgltf.h libgltf/inc/libgltf.h
--- libgltf.org/inc/libgltf.h	2014-05-06 11:56:54.756228299 +0200
+++ libgltf/inc/libgltf.h	2014-05-06 11:57:55.104230666 +0200
@@ -10,6 +10,7 @@
 #include <stddef.h>
 #include "types.h"
 #include "Common.h"
+#include <GL/glew.h>
 #include <glm/glm.hpp>
 using namespace glTF;
  
@@ -66,7 +67,7 @@
 /** Get a bitmap of the screen in the given point in time. */
 extern "C"
 void gltf_renderer_get_bitmap(glTFHandle *handle, double time,
-    char* buffer, long width, long height);
+    char* buffer, GLenum format);
 
 /** Start playing the glTF animation to the current openGLcontext. */
 extern "C"
diff -ur libgltf.org/src/libgltf.cpp libgltf/src/libgltf.cpp
--- libgltf.org/src/libgltf.cpp	2014-05-06 11:56:54.756228299 +0200
+++ libgltf/src/libgltf.cpp	2014-05-06 11:57:00.620228529 +0200
@@ -95,10 +95,10 @@
    
 extern "C"
 void gltf_renderer_get_bitmap(glTFHandle *handle, double time,
-    char* buffer, long width, long height)
+    char* buffer, GLenum format)
 {
     RenderScene* renderScene = (RenderScene*)handle->renderer;
-    renderScene->renderToBuffer(&(handle->viewport), (unsigned char*)buffer);
+    renderScene->renderToBuffer(&(handle->viewport), (unsigned char*)buffer, format);
     return;
 }
 
diff -ur libgltf.org/src/RenderScene.cpp libgltf/src/RenderScene.cpp
--- libgltf.org/src/RenderScene.cpp	2014-05-06 11:56:54.756228299 +0200
+++ libgltf/src/RenderScene.cpp	2014-05-06 11:57:03.852228656 +0200
@@ -859,7 +859,7 @@
     return sCamera;
 }
 
-void RenderScene::renderToBuffer(glTFViewport* pViewport, unsigned char * buffer)
+void RenderScene::renderToBuffer(glTFViewport* pViewport, unsigned char * buffer, GLenum format)
 {
     createRenderObj(pViewport->width, pViewport->height);
     createTextureObj(pViewport->width, pViewport->height);
@@ -876,7 +876,7 @@
     GLenum fbResult = glCheckFramebufferStatus(GL_FRAMEBUFFER);
     if( fbResult != GL_FRAMEBUFFER_COMPLETE )
         result = false;
-    glReadPixels(0, 0, pViewport->width, pViewport->height, GL_BGR, 
+    glReadPixels(0, 0, pViewport->width, pViewport->height, format, 
         GL_UNSIGNED_BYTE, buffer);
     glBindFramebuffer(GL_FRAMEBUFFER, 0);
     glDeleteFramebuffers(1, &mFBOID);
diff -ur libgltf.org/src/RenderScene.h libgltf/src/RenderScene.h
--- libgltf.org/src/RenderScene.h	2014-05-06 11:56:54.756228299 +0200
+++ libgltf/src/RenderScene.h	2014-05-06 11:57:01.524228565 +0200
@@ -124,7 +124,7 @@
     void releaseRender(void* lpParam);
 
     // For bitmap
-    void renderToBuffer(glTFViewport* pViewpoit, unsigned char * buffer);
+    void renderToBuffer(glTFViewport* pViewpoit, unsigned char * buffer, GLenum format);
     
     // For animation
     void startAnimation();
