diff -ur libgltf.org/src/FPSCounter.cpp libgltf/src/FPSCounter.cpp
--- libgltf.org/src/FPSCounter.cpp	2014-06-24 18:10:46.317967903 +0200
+++ libgltf/src/FPSCounter.cpp	2014-06-24 18:10:58.629967382 +0200
@@ -104,7 +104,7 @@
         ++mFrames;
     }
 
-    pFont->printFormattedString(pViewport->width - 100, 30, 15, "%d", mFPS);
+    pFont->printFormattedString(pViewport->width - 40, 10, 15, "%d", mFPS);
 
     glEnable(GL_DEPTH_TEST);
 }
