diff -ur libgltf.org/src/Common.h libgltf/src/Common.h
--- libgltf.org/src/Common.h	2014-05-05 14:31:00.483359159 +0200
+++ libgltf/src/Common.h	2014-05-05 14:34:02.055366903 +0200
@@ -536,8 +536,8 @@
         unsigned int mCount;
         double mMaxTime;
         double mMinTime;
-        std::vector<struct QuatKey> mTimeRorateVec;
-        //std::vector<struct QuatKey> mTimeTransfromVec;
+        std::vector<QuatKey> mTimeRorateVec;
+        //std::vector<QuatKey> mTimeTransfromVec;
     };
 
     class Animation
