--- libgltf/src/Common.h	2014-05-12 09:57:30.211585326 +0200
+++ libgltf.new/src/Common.h	2014-05-12 09:59:20.073120909 +0200
@@ -532,8 +532,8 @@ private:
     unsigned int mCount;
     double mMaxTime;
     double mMinTime;
-    std::vector<struct QuatKey> mTimeRorateVec;
-    //std::vector<struct QuatKey> mTimeTransfromVec;
+    std::vector<QuatKey> mTimeRorateVec;
+    //std::vector<QuatKey> mTimeTransfromVec;
 };
 
 class Animation
