--- misc/build/boost_1_44_0/boost/property_tree/detail/json_parser_read.hpp	2013-04-22 18:22:16.961642320 +0200
+++ misc/build/boost_1_44_0/boost/property_tree/detail/json_parser_read.hpp	2013-04-22 18:22:59.991632602 +0200
@@ -102,7 +102,7 @@
             void operator()(It b, It e) const
             {
                 BOOST_ASSERT(c.stack.size() >= 1);
-                c.stack.back()->push_back(std::make_pair(c.name, Str(b, e)));
+                c.stack.back()->push_back(std::make_pair(c.name, Ptree(Str(b, e))));
                 c.name.clear();
                 c.string.clear();
             }
