--- misc/boost_1_44_0/boost/config/stdlib/libstdcpp3.hpp	2010-06-11 13:16:34.000000000 +0200
+++ misc/build/boost_1_44_0/boost/config/stdlib/libstdcpp3.hpp	2012-07-13 10:31:58.330127342 +0200
@@ -31,7 +31,8 @@
 
 #ifdef __GLIBCXX__ // gcc 3.4 and greater:
 #  if defined(_GLIBCXX_HAVE_GTHR_DEFAULT) \
-        || defined(_GLIBCXX__PTHREADS)
+        || defined(_GLIBCXX__PTHREADS) \
+        || defined(_GLIBCXX_HAS_GTHREADS)
       //
       // If the std lib has thread support turned on, then turn it on in Boost
       // as well.  We do this because some gcc-3.4 std lib headers define _REENTANT
