--- foo/foo/foo/boost/detail/endian.hpp
+++ foo/foo/foo/boost/detail/endian.hpp
@@ -28,8 +28,8 @@
 //
 // Special cases come first:
 //
-#if defined (__GLIBC__)
-// GNU libc offers the helpful header <endian.h> which defines
+#if defined (__GLIBC__) || defined(__ANDROID__)
+// GNU libc and Android's bionic offer the helpful header <endian.h> which defines
 // __BYTE_ORDER
 # include <endian.h>
 # if (__BYTE_ORDER == __LITTLE_ENDIAN)
