--- misc/build/boost_1_44_0/libs/thread/src/win32/makefile.mk
+++ misc/build/boost_1_44_0/libs/thread/src/win32/makefile.mk
@@ -1,1 +1,21 @@
-dummy
+PRJ=..$/..$/..$/..$/..$/..$/..$/..
+
+PRJNAME=boostthread
+TARGET=boostthread
+ENABLE_EXCEPTIONS=TRUE
+LIBTARGET=NO
+
+.IF "$(GUI)"=="WNT"
+
+.INCLUDE :  settings.mk
+
+SLOFILES=	$(SLO)$/thread.obj \
+			$(SLO)$/tss_dll.obj \
+			$(SLO)$/tss_pe.obj
+
+LIB1TARGET=$(SLB)$/$(TARGET).lib
+LIB1ARCHIV=$(LB)$/lib$(TARGET).a
+LIB1OBJFILES=$(SLOFILES)
+
+.INCLUDE :  target.mk
+.ENDIF
