--- misc/build/boost_1_44_0/boost/config/compiler/visualc.hpp
+++ misc/build/boost_1_44_0/boost/config/compiler/visualc.hpp
@@ -101,6 +101,11 @@
 #  define BOOST_NO_ADL_BARRIER
 #endif
 
+#if _MSC_VER >= 1700  // 1700 == VC++ 11.0
+   // Removed in VC11:
+#  define BOOST_NO_0X_HDR_INITIALIZER_LIST
+#endif
+
 
 #if (_MSC_VER <= 1600)
 // MSVC (including the latest checked version) has not yet completely 
@@ -269,8 +269,8 @@
 #error "Compiler not supported or configured - please reconfigure"
 #endif
 //
-// last known and checked version is 1600 (VC10, aka 2010):
-#if (_MSC_VER > 1600)
+// last known and checked version is 1700 (VC12, aka 2012):
+#if (_MSC_VER > 1700)
 #  if defined(BOOST_ASSERT_CONFIG)
 #     error "Unknown compiler version - please run the configure tests and report the results"
 #  else
