--- misc/boost_1_44_0/boost/thread/xtime.hpp	2012-11-05 20:13:28.034300778 -0500
+++ misc/build/boost_1_44_0/boost/thread/xtime.hpp	2012-11-05 20:13:50.856322691 -0500
@@ -20,7 +20,7 @@

 enum xtime_clock_types
 {
-    TIME_UTC=1
+    TIME_UTC_=1
 //    TIME_TAI,
 //    TIME_MONOTONIC,
 //    TIME_PROCESS,
@@ -68,7 +68,7 @@

 inline int xtime_get(struct xtime* xtp, int clock_type)
 {
-    if (clock_type == TIME_UTC)
+    if (clock_type == TIME_UTC_)
     {
         *xtp=get_xtime(get_system_time());
         return clock_type;
