--- misc/boost_1_44_0/boost/logic/tribool.hpp	Sun Nov 25 18:07:19 2007
+++ misc/build/boost_1_44_0/boost/logic/tribool.hpp	Fri May 17 12:03:35 2013
@@ -93,7 +93,7 @@
    *
    * \throws nothrow
    */
-  tribool(bool value) : value(value? true_value : false_value) {}
+  tribool(bool initial_value) : value(initial_value? true_value : false_value) {}
 
   /**
    * Construct a new 3-state boolean value with an indeterminate value.
