--- misc/boost_1_44_0/boost/spirit/home/classic/core/composite/directives.hpp
+++ misc/build/boost_1_44_0/boost/spirit/home/classic/core/composite/directives.hpp
@@ -364,7 +364,7 @@
 
             if (l || r)
             {
-                if (l.length() < r.length() && l || !r)
+                if ((l.length() < r.length() && l) || !r)
                 {
                     scan.first = save;
                     return l;

