--- misc/boost_1_44_0/boost/smart_ptr/make_shared.hpp
+++ misc/build/boost_1_44_0/boost/smart_ptr/make_shared.hpp
@@ -49,7 +49,18 @@
     {
         if( initialized_ )
         {
+#if defined( __GNUC__ )
+
+            // fixes incorrect aliasing warning
+            T * p = reinterpret_cast< T* >( storage_.data_ );
+            p->~T();
+
+#else
+
             reinterpret_cast< T* >( storage_.data_ )->~T();
+
+#endif
+
             initialized_ = false;
         }
     }
