--- src/lib/MSPUBContentChunkType.h
+++ src/lib/MSPUBContentChunkType.h
@@ -27,7 +27,8 @@
   CELLS      = 0x63,
   FONT       = 0x6C,
   IMAGE_2K, //these don't exist in Pub 2k3 so their value in the enum is not used.
-  IMAGE_2K_DATA
+  IMAGE_2K_DATA,
+  MSPUBContentChunkType_dummy=0xFF // MSPUBParser.cpp:2359:9: runtime error: load of value 138, which is not a valid value for type 'libmspub::MSPUBContentChunkType'
 };
 } // namespace libmspub
 
