From 26d33d7cfd93b22e2fa13e6a3e28d133619948cf Mon Sep 17 00:00:00 2001
From: David Tardon <dtardon@redhat.com>
Date: Thu, 4 Jun 2015 12:33:26 +0200
Subject: [PATCH] these functions are implemented in liborcus

---
 include/orcus/spreadsheet/types.hpp | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/include/orcus/spreadsheet/types.hpp b/include/orcus/spreadsheet/types.hpp
index a78340e..bec14c8 100644
--- a/include/orcus/spreadsheet/types.hpp
+++ b/include/orcus/spreadsheet/types.hpp
@@ -22,8 +22,8 @@ typedef unsigned char color_elem_t;
 typedef unsigned short col_width_t;
 typedef unsigned short row_height_t;
 
-ORCUS_SPM_DLLPUBLIC col_width_t get_default_column_width();
-ORCUS_SPM_DLLPUBLIC row_height_t get_default_row_height();
+ORCUS_DLLPUBLIC col_width_t get_default_column_width();
+ORCUS_DLLPUBLIC row_height_t get_default_row_height();
 
 enum border_direction_t
 {
@@ -192,7 +192,7 @@ enum databar_axis_t
  * Convert a string representation of a totals row function name to its
  * equivalent enum value.
  */
-ORCUS_SPM_DLLPUBLIC totals_row_function_t to_totals_row_function_enum(const char* p, size_t n);
+ORCUS_DLLPUBLIC totals_row_function_t to_totals_row_function_enum(const char* p, size_t n);
 
 }}
 
-- 
2.4.1

