From 3a6bb93a382b4d5f1463ee3fdd54cab7048ff996 Mon Sep 17 00:00:00 2001
From: Kohei Yoshida <kohei.yoshida@gmail.com>
Date: Fri, 31 Aug 2018 16:07:06 -0400
Subject: [PATCH] Missing header for std::tolower.

---
 src/parser/sax_token_parser.cpp | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/parser/sax_token_parser.cpp b/src/parser/sax_token_parser.cpp
index 9790b498..44ba5844 100644
--- a/src/parser/sax_token_parser.cpp
+++ b/src/parser/sax_token_parser.cpp
@@ -9,6 +9,7 @@
 #include "orcus/tokens.hpp"
 
 #include <mdds/sorted_string_map.hpp>
+#include <cctype>
 
 namespace orcus {
 
-- 
2.17.1

