diff -up zlib-1.2.5/contrib/minizip/unzip.c.pom zlib-1.2.5/contrib/minizip/unzip.c
--- zlib-1.2.5/contrib/minizip/unzip.c.pom	2010-02-15 12:59:40.000000000 +0100
+++ zlib-1.2.5/contrib/minizip/unzip.c	2011-04-06 12:06:04.000000000 +0200
@@ -1145,7 +1145,7 @@ extern int ZEXPORT unzGetCurrentFileInfo
                                                 szFileName,fileNameBufferSize,
                                                 extraField,extraFieldBufferSize,
                                                 szComment,commentBufferSize);
-    if (err==UNZ_OK)
+    if (pfile_info && err==UNZ_OK)
     {
         pfile_info->version = file_info64.version;
         pfile_info->version_needed = file_info64.version_needed;
