--- src/lib/EtonyekDocument.cpp
+++ src/lib/EtonyekDocument.cpp
@@ -7,6 +7,8 @@
 
 #include <libetonyek/libetonyek.h>
 
+#include <cassert>
+
 #include <boost/scoped_ptr.hpp>
 #include <boost/shared_ptr.hpp>
 #include <boost/logic/tribool.hpp>
@@ -179,7 +181,7 @@
     info.input->seek(0, RVNG_SEEK_SET);
   }
 
-  assert(!isGzipped);
+  assert(bool(!isGzipped));
 
   return probeXMLImpl(info.input, probe, type, info);
 }
