--- src/lib/EtonyekDocument.cpp
+++ src/lib/EtonyekDocument.cpp
@@ -103,12 +103,21 @@
   return false;
 }
 
+namespace
+{
+  void handleError(void * /*arg*/, const char * /*msg*/, xmlParserSeverities /*severity*/, xmlTextReaderLocatorPtr /*locator*/)
+  {
+  }
+}
+
 bool probeXML(DetectionInfo &info)
 {
   const shared_ptr<xmlTextReader> reader(xmlReaderForIO(readFromStream, closeStream, info.m_input.get(), "", 0, 0), xmlFreeTextReader);
   if (!reader)
     return false;
 
+  xmlTextReaderSetErrorHandler(reader.get(), handleError, NULL);
+
   int ret = 0;
   do
   {
