--- misc/lcms2-2.3/include/lcms2.h	2011-12-15 16:45:47.000000000 +0100
+++ misc/build/lcms2-2.3/include/lcms2.h	2012-10-05 14:49:09.721146003 +0200
@@ -30,8 +30,12 @@
 
 // ********** Configuration toggles ****************************************
 
-// Uncomment this one if you are using big endian machines
-// #define CMS_USE_BIG_ENDIAN   1
+#ifndef _WIN32
+#include <endian.h>
+#if __BYTE_ORDER == __BIG_ENDIAN
+#define CMS_USE_BIG_ENDIAN   1
+#endif
+#endif /* !_WIN32 */
 
 // Uncomment this one if your compiler/machine does NOT support the
 // "long long" type.
