--- misc/STLport-4.0/stlport/config/stl_sunpro.h	Mon Jul  6 13:43:35 2009
+++ misc/build/STLport-4.0/stlport/config/stl_sunpro.h	Mon Jul  6 13:56:56 2009
@@ -31,7 +31,8 @@
 #  endif
 # endif
 
-# if (__SUNPRO_CC < 0x600)
+// support SunStudio compiler 12u1 version 5.10 (which results in 0x5100)
+# if (__SUNPRO_CC < 0x600 || ((__SUNPRO_CC | 0x0FFF) > 0x0FFF && __SUNPRO_CC < 0x6000))
 // those are tested and proved not to work...
 #  define __STL_STATIC_ARRAY_BUG 1
 #  define __STL_STATIC_CONST_INIT_BUG 1
@@ -57,7 +58,7 @@
 #  define __STL_NO_FRIEND_TEMPLATES 1
 #endif
 
-# if (__SUNPRO_CC < 0x600)
+# if (__SUNPRO_CC < 0x600 || ((__SUNPRO_CC | 0x0FFF) > 0x0FFF && __SUNPRO_CC < 0x6000))
 #  define __STL_HAS_NO_NEW_C_HEADERS 1 
 #  define __STL_NO_EXPLICIT_FUNCTION_TMPL_ARGS
 #  define __STL_NON_TYPE_TMPL_PARAM_BUG 1
