--- misc/STLport-4.5/stlport/stl/_locale.h	2001-07-06 05:16:18.000000000 +0100
+++ misc/build/STLport-4.5/stlport/stl/_locale.h	2012-02-18 18:17:18.338109337 +0000
@@ -120,7 +120,7 @@
   locale(const locale& __loc, _Facet* __f) : _M_impl(0)
     {
       //      _M_impl = this->_S_copy_impl(__loc._M_impl, __f != 0);
-      new(this) locale(__loc._M_impl, __f != 0);
+      new locale(__loc._M_impl, __f != 0);
       if (__f != 0)
         this->_M_insert(__f, _Facet::id);
     }
