--- misc/libpng/makefile.mk	2011-03-01 00:01:58.221734678 +0100
+++ misc/build/libpng/makefile.mk	2011-02-28 23:52:40.953319424 +0100
@@ -1 +1,49 @@
-dummy
+#*************************************************************************
+#
+#   Copyright according the GNU Public License.
+#
+#*************************************************************************
+
+PRJ=..$/..$/..$/..
+
+PRJNAME=libpng
+TARGET=libpng
+LIBTARGET=NO
+EXTERNAL_WARNINGS_NOT_ERRORS=TRUE
+VISIBILITY_HIDDEN=TRUE
+
+# --- Settings -----------------------------------------------------------
+
+.INCLUDE :  settings.mk
+
+
+.IF "$(GUI)$(CPU)"=="WNTP"
+CFLAGS=$(CFLAGS) -Od
+.ENDIF
+
+# --- Files --------------------------------------------------------
+
+SLOFILES=	 $(SLO)$/png.c 		\
+                $(SLO)$/pngerror.c	\
+                $(SLO)$/pngget.c	\
+                $(SLO)$/pngmem.c	\
+                $(SLO)$/pngpread.c	\
+                $(SLO)$/pngread.c	\
+                $(SLO)$/pngrio.c	\
+                $(SLO)$/pngrtran.c	\
+                $(SLO)$/pngrutil.c	\
+                $(SLO)$/pngset.c	\
+                $(SLO)$/pngtest.c	\
+                $(SLO)$/pngtrans.c	\
+                $(SLO)$/pngvalid.c	\
+                $(SLO)$/pngwio.c	\
+                $(SLO)$/pngwrite.c	\
+                $(SLO)$/pngwtran.c	\
+                $(SLO)$/pngwutil.c
+
+LIB1TARGET=$(SLB)$/$(TARGET).lib
+LIB1ARCHIV=$(LB)$/lib$(TARGET).a
+LIB1OBJFILES=$(SLOFILES)
+
+.INCLUDE :  target.mk
+
