--- src/common/memory.h
+++ src/common/memory.h
@@ -162,7 +162,7 @@
 // Wrapper to use with STL containers
 template <typename T>
 struct PageStdAllocator : public std::allocator<T> {
-  typedef typename std::allocator<T>::pointer pointer;
+  typedef T* pointer;
   typedef typename std::allocator<T>::size_type size_type;
 
   explicit PageStdAllocator(PageAllocator& allocator): allocator_(allocator) {}
