--- curl-7.26.0/lib/setup.h	2009-10-29 05:21:58.000000000 +0900
+++ misc/build/curl-7.26.0/lib/setup.h	2010-03-02 06:03:10.009500000 +0900
@@ -40,6 +40,21 @@
 
 #include "curl_config.h"
 
+#ifdef __MINGW32__
+#undef HAVE_DLFCN_H
+#define HAVE_GETHOSTBYADDR 1
+#define HAVE_GETHOSTBYNAME 1
+#undef HAVE_LIBZ
+#undef HAVE_NETDB_H
+#undef HAVE_POLL_H
+#undef HAVE_SYS_IOCTL_H
+#undef HAVE_SYS_POLL_H
+#undef HAVE_SYS_UIO_H
+#undef HAVE_TERMIOS_H
+#undef HAVE_TERMIO_H
+#undef HAVE_ZLIB_H
+#endif
+
 #else /* HAVE_CONFIG_H */
 
 #ifdef _WIN32_WCE
