--- src/lxml/etree.c
+++ src/lxml/etree.c
@@ -113118,7 +113118,7 @@
  *         self._orig_loader = _register_document_loader() if set_document_loader else NULL
  *         if self._validator is not None:
  */
-  __pyx_v_self->_c_ctxt->sax->serror = __pyx_f_4lxml_5etree__receiveParserError;
+  __pyx_v_self->_c_ctxt->sax->serror = (xmlStructuredErrorFunc)__pyx_f_4lxml_5etree__receiveParserError;
 
   /* "src/lxml/parser.pxi":587
  *         self._doc = None
@@ -117606,7 +117606,7 @@
  *             sax.startElementNs = NULL
  *             sax.endElementNs = NULL
  */
-    __pyx_v_sax->serror = __pyx_f_4lxml_5etree__receiveParserError;
+    __pyx_v_sax->serror = (xmlStructuredErrorFunc)__pyx_f_4lxml_5etree__receiveParserError;
 
     /* "src/lxml/parser.pxi":923
  *             sax.initialized = xmlparser.XML_SAX2_MAGIC
@@ -181690,7 +181690,7 @@
  * 
  *     @cython.final
  */
-  __pyx_v_xpathCtxt->error = __pyx_f_4lxml_5etree__receiveXPathError;
+  __pyx_v_xpathCtxt->error = (xmlStructuredErrorFunc)__pyx_f_4lxml_5etree__receiveXPathError;
 
   /* "src/lxml/extensions.pxi":132
  *         return utf
@@ -218964,7 +218964,7 @@
  *             parser_ctxt, _receiveError, <void*>self._error_log)
  *         _connectGenericErrorLog(self._error_log, xmlerror.XML_FROM_RELAXNGP)
  */
-  xmlRelaxNGSetParserStructuredErrors(__pyx_v_parser_ctxt, __pyx_f_4lxml_5etree__receiveError, ((void *)__pyx_v_self->__pyx_base._error_log));
+  xmlRelaxNGSetParserStructuredErrors(__pyx_v_parser_ctxt, (xmlStructuredErrorFunc)__pyx_f_4lxml_5etree__receiveError, ((void *)__pyx_v_self->__pyx_base._error_log));
 
   /* "src/lxml/relaxng.pxi":92
  *         relaxng.xmlRelaxNGSetParserStructuredErrors(
@@ -219382,7 +219382,7 @@
  *                 valid_ctxt, _receiveError, <void*>self._error_log)
  *             _connectGenericErrorLog(self._error_log, xmlerror.XML_FROM_RELAXNGV)
  */
-    xmlRelaxNGSetValidStructuredErrors(__pyx_v_valid_ctxt, __pyx_f_4lxml_5etree__receiveError, ((void *)__pyx_v_self->__pyx_base._error_log));
+    xmlRelaxNGSetValidStructuredErrors(__pyx_v_valid_ctxt, (xmlStructuredErrorFunc)__pyx_f_4lxml_5etree__receiveError, ((void *)__pyx_v_self->__pyx_base._error_log));
 
     /* "src/lxml/relaxng.pxi":134
  *             relaxng.xmlRelaxNGSetValidStructuredErrors(
@@ -220271,7 +220271,7 @@
  *             parser_ctxt, _receiveError, <void*>self._error_log)
  *         if self._doc is not None:
  */
-  xmlSchemaSetParserStructuredErrors(__pyx_v_parser_ctxt, __pyx_f_4lxml_5etree__receiveError, ((void *)__pyx_v_self->__pyx_base._error_log));
+  xmlSchemaSetParserStructuredErrors(__pyx_v_parser_ctxt, (xmlStructuredErrorFunc)__pyx_f_4lxml_5etree__receiveError, ((void *)__pyx_v_self->__pyx_base._error_log));
 
   /* "src/lxml/xmlschema.pxi":74
  *         xmlschema.xmlSchemaSetParserStructuredErrors(
@@ -220814,7 +220814,7 @@
  *                 valid_ctxt, _receiveError, <void*>self._error_log)
  * 
  */
-    xmlSchemaSetValidStructuredErrors(__pyx_v_valid_ctxt, __pyx_f_4lxml_5etree__receiveError, ((void *)__pyx_v_self->__pyx_base._error_log));
+    xmlSchemaSetValidStructuredErrors(__pyx_v_valid_ctxt, (xmlStructuredErrorFunc)__pyx_f_4lxml_5etree__receiveError, ((void *)__pyx_v_self->__pyx_base._error_log));
 
     /* "src/lxml/xmlschema.pxi":131
  *                 valid_ctxt, _receiveError, <void*>self._error_log)
@@ -221579,7 +221579,7 @@
  *                 self._valid_ctxt, _receiveError, <void*>error_log)
  *         self._sax_plug = xmlschema.xmlSchemaSAXPlug(
  */
-    xmlSchemaSetValidStructuredErrors(__pyx_v_self->_valid_ctxt, __pyx_f_4lxml_5etree__receiveError, ((void *)__pyx_v_error_log));
+    xmlSchemaSetValidStructuredErrors(__pyx_v_self->_valid_ctxt, (xmlStructuredErrorFunc)__pyx_f_4lxml_5etree__receiveError, ((void *)__pyx_v_error_log));
 
     /* "src/lxml/xmlschema.pxi":195
  *                 xmlschema.xmlSchemaSetValidOptions(
@@ -222756,7 +222756,7 @@
  *                 valid_ctxt, _receiveError, <void*>self._error_log)
  *             c_doc = _fakeRootDoc(doc._c_doc, root_node._c_node)
  */
-    xmlSchematronSetValidStructuredErrors(__pyx_v_valid_ctxt, __pyx_f_4lxml_5etree__receiveError, ((void *)__pyx_v_self->__pyx_base._error_log));
+    xmlSchematronSetValidStructuredErrors(__pyx_v_valid_ctxt, (xmlStructuredErrorFunc)__pyx_f_4lxml_5etree__receiveError, ((void *)__pyx_v_self->__pyx_base._error_log));
 
     /* "src/lxml/schematron.pxi":153
  *             schematron.xmlSchematronSetValidStructuredErrors(
