--- afdko/c/mergefonts/source/mergeFonts.c
+++ afdko/c/mergefonts/source/mergeFonts.c
@@ -1177,21 +1177,21 @@
     if (end == 0)
         return;
 
-    while (start < MAX_DICT_ENTRY_LEN) {
-        if ((str[start] == ' ') || (str[start] == '\t') || (str[start] == '(') || (str[start] == '\r') || (str[start] == '\n'))
-            start++;
+    while (end >= 0) {
+        if ((str[end] == ' ') || (str[end] == '\t') || (str[end] == ')') || (str[end] == '\r') || (str[end] == '\n'))
+            end--;
         else
             break;
     }
+    str[end + 1] = 0;
 
-    while (end >= 0) {
-        if ((str[end] == ' ') || (str[end] == '\t') || (str[end] == ')') || (str[end] == '\r') || (str[end] == '\n'))
-            end--;
+    while (start < MAX_DICT_ENTRY_LEN) {
+        if ((str[start] == ' ') || (str[start] == '\t') || (str[start] == '(') || (str[start] == '\r') || (str[start] == '\n'))
+            start++;
         else
             break;
     }
 
-    str[end + 1] = 0;
     if (start > 0)
         memmove(str, &str[start], (end - start) + 2);
     if (strlen(str) == 0) {
