*** misc/db-4.2.52.NC/dist/configure	2003-12-04 06:10:02.000000000 +0900
--- misc/build/db-4.2.52.NC/dist/configure	Thu Dec 20 23:08:25 2007
***************
*** 8301,8307 ****
        export_symbols_cmds='$NM $libobjs $convenience | $global_symbol_pipe | $SED -e '\''/^[BCDGS] /s/.* \([^ ]*\)/\1 DATA/'\'' | $SED -e '\''/^[AITW] /s/.* //'\'' | sort | uniq > $export_symbols'
  
        if $LD --help 2>&1 | grep 'auto-import' > /dev/null; then
!         archive_cmds='$CC -shared $libobjs $deplibs $compiler_flags -o $output_objdir/$soname ${wl}--image-base=0x10000000 ${wl}--out-implib,$lib'
  	# If the export-symbols file already is a .def file (1st line
  	# is EXPORTS), use it as is; otherwise, prepend...
  	archive_expsym_cmds='if test "x`$SED 1q $export_symbols`" = xEXPORTS; then
--- 8301,8307 ----
        export_symbols_cmds='$NM $libobjs $convenience | $global_symbol_pipe | $SED -e '\''/^[BCDGS] /s/.* \([^ ]*\)/\1 DATA/'\'' | $SED -e '\''/^[AITW] /s/.* //'\'' | sort | uniq > $export_symbols'
  
        if $LD --help 2>&1 | grep 'auto-import' > /dev/null; then
!         archive_cmds='$DLLTOOL $add_underscore --export-all-symbols --dllname $soname --output-lib $lib --output-exp $output_objdir/$soname.exp $libobjs; $CC -shared $output_objdir/$soname.exp $libobjs $deplibs $compiler_flags -o $output_objdir/$soname ${wl}--image-base=0x10000000'
  	# If the export-symbols file already is a .def file (1st line
  	# is EXPORTS), use it as is; otherwise, prepend...
  	archive_expsym_cmds='if test "x`$SED 1q $export_symbols`" = xEXPORTS; then
***************
*** 8310,8316 ****
  	  echo EXPORTS > $output_objdir/$soname.def;
  	  cat $export_symbols >> $output_objdir/$soname.def;
  	fi~
! 	$CC -shared $output_objdir/$soname.def $libobjs $deplibs $compiler_flags -o $output_objdir/$soname ${wl}--image-base=0x10000000  ${wl}--out-implib,$lib'
        else
  	ld_shlibs=no
        fi
--- 8310,8316 ----
  	  echo EXPORTS > $output_objdir/$soname.def;
  	  cat $export_symbols >> $output_objdir/$soname.def;
  	fi~
! 	$DLLTOOL $add_underscore --dllname $soname --output-lib $lib --output-exp $output_objdir/$soname.exp --input-def $output_objdir/$soname.def; $CC -shared $output_objdir/$soname.exp $libobjs $deplibs $compiler_flags -o $output_objdir/$soname ${wl}--image-base=0x10000000'
        else
  	ld_shlibs=no
        fi
***************
*** 12077,12091 ****
  
         if test "$pre_test_object_deps_done" = no; then
  	 if test -z "$predep_objects_CXX"; then
! 	   predep_objects_CXX="$p"
  	 else
! 	   predep_objects_CXX="$predep_objects_CXX $p"
  	 fi
         else
  	 if test -z "$postdep_objects_CXX"; then
! 	   postdep_objects_CXX="$p"
  	 else
! 	   postdep_objects_CXX="$postdep_objects_CXX $p"
  	 fi
         fi
         ;;
--- 12077,12091 ----
  
         if test "$pre_test_object_deps_done" = no; then
  	 if test -z "$predep_objects_CXX"; then
! 	   predep_objects_CXX=`cygpath -m "$p"`
  	 else
! 	   predep_objects_CXX="$predep_objects_CXX "`cygpath -m "$p"`
  	 fi
         else
  	 if test -z "$postdep_objects_CXX"; then
! 	   postdep_objects_CXX=`cygpath -m "$p"`
  	 else
! 	   postdep_objects_CXX="$postdep_objects_CXX "`cygpath -m "$p"`
  	 fi
         fi
         ;;
***************
*** 26252,26257 ****
--- 26252,26268 ----
  
  
  
+ ac_cv_sizeof_char_p=`echo $ac_cv_sizeof_char_p | tr -d '\015'`
+ ac_cv_sizeof_unsigned_int=`echo $ac_cv_sizeof_unsigned_int | tr -d '\015'`
+ ac_cv_sizeof_unsigned_char=`echo $ac_cv_sizeof_unsigned_char | tr -d '\015'`
+ ac_cv_sizeof_unsigned_short=`echo $ac_cv_sizeof_unsigned_short | tr -d '\015'`
+ ac_cv_sizeof_unsigned_long=`echo $ac_cv_sizeof_unsigned_long | tr -d '\015'`
+ ac_cv_sizeof_size_t=`echo $ac_cv_sizeof_size_t | tr -d '\015'`
+ ac_cv_sizeof_int=`echo $ac_cv_sizeof_int | tr -d '\015'`
+ ac_cv_sizeof_char=`echo $ac_cv_sizeof_char | tr -d '\015'`
+ ac_cv_sizeof_short=`echo $ac_cv_sizeof_short | tr -d '\015'`
+ ac_cv_sizeof_long=`echo $ac_cv_sizeof_long | tr -d '\015'`
+ 
  echo "$as_me:$LINENO: checking for u_int8_t" >&5
  echo $ECHO_N "checking for u_int8_t... $ECHO_C" >&6
  if test "${ac_cv_type_u_int8_t+set}" = set; then
--- misc/build/db-4.2.52.NC/dist/ltmain.sh.orig	2003-04-27 02:45:38.000000000 +0900
+++ misc/build/db-4.2.52.NC/dist/ltmain.sh	2005-04-08 00:03:16.000000000 +0900
@@ -853,6 +853,7 @@
     lib_search_path=`pwd`
     inst_prefix_dir=
 
+    add_underscore=
     avoid_version=no
     dlfiles=
     dlprefiles=
@@ -1159,6 +1160,11 @@
       prevarg="$arg"
 
       case $arg in
+      -add-underscore)
+	add_underscore="--add-underscore"
+	continue
+	;;
+
       -all-static)
 	if test -n "$link_static_flag"; then
 	  compile_command="$compile_command $link_static_flag"
@@ -4474,7 +4480,7 @@
 	esac
 	# test for cygwin because mv fails w/o .exe extensions
 	case $host in
-	  *cygwin*)
+	  *cygwin* | *mingw*)
 	    exeext=.exe
 	    outputname=`$echo $outputname|${SED} 's,.exe$,,'` ;;
 	  *) exeext= ;;
--- misc/build/db-4.2.52.NC/dist/Makefile.in.orig	2003-11-29 03:50:06.000000000 +0900
+++ misc/build/db-4.2.52.NC/dist/Makefile.in	2005-04-03 13:51:34.000000000 +0900
@@ -45,7 +45,7 @@
 SOLINK=		@MAKEFILE_SOLINK@
 SOFLAGS=	@SOFLAGS@
 LIBMAJOR=	@DB_VERSION_MAJOR@
-LIBVERSION=	@DB_VERSION_MAJOR@.@DB_VERSION_MINOR@
+LIBVERSION=	@DB_VERSION_MAJOR@@DB_VERSION_MINOR@
 
 CPPFLAGS=	-I$(builddir) -I$(srcdir) @CPPFLAGS@
 
@@ -62,18 +62,18 @@
 
 libdb_base=	libdb
 libdb=		$(libdb_base).a
-libdb_version=	$(libdb_base)-$(LIBVERSION).a
-libso=		$(libdb_base)-$(LIBVERSION)@SOSUFFIX@
-libso_target=	$(libdb_base)-$(LIBVERSION).la
+libdb_version=	$(libdb_base)$(LIBVERSION).a
+libso=		$(libdb_base)$(LIBVERSION)@SOSUFFIX@
+libso_target=	$(libdb_base)$(LIBVERSION).la
 libso_default=	$(libdb_base)@SOSUFFIX@
-libso_major=	$(libdb_base)-$(LIBMAJOR)@SOSUFFIX@
+libso_major=	$(libdb_base)$(LIBMAJOR)@SOSUFFIX@
 
 ##################################################
 # C++ API.
 #
 # C++ support is optional, and can be built with static or shared libraries.
 ##################################################
-CXXFLAGS=	-c $(CPPFLAGS) @CXXFLAGS@
+CXXFLAGS=	-c $(SOLARINC) $(CPPFLAGS) -DGXX_INCLUDE_PATH=$(GXX_INCLUDE_PATH) @CXXFLAGS@
 CXX=		@MAKEFILE_CXX@
 CXXLINK=	@MAKEFILE_CXXLINK@
 XSOLINK=	@MAKEFILE_XSOLINK@
@@ -81,11 +81,11 @@
 
 libcxx_base=	libdb_cxx
 libcxx=		$(libcxx_base).a
-libcxx_version=	$(libcxx_base)-$(LIBVERSION).a
-libxso=		$(libcxx_base)-$(LIBVERSION)@SOSUFFIX@
-libxso_target=	$(libcxx_base)-$(LIBVERSION).la
+libcxx_version=	$(libcxx_base)$(LIBVERSION).a
+libxso=		$(libcxx_base)$(LIBVERSION)@SOSUFFIX@
+libxso_target=	$(libcxx_base)$(LIBVERSION).la
 libxso_default=	$(libcxx_base)@SOSUFFIX@
-libxso_major=	$(libcxx_base)-$(LIBMAJOR)@SOSUFFIX@
+libxso_major=	$(libcxx_base)$(LIBMAJOR)@SOSUFFIX@
 
 ##################################################
 # Java API.
@@ -112,12 +112,12 @@
 libj_exjarfile=	dbexamples.jar
 rpc_jarfile=	dbsvc.jar
 libjso_base=	libdb_java
-libjso=		$(libjso_base)-$(LIBVERSION)@JMODSUFFIX@
-libjso_static=	$(libjso_base)-$(LIBVERSION).a
-libjso_target=	$(libjso_base)-$(LIBVERSION).la
+libjso=		$(libjso_base)$(LIBVERSION)@JMODSUFFIX@
+libjso_static=	$(libjso_base)$(LIBVERSION).a
+libjso_target=	$(libjso_base)$(LIBVERSION).la
 libjso_default=	$(libjso_base)@JMODSUFFIX@
-libjso_major=	$(libjso_base)-$(LIBMAJOR)@JMODSUFFIX@
-libjso_g=	$(libjso_base)-$(LIBVERSION)_g@JMODSUFFIX@
+libjso_major=	$(libjso_base)$(LIBMAJOR)@JMODSUFFIX@
+libjso_g=	$(libjso_base)$(LIBVERSION)_g@JMODSUFFIX@
 
 ##################################################
 # TCL API.
@@ -127,11 +127,11 @@
 TCFLAGS=	@TCFLAGS@
 LIBTSO_LIBS=	@LIBTSO_LIBS@
 libtso_base=	libdb_tcl
-libtso=		$(libtso_base)-$(LIBVERSION)@MODSUFFIX@
-libtso_static=	$(libtso_base)-$(LIBVERSION).a
-libtso_target=	$(libtso_base)-$(LIBVERSION).la
+libtso=		$(libtso_base)$(LIBVERSION)@MODSUFFIX@
+libtso_static=	$(libtso_base)$(LIBVERSION).a
+libtso_target=	$(libtso_base)$(LIBVERSION).la
 libtso_default=	$(libtso_base)@MODSUFFIX@
-libtso_major=	$(libtso_base)-$(LIBMAJOR)@MODSUFFIX@
+libtso_major=	$(libtso_base)$(LIBMAJOR)@MODSUFFIX@
 
 ##################################################
 # db_dump185 UTILITY
@@ -666,7 +666,7 @@
 
 # Shared Java library.
 $(libjso_target): $(JAVA_OBJS) $(C_OBJS)
-	$(SOLINK) -shrext @JMODSUFFIX@ $(SOFLAGS) $(LDFLAGS) \
+	$(SOLINK) -shrext @JMODSUFFIX@ -add-underscore $(SOFLAGS) $(LDFLAGS) \
 	    -o $@ $(JAVA_OBJS) $(C_OBJS) $(LIBJSO_LIBS)
 
 # Shared Tcl library.
*** misc/build/db-4.2.52.NC/btree/bt_put.c.orig	2003-11-17 19:58:20.000000000 +0000
--- misc/build/db-4.2.52.NC/btree/bt_put.c	2007-06-17 15:05:09.000000000 +0100
***************
*** 702,708 ****
  
  	/* Sum up all the data items. */
  	for (cnt = 0, first = indx;
! 	    inp[first] == inp[indx] && indx < NUM_ENT(h);
  	    ++cnt, indx += P_INDX) {
  		bk = GET_BKEYDATA(dbp, h, indx + O_INDX);
  		sz += B_TYPE(bk->type) == B_KEYDATA ?
--- 702,708 ----
  
  	/* Sum up all the data items. */
  	for (cnt = 0, first = indx;
! 	    indx < NUM_ENT(h) && inp[first] == inp[indx];
  	    ++cnt, indx += P_INDX) {
  		bk = GET_BKEYDATA(dbp, h, indx + O_INDX);
  		sz += B_TYPE(bk->type) == B_KEYDATA ?
