From cde8d7c47cdea327f8153ab72cde650f1b36f7f8 Mon Sep 17 00:00:00 2001
From: David Tardon <dtardon@redhat.com>
Date: Sun, 3 Nov 2013 12:59:29 +0100
Subject: [PATCH] use correct type

---
 src/lib/HMWJGraph.cxx | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/src/lib/HMWJGraph.cxx b/src/lib/HMWJGraph.cxx
index fca366c..dd83ace 100644
--- a/src/lib/HMWJGraph.cxx
+++ b/src/lib/HMWJGraph.cxx
@@ -2668,7 +2668,7 @@ bool HMWJGraph::sendGroup(HMWJGraphInternal::Group const &group, MWAWPosition po
     return true;
   }
 
-  std::multimap<long, int>::const_iterator fIt;
+  std::map<long, int>::const_iterator fIt;
   int numFrames = int(m_state->m_framesList.size());
   for (size_t c=0; c<group.m_childsList.size(); ++c) {
     long fId=group.m_childsList[c];
@@ -2690,7 +2690,7 @@ bool HMWJGraph::sendGroup(HMWJGraphInternal::Group const &group, MWAWPosition po
 
 bool HMWJGraph::canCreateGraphic(HMWJGraphInternal::Group const &group)
 {
-  std::multimap<long, int>::const_iterator fIt;
+  std::map<long, int>::const_iterator fIt;
   int page = group.m_page;
   int numFrames = int(m_state->m_framesList.size());
   for (size_t c=0; c<group.m_childsList.size(); ++c) {
@@ -2726,7 +2726,7 @@ void HMWJGraph::sendGroup(HMWJGraphInternal::Group const &group, MWAWGraphicList
   if (!listener) return;
   group.m_parsed=true;
   MWAWInputStreamPtr &input= m_parserState->m_input;
-  std::multimap<long, int>::const_iterator fIt;
+  std::map<long, int>::const_iterator fIt;
   int numFrames = int(m_state->m_framesList.size());
   for (size_t c=0; c<group.m_childsList.size(); ++c) {
     long fId=group.m_childsList[c];
@@ -2784,7 +2784,7 @@ void HMWJGraph::sendGroupChild(HMWJGraphInternal::Group const &group, MWAWPositi
   Box2f partialBdBox;
   MWAWPosition partialPos(pos);
   MWAWInputStreamPtr &input= m_parserState->m_input;
-  std::multimap<long, int>::const_iterator fIt;
+  std::map<long, int>::const_iterator fIt;
   int numFrames = int(m_state->m_framesList.size());
   for (size_t c=0; c<numChilds; ++c) {
     long fId=group.m_childsList[c];
-- 
1.8.3.1

