From bc03e5d29fbf3eb3f58b9f91c7e8c748239721ac Mon Sep 17 00:00:00 2001
From: osnola <alonso@loria.fr>
Date: Tue, 6 Jan 2015 15:48:18 +0100
Subject: [PATCH 2/2] ClarisWors/AppleWorks[spreadsheet]: Oops, correct of a
 mistake...

---
 src/lib/ClarisWksDbaseContent.cxx | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/src/lib/ClarisWksDbaseContent.cxx b/src/lib/ClarisWksDbaseContent.cxx
index ea15544..e175c2e 100644
--- a/src/lib/ClarisWksDbaseContent.cxx
+++ b/src/lib/ClarisWksDbaseContent.cxx
@@ -1413,10 +1413,10 @@ void ClarisWksDbaseContent::Record::updateFormulaCells(Vec2i const &removeDelta)
         if (first) {
           MWAW_DEBUG_MSG(("ClarisWksDbaseContent::Record::updateFormulaCells: some cell's positions are bad, remove formula\n"));
           first=false;
-          // revert to the basic cell type
-          m_content.m_contentType=m_valueType;
-          return;
         }
+        // revert to the basic cell type
+        m_content.m_contentType=m_valueType;
+        return;
       }
     }
   }
-- 
2.1.0

