--- misc/xmlsec1-1.2.14/src/c14n.c	2010-03-02 15:46:05.000000000 +0000
+++ misc/build/xmlsec1-1.2.14/src/c14n.c	2010-03-02 15:50:35.000000000 +0000
@@ -406,6 +406,20 @@
     return(0);
 }
 
+#if !defined(LIBXML_VERSION) || LIBXML_VERSION < 20704
+/*
+ * xmlC14NMode:
+ * 
+ * Predefined values for C14N modes
+ *
+ */
+typedef enum {
+    XML_C14N_1_0            = 0,    /* Origianal C14N 1.0 spec */
+    XML_C14N_EXCLUSIVE_1_0  = 1,    /* Exclusive C14N 1.0 spec */
+    XML_C14N_1_1            = 2     /* C14N 1.1 spec */
+} xmlC14NMode;
+#endif
+
 static int 
 xmlSecTransformC14NExecute(xmlSecTransformId id, xmlSecNodeSetPtr nodes, xmlChar** nsList, 
 			   xmlOutputBufferPtr buf) {
