--- build/xmlsec1-1.2.14/win32/Makefile.msvc.old	2010-10-20 00:49:04.671875000 +0200
+++ build/xmlsec1-1.2.14/win32/Makefile.msvc	2010-10-20 00:49:23.406250000 +0200
@@ -351,7 +351,11 @@
 !if "$(DEBUG)" == "1"
 LDFLAGS 		= $(LDFLAGS) /DEBUG
 !else
-LDFLAGS 		= $(LDFLAGS) /OPT:NOWIN98
+!if "$(_NMAKE_VER)" >= "10.00.30319.01"
+LDFLAGS = $(LDFLAGS)
+!else
+LDFLAGS = $(LDFLAGS) /OPT:NOWIN98
+!endif
 !endif
 
 SOLIBS 			= $(LIBS) libxml2.lib 
--- build/xmlsec/win32/Makefile.msvc.old	2012-11-30 11:09:23.130479800 -0500
+++ build/xmlsec/win32/Makefile.msvc	2012-11-30 11:11:06.037550700 -0500
@@ -301,6 +301,10 @@
 CFLAGS 			= $(CFLAGS) /D "HAVE_STDIO_H" /D "HAVE_STDLIB_H"
 CFLAGS 			= $(CFLAGS) /D "HAVE_STRING_H" /D "HAVE_CTYPE_H"
 CFLAGS 			= $(CFLAGS) /D "HAVE_MALLOC_H" /D "HAVE_MEMORY_H"
+CFLAGS 			= $(CFLAGS) $(SOLARINC) -I$(WORKDIR)\UnpackedTarball\xml2\include
+!if "$(MSVC_USE_DEBUG_RUNTIME)" != ""
+CFLAGS 			= $(CFLAGS) /MDd
+!endif
 
 # Optimisation and debug symbols.
 !if "$(DEBUG)" == "1"
