--- build/xmlsec1-1.2.14/win32/Makefile.msvc.old	2010-10-20 00:49:04.671875000 +0200
+++ build/xmlsec1-1.2.14/win32/Makefile.msvc	2010-10-20 00:49:23.406250000 +0200
@@ -351,7 +351,11 @@
 !if "$(DEBUG)" == "1"
 LDFLAGS 		= $(LDFLAGS) /DEBUG
 !else
-LDFLAGS 		= $(LDFLAGS) /OPT:NOWIN98
+!if "$(_NMAKE_VER)" >= "10.00.30319.01"
+LDFLAGS = $(LDFLAGS)
+!else
+LDFLAGS = $(LDFLAGS) /OPT:NOWIN98
+!endif
 !endif
 
 SOLIBS 			= $(LIBS) libxml2.lib 
--- build/xmlsec/win32/Makefile.msvc.old	2012-11-30 11:09:23.130479800 -0500
+++ build/xmlsec/win32/Makefile.msvc	2012-11-30 11:11:06.037550700 -0500
@@ -301,6 +301,10 @@
 CFLAGS 			= $(CFLAGS) /D "HAVE_STDIO_H" /D "HAVE_STDLIB_H"
 CFLAGS 			= $(CFLAGS) /D "HAVE_STRING_H" /D "HAVE_CTYPE_H"
 CFLAGS 			= $(CFLAGS) /D "HAVE_MALLOC_H" /D "HAVE_MEMORY_H"
+CFLAGS 			= $(CFLAGS) -arch:SSE $(SOLARINC) -I$(WORKDIR)\UnpackedTarball\xml2\include -I$(WORKDIR)/UnpackedTarball/icu/source/i18n -I$(WORKDIR)/UnpackedTarball/icu/source/common
+!if "$(MSVC_USE_DEBUG_RUNTIME)" != ""
+CFLAGS 			= $(CFLAGS) /MDd
+!endif
 
 # Optimisation and debug symbols.
 !if "$(DEBUG)" == "1"
diff -ru xmlsec.orig/apps/cmdline.c xmlsec/apps/cmdline.c
--- build/xmlsec.orig/apps/cmdline.c	2009-12-05 22:19:17.000000000 +0100
+++ build/xmlsec/apps/cmdline.c	2015-09-03 23:05:01.003150300 +0200
@@ -7,7 +7,7 @@
  * 
  * Copyright (C) 2002-2003 Aleksey Sanin <aleksey@aleksey.com>
  */
-#if defined(_MSC_VER)
+#if defined(_MSC_VER) && _MSC_VER < 1900
 #define snprintf _snprintf
 #endif
 
diff -ru xmlsec.orig/apps/crypto.c xmlsec/apps/crypto.c
--- build/xmlsec.orig/apps/crypto.c	2009-12-05 22:19:17.000000000 +0100
+++ build/xmlsec/apps/crypto.c	2015-09-03 23:04:36.928472100 +0200
@@ -7,7 +7,7 @@
  * 
  * Copyright (C) 2002-2003 Aleksey Sanin <aleksey@aleksey.com>
  */
-#if defined(_MSC_VER)
+#if defined(_MSC_VER) && _MSC_VER < 1900
 #define snprintf _snprintf
 #endif
 
diff -ru xmlsec.orig/apps/xmlsec.c xmlsec/apps/xmlsec.c
--- build/xmlsec.orig/apps/xmlsec.c	2009-12-05 22:19:17.000000000 +0100
+++ build/xmlsec/apps/xmlsec.c	2015-09-03 23:05:08.057636300 +0200
@@ -9,7 +9,7 @@
 #include <string.h>
 #include <time.h>
 
-#if defined(_MSC_VER)
+#if defined(_MSC_VER) && _MSC_VER < 1900
 #define snprintf _snprintf
 #endif
 
