diff -ur curl-7.26.0/config.guess misc/build/curl-7.26.0/config.guess
--- curl-7.26.0/config.guess	2012-07-02 13:46:52.428053096 +0200
+++ misc/build/curl-7.26.0/config.guess	2012-07-02 13:48:13.644057499 +0200
@@ -565,6 +565,9 @@
 	else
 		IBM_REV=${UNAME_VERSION}.${UNAME_RELEASE}
 	fi
+	if [ "$IBM_REV" == "V7BETA" ]; then
+		IBM_REV=${UNAME_VERSION}.${UNAME_RELEASE}
+	fi
 	echo ${IBM_ARCH}-ibm-aix${IBM_REV}
 	exit ;;
     *:AIX:*:*)
