--- misc/curl-7.19.7/lib/Makefile.vc9	2009-11-04 13:35:39.000000000 +0100
+++ misc/build/curl-7.19.7/lib/Makefile.vc9	2009-11-18 12:32:57.234375000 +0100
@@ -46,13 +46,13 @@
 # Stem for static libs and DLLs
 #
 LIB_NAME       = libcurl
-LIB_NAME_DEBUG = libcurld
+LIB_NAME_DEBUG = libcurl
 
 #
 # Stem for DLL import libs
 #
-IMPLIB_NAME       = libcurl_imp
-IMPLIB_NAME_DEBUG = libcurld_imp
+IMPLIB_NAME       = libcurl
+IMPLIB_NAME_DEBUG = libcurl
 
 !IFNDEF OPENSSL_PATH
 OPENSSL_PATH   = ../../openssl-0.9.8g
