diff -ur curl-7.26.0/configure misc/build/curl-7.26.0/configure
--- curl-7.26.0/configure	2012-07-02 13:28:51.327994494 +0200
+++ misc/build/curl-7.26.0/configure	2012-07-02 13:32:38.754006822 +0200
@@ -2693,8 +2693,8 @@
 
 ac_ext=c
 ac_cpp='$CPP $CPPFLAGS'
-ac_compile='$CC -c $CFLAGS $CPPFLAGS conftest.$ac_ext >&5'
-ac_link='$CC -o conftest$ac_exeext $CFLAGS $CPPFLAGS $LDFLAGS conftest.$ac_ext $LIBS >&5'
+ac_compile='$CC -c $ADDCFLAGS $CFLAGS $CPPFLAGS conftest.$ac_ext >&5'
+ac_link='$CC -o conftest$ac_exeext $ADDCFLAGS $CFLAGS $CPPFLAGS $LDFLAGS conftest.$ac_ext $LIBS >&5'
 ac_compiler_gnu=$ac_cv_c_compiler_gnu
 
 
@@ -3855,8 +3855,8 @@
   ac_save_CPPFLAGS="$CPPFLAGS"
   ac_ext=c
 ac_cpp='$CPP $CPPFLAGS'
-ac_compile='$CC -c $CFLAGS $CPPFLAGS conftest.$ac_ext >&5'
-ac_link='$CC -o conftest$ac_exeext $CFLAGS $CPPFLAGS $LDFLAGS conftest.$ac_ext $LIBS >&5'
+ac_compile='$CC -c $ADDCFLAGS $CFLAGS $CPPFLAGS conftest.$ac_ext >&5'
+ac_link='$CC -o conftest$ac_exeext $ADDCFLAGS $CFLAGS $CPPFLAGS $LDFLAGS conftest.$ac_ext $LIBS >&5'
 ac_compiler_gnu=$ac_cv_c_compiler_gnu
 if test -n "$ac_tool_prefix"; then
   # Extract the first word of "${ac_tool_prefix}gcc", so it can be a program name with args.
@@ -4506,15 +4506,15 @@
   CFLAGS=$ac_save_CFLAGS
 elif test $ac_cv_prog_cc_g = yes; then
   if test "$GCC" = yes; then
-    CFLAGS="-g -O2"
+    CFLAGS="$ADDCFLAGS -g -O2"
   else
-    CFLAGS="-g"
+    CFLAGS="$ADDCFLAGS -g"
   fi
 else
   if test "$GCC" = yes; then
-    CFLAGS="-O2"
+    CFLAGS="$ADDCFLAGS -O2"
   else
-    CFLAGS=
+    CFLAGS="$ADDCFLAGS"
   fi
 fi
 { $as_echo "$as_me:${as_lineno-$LINENO}: checking for $CC option to accept ISO C89" >&5
diff -ur curl-7.26.0/lib/setup.h misc/build/curl-7.26.0/lib/setup.h
--- curl-7.26.0/lib/setup.h	2012-07-02 13:28:51.322994494 +0200
+++ misc/build/curl-7.26.0/lib/setup.h	2012-07-02 13:41:15.652034841 +0200
@@ -231,6 +231,7 @@
 #    include <winsock2.h>
 #    ifdef HAVE_WS2TCPIP_H
 #       include <ws2tcpip.h>
+#       include <wspiapi.h>
 #    endif
 #  else
 #    ifdef HAVE_WINSOCK_H
diff -ur curl-7.26.0/ltmain.sh misc/build/curl-7.26.0/ltmain.sh
--- curl-7.26.0/ltmain.sh	2012-07-02 13:28:51.298994493 +0200
+++ misc/build/curl-7.26.0/ltmain.sh	2012-07-02 13:42:46.511039769 +0200
@@ -7334,13 +7334,13 @@
 	  #
 	  case $version_type in
 	  # correct linux to gnu/linux during the next big refactor
-	  darwin|linux|osf|windows|none)
+	  darwin|linux|osf|windows|freebsd-aout|freebsd-elf|none)
 	    func_arith $number_major + $number_minor
 	    current=$func_arith_result
 	    age="$number_minor"
 	    revision="$number_revision"
 	    ;;
-	  freebsd-aout|freebsd-elf|qnx|sunos)
+	  qnx|sunos)
 	    current="$number_major"
 	    revision="$number_minor"
 	    age="0"
@@ -7420,8 +7420,8 @@
 	  ;;
 
 	freebsd-elf)
-	  major=".$current"
-	  versuffix=".$current"
+	  major=.`expr $current - $age`
+	  versuffix="$major"
 	  ;;
 
 	irix | nonstopux)
