diff -ur misc/curl-7.49.1/winbuild/Makefile.msvc.names misc/build/curl-7.49.1/winbuild/Makefile.msvc.names
--- misc/curl-7.49.1/winbuild/Makefile.msvc.names	2016-03-22 00:15:38.000000000 -0700
+++ misc/build/curl-7.49.1/winbuild/Makefile.msvc.names	2016-07-19 15:35:28.465823000 -0700
@@ -43,7 +43,7 @@
 !ENDIF
 
 !IF !DEFINED(LIB_NAME_STA_DBG) || "$(LIB_NAME_STA_DBG)" == ""
-LIB_NAME_STA_DBG = $(LIB_NAME_STA_REL)d
+LIB_NAME_STA_DBG = $(LIB_NAME_STA_REL)
 !ENDIF
 
 !IF !DEFINED(LIB_NAME_DYN_REL) || "$(LIB_NAME_DYN_REL)" == ""
@@ -51,7 +51,7 @@
 !ENDIF
 
 !IF !DEFINED(LIB_NAME_DYN_DBG) || "$(LIB_NAME_DYN_DBG)" == ""
-LIB_NAME_DYN_DBG = $(LIB_NAME_DYN_REL)d
+LIB_NAME_DYN_DBG = $(LIB_NAME_DYN_REL)
 !ENDIF
 
 # --------------------------------------------
@@ -59,11 +59,11 @@
 # --------------------------------------------
 
 !IF !DEFINED(LIB_NAME_IMP_REL) || "$(LIB_NAME_IMP_REL)" == ""
-LIB_NAME_IMP_REL = $(LIB_NAME_DYN_REL)_imp
+LIB_NAME_IMP_REL = $(LIB_NAME_DYN_REL)
 !ENDIF
 
 !IF !DEFINED(LIB_NAME_IMP_DBG) || "$(LIB_NAME_IMP_DBG)" == ""
-LIB_NAME_IMP_DBG = $(LIB_NAME_DYN_DBG)_imp
+LIB_NAME_IMP_DBG = $(LIB_NAME_DYN_DBG)
 !ENDIF
 
 # --------------------------------------
