--- misc/vigra1.6.0/include/vigra/basicimage.hxx	2012-01-19 22:49:11.232074631 +0100
+++ misc/build/vigra1.6.0/include/vigra/basicimage.hxx	2012-01-19 22:48:12.997439136 +0100
@@ -72,7 +72,7 @@
         difference_type offset_;
     };
 
-    static void initialize(BaseType &) {}
+    static void initialize(__attribute__ ((unused)) BaseType &) {}
 
     static reference dereference(BaseType const & d)
         { return const_cast<reference>(*(*d.line_start_ + d.offset_)); }
--- misc/vigra1.6.0/include/vigra/diff2d.hxx	2012-01-19 22:49:11.280074335 +0100
+++ misc/build/vigra1.6.0/include/vigra/diff2d.hxx	2012-01-19 22:49:00.129144170 +0100
@@ -60,7 +60,7 @@
     typedef Diff const *                    pointer;
     typedef std::random_access_iterator_tag iterator_category;
 
-    static void initialize(BaseType &) {}
+    static void initialize(__attribute__ ((unused)) BaseType &) {}
 
     static reference dereference(BaseType const & d)
         { return d; }
@@ -102,7 +102,7 @@
     typedef Diff const *                    pointer;
     typedef std::random_access_iterator_tag iterator_category;
 
-    static void initialize(BaseType & /*d*/) {}
+    static void initialize(__attribute__ ((unused)) BaseType & /*d*/) {}
 
     static reference dereference(BaseType const & d)
         { return d; }
