--- misc/zlib-1.2.7/makefile.mk	Fri Mar 14 10:17:06 2008
+++ misc/build/zlib-1.2.7/makefile.mk	Fri Mar 14 10:16:56 2008
@@ -1 +1,64 @@
-dummy
+#*************************************************************************
+#
+#   Copyright according the GNU Public License.
+#
+#*************************************************************************
+
+PRJ=..$/..$/..$/..
+
+PRJNAME=zlib
+TARGET=zlib
+LIBTARGET=NO
+EXTERNAL_WARNINGS_NOT_ERRORS=TRUE
+UWINAPILIB=
+
+# --- Settings -----------------------------------------------------
+
+.INCLUDE :  settings.mk
+
+# --- Files --------------------------------------------------------
+
+INCEXT=contrib$/minizip
+
+SLOFILES=	$(SLO)$/adler32.obj		\
+			$(SLO)$/compress.obj	\
+			$(SLO)$/deflate.obj		\
+			$(SLO)$/crc32.obj		\
+			$(SLO)$/inffast.obj		\
+			$(SLO)$/inflate.obj		\
+			$(SLO)$/inftrees.obj	\
+			$(SLO)$/trees.obj		\
+			$(SLO)$/zutil.obj
+
+
+LIB1TARGET=$(SLB)$/$(TARGET).lib
+LIB1ARCHIV=$(LB)$/lib$(TARGET).a
+LIB1OBJFILES=$(SLOFILES)
+
+.IF "$(BUILD_X64)"!=""
+SLOFILES_X64=	$(SLO_X64)$/adler32.obj		\
+			$(SLO_X64)$/compress.obj	\
+			$(SLO_X64)$/deflate.obj		\
+			$(SLO_X64)$/crc32.obj		\
+			$(SLO_X64)$/inffast.obj		\
+			$(SLO_X64)$/inflate.obj		\
+			$(SLO_X64)$/inftrees.obj	\
+			$(SLO_X64)$/trees.obj		\
+			$(SLO_X64)$/zutil.obj
+
+LIB1TARGET_X64=$(SLB_X64)$/$(TARGET).lib
+LIB1OBJFILES_X64=$(SLOFILES_X64)
+.ENDIF # "$(BUILD_X64)"!=""
+
+
+# --- Targets ------------------------------------------------------
+
+$(MISC)$/%.c : contrib$/minizip$/%.c
+	@echo ------------------------------
+	@echo Making: $@
+    @$(COPY) $< $@
+
+.INCLUDE :  set_wntx64.mk
+.INCLUDE :  target.mk
+.INCLUDE :  tg_wntx64.mk
+
