*** misc/lp_solve_5.5/lpsolve55/ccc	Sat Jun 11 21:27:18 2005
--- misc/build/lp_solve_5.5/lpsolve55/ccc	Thu Mar 13 11:05:18 2008
***************
*** 1,5 ****
  src='../lp_MDO.c ../shared/commonlib.c ../shared/mmio.c ../shared/myblas.c ../ini.c ../fortify.c ../colamd/colamd.c ../lp_rlp.c ../lp_crash.c ../bfp/bfp_LUSOL/lp_LUSOL.c ../bfp/bfp_LUSOL/LUSOL/lusol.c ../lp_Hash.c ../lp_lib.c ../lp_wlp.c ../lp_matrix.c ../lp_mipbb.c ../lp_MPS.c ../lp_params.c ../lp_presolve.c ../lp_price.c ../lp_pricePSE.c ../lp_report.c ../lp_scale.c ../lp_simplex.c ../lp_SOS.c ../lp_utils.c ../yacc_read.c'
! c=cc
  
  def=
  so=
--- 1,5 ----
  src='../lp_MDO.c ../shared/commonlib.c ../shared/mmio.c ../shared/myblas.c ../ini.c ../fortify.c ../colamd/colamd.c ../lp_rlp.c ../lp_crash.c ../bfp/bfp_LUSOL/lp_LUSOL.c ../bfp/bfp_LUSOL/LUSOL/lusol.c ../lp_Hash.c ../lp_lib.c ../lp_wlp.c ../lp_matrix.c ../lp_mipbb.c ../lp_MPS.c ../lp_params.c ../lp_presolve.c ../lp_price.c ../lp_pricePSE.c ../lp_report.c ../lp_scale.c ../lp_simplex.c ../lp_SOS.c ../lp_utils.c ../yacc_read.c'
! c=$CC
  
  def=
  so=
*** misc/lp_solve_5.5/lpsolve55/ccc.osx	Thu Jun 23 22:53:08 2005
--- misc/build/lp_solve_5.5/lpsolve55/ccc.osx	Fri Mar  7 09:57:18 2008
***************
*** 3,8 ****
--- 3,12 ----
  
  def=
  so=
+ extra_cflags=$EXTRA_CFLAGS
+ extra_cdefs=$EXTRA_CDEFS
+ extra_linkflags=$EXTRA_LINKFLAGS
+ 
  if [ "$PLATFORM" = "SCO_UNIX" ]
  then def='-DLoadInverseLib=0 -DLoadLanguageLib=0 -D__HYPER=long'
  else dl=-ldl
***************
*** 11,23 ****
  
  opts='-idirafter /usr/include/sys -O3 -DINTEGERTIME -Wno-long-double'
  
! $c -s -c -I.. -I../shared -I../bfp -I../bfp/bfp_LUSOL -I../bfp/bfp_LUSOL/LUSOL -I../colamd $opts $def -DYY_NEVER_INTERACTIVE -DPARSER_LP -DINVERSE_ACTIVE=INVERSE_LUSOL -DRoleIsExternalInvEngine $src
  libtool -static -o liblpsolve55.a `echo $src|sed s/[.]c/.o/g|sed 's/[^ ]*\///g'`
  
  if [ "$so" != "" ]
  then
!   $c -fPIC -fno-common -s -c -I.. -I../shared -I../bfp -I../bfp/bfp_LUSOL -I../bfp/bfp_LUSOL/LUSOL -I../colamd -I. $opts -DYY_NEVER_INTERACTIVE -DPARSER_LP -DINVERSE_ACTIVE=INVERSE_LUSOL -DRoleIsExternalInvEngine $src
!   $c -dynamiclib liblpsolve55.a -compatibility_version 5.5.0 -current_version 5.5.0 -o liblpsolve55.dylib `echo $src|sed s/[.]c/.o/g|sed 's/[^ ]*\///g'` -lc
  fi
  
  rm *.o 2>/dev/null
--- 15,27 ----
  
  opts='-idirafter /usr/include/sys -O3 -DINTEGERTIME -Wno-long-double'
  
! $c $extra_cflags -s -c -I.. -I../shared -I../bfp -I../bfp/bfp_LUSOL -I../bfp/bfp_LUSOL/LUSOL -I../colamd $opts $def -DYY_NEVER_INTERACTIVE -DPARSER_LP -DINVERSE_ACTIVE=INVERSE_LUSOL -DRoleIsExternalInvEngine $extra_cdefs $src
  libtool -static -o liblpsolve55.a `echo $src|sed s/[.]c/.o/g|sed 's/[^ ]*\///g'`
  
  if [ "$so" != "" ]
  then
!   $c $extra_cflags -fPIC -fno-common -s -c -I.. -I../shared -I../bfp -I../bfp/bfp_LUSOL -I../bfp/bfp_LUSOL/LUSOL -I../colamd -I. $opts -DYY_NEVER_INTERACTIVE -DPARSER_LP -DINVERSE_ACTIVE=INVERSE_LUSOL -DRoleIsExternalInvEngine $extra_cdefs $src
!   $c $extra_linkflags -dynamiclib liblpsolve55.a -compatibility_version 5.5.0 -current_version 5.5.0 -o liblpsolve55.dylib `echo $src|sed s/[.]c/.o/g|sed 's/[^ ]*\///g'` -lc
  fi
  
  rm *.o 2>/dev/null
*** misc/lp_solve_5.5/lpsolve55/ccc.solaris	Fri Mar  7 09:57:13 2008
--- misc/build/lp_solve_5.5/lpsolve55/ccc.solaris	Fri Mar  7 09:57:18 2008
***************
*** 1 ****
! dummy
--- 1,25 ----
! src='../lp_MDO.c ../shared/commonlib.c ../shared/mmio.c ../shared/myblas.c ../ini.c ../fortify.c ../colamd/colamd.c ../lp_rlp.c ../lp_crash.c ../bfp/bfp_LUSOL/lp_LUSOL.c ../bfp/bfp_LUSOL/LUSOL/lusol.c ../lp_Hash.c ../lp_lib.c ../lp_wlp.c ../lp_matrix.c ../lp_mipbb.c ../lp_MPS.c ../lp_params.c ../lp_presolve.c ../lp_price.c ../lp_pricePSE.c ../lp_report.c ../lp_scale.c ../lp_simplex.c ../lp_SOS.c ../lp_utils.c ../yacc_read.c'
! c=cc
! 
! def=
! so=
! if [ "$PLATFORM" = "SCO_UNIX" ]
! then def='-dy -K PIC -DNOLONGLONG'
!      dl=-lc
! else dl=-ldl
!      so=y
! fi
! 
! opts='-xO3'
! 
! $c -s -c -I.. -I../shared -I../bfp -I../bfp/bfp_LUSOL -I../bfp/bfp_LUSOL/LUSOL -I../colamd $opts $def -DYY_NEVER_INTERACTIVE -DPARSER_LP -DINVERSE_ACTIVE=INVERSE_LUSOL -DRoleIsExternalInvEngine $src
! ar rv liblpsolve55.a `echo $src|sed s/[.]c/.o/g|sed 's/[^ ]*\///g'`
! ranlib liblpsolve55.a
! 
! if [ "$so" != "" ]
! then
!   $c -K pic -s -c -I.. -I../shared -I../bfp -I../bfp/bfp_LUSOL -I../bfp/bfp_LUSOL/LUSOL -I../colamd -I. $opts -DYY_NEVER_INTERACTIVE -DPARSER_LP -DINVERSE_ACTIVE=INVERSE_LUSOL -DRoleIsExternalInvEngine $src
!   $c -G -Bsymbolic -h liblpsolve55.so -o liblpsolve55.so `echo $src|sed s/[.]c/.o/g|sed 's/[^ ]*\///g'` $SOLARLIB -lc -lm -ldl
! fi
! 
! rm *.o 2>/dev/null
