--- /dev/null	2010-07-28 04:47:47.000000000 -0500
+++ misc/build/lp_solve_5.5/lpsolve55/ccc.aix.gcc	2010-06-29 17:44:53.000000000 -0500
@@ -0,0 +1,25 @@
+src='../lp_MDO.c ../shared/commonlib.c ../shared/mmio.c ../shared/myblas.c ../ini.c ../fortify.c ../colamd/colamd.c ../lp_rlp.c ../lp_crash.c ../bfp/bfp_LUSOL/lp_LUSOL.c ../bfp/bfp_LUSOL/LUSOL/lusol.c ../lp_Hash.c ../lp_lib.c ../lp_wlp.c ../lp_matrix.c ../lp_mipbb.c ../lp_MPS.c ../lp_params.c ../lp_presolve.c ../lp_price.c ../lp_pricePSE.c ../lp_report.c ../lp_scale.c ../lp_simplex.c ../lp_SOS.c ../lp_utils.c ../yacc_read.c'
+c=$CC
+
+def=
+so=
+if [ "$PLATFORM" = "SCO_UNIX" ]
+then def='-dy -K PIC -DNOLONGLONG'
+     dl=-lc
+else dl=-ldl
+     so=y
+fi
+
+opts='-O3'
+
+$c -s -c -I.. -I../shared -I../bfp -I../bfp/bfp_LUSOL -I../bfp/bfp_LUSOL/LUSOL -I../colamd $opts $def -DYY_NEVER_INTERACTIVE -DPARSER_LP -DINVERSE_ACTIVE=INVERSE_LUSOL -DRoleIsExternalInvEngine $src
+ar rv liblpsolve55.a `echo $src|sed s/[.]c/.o/g|sed 's/[^ ]*\///g'`
+ranlib liblpsolve55.a
+
+if [ "$so" != "" ]
+then
+  $c -fpic -s -c -I.. -I../shared -I../bfp -I../bfp/bfp_LUSOL -I../bfp/bfp_LUSOL/LUSOL -I../colamd -I. $opts -DYY_NEVER_INTERACTIVE -DPARSER_LP -DINVERSE_ACTIVE=INVERSE_LUSOL -DRoleIsExternalInvEngine $src
+  $c -shared $lpsolve_LDFLAGS -o liblpsolve55.so `echo $src|sed s/[.]c/.o/g|sed 's/[^ ]*\///g'` -lc -lm -ldl
+fi
+
+rm *.o 2>/dev/null
--- misc/lp_solve_5.5/lp_lib.h	2010-07-28 05:15:10.000000000 -0500
+++ misc/build/lp_solve_5.5/lp_lib.h	2010-07-28 05:15:55.000000000 -0500
@@ -56,7 +56,7 @@
 /* Define user program feature option switches                               */
 /* ------------------------------------------------------------------------- */
 
-#if !defined _WINDOWS && !defined _WIN32 && !defined WIN32
+#if !defined _WINDOWS && !defined _WIN32 && !defined WIN32 && !defined(_AIX)
 # define _isnan(x) FALSE
 #endif
 
