--- misc/nss-3.12.8/mozilla/nsprpub/configure	2010-08-18 07:23:01.000000000 -0500
+++ misc/build/nss-3.12.8/mozilla/nsprpub/configure	2010-08-18 07:33:11.000000000 -0500
@@ -2973,7 +2973,8 @@
 #define SYSV 1
 EOF
 
-    DSO_LDOPTS='-brtl -bnortllib -bM:SRE -bnoentry -bexpall -blibpath:/usr/lib:/lib'
+    MKSHLIB='$(CCC) -shared $(DSO_LDOPTS) -o $@'
+    DSO_LDOPTS='-Wl,-brtl -Wl,-bnortllib -Wl,-blibpath:/usr/lib:/lib'
     ac_safe=`echo "sys/atomic_op.h" | sed 'y%./+-%__p_%'`
 echo $ac_n "checking for sys/atomic_op.h""... $ac_c" 1>&6
 echo "configure:2980: checking for sys/atomic_op.h" >&5
@@ -3102,7 +3103,6 @@
         AIX_LINK_OPTS='-brtl -bnso -berok'
         ;;
     esac
-    CFLAGS="$CFLAGS -qro -qroconst"
     AIX_WRAP='$(DIST)/lib/aixwrap.o'
     AIX_TMP='./_aix_tmp.o'
     if test -n "$USE_64"; then
--- misc/nss-3.12.8/mozilla/nsprpub/configure.in	2010-08-18 07:23:01.000000000 -0500
+++ misc/build/nss-3.12.8/mozilla/nsprpub/configure.in	2010-08-18 07:32:36.000000000 -0500
@@ -831,7 +831,8 @@
     AC_DEFINE(XP_UNIX)
     AC_DEFINE(AIX)
     AC_DEFINE(SYSV)
-    DSO_LDOPTS='-brtl -bnortllib -bM:SRE -bnoentry -bexpall -blibpath:/usr/lib:/lib'
+    DSO_LDOPTS='-Wl,-brtl -Wl,-bnortllib -Wl,-blibpath:/usr/lib:/lib'
+    MKSHLIB='$(CCC) -shared $(DSO_LDOPTS) -o $@'
     AC_CHECK_HEADER(sys/atomic_op.h, AC_DEFINE(AIX_HAVE_ATOMIC_OP_H))
     case "${target_os}" in
     aix3.2*)
@@ -871,10 +872,9 @@
         AC_DEFINE(HAVE_SOCKLEN_T)
         AC_DEFINE(HAVE_FCNTL_FILE_LOCKING)
         USE_IPV6=1
-        AIX_LINK_OPTS='-brtl -bnso -berok'
+        AIX_LINK_OPTS='-Wl,-brtl -Wl,-bnso -Wl,-berok'
         ;;
     esac
-    CFLAGS="$CFLAGS -qro -qroconst"
     AIX_WRAP='$(DIST)/lib/aixwrap.o'
     AIX_TMP='./_aix_tmp.o'
     if test -n "$USE_64"; then
--- misc/nss-3.12.8/mozilla/nsprpub/pr/src/Makefile.in	2010-08-18 07:23:04.000000000 -0500
+++ misc/build/nss-3.12.8/mozilla/nsprpub/pr/src/Makefile.in	2010-08-18 07:24:49.000000000 -0500
@@ -110,7 +110,6 @@
 endif
 
 ifeq ($(OS_ARCH),AIX)
-DSO_LDOPTS	+= -binitfini::_PR_Fini
 OS_LIBS		= -lodm -lcfg
 ifeq ($(CLASSIC_NSPR),1)
 ifeq ($(OS_RELEASE),4.1)
--- misc/nss-3.12.8/mozilla/security/coreconf/AIX.mk	2010-08-18 07:23:06.000000000 -0500
+++ misc/build/nss-3.12.8/mozilla/security/coreconf/AIX.mk	2010-08-18 07:26:26.000000000 -0500
@@ -62,35 +62,30 @@
 
 DEFAULT_COMPILER = xlc_r
 
-CC		= xlc_r
-CCC		= xlC_r
+CCC		= $(CXX)
 
 CPU_ARCH	= rs6000
 
 RANLIB		= ranlib
 
 OS_CFLAGS	= -DAIX -DSYSV
-OS_LIBS 	+= -blibpath:/usr/lib:/lib -lc -lm
+OS_LIBS 	+= -Wl,-blibpath:/usr/lib:/lib -lc -lm
 
-DSO_LDOPTS	= -brtl -bnortllib -bM:SRE -bnoentry
-MKSHLIB 	= $(LD) $(DSO_LDOPTS) -blibpath:/usr/lib:/lib -lc -lm
+DSO_LDOPTS	= -Wl,-brtl -Wl,-bnortllib
+MKSHLIB 	= $(CC) --shared $(DSO_LDOPTS) -Wl,-blibpath:/usr/lib:/lib -lc -lm
 
 AIX_WRAP	= $(DIST)/lib/aixwrap.o
 AIX_TMP		= $(OBJDIR)/_aix_tmp.o
 
 ifdef MAPFILE
-DSO_LDOPTS	+= -bexport:$(MAPFILE)
+DSO_LDOPTS	+= -Wl,-bexport:$(MAPFILE)
 else
-DSO_LDOPTS	+= -bexpall
+DSO_LDOPTS	+= -Wl,-bexpall
 endif
 
 PROCESS_MAP_FILE = grep -v ';+' $< | grep -v ';-' | \
                 sed -e 's; DATA ;;' -e 's,;;,,' -e 's,;.*,,' > $@
 
-ifdef BUILD_OPT
-	OPTIMIZER += -qmaxmem=-1
-endif
-
 ifeq ($(USE_64), 1)
 	OS_CFLAGS	+= -DAIX_64BIT
 	OBJECT_MODE=64
--- misc/nss-3.12.8/mozilla/security/nss/cmd/platlibs.mk	2010-08-18 07:23:18.000000000 -0500
+++ misc/build/nss-3.12.8/mozilla/security/nss/cmd/platlibs.mk	2010-08-18 07:26:48.000000000 -0500
@@ -173,7 +173,7 @@
 	$(NULL)
 
 ifeq ($(OS_ARCH), AIX) 
-EXTRA_SHARED_LIBS += -brtl 
+EXTRA_SHARED_LIBS += -Wl,-brtl 
 endif
 
 # $(PROGRAM) has NO explicit dependencies on $(EXTRA_SHARED_LIBS)
@@ -223,7 +223,7 @@
 	$(NULL)
 
 ifeq ($(OS_ARCH), AIX) 
-EXTRA_SHARED_LIBS += -brtl 
+EXTRA_SHARED_LIBS += -Wl,-brtl 
 endif
 
 # $(PROGRAM) has NO explicit dependencies on $(EXTRA_SHARED_LIBS)
--- misc/nss-3.12.8/mozilla/security/nss/cmd/shlibsign/mangle/Makefile	2010-08-18 07:23:20.000000000 -0500
+++ misc/build/nss-3.12.8/mozilla/security/nss/cmd/shlibsign/mangle/Makefile	2010-08-18 07:42:44.000000000 -0500
@@ -75,7 +75,9 @@
 
 endif
 
-
+ifeq ($(OS), AIX)
+EXTRA_SHARED_LIBS += -lpthreads
+endif
 
 #######################################################################
 # (5) Execute "global" rules. (OPTIONAL)                              #
--- misc/nss-3.12.8/mozilla/security/nss/cmd/shlibsign/Makefile	2010-08-18 08:00:17.000000000 -0500
+++ misc/build/nss-3.12.8/mozilla/security/nss/cmd/shlibsign/Makefile	2010-08-18 07:55:55.000000000 -0500
@@ -75,6 +75,9 @@
 
 endif
 
+ifeq ($(OS), AIX)
+EXTRA_SHARED_LIBS += -lpthreads
+endif
 
 # sign any and all shared libraries that contain the word freebl
 
