diff -uNrp misc/nss-3.14.4/mozilla/security/coreconf/Linux.mk misc/build/nss-3.14.4/mozilla/security/coreconf/Linux.mk
--- misc/nss-3.14.4/mozilla/security/coreconf/Linux.mk	2011-12-14 14:06:50.112354384 -0300
+++ misc/build/nss-3.14.4/mozilla/security/coreconf/Linux.mk	2011-12-14 14:07:30.978358265 -0300
@@ -174,3 +174,18 @@ RPATH = -Wl,-rpath,'$$ORIGIN:/opt/sun/pr
 endif
 endif
 
+OS_REL_CFLAGS   += -DLINUX2_1
+MKSHLIB         = $(CC) $(DSO_LDOPTS) -Wl,-soname -Wl,$(@:$(OBJDIR)/%.so=%.so) $(RPATH)
+
+ifdef MAPFILE
+	MKSHLIB += -Wl,--version-script,$(MAPFILE)
+endif
+PROCESS_MAP_FILE = grep -v ';-' $< | \
+        sed -e 's,;+,,' -e 's; DATA ;;' -e 's,;;,,' -e 's,;.*,;,' > $@
+
+ifeq ($(OS_RELEASE),2.4)
+# Softoken 3.13 uses NO_FORK_CHECK only.
+# Softoken 3.12 uses NO_FORK_CHECK and NO_CHECK_FORK.
+# Don't use NO_CHECK_FORK in new code.
+DEFINES += -DNO_FORK_CHECK -DNO_CHECK_FORK
+endif
