--- misc/nss-3.14.4/mozilla/security/coreconf/Linux.mk	2013-11-01 14:39:28.195966851 +0100
+++ misc/build/nss-3.14.4/mozilla/security/coreconf/Linux.mk	2013-11-01 14:39:10.246610849 +0100
@@ -131,7 +131,7 @@
 # -ansi on platforms like Android where the system headers are C99 and do
 # not build with -ansi.
 STANDARDS_CFLAGS	= -D_POSIX_SOURCE -D_BSD_SOURCE -D_XOPEN_SOURCE
-OS_CFLAGS		= $(STANDARDS_CFLAGS) $(DSO_CFLAGS) $(OS_REL_CFLAGS) $(ARCHFLAG) -Wall -Werror-implicit-function-declaration -Wno-switch -pipe -DLINUX -Dlinux -DHAVE_STRERROR
+OS_CFLAGS		= $(STANDARDS_CFLAGS) $(DSO_CFLAGS) $(OS_REL_CFLAGS) $(ARCHFLAG) -Wall -Werror-implicit-function-declaration -Wno-switch -pipe -DLINUX -Dlinux -DHAVE_STRERROR -DHAVE_UNISTD_H
 OS_LIBS			= $(OS_PTHREAD) -ldl -lc
 
 ifdef USE_PTHREADS
