--- misc/hunspell-1.2.9/src/hunspell/affixmgr.cxx	2010-02-27 12:59:53.000000000 +0100
+++ misc/build/hunspell-1.2.9/src/hunspell/affixmgr.cxx	2011-01-06 16:18:23.361159724 +0100
@@ -1542,7 +1542,7 @@
         }
 
         if (!rv) {
-            if (compoundflag && 
+            if (compoundflag && !words && 
              !(rv = prefix_check(st, i, hu_mov_rule ? IN_CPD_OTHER : IN_CPD_BEGIN, compoundflag))) {
                 if ((rv = suffix_check(st, i, 0, NULL, NULL, 0, NULL,
                         FLAG_NULL, compoundflag, hu_mov_rule ? IN_CPD_OTHER : IN_CPD_BEGIN)) && !hu_mov_rule &&
