--- src/ne_gnutls.c	2009-12-02 23:40:41.000000000 +0200
+++ src/ne_gnutls.c	2013-03-29 13:41:00.429857276 +0200
@@ -48,6 +48,26 @@
 #include <iconv.h>
 #endif
 
+
+/* GnuTLS removed these symbols from 2.99.0.
+ *
+ * This is a dodgy solution to avoid further patching of neon
+ * sources */
+#if LIBGNUTLS_VERSION_NUMBER >= 0x026300
+# define gnutls_pkcs12          gnutls_pkcs12_t
+# define gnutls_pkcs12_bag      gnutls_pkcs12_bag_t
+# define gnutls_pkcs12_bag_type gnutls_pkcs12_bag_type_t
+# define gnutls_certificate_verify_peers(_ssl) ({ \
+    int __status_ ## __LINE__ = 0; \
+    gnutls_certificate_verify_peers2((_ssl), &__status_ ## __LINE__) == 0 \
+        ? __status_ ## __LINE__ : -1; \
+  })
+#elif LIBGNUTLS_VERSION_NUMBER >= 0x020303
+/* GnuTLS had these symbols since 2.3.3. */
+# define HAVE_GNUTLS_CERTIFICATE_GET_X509_CAS
+#endif /* LIBGNUTLS_VERSION_NUMBER */
+
+
 #include "ne_ssl.h"
 #include "ne_string.h"
 #include "ne_session.h"
